<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title> Vaccine Trials</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Zehui Bai</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="02-Clinical_Experience.html">
    <span class="fa fa-user"></span>
     
    Clinical Experience
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-sliders"></span>
     
    Sample Size
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="03-SSC-Everything-to-Know-About-Sample-Size-Determination.html">Everything to Know About Sample Size Determination</a>
    </li>
    <li>
      <a href="03-SSC-Choosing-the-Effect-Size-for-Sample-Size-Calculations.html">Choosing the Effect Size</a>
    </li>
    <li>
      <a href="03-SSC-Biosimilar-Trials.html">Statistical Considerations for the Design and Analysis of Biosimilar Trials</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-and-Power-for-Non-Parametric-Analysis.html">Sample Size and Power for Non-Parametric Analysis</a>
    </li>
    <li>
      <a href="03-SSC-Power-for-Complex-Hypotheses.html">Power for Complex Hypotheses</a>
    </li>
    <li>
      <a href="03-SSC-Alternatives-to-Power.html">Bayesian methods - Alternatives to Power</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-for-Pilot-Studies.html">Sample Size for Pilot Studies</a>
    </li>
    <li>
      <a href="03-SSC-Case-Continuous-Endpoint.html">Sample Size Determination for Continuous Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Categorical-Endpoint.html">Sample Size Determination for Categorical Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-Determination-for-Counts-and-Rates.html">Sample Size Determination for Counts and Rates</a>
    </li>
    <li>
      <a href="03-SSC-Case-Survival-Endpoint.html">Sample Size Determination for Survival Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Repeated-Measures.html">Sample Size Determination for Repeated Measures</a>
    </li>
    <li>
      <a href="03-SSC-IA-Sequential-Design.html">Statistical Considerations for Group Sequential Design</a>
    </li>
    <li>
      <a href="03-SSC-IA-Adaptive-Design.html">Statistical Considerations for Adaptive Design</a>
    </li>
    <li>
      <a href="03-SSC-Multiple-Test.html">Sample Size for Multiple Test</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    Study Design
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04-Design-General-Consideration.html">General Consideration</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Framework.html">Estimands Framework</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Practice.html">Estimands and Sensitivity Analyses</a>
    </li>
    <li>
      <a href="04-Design-Phase-I-Trials---Design-Considerations.html">Phase I Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-II-Trials---Design-Considerations.html">Phase II Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-III-Trials---Design-Considerations.html">Phase III Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-IV-Trials---Design-Considerations.html">Phase IV Trials - Design Considerations for Post Marketing Surveillance</a>
    </li>
    <li>
      <a href="04-Design-Complex-Sequential-Trials.html">Complex Sequential Analysis Trials</a>
    </li>
    <li>
      <a href="04-Design-Adaptive-Clinical-Trials.html">Adaptive Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Bayesian-Clinical-Trials.html">Bayesian Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Dose-Escalation-and-Stratification-Designs.html">Dose Escalation and Stratification Designs in Early Oncology Development</a>
    </li>
    <li>
      <a href="04-Design-Single-Arm-Clinical-Trials.html">Single Arm Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-Design-and-Evaluation.html">Diagnostic Study-Design and Evaluation</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-MRMC.html">Diagnostic Study-Multireader Multicase (MRMC)</a>
    </li>
    <li>
      <a href="04-Design-Vaccine-Design.html">Vaccine Trials</a>
    </li>
    <li>
      <a href="04-Design-Externally-Controlled-Trials.html">Externally Controlled Trials</a>
    </li>
    <li>
      <a href="04-Design-Noninferiority-Trials.html">Noninferiority Trials</a>
    </li>
    <li>
      <a href="04-Design-Bioequivalence-and-Biosimilar-Trials.html">Bioequivalence and Biosimilar Trials</a>
    </li>
    <li>
      <a href="04-Design-Exploring-Survival-Analysis-Designs-for-Clinical-Trials.html">Exploring Survival Analysis Designs for Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Hierarchical-composite-endpoints.html">Hierarchical Composite Endpoints</a>
    </li>
    <li>
      <a href="04-Design-Regulatory-Submission.html">Regulatory Submission from Stats Perspective</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Data Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="05-Plot-Time-To-Event.html">Time to Event Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-PRO-Data.html">Patient Reported Outcome Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-SSC-and-Power-Calculation.html">Sample Size and Power Calculations Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Correlation.html">Correlation Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Model-Table.html">Clinical Data and Model visualization</a>
    </li>
    <li>
      <a href="05-Plot-ScatterPlot.html">Scatter and Line Plot</a>
    </li>
    <li>
      <a href="05-Plot-BarPlot.html">Bar Chart</a>
    </li>
    <li>
      <a href="05-Plot-PieChart.html">Pie Chart</a>
    </li>
    <li>
      <a href="05-Plot-BoxPlot.html">Box Plot</a>
    </li>
    <li>
      <a href="05-Plot-Histogram.html">Histogram</a>
    </li>
    <li>
      <a href="05-Plot-Forest-Plot.html">Forest Plot</a>
    </li>
    <li>
      <a href="05-Plot-Flow-Chart.html">Flow Chart</a>
    </li>
    <li>
      <a href="05-Plot-Some-Interesting.html">Some Interesting Plots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-calculator"></span>
     
    Statistical Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06-Analysis-Linear-Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Logistic-Regression.html">Logistic Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Mixed-Model.html">Mixed Model</a>
    </li>
    <li>
      <a href="06-Analysis-MMRM.html">Mixed Model Repeated Measures</a>
    </li>
    <li>
      <a href="06-Analysis-GEE.html">Generalized Estimating Equation</a>
    </li>
    <li>
      <a href="06-Analysis-ANOVA.html">Analysis of Variance</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Analysis.html">Survival Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Competing-Risk.html">Survival Analysis - Competing Risk</a>
    </li>
    <li>
      <a href="06-Analysis-Missing-Data.html">Missing Data Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-MI-Implementation.html">Multiple Imputation Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SSD.html">Safety Signal Detection and Evaluation</a>
    </li>
    <li>
      <a href="06-Analysis-Meta-Analysis.html">Meta Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-PK-and-PD.html">PK and PD Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Time-Series-Analysis.html">Time Series Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SEM.html">Structural Equation Modeling</a>
    </li>
    <li>
      <a href="06-Analysis-Factor-Analysis.html">Factor Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Propensity-Score.html">Propensity Score Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Machine Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="07-ML-Bayesian-Theory.html">Bayesian Theory</a>
    </li>
    <li>
      <a href="07-ML-Bayesian-Analysis.html">Bayesian Analysis</a>
    </li>
    <li>
      <a href="07-ML-Regularization-Penalized-Regression.html">Regularization Penalized Regression</a>
    </li>
    <li>
      <a href="07-ML-Loss-Regression.html">Loss Functions in Machine Learning</a>
    </li>
    <li>
      <a href="07-ML-PCA.html">Principal Component Analysis</a>
    </li>
    <li>
      <a href="07-ML-KNN.html">K-Nearest Neighbors</a>
    </li>
    <li>
      <a href="07-ML-SVM.html">Support Vector Machine</a>
    </li>
    <li>
      <a href="07-ML-Tree-Models.html">Tree Models</a>
    </li>
    <li>
      <a href="07-ML-LDA.html">Linear Discriminant Analysis</a>
    </li>
    <li>
      <a href="07-ML-Cluster-Analysis.html">Cluster Analysis</a>
    </li>
    <li>
      <a href="07-ML-Neural-Networks.html">Neural Network</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-stethoscope"></span>
     
    Clinical Analysis (ADaM)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="08-Clinical-Disposition-Baseline.html">Disposition and Baseline</a>
    </li>
    <li>
      <a href="08-Clinical-Efficacy.html">Efficacy Analysis</a>
    </li>
    <li>
      <a href="08-Clinical-Pharmacokinetics-Analysis.html">Pharmacokinetics Analysis</a>
    </li>
    <li>
      <a href="08-Clinical-Adverse-Event.html">Adverse Event Visualization</a>
    </li>
    <li>
      <a href="08-Clinical-Laboratory-Data.html">Laboratory Data Visualization</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="08-CV.html">
    <span class="fa fa-file-pdf-o"></span>
     
    CV
  </a>
</li>
<li>
  <a href="mailto:zehuibai@outlook.com">
    <span class="fa fa-envelope-o"></span>
     
    Contact me
  </a>
</li>
<li>
  <a href="https://github.com/Zehuibai">
    <span class="fa fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><p><img src="logo.png"
style="width:3in" /><br />
Vaccine Trials</p></h1>

</div>


<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<div id="phases-of-vaccine-trials" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Phases of Vaccine
Trials</h2>
<p>Vaccine trials are a critical component of public health, designed to
ensure that vaccines are both safe and effective before widespread use.
They are uniquely complex compared to other clinical trials due to the
preventive nature of vaccines and their administration to healthy
populations. Below is a detailed explanation of vaccine trial designs
and the challenges they present.</p>
<p>Vaccine development typically progresses through several phases:</p>
<ul>
<li><p><strong>Phase I</strong>: Involves a small group of healthy
volunteers to assess safety, determine appropriate dosage, and identify
any immediate adverse effects.</p></li>
<li><p><strong>Phase II</strong>: Expands the participant pool to
include individuals representative of the target population, focusing on
immunogenicity (the ability to provoke an immune response) and continued
safety assessment.</p></li>
<li><p><strong>Phase III</strong>: Encompasses large-scale trials with
thousands to tens of thousands of participants to evaluate the vaccine’s
efficacy in preventing the disease and to monitor for rare side effects.
Given that many vaccinated individuals may not be exposed to the
pathogen, large sample sizes are necessary to detect statistically
significant differences between vaccinated and unvaccinated
groups.</p></li>
<li><p><strong>Post-Marketing Surveillance (Phase IV)</strong>: After
regulatory approval, ongoing monitoring continues to detect any
long-term or rare adverse events in the general population.</p></li>
</ul>
<p><strong>Safety Considerations</strong></p>
<p>Safety is paramount in vaccine trials due to their administration to
healthy individuals, including vulnerable populations like children and
the elderly. Safety assessments occur at all trial phases, with
particular emphasis during Phase III and post-marketing surveillance.
Regulatory agencies employ systems such as the Vaccine Adverse Event
Reporting System (VAERS) and the Vaccine Safety Datalink (VSD) to
monitor and evaluate adverse events continuously.</p>
<p><img src="02_Plots/Vaccine/Vaccine-COVID19Development.jpg" /></p>
<p>To respond to the COVID-19 emergency, traditional timelines were
compressed. Normally, vaccine development might take ten years, but
several COVID-19 vaccines reached approval in just 18 to 20 months. This
was achieved through parallel trial phases, early manufacturing
scale-up, and the integration of computational tools. For example,
Moderna’s mRNA vaccine entered Phase I trials within ten weeks of the
virus genome being published.</p>
</div>
<div id="challenges-of-vaccine-trial-design" class="section level2"
number="1.2">
<h2><span class="header-section-number">1.2</span> Challenges of Vaccine
Trial Design</h2>
<p>Vaccine trials are uniquely challenging because they aim to prevent
diseases that may occur infrequently. This rarity of cases (e.g., a 1%
attack rate) means that trials must involve very large sample
sizes—often tens of thousands of participants—to accumulate enough
disease events to properly assess efficacy. Unlike therapeutics, which
are given only to sick patients, vaccines are administered to healthy
individuals across a wide population, making safety concerns even more
critical.</p>
<p>Due to this wide rollout, there is a strong emphasis on ensuring both
short- and long-term safety. Vaccine trials are longer and more
extensive than many other types of trials, often including
post-marketing surveillance to detect rare adverse events that may not
appear during the Phase III trial. Safety is assessed continuously
throughout the process.</p>
<p>The design of vaccine trials also introduces unique terminology and
methods. While concepts like attack rate, incidence, and vaccine
efficacy are standard in vaccine studies, they often parallel familiar
statistical concepts like proportions and time-to-event analysis found
in other areas. The language around vaccine studies can seem specialized
but usually maps back to more general clinical trial methodology.</p>
<p><strong>Challenges in Vaccine Trials</strong></p>
<p>Several challenges are inherent to vaccine trials:</p>
<ul>
<li><p><strong>Large Sample Sizes and Long Durations</strong>: To detect
rare adverse events and ensure statistical power, vaccine trials often
require large participant numbers and extended follow-up periods,
sometimes spanning 5-10 years.</p></li>
<li><p><strong>Endpoint Selection</strong>: Determining appropriate
endpoints is complex. Endpoints may include disease incidence, severity,
or composite measures combining multiple outcomes. The choice impacts
trial design and statistical analysis.</p></li>
<li><p><strong>Adaptive Designs</strong>: While adaptive trial designs
can offer flexibility and efficiency, their application in vaccine
trials is limited due to the emphasis on safety and the complexity of
implementing changes mid-trial.</p></li>
</ul>
<p><img src="02_Plots/Vaccine/Vaccine-Development.png" /></p>
</div>
</div>
<div
id="standard-statistical-methods-for-the-analysis-of-immunogenicity-data"
class="section level1" number="2">
<h1><span class="header-section-number">2</span> Standard Statistical
Methods for the Analysis of Immunogenicity Data</h1>
<p>Historically, vaccine immunogenicity studies have primarily focused
on evaluating the humoral immune response, specifically through the
measurement of serum antibody levels. In recent years, however, there
has been a growing emphasis on also assessing cellular immunity in such
trials. Despite this development, the use of cellular immunity endpoints
in vaccine trials largely remains at the investigational stage.</p>
<p>To summarize both humoral and cellular immunity data, four standard
statistical measures are commonly used:</p>
<ol style="list-style-type: decimal">
<li><strong>Geometric Mean Response</strong> (GMR)</li>
<li><strong>Geometric Mean Fold Increase</strong> (GMFI)</li>
<li><strong>Seroprotection Rate</strong></li>
<li><strong>Seroconversion Rate</strong></li>
</ol>
<p>The <strong>GMR</strong> and <strong>Seroprotection Rate</strong>
provide absolute measures of immunogenicity, while the
<strong>GMFI</strong> and <strong>Seroconversion Rate</strong> capture
within-individual increases following vaccination.</p>
<div id="geometric-mean-titres-and-concentrations"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Geometric Mean Titres
and Concentrations</h2>
<div
id="geometric-mean-titres-gmt-and-geometric-mean-concentrations-gmc"
class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Geometric Mean
Titres (GMT) and Geometric Mean Concentrations (GMC)</h3>
<p>Post-vaccination immunogenicity data, whether for humoral or cellular
responses, tend to exhibit right-skewed distributions. However, applying
a logarithmic transformation to these values often results in an
approximately normal distribution, which allows the use of standard
parametric statistical methods. For interpretability, the geometric
means and confidence intervals are computed by taking the antilog of
estimates derived from log-transformed data.</p>
<p>The <strong>Geometric Mean (GM)</strong> is the standard summary
statistic for immunogenicity data:</p>
<p>For a set of <em>n</em> independent and identically distributed
(i.i.d.) immunogenicity values <em>v₁, v₂, …, vₙ</em>, the geometric
mean is defined as:</p>
<p><span class="math display">\[
GM = (v_1 \times v_2 \times \ldots \times v_n)^{1/n}
\]</span></p>
<p>Alternatively, it can be expressed as:</p>
<p><span class="math display">\[
GM = \exp \left( \frac{1}{n} \sum_{i=1}^n \log_e v_i \right)
\]</span></p>
<p>The logarithmic base used for the transformation can vary, commonly
log₂ or log₁₀, but consistency is crucial. For example, when log₁₀ is
used:</p>
<p><span class="math display">\[
GM = 10^{\frac{1}{n} \sum_{i=1}^n \log_{10} v_i}
\]</span></p>
<hr />
</div>
<div id="geometric-mean-titres-for-antibody-dilutions"
class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Geometric Mean
Titres for Antibody Dilutions</h3>
<p>In immunogenicity studies using antibody titres based on reciprocal
two-fold serial dilutions (e.g., 1:8, 1:16, 1:32, etc.), a convenient
log transformation is applied:</p>
<p><span class="math display">\[
u_i = \log_2 \left( \frac{t_i}{D/2} \right)
\]</span></p>
<p>Where <em>tᵢ</em> is the observed titre and <em>D</em> is the lowest
dilution tested. The transformed <em>uᵢ</em> values correspond to
dilution steps (1, 2, 3, etc.), simplifying interpretation. The
geometric mean of these titres is then:</p>
<p><span class="math display">\[
GM = \left( \frac{D}{2} \right)^{\frac{1}{n} \sum_{i=1}^n u_i}
\]</span></p>
<p><strong>Illustrative Example: Rubella HI Test</strong></p>
<p>Rubella virus infection, while often mild in adults, poses
significant risks during pregnancy. A haemagglutination inhibition (HI)
test is frequently performed to assess immunity. Consider HI titres with
a lowest tested dilution of 1:8, resulting in possible titres of 8, 16,
32, 64, etc.</p>
<p>Using the standard log₂ transformation:</p>
<ul>
<li>A titre of 8 becomes 1</li>
<li>16 becomes 2</li>
<li>32 becomes 3</li>
<li>64 becomes 4</li>
</ul>
<p>For titres 8, 8, 16, 32, 64, the GMT is computed as:</p>
<p><span class="math display">\[
GMT = 4 \times 2^{(1+1+2+3+4)/5} = 18.379
\]</span></p>
<p>In this transformation, differences in log-transformed values
directly correspond to dilution steps, simplifying interpretation.</p>
<hr />
</div>
<div id="geometric-standard-deviation-gsd-and-confidence-intervals"
class="section level3" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Geometric Standard
Deviation (GSD) and Confidence Intervals</h3>
<p>The <strong>Geometric Standard Deviation (GSD)</strong> is derived by
taking the exponential of the standard deviation (SD) of the
log-transformed data:</p>
<p><span class="math display">\[
GSD = \exp(SD)
\]</span></p>
<p>This allows calculation of the two-sided 95% confidence interval for
the underlying geometric mean (denoted as <em>ē</em>) using:</p>
<p><span class="math display">\[
LCL_e = \frac{GMT}{GSD^{t_{n-1, 1-\alpha/2} / \sqrt{n}}}  
\]</span></p>
<p><span class="math display">\[
UCL_e = GMT \times GSD^{t_{n-1, 1-\alpha/2} / \sqrt{n}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>tₙ₋₁,₁₋α/2</em> is the corresponding percentile of the Student’s
t-distribution with <em>(n-1)</em> degrees of freedom.</li>
</ul>
<p><strong>Rubella HI Test Example: Confidence Interval
Calculation</strong></p>
<p>Continuing the Rubella example:</p>
<ul>
<li>Sample SD of the log-transformed titres: <strong>0.904</strong></li>
<li>GSD = <strong>exp(0.904) = 2.469</strong></li>
<li>Using <em>t₄, 0.975 = 2.776</em> for a 95% confidence interval</li>
</ul>
<p>The lower and upper confidence limits for the GMT are calculated
as:</p>
<p><span class="math display">\[
LCL_e = \frac{18.379}{2.469^{(2.776/\sqrt{5})}} = 5.98  
\]</span></p>
<p><span class="math display">\[
UCL_e = 18.379 \times 2.469^{(2.776/\sqrt{5})} = 56.4  
\]</span></p>
</div>
<div id="single-vaccine-group" class="section level3" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> Single Vaccine
Group</h3>
<p>When immunogenicity values (<em>vi</em>) are log-transformed as
<em>ui = logₑ(vi)</em> and the <em>ui</em> values follow a normal
distribution with mean <strong>μ</strong> and variance
<strong>σ²</strong>, the <strong>arithmetic mean</strong> of the
<em>ui</em> values serves as a point estimate for μ. Consequently, the
<strong>geometric mean titre (GMT)</strong> or <strong>geometric mean
concentration (GMC)</strong> is estimated as:</p>
<p><span class="math display">\[
GMT = e^{\bar{u}}
\]</span></p>
<p>The distribution of the original <em>vi</em> values is thus
log-normal.</p>
<p>Since the <em>ui</em> values are normally distributed, confidence
intervals for μ can be calculated using the one-sample <em>t</em>-test.
The antilog of these limits provides the corresponding confidence
intervals for the geometric mean <strong>e</strong></p>
<p>It is important to note that the <strong>mean</strong> of the
log-normal distribution (expected value of <em>vi</em>) is not
<strong>e</strong>, but:</p>
<p><span class="math display">\[
E(vi) = e^{μ + \frac{σ^2}{2}}
\]</span></p>
<p>A key property of the log-normal distribution is that its
<strong>geometric mean</strong> and <strong>median</strong> are both
equal to <strong>e</strong>.</p>
<p><strong>Example Single Vaccine Group</strong></p>
<p>Consider six IFN-γ (Th1-type) immunogenicity measurements in IU/L:
<strong>3.51, 9.24, 13.7, 35.2, 47.4, 57.5</strong></p>
<p>Their natural logarithms are: <strong>1.256, 2.224, 2.617, 3.561,
3.859, 4.052</strong></p>
<p>The arithmetic mean of these log-transformed values is
<strong>2.928</strong>, with a standard error of <strong>0.444</strong>.
The geometric mean concentration (GMC) is thus:</p>
<p><span class="math display">\[
GMC = e^{2.928} = 18.7 \, \text{IU/L}
\]</span></p>
<p>Using a two-sided 95% confidence interval for μ with
<em>t</em>(0.975, 5) = 2.571:</p>
<p><span class="math display">\[
\text{Lower limit} = 2.928 - 2.571(0.444) = 1.786  
\text{Upper limit} = 2.928 + 2.571(0.444) = 4.070  
\]</span></p>
<p>Exponentiating these limits yields the 95% confidence interval for
the underlying geometric mean:</p>
<p><span class="math display">\[
e^{1.786} = 5.97 \quad \text{and} \quad e^{4.070} = 58.6  
\]</span></p>
<p>As expected with geometric means, these confidence intervals are
asymmetrical.</p>
</div>
<div id="two-vaccine-groups" class="section level3" number="2.1.5">
<h3><span class="header-section-number">2.1.5</span> Two Vaccine
Groups</h3>
<p>When comparing two vaccine groups, statistical inference is performed
using a <strong>two-sample <em>t</em>-test</strong> applied to the
log-transformed immunogenicity values. Point and interval estimates for
the difference in means <strong>δ = μ₁ - μ₀</strong> can be transformed
back to estimates for the <strong>geometric mean ratio (GMR)</strong>,
representing the ratio of the geometric means of the two groups:</p>
<p><span class="math display">\[
GMR = \frac{GM₁}{GM₀} = e^{\bar{u₁} - \bar{u₀}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>GM₁</em> is the geometric mean of the investigational vaccine
group</li>
<li><em>GM₀</em> is the geometric mean of the control group</li>
<li><em>ū₁</em>, <em>ū₀</em> are the means of the log-transformed values
for the respective groups</li>
</ul>
<p>The <em>p</em>-value from the two-sample <em>t</em>-test testing
<strong>δ = 0</strong> corresponds to a test of whether <strong>GMR =
1</strong>, i.e., whether the geometric means differ significantly
between the groups.</p>
<p><strong>Example: Two Vaccine Groups</strong></p>
<p>Assume the six IFN-γ values mentioned earlier are compared to a
second group of six values, with:</p>
<ul>
<li>Arithmetic mean of log-transformed values:
<strong>2.754</strong></li>
<li>Standard error: <strong>0.512</strong></li>
</ul>
<p>The geometric mean concentration of the control group is:</p>
<p><span class="math display">\[
GMC₀ = e^{2.754} = 15.7 \, \text{IU/L}
\]</span></p>
<p>The geometric mean ratio (GMR) comparing the investigational group to
the control group is:</p>
<p><span class="math display">\[
GMR = \frac{18.7}{15.7} = e^{2.928 - 2.754} = 1.19  
\]</span></p>
<p>The estimated standard error of the difference is
<strong>0.677</strong>. The 95% confidence interval for the underlying
GMR is:</p>
<p><span class="math display">\[
\text{Lower limit} = e^{1.19 - 2.228(0.677)} = 0.73  
\]</span></p>
<p><span class="math display">\[
\text{Upper limit} = e^{1.19 + 2.228(0.677)} = 14.9  
\]</span></p>
<p>This provides an asymmetric confidence interval for the ratio of
geometric means, consistent with the properties of log-normal
distributions.</p>
</div>
</div>
<div id="geometric-mean-fold-increase" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> Geometric Mean Fold
Increase</h2>
<p>In some infectious disease settings, participants may already have
measurable pre-vaccination immune responses due to previous infections
or vaccinations. A typical example is <strong>influenza</strong>, where
most adults have been exposed to influenza viruses during their
lifetime. In such cases, post-vaccination immune responses reflect both
the effect of vaccination and existing pre-vaccination immunity.</p>
<p>For these scenarios, an alternative to geometric mean titres (GMT) or
concentrations (GMC) is the <strong>Geometric Mean Fold Increase
(gMFI)</strong>, also referred to as <strong>Mean Fold Increase</strong>
or <strong>Geometric Mean Fold Rise</strong>, which specifically
quantifies the relative increase in immune response following
vaccination at the individual level.</p>
<div id="single-arm" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Single Arm</h3>
<p><strong>Definition and Calculation</strong></p>
<p>For each individual:</p>
<p><span class="math display">\[
f_i = \frac{v_{post}}{v_{pre}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>vₚᵣₑ</em> = pre-vaccination immunogenicity value</li>
<li><em>vₚₒₛₜ</em> = post-vaccination immunogenicity value</li>
<li><em>fᵢ</em> = fold increase</li>
</ul>
<p>The <strong>Geometric Mean Fold Increase (gMFI)</strong> across
<em>n</em> individuals is calculated as:</p>
<p><span class="math display">\[
gMFI = \exp \left( \frac{1}{n} \sum_{j=1}^{n} \log_e f_{ij} \right)
\]</span></p>
<p>Alternatively, if both pre- and post-vaccination data are available
for all subjects:</p>
<p><span class="math display">\[
gMFI = \frac{GM_{post}}{GM_{pre}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>GMₚₒₛₜ</em> = geometric mean of post-vaccination values</li>
<li><em>GMₚᵣₑ</em> = geometric mean of pre-vaccination values</li>
</ul>
<p><strong>Note:</strong> This relationship holds only if both pre- and
post-vaccination data are available for all individuals.</p>
<hr />
<p><strong>Example-Single Group gMFI Calculation</strong></p>
<p>In an influenza vaccine trial, pre- and post-vaccination antibody
titres for six subjects are recorded as:</p>
<table>
<thead>
<tr class="header">
<th>Subject</th>
<th>Pre-Vaccination</th>
<th>Post-Vaccination</th>
<th>Fold Increase</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>5</td>
<td>40</td>
<td>8</td>
</tr>
<tr class="even">
<td>2</td>
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="odd">
<td>3</td>
<td>10</td>
<td>160</td>
<td>16</td>
</tr>
<tr class="even">
<td>4</td>
<td>10</td>
<td>320</td>
<td>32</td>
</tr>
<tr class="odd">
<td>5</td>
<td>20</td>
<td>80</td>
<td>4</td>
</tr>
<tr class="even">
<td>6</td>
<td>20</td>
<td>640</td>
<td>32</td>
</tr>
</tbody>
</table>
<p>The calculated statistics are:</p>
<ul>
<li><strong>GMₚᵣₑ</strong> = 10.0</li>
<li><strong>GMₚₒₛₜ</strong> = 142.5</li>
<li><strong>gMFI</strong> = 14.25</li>
<li><strong>Geometric Standard Deviation (GSD)</strong> = 2.249</li>
</ul>
<p>The confidence intervals for the gMFI can be calculated using the
same formulas applied to geometric means, as described earlier.</p>
<hr />
</div>
<div id="comparison-of-two-groups-gmfi-and-gmfr" class="section level3"
number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Comparison of Two
Groups: gMFI and gMFR</h3>
<p>For comparing two vaccine groups, the <strong>Geometric Mean Fold
Ratio (gMFR)</strong> is used, representing the ratio of fold increases
between the groups.</p>
<p>The relationship between gMFR and geometric means is expressed
as:</p>
<p><span class="math display">\[
gMFR = \frac{gMFI_1}{gMFI_0} = \frac{GMR_{post}}{GMR_{pre}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>gMFI₁</em>, <em>gMFI₀</em> = gMFI of investigational and control
groups, respectively</li>
<li><em>GMRₚₒₛₜ</em> = geometric mean ratio of post-vaccination
values</li>
<li><em>GMRₚᵣₑ</em> = geometric mean ratio of pre-vaccination
values</li>
</ul>
<p>In randomized trials with no significant baseline imbalance (i.e.,
<em>GMRₚᵣₑ ≈ 1</em>), the gMFR effectively approximates
<em>GMRₚₒₛₜ</em>. Thus, the analysis of fold increases yields results
nearly identical to those obtained from analyzing post-vaccination
immunogenicity values alone.</p>
<hr />
<p><strong>Example: Two Group Comparison</strong></p>
<p><strong>Experimental Group:</strong></p>
<table>
<thead>
<tr class="header">
<th>Pre-Vaccination</th>
<th>Post-Vaccination</th>
<th>Fold Increase</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>5</td>
<td>40</td>
<td>8</td>
</tr>
<tr class="even">
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="odd">
<td>10</td>
<td>80</td>
<td>8</td>
</tr>
<tr class="even">
<td>10</td>
<td>80</td>
<td>8</td>
</tr>
</tbody>
</table>
<p><strong>Control Group:</strong></p>
<table>
<thead>
<tr class="header">
<th>Pre-Vaccination</th>
<th>Post-Vaccination</th>
<th>Fold Increase</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="even">
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="odd">
<td>10</td>
<td>80</td>
<td>8</td>
</tr>
<tr class="even">
<td>10</td>
<td>160</td>
<td>16</td>
</tr>
</tbody>
</table>
<p>The calculated statistics are:</p>
<ul>
<li><em>GMₚᵣₑ</em> = 7.1 (both groups)</li>
<li><em>GMₚₒₛₜ</em> Experimental = 67.27</li>
<li><em>GMₚₒₛₜ</em> Control = 95.14</li>
<li><em>gMFI₁</em> = 9.51</li>
<li><em>gMFI₀</em> = 13.45</li>
<li><em>gMFR</em> = 1.4</li>
<li><em>GMRₚₒₛₜ</em> = 1.4</li>
</ul>
<p>The gMFR and GMRₚₒₛₜ are identical due to the absence of baseline
imbalance.</p>
<p><strong>Efficiency Considerations</strong></p>
<p>When no baseline imbalance exists:</p>
<ul>
<li>Using fold increases introduces unnecessary variability because fold
increases mathematically involve the difference between log-transformed
post- and pre-vaccination values.</li>
<li>If the correlation between pre- and post-vaccination values is weak
(e.g., correlation &lt; 0.5), the variance of fold increases is larger
than the variance of post-vaccination values alone.</li>
<li>As a result, analyses based on fold increases often produce larger
<em>p</em>-values and wider confidence intervals compared to analyses
based on post-vaccination data alone.</li>
</ul>
</div>
<div id="misconception-do-fold-increases-correct-for-baseline-imbalance"
class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Misconception: Do
Fold Increases Correct for Baseline Imbalance?</h3>
<p>It is a common but incorrect belief that analyzing fold increases
eliminates bias due to baseline imbalance. In reality:</p>
<ul>
<li>On the log scale:</li>
</ul>
<p><span class="math display">\[
\log(f_i) = \log(v_{post}) - \log(v_{pre})
\]</span></p>
<ul>
<li>Change scores (including fold increases) remain correlated with
baseline values, often negatively.</li>
<li>For bioassays, higher pre-vaccination values often result in smaller
observed fold increases.</li>
<li>If there is a baseline imbalance, with one group having lower
pre-vaccination values, this group will tend to show artificially
inflated fold increases, introducing bias <strong>in favor of the group
with lower pre-vaccination values</strong>.</li>
</ul>
<p>Thus, fold increase analysis does <strong>not</strong> inherently
correct for baseline imbalance. The appropriate statistical method for
addressing baseline imbalance is <strong>analysis of covariance
(ANCOVA)</strong>, which explicitly adjusts for pre-vaccination
differences.</p>
</div>
</div>
<div id="two-seroresponse-rates" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Two Seroresponse
Rates</h2>
<p>In vaccine clinical trials and immunogenicity studies, it is common
to assess the immune response using binary endpoints that reflect the
presence or increase of protective antibodies following vaccination. Two
widely used binary measures are the <em>seroprotection rate</em> and the
<em>seroconversion rate</em>. Both provide critical information
regarding the effectiveness of a vaccine but capture different
immunological phenomena.</p>
<div id="seroprotection-rate" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> 1. Seroprotection
Rate</h3>
<p>The seroprotection rate is defined as the proportion of vaccinated
individuals whose antibody levels reach or exceed a predetermined
threshold that is considered to be associated with protection from
infection or disease. This threshold is referred to as the
<strong>threshold of protection</strong>.</p>
<p>For many infectious diseases, scientific research has identified
specific antibody levels that are assumed to provide protection. If an
individual’s antibody level is equal to or higher than this threshold
after vaccination, the person is classified as
<em>seroprotected</em>.</p>
<p>For example:</p>
<ul>
<li>In <strong>influenza</strong> vaccine studies, an anti-hemagglutinin
(anti-HA) antibody titre of <strong>40 or higher</strong> is generally
accepted as the threshold for seroprotection.</li>
<li>For <strong>diphtheria</strong> and <strong>tetanus</strong>, the
commonly cited threshold is an anti-diphtheria or anti-tetanus antibody
concentration of <strong>0.1 international units per millilitre
(IU/ml)</strong>.</li>
</ul>
<p>The seroprotection rate is thus a binary measure: individuals are
either classified as seroprotected or not seroprotected based on their
antibody levels. The seroprotection rate expresses the percentage of
vaccinated individuals who meet this criterion.</p>
<p>It is important to note that the phrase <em>associated with
protection</em> does not always imply complete immunity. In many cases,
this association indicates a <strong>moderate to high
probability</strong> of protection rather than absolute prevention of
disease. For instance, in the case of influenza, it is estimated that an
anti-HA antibody titre of 40 corresponds to approximately a <strong>50%
probability</strong> of being protected against infection. Therefore,
while the seroprotection threshold is useful for evaluating vaccine
responses, it should not be interpreted as guaranteeing full
immunity.</p>
</div>
<div id="seroconversion-rate" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> 2. Seroconversion
Rate</h3>
<p>The seroconversion rate refers to the proportion of individuals whose
antibody status changes in response to vaccination, indicating the
development or significant increase of detectable antibodies.</p>
<p>According to <strong>Stedman’s Medical Dictionary</strong>,
seroconversion is defined as the appearance of detectable, specific
antibodies in the blood as a result of infection or immunization.</p>
<p>An individual’s antibody status, also called
<strong>serostatus</strong>, is classified as follows:</p>
<ul>
<li><strong>Seronegative</strong>: No detectable antibodies are
present.</li>
<li><strong>Seropositive</strong>: Detectable antibodies are
present.</li>
</ul>
<p>If a person is seronegative before vaccination and becomes
seropositive after vaccination, that individual is considered to have
<em>seroconverted</em>.</p>
<p>Different studies may apply slightly different definitions of
seroconversion, depending on the disease and the vaccine being
evaluated.</p>
<p>Examples of seroconversion definitions include:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Absolute Appearance of Detectable Antibodies</strong></p>
<p>In a study of a live attenuated human rotavirus vaccine conducted by
Vesikari and colleagues, seroconversion was defined as the appearance of
serum IgA antibodies to rotavirus at a titre of <strong>20 units per
millilitre or higher</strong> in infants who were seronegative at
baseline. The proportion of infants who seroconverted varied between
<strong>73% and 96%</strong>, depending on the vaccine dose.</p></li>
<li><p><strong>Fourfold Rise in Antibody Levels</strong></p>
<p>Another common definition of seroconversion is based on a
<strong>fourfold increase</strong> in antibody levels following
vaccination. This definition is particularly relevant when some
individuals may already be seropositive at the time of enrollment.</p>
<p>For example, in a study evaluating a virus-like particle vaccine for
human papillomavirus (<strong>HPV</strong>), Harro and colleagues
defined seropositivity as an ELISA antibody titre equal to or greater
than the reactivity of a standard pooled serum. Among the 72 female
participants, six were seropositive at baseline. In this study,
seroconversion was defined as a <strong>fourfold or greater
rise</strong> in antibody titre after vaccination.</p></li>
<li><p><strong>Combined Definition</strong></p>
<p>Many clinical studies, particularly influenza vaccine trials, adopt a
combined approach to defining seroconversion. This definition takes into
account both baseline serostatus and post-vaccination changes:</p>
<ul>
<li>For individuals with a baseline antibody titre <strong>below
10</strong> (seronegative), seroconversion is defined as reaching a
<strong>post-vaccination titre of 40 or higher</strong>, which is
consistent with the seroprotection threshold in influenza studies.</li>
<li>For individuals with a baseline antibody titre <strong>10 or
higher</strong> (seropositive), seroconversion is defined as achieving
at least a <strong>fourfold rise</strong> in titre after
vaccination.</li>
</ul></li>
</ol>
<p>It is important to highlight that such definitions may vary slightly
between studies. For instance, some studies distinguish between becoming
seropositive (titre ≥10) and achieving seroprotection (titre ≥40),
especially in the context of influenza vaccines.</p>
<p>Regardless of the specific definition applied, seroconversion remains
a binary endpoint. Individuals either meet the criteria for
seroconversion or they do not. The <strong>seroconversion rate</strong>
represents the proportion or percentage of study participants who
seroconverted after vaccination.</p>
</div>
</div>
<div id="analysis-of-proportions" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Analysis of
Proportions</h2>
</div>
<div id="multiple-co-primary-endpoints-and-the-intersectionunion-test"
class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Multiple Co-Primary
Endpoints and the Intersection–Union Test</h2>
</div>
<div id="reverse-cumulative-distribution-plot" class="section level2"
number="2.6">
<h2><span class="header-section-number">2.6</span> Reverse Cumulative
Distribution Plot</h2>
</div>
</div>
<div id="vaccine-efficacy" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Vaccine Efficacy</h1>
<div id="assessing-vaccine-efficacy" class="section level2"
number="3.1">
<h2><span class="header-section-number">3.1</span> Assessing Vaccine
Efficacy</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Definition and Formula</strong> Vaccine Efficacy (VE)
measures how well a vaccine protects individuals from disease. It is
defined as:</p>
<p><strong>VE = 1 - Risk Ratio (RR)</strong> Where:</p>
<ul>
<li><p><strong>RR (Risk Ratio)</strong> = ARV / ARU</p>
<ul>
<li><strong>ARV</strong>: Attack rate in vaccinated group</li>
<li><strong>ARU</strong>: Attack rate in unvaccinated (control/placebo)
group</li>
</ul></li>
</ul>
<p>Interpreting VE:</p>
<ul>
<li>The range of VE is from negative infinity to 1.</li>
<li>A VE of <strong>1</strong> (or 100%) means <strong>complete
protection</strong> — no cases occurred in the vaccinated group.</li>
<li>A VE of <strong>0</strong> means <strong>no effect</strong> — the
vaccine does not reduce the risk.</li>
<li>A <strong>negative VE</strong> suggests harm (higher risk in the
vaccinated group than unvaccinated).</li>
</ul></li>
<li><p><strong>Multiple Testing Approaches</strong> The choice of
statistical method to estimate VE depends on how disease occurrence is
modeled:</p>
<ul>
<li><strong>Counts</strong>: Number of events (e.g., infections) across
follow-up time.</li>
<li><strong>Proportions</strong>: Percentage of individuals infected in
each group.</li>
<li><strong>Time-to-Event</strong>: Time from vaccination to disease
onset (e.g., using survival analysis).</li>
</ul></li>
<li><p><strong>VE as Primary Endpoint</strong> In vaccine trials, VE is
typically the <strong>primary efficacy endpoint</strong>, alongside key
<strong>safety endpoints</strong>.</p>
<ul>
<li>Trials may define a “case” not just based on lab-confirmed
infection, but also by <strong>symptom presentation</strong>.</li>
<li>This is particularly relevant when mild/asymptomatic cases are
common or underreported.</li>
</ul></li>
</ol>
</div>
<div id="vaccine-methods-ssd-sample-size-determination-examples"
class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Vaccine Methods / SSD
(Sample Size Determination) Examples</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Confidence Interval for VE</strong></p>
<ul>
<li>Based on classical <strong>case-control</strong> or
<strong>cohort</strong> designs.</li>
<li>Reference: <strong>O’Neill (1988)</strong>.</li>
<li>Focused on estimating VE along with its confidence interval to
quantify uncertainty.</li>
</ul></li>
<li><p><strong>One Proportion Tests</strong></p>
<ul>
<li>Applied when comparing observed event proportion against a known
benchmark (e.g., historical attack rate).</li>
<li>Often used in <strong>event-driven designs</strong> (study continues
until a set number of events occurs).</li>
<li>Uses standard binomial tests.</li>
</ul></li>
<li><p><strong>Two Proportion Tests</strong></p>
<ul>
<li>Directly compares attack rates between vaccinated and control
groups.</li>
<li>Computes VE using the formula: <strong>VE = 1 – RR</strong>, and
then applies tests like the <strong>z-test for
proportions</strong>.</li>
</ul></li>
<li><p><strong>Composite Models</strong></p>
<ul>
<li>Consider multiple outcomes (e.g., disease <strong>incidence and
severity</strong>) together.</li>
<li>Reference: <strong>Callegaro (2020)</strong>.</li>
<li>Used to account for the <strong>public health relevance</strong> of
various symptoms, not just whether infection occurred.</li>
</ul></li>
<li><p><strong>Count Models (Poisson/Negative Binomial)</strong></p>
<ul>
<li>Appropriate when tracking <strong>event counts over time</strong>
(e.g., number of infections/person-time).</li>
<li>Poisson models work when events are rare and follow a constant
rate.</li>
<li>Negative Binomial (NB) used when there is overdispersion (more
variability than Poisson allows).</li>
<li>Applies both to standard and rare-event settings.</li>
</ul></li>
<li><p><strong>Cox Regression</strong></p>
<ul>
<li>A <strong>time-to-event model</strong> commonly used for survival
analysis.</li>
<li>Used in <strong>non-inferiority (NI)</strong> or <strong>superiority
(SM)</strong> hypotheses testing.</li>
<li>Can adjust for covariates and censoring (e.g., if participants drop
out or are uninfected at study end).</li>
</ul></li>
<li><p><strong>Cluster Randomized Designs</strong></p>
<ul>
<li>Applied when randomizing <strong>groups</strong> (e.g., households,
villages) instead of individuals.</li>
<li>Often used in <strong>public health</strong> or <strong>epidemic
settings</strong>.</li>
<li>Statistical methods account for intra-cluster correlation.</li>
<li>Reference: <strong>Hayes &amp; Bennett (1999)</strong>.</li>
</ul></li>
</ol>
<hr />
<p><img src="02_Plots/Vaccine/Vaccine-COVID19Example.png" /></p>
</div>
<div id="vaccine-efficacy-hypothesis-into-binomial-testing-framework"
class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Vaccine Efficacy
Hypothesis into Binomial Testing Framework</h2>
<p><strong>Goal:</strong> Compare two hypotheses:</p>
<ul>
<li><strong>Null Hypothesis (H₀): VE ≤ 0.3</strong> (vaccine provides
low or no protection)</li>
<li><strong>Alternative Hypothesis (H₁): VE = 0.6</strong> (vaccine
provides strong protection)</li>
</ul>
<p><strong>Step 1: Define Probabilities</strong></p>
<ul>
<li>Let <strong>πₚₗₐcₑbₒ</strong> be the probability of getting
symptomatic COVID-19 in the <strong>placebo group</strong></li>
<li>Let <strong>πᵥₐccᵢₙₑ</strong> be the same for the <strong>vaccine
group</strong></li>
</ul>
<p><strong>Step 2: Conditional Probability for Vaccine Case</strong>
Derive the probability that a <strong>random case</strong> (symptomatic
COVID-19) comes from the vaccine group, under a given hypothesis
(VE):</p>
<p><span class="math display">\[
P_{H_x}(Vaccine\ Case\ |\ Case) = \frac{π_{Placebo}(1 -
VE)}{π_{Placebo}(1 - VE) + π_{Placebo}} = \frac{0.08(1 - VE)}{0.08(1 -
VE) + 0.08}
\]</span></p>
<p><strong>Step 3: Plug in VE under each hypothesis</strong></p>
<ul>
<li><p>Under <strong>H₀ (VE = 0.3)</strong>, the conditional probability
becomes:</p>
<p><span class="math display">\[
P_{H₀} = \frac{0.08(1 - 0.3)}{0.08(1 - 0.3) + 0.08} = 0.4118
\]</span></p></li>
<li><p>Under <strong>H₁ (VE = 0.6)</strong>:</p>
<p><span class="math display">\[
P_{H₁} = \frac{0.08(1 - 0.6)}{0.08(1 - 0.6) + 0.08} = 0.2857
\]</span></p></li>
</ul>
<p><strong>Step 4: Modeling the Number of Vaccine Cases</strong>
Assume:</p>
<ul>
<li><strong>N</strong> is the total number of symptomatic cases
observed.</li>
<li><strong>V</strong> is the number of those that are in the vaccine
group.</li>
</ul>
<p>Then:</p>
<ul>
<li><strong>V ~ Binomial(N, Pₕₓ)</strong> depending on which hypothesis
is true.</li>
<li><strong>1 – β = P(V ≥ c | H₁) = 0.9</strong>, meaning we want
<strong>90% power</strong> to detect a VE of 0.6 if it’s true.</li>
</ul>
</div>
</div>
<div id="vaccine-safety" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Vaccine Safety</h1>
<div id="introduction-1" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Introduction</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Focus on Adverse Events</strong> Vaccine safety revolves
around the monitoring of <strong>adverse events</strong> (AEs) that
occur after vaccination. These events could range from mild
(e.g. injection site reactions) to severe or life-threatening
(e.g. anaphylaxis or myocarditis). Because vaccines are administered to
healthy people, including children and vulnerable groups, safety is held
to an especially high standard.</p></li>
<li><p><strong>Primary Safety Endpoints</strong> During clinical trials
(especially Phase III), specific <strong>safety endpoints</strong> are
defined—typically focusing on <strong>severe adverse events
(SAEs)</strong>. These endpoints are carefully monitored in parallel
with efficacy.</p></li>
<li><p><strong>Post-Marketing Surveillance</strong> Some AEs may be too
rare to detect even in large trials (e.g., with frequencies of less than
1 in 10,000). These events are captured through <strong>post-market
surveillance</strong>, which continues after the vaccine is approved and
in widespread use.</p>
<ul>
<li>This includes <strong>Phase IV studies</strong> and ongoing data
collection from <strong>vaccine safety networks</strong> (e.g., VAERS in
the U.S.).</li>
<li>The goal is to identify <strong>unexpected or delayed
effects</strong> that only become evident with broader population
exposure.</li>
</ul></li>
<li><p><strong>Use of Self-Controlled Studies</strong></p>
<p>There’s a practical and ethical limitation post-approval:</p>
<blockquote>
<p>You can’t maintain a randomized control group once a vaccine is
proven effective, since it would be unethical to deny the vaccine to
people who need it.</p>
</blockquote>
<p>As a result, researchers rely on <strong>observational
methods</strong> that don’t require a control group — this is where
<strong>SCCS</strong> comes in.</p>
<p>Among the most important tools in post-marketing safety surveillance
are <strong>self-controlled designs</strong>, such as the
<strong>Self-Controlled Case Series (SCCS)</strong>. These are useful
for evaluating whether the rate of adverse events increases shortly
after vaccination compared to other periods in the same
individual.</p></li>
</ol>
</div>
<div id="self-controlled-case-studies" class="section level2"
number="4.2">
<h2><span class="header-section-number">4.2</span> Self Controlled Case
Studies</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Overview</strong> SCCS is a
<strong>within-person</strong> comparison method that estimates the
<strong>relative incidence (RI)</strong> of an adverse event during
defined risk windows (e.g., 0–28 days post-vaccination) versus control
periods.</p>
<ul>
<li>It uses <strong>only individuals who experienced the adverse
event</strong> (i.e., case-only design).</li>
<li>The benefit is that <strong>time-invariant confounders</strong>
(e.g., genetics, baseline health status) are automatically controlled
for, since each person serves as their own control.</li>
</ul></li>
<li><p><strong>How It Works</strong> The observation period for each
case is divided into:</p>
<ul>
<li><strong>Risk period(s)</strong>: When the event is suspected to be
more likely due to vaccination.</li>
<li><strong>Control period(s)</strong>: Baseline period when the event
risk is assumed to be unaffected by the vaccine.</li>
</ul>
<p>The <strong>RI</strong> is the ratio of event incidence in the risk
period vs. the control period.</p></li>
<li><p><strong>Statistical Methods</strong> Estimation methods
include:</p>
<ul>
<li><strong>Likelihood functions</strong> based on binomial models.</li>
<li>Signed root likelihood ratios and log-relative incidence
estimates.</li>
<li>Modifications for <strong>age effects</strong>, which are common
confounders in vaccine safety.</li>
</ul>
<p>The formula for RI without age effects is based on a likelihood
function <span class="math inline">\(l(\rho)\)</span>, where:</p>
<ul>
<li><span class="math inline">\(x\)</span>: number of events in the risk
window,</li>
<li><span class="math inline">\(n_1\)</span>: total number of
events,</li>
<li><span class="math inline">\(r\)</span>: proportion of time at
risk.</li>
</ul>
<p>With age effects, the likelihood becomes more complex, including
terms to account for age groups and event timing (e.g., <span
class="math inline">\(\delta_j\)</span> and <span
class="math inline">\(\beta\)</span> in the likelihood).</p></li>
<li><p><strong>Why SCCS Is Useful</strong> SCCS is particularly
effective for vaccine safety because:</p>
<ul>
<li>It <strong>requires only case data</strong>, which is often more
readily available.</li>
<li>It handles <strong>time-varying exposures</strong> (like vaccines)
and <strong>rare events</strong>.</li>
<li>It is robust against confounders that do not change over time (e.g.,
sex, ethnicity).</li>
</ul></li>
</ol>
<p><strong>Example</strong></p>
<p>“Miller et al. [13] studied the association between measles, mumps,
rubella (MMR) vaccine and idiopathic thrombocytopenic purpura (ITP)
(abnormal bleeding into the skin due to low blood platelet count) in
children aged 12–23 months during the period from October 1991 to
September 1994 within 42 days of receiving the vaccine.”</p>
<p>“The observation period includes the ages 366–730 days, which we
subdivide into J = 4 periods of lengths e1 = e2 = e3 = 91 days, and e4 =
92 days. We take the proportions vaccinated in each of these age
intervals to be p1 = 0.6, p2 = 0.2, p3 = 0.05, p4 = 0.05. We take the
age effects to be e1 = 1, e2 = 0.6, e3 = e4 = 0.4. The risk period is e∗
= 42 days. We set ⍴ = 3, z⍺ = 1.96 and zꞵ = 0.8416 for 80 per cent power
to detect a relative incidence of 3 at the 5 per cent significance
level.”</p>
<table>
<thead>
<tr class="header">
<th>Parameter</th>
<th>Overall</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Number of Periods</td>
<td>4</td>
</tr>
<tr class="even">
<td>Observation Periods</td>
<td>91, 91, 91, 92 days</td>
</tr>
<tr class="odd">
<td>Risk Period</td>
<td>42 days</td>
</tr>
<tr class="even">
<td>P(Exposure)</td>
<td>0.6, 0.2, 0.05, 0.05</td>
</tr>
<tr class="odd">
<td>Age Effects</td>
<td>1, 0.6, 0.4, 0.4</td>
</tr>
<tr class="even">
<td>Relative Incidence</td>
<td>3</td>
</tr>
<tr class="odd">
<td>Alpha (2-sided)</td>
<td>0.05</td>
</tr>
<tr class="even">
<td>Power</td>
<td>80%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="vaccine-durability" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Vaccine Durability</h1>
<div id="deferred-vaccine-designs" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Deferred Vaccine
Designs</h2>
<p><strong>Ethical Considerations</strong> Once a vaccine has been shown
to be effective in early trials, continuing to keep participants in a
placebo group becomes ethically problematic. Therefore:</p>
<ul>
<li>Placebo recipients are typically <strong>offered the
vaccine</strong> after efficacy is established.</li>
<li>This transition must be handled carefully to <strong>preserve the
integrity of the trial</strong>, particularly in long-term (durability)
follow-up.</li>
</ul>
<hr />
<p><strong>Crossover Design</strong> To address this ethical issue while
still enabling long-term efficacy assessment, trials often use a
<strong>crossover design</strong>:</p>
<ul>
<li><p><strong>All participants eventually receive the vaccine</strong>,
either at the beginning or after the placebo phase.</p></li>
<li><p>This setup:</p>
<ul>
<li>Maintains <strong>blinding</strong> in the early phases (so
participants and investigators don’t know who has which treatment).</li>
<li>Still allows for <strong>comparisons</strong> of initial protection
and long-term durability across groups.</li>
</ul></li>
</ul>
<hr />
<p><strong>Less Precision Compared to Standard Designs</strong>
Crossover and ethical adaptations can result in <strong>less precise
efficacy estimates</strong>, especially over time:</p>
<ul>
<li>The estimates for long-term efficacy (durability) may be
<strong>less reliable</strong> than those in traditional randomized
controlled trials.</li>
<li>In durability trials, any early misestimations can undermine later
results—this is described as the study being only as <strong>strong as
its weakest link</strong>.</li>
</ul>
<hr />
<p><strong>Subgroup Safety and Harm Analysis</strong> Durability trials
also provide opportunities for <strong>subgroup analysis</strong>, such
as:</p>
<ul>
<li>Assessing whether <strong>efficacy or safety</strong> varies by
<strong>demographic or clinical characteristics</strong> (e.g., age,
gender, comorbidities).</li>
<li>These analyses help understand <strong>differential vaccine
performance</strong>, contributing to more personalized or targeted
vaccination strategies.</li>
</ul>
<p><img src="02_Plots/Vaccine/Vaccine-Durability.png" /></p>
</div>
</div>
<div id="reference" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Reference</h1>
<p>Nauta, J. (2011). Statistics in clinical vaccine trials.
Springer-Verlag.</p>
<p>nQuery Webinar, Sample Sizes for Vaccine Trials, Trial Designs for
Assessing Vaccine Efficacy, Safety &amp; Durability, <a
href="https://www.statsols.com/guides/sample-sizes-for-vaccine-trials"
class="uri">https://www.statsols.com/guides/sample-sizes-for-vaccine-trials</a></p>
<p>nQuery Webinar, Design and Sample Size for Vaccine Trials, Case Study
of COVID-19, <a
href="https://www.statsols.com/guides/design-and-sample-size-for-vaccine-trials"
class="uri">https://www.statsols.com/guides/design-and-sample-size-for-vaccine-trials</a></p>
<p>Fleiss, J.L., Levin, B. and Paik, M.C., 2013. Statistical methods for
rates and proportions. John Wiley &amp; Sons.</p>
<p>Halloran, M.E., Longini, I.M., Struchiner, C.J. and Longini, I.M.,
2010. Design and analysis of vaccine studies (Vol. 18). New York, NY:
Springer.</p>
<p>Singh, K. and Mehta, S., 2016. The clinical development process for a
novel preventive vaccine: An overview. Journal of postgraduate medicine,
62(1), p.4.</p>
<p>O’Neill, R.T., 1988. On sample sizes to estimate the protective
efficacy of a vaccine. Statistics in Medicine, 7(12), pp.1279-1288.</p>
<p>Chan, I.S. and Bohidar, N.R., 1998. Exact power and sample size for
vaccine efficacy studies. Communications in Statistics-Theory and
Methods, 27(6), pp.1305-1322.</p>
<p>Gu, K., Ng, H.K.T., Tang, M.L. and Schucany, W.R., 2008. Testing the
ratio of two poisson rates. Biometrical Journal: Journal of Mathematical
Methods in Biosciences, 50(2), pp.283-298.</p>
<p>Schoenfeld, D.A., 1983. Sample-size formula for the
proportional-hazards regression model. Biometrics, pp.499-503.</p>
<p>Callegaro, A., Curran, D. and Matthews, S., 2020. Burden‐of‐illness
vaccine efficacy. Pharmaceutical statistics, 19(5), pp.636-645.</p>
<p>Chow, S.C., Shao, J., Wang, H. and Lokhnygina, Y., 2017. Sample size
calculations in clinical research. CRC press.</p>
<div id="vaccine-efficacy-1" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Vaccine Efficacy</h2>
<p>Polack, F.P., Thomas, S.J., Kitchin, N., Absalon, J., Gurtman, A.,
Lockhart, S., Perez, J.L., Pérez Marc, G., Moreira, E.D., Zerbini, C.
and Bailey, R., 2020. Safety and efficacy of the BNT162b2 mRNA Covid-19
vaccine. New England Journal of Medicine, 383(27), pp.2603-2615.</p>
<p>Baden, L.R., El Sahly, H.M., Essink, B., Kotloff, K., Frey, S.,
Novak, R., Diemert, D., Spector, S.A., Rouphael, N., Creech, C.B. and
McGettigan, J., 2021. Efficacy and safety of the mRNA-1273 SARS-CoV-2
vaccine. New England Journal of Medicine, 384(5), pp.403-416.</p>
<p>Voysey, M., Clemens, S.A.C., Madhi, S.A., Weckx, L.Y., Folegatti,
P.M., Aley, P.K., Angus, B., Baillie, V.L., Barnabas, S.L., Bhorat, Q.E.
and Bibi, S., 2021. Safety and efficacy of the ChAdOx1 nCoV-19 vaccine
(AZD1222) against SARS-CoV-2: an interim analysis of four randomised
controlled trials in Brazil, South Africa, and the UK. The Lancet,
397(10269), pp.99-111.</p>
<p>Sadoff, J., Le Gars, M., Shukarev, G., Heerwegh, D., Truyers, C., de
Groot, A.M., Stoop, J., Tete, S., Van Damme, W., Leroux-Roels, I. and
Berghmans, P.J., 2021. Interim Results of a Phase 1–2a Trial of Ad26.
COV2. S Covid-19 Vaccine. New England Journal of Medicine.</p>
<p>Logunov, D.Y., Dolzhikova, I.V., Shcheblyakov, D.V., Tukhvatulin,
A.I., Zubkova, O.V., Dzharullaeva, A.S., Kovyrshina, A.V., Lubenets,
N.L., Grousova, D.M., Erokhova, A.S. and Botikov, A.G., 2021. Safety and
efficacy of an rAd26 and rAd5 vector-based heterologous prime-boost
COVID-19 vaccine: an interim analysis of a randomised controlled phase 3
trial in Russia. The Lancet, 397(10275), pp.671-681.</p>
<p>Uttarilli, Anusha, Amalakanti, Sridhar, Kommoju, Phaneeswara-Rao,
Sharma, Srihari, Goyal, Pankaj, Manjunath, Gowrang Kasaba, Upadhayay,
Vineet, Parveen, Alisha, Tandon, Ravi, Prasad, Kumar Suranjit, Dakal,
Tikam Chand, Ben Shlomo, Izhar, Yousef, Malik, Neerathilingam, Muniasamy
and Kumar, Abhishek. “Super-rapid race for saving lives by developing
COVID-19 vaccines” Journal of Integrative Bioinformatics, vol. 18, no.
1, 2021, pp. 27-43. <a href="https://doi.org/10.1515/jib-2021-0002"
class="uri">https://doi.org/10.1515/jib-2021-0002</a></p>
</div>
<div id="vaccine-safety-1" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Vaccine Safety</h2>
<p>Petersen, I., Douglas, I. and Whitaker, H., 2016. Self controlled
case series methods: an alternative to standard epidemiological study
designs. BMJ, 354.Musonda, P., Paddy Farrington, C. and Whitaker, H.J.,
2006. Sample sizes for self‐controlled case series studies. Statistics
in Medicine, 25(15), pp.2618-2631.</p>
<p>Farrington CP. 1995. Relative incidence estimation from case series
for vaccine safety evaluation. Biometrics 51:228-235. page 228.Dodd, C.,
Andrews, N., Petousis-Harris, H., Sturkenboom, M., Omer, S.B. and Black,
S., 2021. Methodological frontiers in vaccine safety: qualifying
available evidence for rare events, use of distributed data networks to
monitor vaccine safety issues, and monitoring the safety of pregnancy
interventions. BMJ global health, 6(Suppl 2), p.e003540. Knipe, D.M.,
Levy, O., Fitzgerald, K.A. and Mühlberger, E., 2020. Ensuring vaccine
safety. Science, 370(6522), pp.1274-1275.</p>
<p>Matz, K.M., Marzi, A. and Feldmann, H., 2019. Ebola vaccine trials:
progress in vaccine safety and immunogenicity. Expert review of
vaccines, 18(12), pp.1229-1242.</p>
<p>Pormohammad, A., Zarei, M., Ghorbani, S., Mohammadi, M., Razizadeh,
M.H., Turner, D.L. and Turner, R.J., 2021. Efficacy and safety of
COVID-19 vaccines: a systematic review and meta-analysis of randomized
clinical trials. Vaccines, 9(5), p.467.</p>
</div>
<div id="vaccine-durability-1" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Vaccine
Durability</h2>
<p>Choe, P.G., Kang, C.K., Suh, H.J., Jung, J., Song, K.H., Bang, J.H.,
Kim, E.S., Kim, H.B., Park, S.W., Kim, N.J. and Park, W.B., 2021. Waning
antibody responses in asymptomatic and symptomatic SARS-CoV-2 infection.
Emerging infectious diseases, 27(1), p.327.</p>
<p>Nason, M. and Follmann, D., 2010. Design and analysis of crossover
trials for absorbing binary endpoints. Biometrics, 66(3),
pp.958-965.</p>
<p>Makubate, B. and Senn, S., 2010. Planning and analysis of cross‐over
trials in infertility. Statistics in medicine, 29(30), pp.3203-3210.</p>
<p>Follmann, D., 2006. Augmented designs to assess immune response in
vaccine trials. Biometrics, 62(4), pp.1161-1169.</p>
<p>Palin, A.C., Alter, G., Crotty, S., Ellebedy, A.H., Lane, M.C., Lee,
F.E.H., Locci, M., Malaspina, A., Mallia, C., McElrath, M.J. and
Pulendran, B., 2022. The persistence of memory: defining, engineering,
and measuring vaccine durability. Nature immunology, 23(12),
pp.1665-1668.Fintzi, J. and Follmann, D., 2021. Assessing vaccine
durability in randomized trials following placebo crossover. Statistics
in Medicine, 40(27), pp.5983-6007.</p>
<p>Follmann, D., Fintzi, J., Fay, M.P., Janes, H.E., Baden, L.R., El
Sahly, H.M., Fleming, T.R., Mehrotra, D.V., Carpp, L.N., Juraska, M. and
Benkeser, D., 2021. A deferred-vaccination design to assess durability
of COVID-19 vaccine effect after the placebo group is vaccinated. Annals
of internal medicine, 174(8), pp.1118-1125.</p>
<p>Lin, D.Y., Zeng, D. and Gilbert, P.B., 2021. Evaluating the long-term
efficacy of coronavirus disease 2019 (COVID-19) vaccines. Clinical
Infectious Diseases, 73(10), pp.1927-1939.</p>
<p>Lipsitch, M., 2019. Challenges of vaccine effectiveness and waning
studies. Clinical Infectious Diseases, 68(10), pp.1631-1633.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
