<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title> Vaccine Trials</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Zehui Bai</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="02-Clinical_Experience.html">
    <span class="fa fa-user"></span>
     
    Clinical Experience
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-sliders"></span>
     
    Sample Size
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="03-SSC-Everything-to-Know-About-Sample-Size-Determination.html">Everything to Know About Sample Size Determination</a>
    </li>
    <li>
      <a href="03-SSC-Choosing-the-Effect-Size-for-Sample-Size-Calculations.html">Choosing the Effect Size</a>
    </li>
    <li>
      <a href="03-SSC-Biosimilar-Trials.html">Statistical Considerations for the Design and Analysis of Biosimilar Trials</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-and-Power-for-Non-Parametric-Analysis.html">Sample Size and Power for Non-Parametric Analysis</a>
    </li>
    <li>
      <a href="03-SSC-Power-for-Complex-Hypotheses.html">Power for Complex Hypotheses</a>
    </li>
    <li>
      <a href="03-SSC-Alternatives-to-Power.html">Bayesian methods - Alternatives to Power</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-for-Pilot-Studies.html">Sample Size for Pilot Studies</a>
    </li>
    <li>
      <a href="03-SSC-Case-Continuous-Endpoint.html">Sample Size Determination for Continuous Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Categorical-Endpoint.html">Sample Size Determination for Categorical Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-Determination-for-Counts-and-Rates.html">Sample Size Determination for Counts and Rates</a>
    </li>
    <li>
      <a href="03-SSC-Case-Survival-Endpoint.html">Sample Size Determination for Survival Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Repeated-Measures.html">Sample Size Determination for Repeated Measures</a>
    </li>
    <li>
      <a href="03-SSC-IA-Sequential-Design.html">Statistical Considerations for Group Sequential Design</a>
    </li>
    <li>
      <a href="03-SSC-IA-Adaptive-Design.html">Statistical Considerations for Adaptive Design</a>
    </li>
    <li>
      <a href="03-SSC-Multiple-Test.html">Sample Size for Multiple Test</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    Study Design
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04-Design-General-Consideration.html">General Consideration</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Framework.html">Estimands Framework</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Practice.html">Estimands and Sensitivity Analyses</a>
    </li>
    <li>
      <a href="04-Design-Phase-I-Trials---Design-Considerations.html">Phase I Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-II-Trials---Design-Considerations.html">Phase II Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-III-Trials---Design-Considerations.html">Phase III Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-IV-Trials---Design-Considerations.html">Phase IV Trials - Design Considerations for Post Marketing Surveillance</a>
    </li>
    <li>
      <a href="04-Design-Complex-Sequential-Trials.html">Complex Sequential Analysis Trials</a>
    </li>
    <li>
      <a href="04-Design-Adaptive-Clinical-Trials.html">Adaptive Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Bayesian-Clinical-Trials.html">Bayesian Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Dose-Escalation-and-Stratification-Designs.html">Dose Escalation and Stratification Designs in Early Oncology Development</a>
    </li>
    <li>
      <a href="04-Design-Single-Arm-Clinical-Trials.html">Single Arm Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-Design-and-Evaluation.html">Diagnostic Study-Design and Evaluation</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-MRMC.html">Diagnostic Study-Multireader Multicase (MRMC)</a>
    </li>
    <li>
      <a href="04-Design-Vaccine-Design.html">Vaccine Trials</a>
    </li>
    <li>
      <a href="04-Design-Externally-Controlled-Trials.html">Externally Controlled Trials</a>
    </li>
    <li>
      <a href="04-Design-Noninferiority-Trials.html">Noninferiority Trials</a>
    </li>
    <li>
      <a href="04-Design-Bioequivalence-and-Biosimilar-Trials.html">Bioequivalence and Biosimilar Trials</a>
    </li>
    <li>
      <a href="04-Design-Exploring-Survival-Analysis-Designs-for-Clinical-Trials.html">Exploring Survival Analysis Designs for Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Hierarchical-composite-endpoints.html">Hierarchical Composite Endpoints</a>
    </li>
    <li>
      <a href="04-Design-Regulatory-Submission.html">Regulatory Submission from Stats Perspective</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Data Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="05-Plot-Time-To-Event.html">Time to Event Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-PRO-Data.html">Patient Reported Outcome Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-SSC-and-Power-Calculation.html">Sample Size and Power Calculations Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Correlation.html">Correlation Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Model-Table.html">Clinical Data and Model visualization</a>
    </li>
    <li>
      <a href="05-Plot-ScatterPlot.html">Scatter and Line Plot</a>
    </li>
    <li>
      <a href="05-Plot-BarPlot.html">Bar Chart</a>
    </li>
    <li>
      <a href="05-Plot-PieChart.html">Pie Chart</a>
    </li>
    <li>
      <a href="05-Plot-BoxPlot.html">Box Plot</a>
    </li>
    <li>
      <a href="05-Plot-Histogram.html">Histogram</a>
    </li>
    <li>
      <a href="05-Plot-Forest-Plot.html">Forest Plot</a>
    </li>
    <li>
      <a href="05-Plot-Flow-Chart.html">Flow Chart</a>
    </li>
    <li>
      <a href="05-Plot-Some-Interesting.html">Some Interesting Plots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-calculator"></span>
     
    Statistical Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06-Analysis-Linear-Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Logistic-Regression.html">Logistic Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Mixed-Model.html">Mixed Model</a>
    </li>
    <li>
      <a href="06-Analysis-MMRM.html">Mixed Model Repeated Measures</a>
    </li>
    <li>
      <a href="06-Analysis-GEE.html">Generalized Estimating Equation</a>
    </li>
    <li>
      <a href="06-Analysis-ANOVA.html">Analysis of Variance</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Analysis.html">Survival Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Competing-Risk.html">Survival Analysis - Competing Risk</a>
    </li>
    <li>
      <a href="06-Analysis-Missing-Data.html">Missing Data Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-MI-Implementation.html">Multiple Imputation Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SSD.html">Safety Signal Detection and Evaluation</a>
    </li>
    <li>
      <a href="06-Analysis-Meta-Analysis.html">Meta Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-PK-and-PD.html">PK and PD Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Time-Series-Analysis.html">Time Series Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SEM.html">Structural Equation Modeling</a>
    </li>
    <li>
      <a href="06-Analysis-Factor-Analysis.html">Factor Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Propensity-Score.html">Propensity Score Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Machine Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="07-ML-Bayesian-Theory.html">Bayesian Theory</a>
    </li>
    <li>
      <a href="07-ML-Bayesian-Analysis.html">Bayesian Analysis</a>
    </li>
    <li>
      <a href="07-ML-Regularization-Penalized-Regression.html">Regularization Penalized Regression</a>
    </li>
    <li>
      <a href="07-ML-Loss-Regression.html">Loss Functions in Machine Learning</a>
    </li>
    <li>
      <a href="07-ML-PCA.html">Principal Component Analysis</a>
    </li>
    <li>
      <a href="07-ML-KNN.html">K-Nearest Neighbors</a>
    </li>
    <li>
      <a href="07-ML-SVM.html">Support Vector Machine</a>
    </li>
    <li>
      <a href="07-ML-Tree-Models.html">Tree Models</a>
    </li>
    <li>
      <a href="07-ML-LDA.html">Linear Discriminant Analysis</a>
    </li>
    <li>
      <a href="07-ML-Cluster-Analysis.html">Cluster Analysis</a>
    </li>
    <li>
      <a href="07-ML-Neural-Networks.html">Neural Network</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-stethoscope"></span>
     
    Clinical Analysis (ADaM)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="08-Clinical-Disposition-Baseline.html">Disposition and Baseline</a>
    </li>
    <li>
      <a href="08-Clinical-Efficacy.html">Efficacy Analysis</a>
    </li>
    <li>
      <a href="08-Clinical-Pharmacokinetics-Analysis.html">Pharmacokinetics Analysis</a>
    </li>
    <li>
      <a href="08-Clinical-Adverse-Event.html">Adverse Event Visualization</a>
    </li>
    <li>
      <a href="08-Clinical-Laboratory-Data.html">Laboratory Data Visualization</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="08-CV.html">
    <span class="fa fa-file-pdf-o"></span>
     
    CV
  </a>
</li>
<li>
  <a href="mailto:zehuibai@outlook.com">
    <span class="fa fa-envelope-o"></span>
     
    Contact me
  </a>
</li>
<li>
  <a href="https://github.com/Zehuibai">
    <span class="fa fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><p><img src="logo.png"
style="width:3in" /><br />
Vaccine Trials</p></h1>

</div>


<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<div id="phases-of-vaccine-trials" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Phases of Vaccine
Trials</h2>
<p>Vaccine trials are a critical component of public health, designed to
ensure that vaccines are both safe and effective before widespread use.
They are uniquely complex compared to other clinical trials due to the
preventive nature of vaccines and their administration to healthy
populations. Below is a detailed explanation of vaccine trial designs
and the challenges they present.</p>
<p>Vaccine development typically progresses through several phases:</p>
<ul>
<li><p><strong>Phase I</strong>: Involves a small group of healthy
volunteers to assess safety, determine appropriate dosage, and identify
any immediate adverse effects.</p></li>
<li><p><strong>Phase II</strong>: Expands the participant pool to
include individuals representative of the target population, focusing on
immunogenicity (the ability to provoke an immune response) and continued
safety assessment.</p></li>
<li><p><strong>Phase III</strong>: Encompasses large-scale trials with
thousands to tens of thousands of participants to evaluate the vaccine’s
efficacy in preventing the disease and to monitor for rare side effects.
Given that many vaccinated individuals may not be exposed to the
pathogen, large sample sizes are necessary to detect statistically
significant differences between vaccinated and unvaccinated
groups.</p></li>
<li><p><strong>Post-Marketing Surveillance (Phase IV)</strong>: After
regulatory approval, ongoing monitoring continues to detect any
long-term or rare adverse events in the general population.</p></li>
</ul>
<p><strong>Safety Considerations</strong></p>
<p>Safety is paramount in vaccine trials due to their administration to
healthy individuals, including vulnerable populations like children and
the elderly. Safety assessments occur at all trial phases, with
particular emphasis during Phase III and post-marketing surveillance.
Regulatory agencies employ systems such as the Vaccine Adverse Event
Reporting System (VAERS) and the Vaccine Safety Datalink (VSD) to
monitor and evaluate adverse events continuously.</p>
<p><img src="./02_Plots/Vaccine/Vaccine-COVID19Development.jpg" /></p>
<p>To respond to the COVID-19 emergency, traditional timelines were
compressed. Normally, vaccine development might take ten years, but
several COVID-19 vaccines reached approval in just 18 to 20 months. This
was achieved through parallel trial phases, early manufacturing
scale-up, and the integration of computational tools. For example,
Moderna’s mRNA vaccine entered Phase I trials within ten weeks of the
virus genome being published.</p>
</div>
<div id="challenges-of-vaccine-trial-design" class="section level2"
number="1.2">
<h2><span class="header-section-number">1.2</span> Challenges of Vaccine
Trial Design</h2>
<p>Vaccine trials are uniquely challenging because they aim to prevent
diseases that may occur infrequently. This rarity of cases (e.g., a 1%
attack rate) means that trials must involve very large sample
sizes—often tens of thousands of participants—to accumulate enough
disease events to properly assess efficacy. Unlike therapeutics, which
are given only to sick patients, vaccines are administered to healthy
individuals across a wide population, making safety concerns even more
critical.</p>
<p>Due to this wide rollout, there is a strong emphasis on ensuring both
short- and long-term safety. Vaccine trials are longer and more
extensive than many other types of trials, often including
post-marketing surveillance to detect rare adverse events that may not
appear during the Phase III trial. Safety is assessed continuously
throughout the process.</p>
<p>The design of vaccine trials also introduces unique terminology and
methods. While concepts like attack rate, incidence, and vaccine
efficacy are standard in vaccine studies, they often parallel familiar
statistical concepts like proportions and time-to-event analysis found
in other areas. The language around vaccine studies can seem specialized
but usually maps back to more general clinical trial methodology.</p>
<p><strong>Challenges in Vaccine Trials</strong></p>
<p>Several challenges are inherent to vaccine trials:</p>
<ul>
<li><p><strong>Large Sample Sizes and Long Durations</strong>: To detect
rare adverse events and ensure statistical power, vaccine trials often
require large participant numbers and extended follow-up periods,
sometimes spanning 5-10 years.</p></li>
<li><p><strong>Endpoint Selection</strong>: Determining appropriate
endpoints is complex. Endpoints may include disease incidence, severity,
or composite measures combining multiple outcomes. The choice impacts
trial design and statistical analysis.</p></li>
<li><p><strong>Adaptive Designs</strong>: While adaptive trial designs
can offer flexibility and efficiency, their application in vaccine
trials is limited due to the emphasis on safety and the complexity of
implementing changes mid-trial.</p></li>
</ul>
<p><img src="./02_Plots/Vaccine/Vaccine-Development.png" /></p>
</div>
</div>
<div id="statistical-methods-for-the-analysis-of-immunogenicity-data"
class="section level1" number="2">
<h1><span class="header-section-number">2</span> Statistical Methods for
the Analysis of Immunogenicity Data</h1>
<p>Historically, vaccine immunogenicity studies have primarily focused
on evaluating the humoral immune response, specifically through the
measurement of serum antibody levels. In recent years, however, there
has been a growing emphasis on also assessing cellular immunity in such
trials. Despite this development, the use of cellular immunity endpoints
in vaccine trials largely remains at the investigational stage.</p>
<p>To summarize both humoral and cellular immunity data, four standard
statistical measures are commonly used:</p>
<ol style="list-style-type: decimal">
<li><strong>Geometric Mean Response</strong> (GMR)</li>
<li><strong>Geometric Mean Fold Increase</strong> (GMFI)</li>
<li><strong>Seroprotection Rate</strong></li>
<li><strong>Seroconversion Rate</strong></li>
</ol>
<p>The <strong>GMR</strong> and <strong>Seroprotection Rate</strong>
provide absolute measures of immunogenicity, while the
<strong>GMFI</strong> and <strong>Seroconversion Rate</strong> capture
within-individual increases following vaccination.</p>
<div id="geometric-mean-titres-and-concentrations"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Geometric Mean Titres
and Concentrations</h2>
<div
id="geometric-mean-titres-gmt-and-geometric-mean-concentrations-gmc"
class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Geometric Mean
Titres (GMT) and Geometric Mean Concentrations (GMC)</h3>
<p>Post-vaccination immunogenicity data, whether for humoral or cellular
responses, tend to exhibit right-skewed distributions. However, applying
a logarithmic transformation to these values often results in an
approximately normal distribution, which allows the use of standard
parametric statistical methods. For interpretability, the geometric
means and confidence intervals are computed by taking the antilog of
estimates derived from log-transformed data.</p>
<p>The <strong>Geometric Mean (GM)</strong> is the standard summary
statistic for immunogenicity data:</p>
<p>For a set of <em>n</em> independent and identically distributed
(i.i.d.) immunogenicity values <em>v₁, v₂, …, vₙ</em>, the geometric
mean is defined as:</p>
<p><span class="math display">\[
GM = (v_1 \times v_2 \times \ldots \times v_n)^{1/n}
\]</span></p>
<p>Alternatively, it can be expressed as:</p>
<p><span class="math display">\[
GM = \exp \left( \frac{1}{n} \sum_{i=1}^n \log_e v_i \right)
\]</span></p>
<p>The logarithmic base used for the transformation can vary, commonly
log₂ or log₁₀, but consistency is crucial. For example, when log₁₀ is
used:</p>
<p><span class="math display">\[
GM = 10^{\frac{1}{n} \sum_{i=1}^n \log_{10} v_i}
\]</span></p>
<hr />
</div>
<div id="geometric-mean-titres-for-antibody-dilutions"
class="section level3" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Geometric Mean
Titres for Antibody Dilutions</h3>
<p>In immunogenicity studies using antibody titres based on reciprocal
two-fold serial dilutions (e.g., 1:8, 1:16, 1:32, etc.), a convenient
log transformation is applied:</p>
<p><span class="math display">\[
u_i = \log_2 \left( \frac{t_i}{D/2} \right)
\]</span></p>
<p>Where <em>tᵢ</em> is the observed titre and <em>D</em> is the lowest
dilution tested. The transformed <em>uᵢ</em> values correspond to
dilution steps (1, 2, 3, etc.), simplifying interpretation. The
geometric mean of these titres is then:</p>
<p><span class="math display">\[
GM = \left( \frac{D}{2} \right)^{\frac{1}{n} \sum_{i=1}^n u_i}
\]</span></p>
<p><strong>Illustrative Example: Rubella HI Test</strong></p>
<p>Rubella virus infection, while often mild in adults, poses
significant risks during pregnancy. A haemagglutination inhibition (HI)
test is frequently performed to assess immunity. Consider HI titres with
a lowest tested dilution of 1:8, resulting in possible titres of 8, 16,
32, 64, etc.</p>
<p>Using the standard log₂ transformation:</p>
<ul>
<li>A titre of 8 becomes 1</li>
<li>16 becomes 2</li>
<li>32 becomes 3</li>
<li>64 becomes 4</li>
</ul>
<p>For titres 8, 8, 16, 32, 64, the GMT is computed as:</p>
<p><span class="math display">\[
GMT = 4 \times 2^{(1+1+2+3+4)/5} = 18.379
\]</span></p>
<p>In this transformation, differences in log-transformed values
directly correspond to dilution steps, simplifying interpretation.</p>
<hr />
</div>
<div id="geometric-standard-deviation-gsd-and-confidence-intervals"
class="section level3" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Geometric Standard
Deviation (GSD) and Confidence Intervals</h3>
<p>The <strong>Geometric Standard Deviation (GSD)</strong> is derived by
taking the exponential of the standard deviation (SD) of the
log-transformed data:</p>
<p><span class="math display">\[
GSD = \exp(SD)
\]</span></p>
<p>This allows calculation of the two-sided 95% confidence interval for
the underlying geometric mean (denoted as <em>ē</em>) using:</p>
<p><span class="math display">\[
LCL_e = \frac{GMT}{GSD^{t_{n-1, 1-\alpha/2} / \sqrt{n}}}  
\]</span></p>
<p><span class="math display">\[
UCL_e = GMT \times GSD^{t_{n-1, 1-\alpha/2} / \sqrt{n}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>tₙ₋₁,₁₋α/2</em> is the corresponding percentile of the Student’s
t-distribution with <em>(n-1)</em> degrees of freedom.</li>
</ul>
<p><strong>Rubella HI Test Example: Confidence Interval
Calculation</strong></p>
<p>Continuing the Rubella example:</p>
<ul>
<li>Sample SD of the log-transformed titres: <strong>0.904</strong></li>
<li>GSD = <strong>exp(0.904) = 2.469</strong></li>
<li>Using <em>t₄, 0.975 = 2.776</em> for a 95% confidence interval</li>
</ul>
<p>The lower and upper confidence limits for the GMT are calculated
as:</p>
<p><span class="math display">\[
LCL_e = \frac{18.379}{2.469^{(2.776/\sqrt{5})}} = 5.98  
\]</span></p>
<p><span class="math display">\[
UCL_e = 18.379 \times 2.469^{(2.776/\sqrt{5})} = 56.4  
\]</span></p>
</div>
<div id="single-vaccine-group" class="section level3" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> Single Vaccine
Group</h3>
<p>When immunogenicity values (<em>vi</em>) are log-transformed as
<em>ui = logₑ(vi)</em> and the <em>ui</em> values follow a normal
distribution with mean <strong>μ</strong> and variance
<strong>σ²</strong>, the <strong>arithmetic mean</strong> of the
<em>ui</em> values serves as a point estimate for μ. Consequently, the
<strong>geometric mean titre (GMT)</strong> or <strong>geometric mean
concentration (GMC)</strong> is estimated as:</p>
<p><span class="math display">\[
GMT = e^{\bar{u}}
\]</span></p>
<p>The distribution of the original <em>vi</em> values is thus
log-normal.</p>
<p>Since the <em>ui</em> values are normally distributed, confidence
intervals for μ can be calculated using the one-sample <em>t</em>-test.
The antilog of these limits provides the corresponding confidence
intervals for the geometric mean <strong>e</strong></p>
<p>It is important to note that the <strong>mean</strong> of the
log-normal distribution (expected value of <em>vi</em>) is not
<strong>e</strong>, but:</p>
<p><span class="math display">\[
E(vi) = e^{μ + \frac{σ^2}{2}}
\]</span></p>
<p>A key property of the log-normal distribution is that its
<strong>geometric mean</strong> and <strong>median</strong> are both
equal to <strong>e</strong>.</p>
<p><strong>Example Single Vaccine Group</strong></p>
<p>Consider six IFN-γ (Th1-type) immunogenicity measurements in IU/L:
<strong>3.51, 9.24, 13.7, 35.2, 47.4, 57.5</strong></p>
<p>Their natural logarithms are: <strong>1.256, 2.224, 2.617, 3.561,
3.859, 4.052</strong></p>
<p>The arithmetic mean of these log-transformed values is
<strong>2.928</strong>, with a standard error of <strong>0.444</strong>.
The geometric mean concentration (GMC) is thus:</p>
<p><span class="math display">\[
GMC = e^{2.928} = 18.7 \, \text{IU/L}
\]</span></p>
<p>Using a two-sided 95% confidence interval for μ with
<em>t</em>(0.975, 5) = 2.571:</p>
<p><span class="math display">\[
\text{Lower limit} = 2.928 - 2.571(0.444) = 1.786  
\text{Upper limit} = 2.928 + 2.571(0.444) = 4.070  
\]</span></p>
<p>Exponentiating these limits yields the 95% confidence interval for
the underlying geometric mean:</p>
<p><span class="math display">\[
e^{1.786} = 5.97 \quad \text{and} \quad e^{4.070} = 58.6  
\]</span></p>
<p>As expected with geometric means, these confidence intervals are
asymmetrical.</p>
</div>
<div id="two-vaccine-groups" class="section level3" number="2.1.5">
<h3><span class="header-section-number">2.1.5</span> Two Vaccine
Groups</h3>
<p>When comparing two vaccine groups, statistical inference is performed
using a <strong>two-sample <em>t</em>-test</strong> applied to the
log-transformed immunogenicity values. Point and interval estimates for
the difference in means <strong>δ = μ₁ - μ₀</strong> can be transformed
back to estimates for the <strong>geometric mean ratio (GMR)</strong>,
representing the ratio of the geometric means of the two groups:</p>
<p><span class="math display">\[
GMR = \frac{GM₁}{GM₀} = e^{\bar{u₁} - \bar{u₀}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>GM₁</em> is the geometric mean of the investigational vaccine
group</li>
<li><em>GM₀</em> is the geometric mean of the control group</li>
<li><em>ū₁</em>, <em>ū₀</em> are the means of the log-transformed values
for the respective groups</li>
</ul>
<p>The <em>p</em>-value from the two-sample <em>t</em>-test testing
<strong>δ = 0</strong> corresponds to a test of whether <strong>GMR =
1</strong>, i.e., whether the geometric means differ significantly
between the groups.</p>
<p><strong>Example: Two Vaccine Groups</strong></p>
<p>Assume the six IFN-γ values mentioned earlier are compared to a
second group of six values, with:</p>
<ul>
<li>Arithmetic mean of log-transformed values:
<strong>2.754</strong></li>
<li>Standard error: <strong>0.512</strong></li>
</ul>
<p>The geometric mean concentration of the control group is:</p>
<p><span class="math display">\[
GMC₀ = e^{2.754} = 15.7 \, \text{IU/L}
\]</span></p>
<p>The geometric mean ratio (GMR) comparing the investigational group to
the control group is:</p>
<p><span class="math display">\[
GMR = \frac{18.7}{15.7} = e^{2.928 - 2.754} = 1.19  
\]</span></p>
<p>The estimated standard error of the difference is
<strong>0.677</strong>. The 95% confidence interval for the underlying
GMR is:</p>
<p><span class="math display">\[
\text{Lower limit} = e^{1.19 - 2.228(0.677)} = 0.73  
\]</span></p>
<p><span class="math display">\[
\text{Upper limit} = e^{1.19 + 2.228(0.677)} = 14.9  
\]</span></p>
<p>This provides an asymmetric confidence interval for the ratio of
geometric means, consistent with the properties of log-normal
distributions.</p>
</div>
</div>
<div id="geometric-mean-fold-increase" class="section level2"
number="2.2">
<h2><span class="header-section-number">2.2</span> Geometric Mean Fold
Increase</h2>
<p>In some infectious disease settings, participants may already have
measurable pre-vaccination immune responses due to previous infections
or vaccinations. A typical example is <strong>influenza</strong>, where
most adults have been exposed to influenza viruses during their
lifetime. In such cases, post-vaccination immune responses reflect both
the effect of vaccination and existing pre-vaccination immunity.</p>
<p>For these scenarios, an alternative to geometric mean titres (GMT) or
concentrations (GMC) is the <strong>Geometric Mean Fold Increase
(gMFI)</strong>, also referred to as <strong>Mean Fold Increase</strong>
or <strong>Geometric Mean Fold Rise</strong>, which specifically
quantifies the relative increase in immune response following
vaccination at the individual level.</p>
<div id="single-arm" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Single Arm</h3>
<p><strong>Definition and Calculation</strong></p>
<p>For each individual:</p>
<p><span class="math display">\[
f_i = \frac{v_{post}}{v_{pre}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>vₚᵣₑ</em> = pre-vaccination immunogenicity value</li>
<li><em>vₚₒₛₜ</em> = post-vaccination immunogenicity value</li>
<li><em>fᵢ</em> = fold increase</li>
</ul>
<p>The <strong>Geometric Mean Fold Increase (gMFI)</strong> across
<em>n</em> individuals is calculated as:</p>
<p><span class="math display">\[
gMFI = \exp \left( \frac{1}{n} \sum_{j=1}^{n} \log_e f_{ij} \right)
\]</span></p>
<p>Alternatively, if both pre- and post-vaccination data are available
for all subjects:</p>
<p><span class="math display">\[
gMFI = \frac{GM_{post}}{GM_{pre}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>GMₚₒₛₜ</em> = geometric mean of post-vaccination values</li>
<li><em>GMₚᵣₑ</em> = geometric mean of pre-vaccination values</li>
</ul>
<p><strong>Note:</strong> This relationship holds only if both pre- and
post-vaccination data are available for all individuals.</p>
<hr />
<p><strong>Example-Single Group gMFI Calculation</strong></p>
<p>In an influenza vaccine trial, pre- and post-vaccination antibody
titres for six subjects are recorded as:</p>
<table>
<thead>
<tr class="header">
<th>Subject</th>
<th>Pre-Vaccination</th>
<th>Post-Vaccination</th>
<th>Fold Increase</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>5</td>
<td>40</td>
<td>8</td>
</tr>
<tr class="even">
<td>2</td>
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="odd">
<td>3</td>
<td>10</td>
<td>160</td>
<td>16</td>
</tr>
<tr class="even">
<td>4</td>
<td>10</td>
<td>320</td>
<td>32</td>
</tr>
<tr class="odd">
<td>5</td>
<td>20</td>
<td>80</td>
<td>4</td>
</tr>
<tr class="even">
<td>6</td>
<td>20</td>
<td>640</td>
<td>32</td>
</tr>
</tbody>
</table>
<p>The calculated statistics are:</p>
<ul>
<li><strong>GMₚᵣₑ</strong> = 10.0</li>
<li><strong>GMₚₒₛₜ</strong> = 142.5</li>
<li><strong>gMFI</strong> = 14.25</li>
<li><strong>Geometric Standard Deviation (GSD)</strong> = 2.249</li>
</ul>
<p>The confidence intervals for the gMFI can be calculated using the
same formulas applied to geometric means, as described earlier.</p>
<hr />
</div>
<div id="comparison-of-two-groups-gmfi-and-gmfr" class="section level3"
number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Comparison of Two
Groups: gMFI and gMFR</h3>
<p>For comparing two vaccine groups, the <strong>Geometric Mean Fold
Ratio (gMFR)</strong> is used, representing the ratio of fold increases
between the groups.</p>
<p>The relationship between gMFR and geometric means is expressed
as:</p>
<p><span class="math display">\[
gMFR = \frac{gMFI_1}{gMFI_0} = \frac{GMR_{post}}{GMR_{pre}}
\]</span></p>
<p>Where:</p>
<ul>
<li><em>gMFI₁</em>, <em>gMFI₀</em> = gMFI of investigational and control
groups, respectively</li>
<li><em>GMRₚₒₛₜ</em> = geometric mean ratio of post-vaccination
values</li>
<li><em>GMRₚᵣₑ</em> = geometric mean ratio of pre-vaccination
values</li>
</ul>
<p>In randomized trials with no significant baseline imbalance (i.e.,
<em>GMRₚᵣₑ ≈ 1</em>), the gMFR effectively approximates
<em>GMRₚₒₛₜ</em>. Thus, the analysis of fold increases yields results
nearly identical to those obtained from analyzing post-vaccination
immunogenicity values alone.</p>
<hr />
<p><strong>Example: Two Group Comparison</strong></p>
<p><strong>Experimental Group:</strong></p>
<table>
<thead>
<tr class="header">
<th>Pre-Vaccination</th>
<th>Post-Vaccination</th>
<th>Fold Increase</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>5</td>
<td>40</td>
<td>8</td>
</tr>
<tr class="even">
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="odd">
<td>10</td>
<td>80</td>
<td>8</td>
</tr>
<tr class="even">
<td>10</td>
<td>80</td>
<td>8</td>
</tr>
</tbody>
</table>
<p><strong>Control Group:</strong></p>
<table>
<thead>
<tr class="header">
<th>Pre-Vaccination</th>
<th>Post-Vaccination</th>
<th>Fold Increase</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="even">
<td>5</td>
<td>80</td>
<td>16</td>
</tr>
<tr class="odd">
<td>10</td>
<td>80</td>
<td>8</td>
</tr>
<tr class="even">
<td>10</td>
<td>160</td>
<td>16</td>
</tr>
</tbody>
</table>
<p>The calculated statistics are:</p>
<ul>
<li><em>GMₚᵣₑ</em> = 7.1 (both groups)</li>
<li><em>GMₚₒₛₜ</em> Experimental = 67.27</li>
<li><em>GMₚₒₛₜ</em> Control = 95.14</li>
<li><em>gMFI₁</em> = 9.51</li>
<li><em>gMFI₀</em> = 13.45</li>
<li><em>gMFR</em> = 1.4</li>
<li><em>GMRₚₒₛₜ</em> = 1.4</li>
</ul>
<p>The gMFR and GMRₚₒₛₜ are identical due to the absence of baseline
imbalance.</p>
<p><strong>Efficiency Considerations</strong></p>
<p>When no baseline imbalance exists:</p>
<ul>
<li>Using fold increases introduces unnecessary variability because fold
increases mathematically involve the difference between log-transformed
post- and pre-vaccination values.</li>
<li>If the correlation between pre- and post-vaccination values is weak
(e.g., correlation &lt; 0.5), the variance of fold increases is larger
than the variance of post-vaccination values alone.</li>
<li>As a result, analyses based on fold increases often produce larger
<em>p</em>-values and wider confidence intervals compared to analyses
based on post-vaccination data alone.</li>
</ul>
</div>
<div id="misconception-do-fold-increases-correct-for-baseline-imbalance"
class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Misconception: Do
Fold Increases Correct for Baseline Imbalance?</h3>
<p>It is a common but incorrect belief that analyzing fold increases
eliminates bias due to baseline imbalance. In reality:</p>
<ul>
<li>On the log scale:</li>
</ul>
<p><span class="math display">\[
\log(f_i) = \log(v_{post}) - \log(v_{pre})
\]</span></p>
<ul>
<li>Change scores (including fold increases) remain correlated with
baseline values, often negatively.</li>
<li>For bioassays, higher pre-vaccination values often result in smaller
observed fold increases.</li>
<li>If there is a baseline imbalance, with one group having lower
pre-vaccination values, this group will tend to show artificially
inflated fold increases, introducing bias <strong>in favor of the group
with lower pre-vaccination values</strong>.</li>
</ul>
<p>Thus, fold increase analysis does <strong>not</strong> inherently
correct for baseline imbalance. The appropriate statistical method for
addressing baseline imbalance is <strong>analysis of covariance
(ANCOVA)</strong>, which explicitly adjusts for pre-vaccination
differences.</p>
</div>
</div>
<div id="two-seroresponse-rates" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Two Seroresponse
Rates</h2>
<p>In vaccine clinical trials and immunogenicity studies, it is common
to assess the immune response using binary endpoints that reflect the
presence or increase of protective antibodies following vaccination. Two
widely used binary measures are the <em>seroprotection rate</em> and the
<em>seroconversion rate</em>. Both provide critical information
regarding the effectiveness of a vaccine but capture different
immunological phenomena.</p>
<div id="seroprotection-rate" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> 1. Seroprotection
Rate</h3>
<p>The seroprotection rate is defined as the proportion of vaccinated
individuals whose antibody levels reach or exceed a predetermined
threshold that is considered to be associated with protection from
infection or disease. This threshold is referred to as the
<strong>threshold of protection</strong>.</p>
<p>For many infectious diseases, scientific research has identified
specific antibody levels that are assumed to provide protection. If an
individual’s antibody level is equal to or higher than this threshold
after vaccination, the person is classified as
<em>seroprotected</em>.</p>
<p>For example:</p>
<ul>
<li>In <strong>influenza</strong> vaccine studies, an anti-hemagglutinin
(anti-HA) antibody titre of <strong>40 or higher</strong> is generally
accepted as the threshold for seroprotection.</li>
<li>For <strong>diphtheria</strong> and <strong>tetanus</strong>, the
commonly cited threshold is an anti-diphtheria or anti-tetanus antibody
concentration of <strong>0.1 international units per millilitre
(IU/ml)</strong>.</li>
</ul>
<p>The seroprotection rate is thus a binary measure: individuals are
either classified as seroprotected or not seroprotected based on their
antibody levels. The seroprotection rate expresses the percentage of
vaccinated individuals who meet this criterion.</p>
<p>It is important to note that the phrase <em>associated with
protection</em> does not always imply complete immunity. In many cases,
this association indicates a <strong>moderate to high
probability</strong> of protection rather than absolute prevention of
disease. For instance, in the case of influenza, it is estimated that an
anti-HA antibody titre of 40 corresponds to approximately a <strong>50%
probability</strong> of being protected against infection. Therefore,
while the seroprotection threshold is useful for evaluating vaccine
responses, it should not be interpreted as guaranteeing full
immunity.</p>
</div>
<div id="seroconversion-rate" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> 2. Seroconversion
Rate</h3>
<p>The seroconversion rate refers to the proportion of individuals whose
antibody status changes in response to vaccination, indicating the
development or significant increase of detectable antibodies.</p>
<p>According to <strong>Stedman’s Medical Dictionary</strong>,
seroconversion is defined as the appearance of detectable, specific
antibodies in the blood as a result of infection or immunization.</p>
<p>An individual’s antibody status, also called
<strong>serostatus</strong>, is classified as follows:</p>
<ul>
<li><strong>Seronegative</strong>: No detectable antibodies are
present.</li>
<li><strong>Seropositive</strong>: Detectable antibodies are
present.</li>
</ul>
<p>If a person is seronegative before vaccination and becomes
seropositive after vaccination, that individual is considered to have
<em>seroconverted</em>.</p>
<p>Different studies may apply slightly different definitions of
seroconversion, depending on the disease and the vaccine being
evaluated.</p>
<p>Examples of seroconversion definitions include:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Absolute Appearance of Detectable Antibodies</strong></p>
<p>In a study of a live attenuated human rotavirus vaccine conducted by
Vesikari and colleagues, seroconversion was defined as the appearance of
serum IgA antibodies to rotavirus at a titre of <strong>20 units per
millilitre or higher</strong> in infants who were seronegative at
baseline. The proportion of infants who seroconverted varied between
<strong>73% and 96%</strong>, depending on the vaccine dose.</p></li>
<li><p><strong>Fourfold Rise in Antibody Levels</strong></p>
<p>Another common definition of seroconversion is based on a
<strong>fourfold increase</strong> in antibody levels following
vaccination. This definition is particularly relevant when some
individuals may already be seropositive at the time of enrollment.</p>
<p>For example, in a study evaluating a virus-like particle vaccine for
human papillomavirus (<strong>HPV</strong>), Harro and colleagues
defined seropositivity as an ELISA antibody titre equal to or greater
than the reactivity of a standard pooled serum. Among the 72 female
participants, six were seropositive at baseline. In this study,
seroconversion was defined as a <strong>fourfold or greater
rise</strong> in antibody titre after vaccination.</p></li>
<li><p><strong>Combined Definition</strong></p>
<p>Many clinical studies, particularly influenza vaccine trials, adopt a
combined approach to defining seroconversion. This definition takes into
account both baseline serostatus and post-vaccination changes:</p>
<ul>
<li>For individuals with a baseline antibody titre <strong>below
10</strong> (seronegative), seroconversion is defined as reaching a
<strong>post-vaccination titre of 40 or higher</strong>, which is
consistent with the seroprotection threshold in influenza studies.</li>
<li>For individuals with a baseline antibody titre <strong>10 or
higher</strong> (seropositive), seroconversion is defined as achieving
at least a <strong>fourfold rise</strong> in titre after
vaccination.</li>
</ul></li>
</ol>
<p>It is important to highlight that such definitions may vary slightly
between studies. For instance, some studies distinguish between becoming
seropositive (titre ≥10) and achieving seroprotection (titre ≥40),
especially in the context of influenza vaccines.</p>
<p>Regardless of the specific definition applied, seroconversion remains
a binary endpoint. Individuals either meet the criteria for
seroconversion or they do not. The <strong>seroconversion rate</strong>
represents the proportion or percentage of study participants who
seroconverted after vaccination.</p>
</div>
</div>
<div id="analysis-of-proportions" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Analysis of
Proportions</h2>
<p>This section discusses statistical methods for analyzing
<strong>binary endpoints</strong> in clinical vaccine trials—most
notably <strong>seroprotection</strong> and
<strong>seroconversion</strong> rates. However, these methods are
broadly applicable to other binary outcomes, such as:</p>
<ul>
<li>Proportion of subjects reporting local/systemic reactions</li>
<li>Proportion experiencing adverse vaccine events</li>
<li>Proportion remaining disease-free post-vaccination</li>
</ul>
<div id="single-proportion-analysis" class="section level3"
number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Single Proportion
Analysis</h3>
<p>To test a null hypothesis about a single proportion (e.g.,
probability of seroprotection):</p>
<ul>
<li><p><strong>Distribution</strong>: Use the Binomial Distribution
<span class="math inline">\(B(n, \pi)\)</span>, where:</p>
<ul>
<li><span class="math inline">\(n\)</span>: number of observations</li>
<li><span class="math inline">\(\pi\)</span>: assumed proportion under
<span class="math inline">\(H_0\)</span></li>
</ul></li>
<li><p><strong>One-sided test</strong>: <span class="math inline">\(H_0:
\pi = \pi_0\)</span> vs. <span class="math inline">\(H_1: \pi &gt;
\pi_0\)</span> Compute <span class="math inline">\(P(S \geq s \mid
\pi_0)\)</span>, with <span class="math inline">\(S \sim B(n,
\pi_0)\)</span></p></li>
<li><p><strong>In SAS</strong>: Use <code>PROBBNML(π; n; m)</code> to
calculate tail probabilities.</p></li>
</ul>
<p>To calculate confidence Intervals for a Single Proportion</p>
<p>Three common methods:</p>
<ol style="list-style-type: decimal">
<li>Clopper–Pearson (Exact)</li>
</ol>
<ul>
<li>Based on binomial tail probabilities</li>
<li>Conservative: always ≥ nominal coverage (e.g., 95%)</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Wald Method (Asymptotic)</li>
</ol>
<ul>
<li>Simple, but poor performance near <span class="math inline">\(\pi
\approx 0\)</span> or <span class="math inline">\(\pi \approx
1\)</span></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Wilson Method (Asymptotic)</li>
</ol>
<ul>
<li>Better average coverage than Wald</li>
<li>Confidence interval obtained by solving a quadratic inequality</li>
</ul>
</div>
<div id="comparing-two-proportions" class="section level3"
number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Comparing Two
Proportions</h3>
<p>Two main comparison metrics:</p>
<ul>
<li><strong>Rate Difference (RD)</strong>: <span
class="math inline">\(r_1 - r_0\)</span></li>
<li><strong>Rate Ratio (RR)</strong>: <span class="math inline">\(r_1 /
r_0\)</span></li>
</ul>
<p><strong>Hypothesis Testing</strong></p>
<ul>
<li><p>Test <span class="math inline">\(H_0: RD = 0\)</span> or <span
class="math inline">\(H_0: RR = 1\)</span></p></li>
<li><p>Use:</p>
<ul>
<li>Pearson’s Chi-Square</li>
<li>Fisher’s Exact Test (for small samples)</li>
<li>Suissa and Shuster Test (exact, more powerful)</li>
</ul></li>
</ul>
<p>To calculate Confidence Intervals</p>
<p>For Rate Difference:</p>
<ul>
<li><p><strong>Wald-type</strong>: <span class="math inline">\(RD \pm
z_{\alpha/2} \cdot SE(RD)\)</span></p>
<blockquote>
<p>Risk of inconsistent test statistic and interval due to differing
standard error assumptions.</p>
</blockquote></li>
<li><p><strong>Wilson-type (Miettinen and Nurminen)</strong>: Solves for
values of RD satisfying:</p>
<p><span class="math display">\[
\frac{RD - RD_0}{SE^*(RD_0)} = \pm z_{\alpha/2}
\]</span></p>
<p>Iterative solution using constrained MLEs.</p></li>
</ul>
<p>For Rate Ratio:</p>
<ul>
<li><p><strong>Wald-type (log-transform)</strong>:</p>
<p><span class="math display">\[
CI = \exp\left[ \log(RR) \pm z_{\alpha/2} \cdot SE(\log RR) \right]
\]</span></p></li>
<li><p><strong>Wilson-type</strong>: Also iterative, using constrained
MLEs.</p></li>
</ul>
</div>
<div id="suissa-and-shuster-exact-test" class="section level3"
number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Suissa and Shuster
Exact Test</h3>
<p>The Suissa and Shuster Exact Test is a statistical method for
comparing two proportions, such as two groups in a clinical trial (e.g.,
treatment vs. placebo). Unlike Fisher’s Exact Test, which conditions on
the margins (i.e., the row and column totals in a 2x2 contingency
table), the Suissa and Shuster test conditions only on the sample sizes
of the two groups, not on the number of events observed. This makes it a
more flexible and often more powerful test than Fisher’s test,
particularly when you want to test hypotheses beyond the standard
“proportions are equal.”</p>
<p><strong>Key Characteristics:</strong></p>
<ol style="list-style-type: decimal">
<li><p><strong>Exact Test</strong>: It does not rely on large-sample
approximations. The test remains valid regardless of the sample
size.</p></li>
<li><p><strong>More Powerful than Fisher’s Test</strong>: Studies have
shown that the Suissa and Shuster test typically has greater power,
meaning it’s more likely to detect a true difference when one
exists.</p></li>
<li><p><strong>Supports More General Hypotheses</strong>:</p>
<ul>
<li>It can test whether the difference in proportions equals a specific
value other than zero (i.e., H₀: RD = Δ₀).</li>
<li>It can test whether the relative risk equals a value other than 1
(i.e., H₀: RR = RR₀).</li>
<li>Fisher’s test does not support this directly.</li>
</ul></li>
<li><p><strong>Enables Exact Confidence Intervals</strong>: Since it
supports testing a range of null hypotheses, it can be used to derive
exact confidence intervals by identifying the smallest and largest
parameter values that would not be rejected at a given significance
level.</p></li>
</ol>
<p><strong>How the Test Works (Conceptually):</strong></p>
<ul>
<li><p>First, compute a Z-statistic based on your observed data and the
hypothesis you’re testing. The formula depends on whether you’re testing
a difference in proportions (risk difference) or a relative
risk.</p></li>
<li><p>Then, generate all possible 2x2 contingency tables for the fixed
sample sizes. For each of these tables, calculate the same type of
Z-statistic.</p></li>
<li><p>The p-value is determined by finding the maximum tail probability
over a range of assumed null parameter values (e.g., for RD or RR). This
is why it’s called an “exact” test—it integrates over possible nuisance
parameter values rather than assuming a fixed one.</p></li>
<li><p>In practice, the test is implemented by discretizing the possible
values of the nuisance parameter (e.g., from 0 to 1 in small steps like
0.001) and computing the p-value at each step.</p></li>
</ul>
<p><strong>Procedure (Chan, 1998):</strong></p>
<ol style="list-style-type: decimal">
<li><p>Compute Z-statistic under null</p></li>
<li><p>Enumerate all 2x2 tables and calculate <span
class="math inline">\(Z_{ij}\)</span></p></li>
<li><p>Compute P-value:</p>
<p><span class="math display">\[
P = \max_{\pi_0 \in D} \Pr(|Z| \geq |Z_{obs}| \mid \pi_0)
\]</span></p></li>
</ol>
<blockquote>
<p><em>Where <span class="math inline">\(D\)</span> is the domain of
nuisance parameter <span class="math inline">\(\pi_0\)</span>, divided
into small increments.</em></p>
</blockquote>
<p><strong>Why it is Useful in Vaccine Trials or Safety
Studies:</strong></p>
<ul>
<li><p>Vaccine efficacy trials often deal with low event rates (e.g.,
infection rates), where approximations like the chi-square test or Wald
test may perform poorly. The Suissa and Shuster test avoids these
problems.</p></li>
<li><p>It remains accurate when group sizes are small or event rates are
extreme (close to 0 or 1).</p></li>
</ul>
<p><strong>Example Use:</strong></p>
<p>Imagine you conduct a study with 15 participants in each group. The
vaccine group has 7 people who develop illness, while the placebo group
has 12. Using Fisher’s exact test, the two-sided p-value might be around
0.128. But using the Suissa and Shuster test, the p-value might be lower
(e.g., 0.068), suggesting a stronger signal.</p>
<p>This difference in power can be crucial when working with small
sample sizes or in early-phase clinical trials.</p>
<table>
<colgroup>
<col width="24%" />
<col width="23%" />
<col width="21%" />
<col width="30%" />
</colgroup>
<thead>
<tr class="header">
<th>Method</th>
<th>Use Case</th>
<th>Pros</th>
<th>Cons</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Clopper–Pearson</strong></td>
<td>Single proportion</td>
<td>Exact, conservative</td>
<td>Wider intervals</td>
</tr>
<tr class="even">
<td><strong>Wald</strong></td>
<td>Single/two proportions</td>
<td>Simple</td>
<td>Poor performance near 0 or 1</td>
</tr>
<tr class="odd">
<td><strong>Wilson</strong></td>
<td>Single/two proportions</td>
<td>Good coverage</td>
<td>Slightly more complex</td>
</tr>
<tr class="even">
<td><strong>Miettinen–Nurminen</strong></td>
<td>Rate diff./ratio CI</td>
<td>Accurate, asymptotic</td>
<td>Needs iteration</td>
</tr>
<tr class="odd">
<td><strong>Suissa–Shuster</strong></td>
<td>Exact comparison</td>
<td>Powerful, exact</td>
<td>Computationally intensive</td>
</tr>
<tr class="even">
<td><strong>Jewell’s Correction</strong></td>
<td>RR bias correction</td>
<td>Effective</td>
<td>No gain in CI coverage</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="multiple-co-primary-endpoints-and-the-intersectionunion-test"
class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> Multiple Co-Primary
Endpoints and the Intersection–Union Test</h2>
<p>In clinical vaccine trials, researchers often need to assess
<strong>more than one key outcome</strong>, or <strong>co-primary
endpoints</strong>, to determine whether a vaccine is effective. These
endpoints may reflect different aspects of vaccine performance—such as
protection against various disease strains, or different types of immune
response (e.g., antibody levels and seroprotection).</p>
<p>Co-primary endpoints are <strong>multiple main outcomes that must all
be met</strong> to declare the trial a success. For example:</p>
<ul>
<li><p>A <strong>combination vaccine</strong> targeting multiple
serotypes (e.g., pneumococcal vaccine for 10 serotypes) might require
showing superiority against <strong>each serotype</strong> individually.
That leads to <strong>10 co-primary endpoints</strong>.</p></li>
<li><p>A vaccine might need to prove superiority on
<strong>both</strong>:</p>
<ul>
<li><strong>Seroprotection rate</strong> (the proportion of subjects
above a protective antibody threshold),</li>
<li>and the <strong>geometric mean response</strong> (a measure of
average immune strength).</li>
</ul></li>
</ul>
<p>In these cases, a statistically sound method is required to
<strong>account for multiple testing</strong>, which increases the
chance of false positives.</p>
<hr />
<p><strong>The Multiplicity Problem</strong></p>
<p>When testing multiple hypotheses (i.e., each endpoint), the chance of
<strong>at least one false positive</strong> increases. Simply using a
significance level (α) of 0.05 for each endpoint will no longer maintain
an overall type I error rate of 5%.</p>
<p>The <strong>Intersection–Union Test (IU test)</strong> is a
conservative but commonly used method to address this issue in
regulatory trials.</p>
<ul>
<li>For each endpoint (e.g., each serotype or measure), define a
<strong>null hypothesis</strong> <span
class="math inline">\(H_{0i}\)</span>: the experimental vaccine is not
superior to control on endpoint <em>i</em>.</li>
<li>Each null hypothesis is tested <strong>individually at significance
level α</strong> (usually 0.05).</li>
<li><strong>Only if all</strong> individual hypotheses are rejected, can
the overall conclusion of vaccine superiority be claimed.</li>
</ul>
<p><strong>Mathematically</strong>:</p>
<ul>
<li><p>Let there be <em>k</em> co-primary endpoints.</p></li>
<li><p>The <strong>overall null hypothesis</strong> is the
<strong>union</strong> of all individual nulls:</p>
<p><span class="math display">\[
H_0 = H_{01} \cup H_{02} \cup \cdots \cup H_{0k}
\]</span></p></li>
<li><p>The <strong>alternative hypothesis</strong> is their
<strong>intersection</strong>:</p>
<p><span class="math display">\[
H_1 = H_{11} \cap H_{12} \cap \cdots \cap H_{1k}
\]</span></p></li>
<li><p>This is why it’s called the <strong>Intersection–Union
Test</strong>: to reject the union of nulls, all individual nulls must
be rejected (i.e., prove all endpoints simultaneously).</p></li>
</ul>
<hr />
<p><strong>Advantages of the IU Test</strong></p>
<ul>
<li><strong>Simplicity</strong>: Easy to implement and explain, even to
non-statistical stakeholders.</li>
<li><strong>No adjustment to α is required</strong> because the approach
itself controls the family-wise error rate by construction.</li>
</ul>
<hr />
<p><strong>Disadvantages of the IU Test</strong></p>
<ul>
<li><p><strong>Conservativeness</strong>: The IU test is often
<strong>too strict</strong>, meaning it may <strong>fail to detect true
effects</strong>, especially if endpoints are <strong>not strongly
correlated</strong>.</p>
<ul>
<li>If endpoints are <strong>independent</strong>, the overall
significance level becomes <span
class="math inline">\(\alpha^k\)</span>, which is very small for large
<em>k</em>.</li>
<li>For example, with α = 0.05 and 5 endpoints, the joint level is <span
class="math inline">\(0.05^5 = 3.1 \times 10^{-7}\)</span>.</li>
</ul></li>
</ul>
<hr />
<p><strong>Alternatives to IU: The Min Test</strong></p>
<p>When the outcomes (endpoints) are assumed to follow a
<strong>multivariate normal distribution</strong>, an alternative is the
<strong>Min Test</strong>, which is:</p>
<ul>
<li><p>Based on the <strong>minimum</strong> of the individual test
statistics:</p>
<p><span class="math display">\[
Z_{\text{min}} = \min(Z_1, Z_2, ..., Z_k)
\]</span></p></li>
<li><p>More <strong>statistically powerful</strong> than the IU test
under some conditions.</p></li>
<li><p><strong>Problem</strong>: The test distribution depends on the
<strong>covariance structure</strong> of the endpoints, making it
<strong>computationally complex</strong> and <strong>hard to
generalize</strong>.</p></li>
</ul>
<hr />
<p><strong>Conclusion</strong></p>
<p>The IU Test is widely accepted in regulatory settings due to its
<strong>simplicity and conservativeness</strong>, but it may
<strong>lack statistical power</strong>, especially in cases with many
co-primary endpoints. More powerful alternatives exist, such as the Min
Test or other <strong>multivariate modeling approaches</strong>, but
they require <strong>more assumptions and computational
effort</strong>.</p>
<p>Ultimately, the choice depends on <strong>regulatory
expectations</strong>, <strong>study design</strong>, and the
<strong>importance of detecting modest effects across multiple
endpoints</strong>. * <strong>Chuang-Stein et al.</strong> discuss power
and strategy for designing trials with multiple primary endpoints. *
<strong>Senn and Bretz</strong> explore the implications of multiple
testing on drug development and offer insights into alternative
designs.</p>
</div>
<div id="reverse-cumulative-distribution-plot" class="section level2"
number="2.6">
<h2><span class="header-section-number">2.6</span> Reverse Cumulative
Distribution Plot</h2>
<p>The Reverse Cumulative Distribution (RCD) Plot is a graphical tool
commonly used to display and compare the distribution of immunogenicity
values between different vaccine groups. This visualization is
particularly useful in vaccine trials where understanding the entire
distribution—not just summary statistics like the mean or median—is
crucial.</p>
<p><strong>Concept and Axes</strong></p>
<ul>
<li><strong>X-axis</strong>: Represents immunogenicity values, usually
plotted on a logarithmic scale.</li>
<li><strong>Y-axis</strong>: Indicates the percentage of subjects who
have an immunogenicity value greater than or equal to the corresponding
x-value.</li>
</ul>
<p>This format means that the plot shows the <em>reverse</em> of the
usual cumulative distribution. Instead of showing the proportion of
subjects <em>below</em> a certain value, it shows the proportion
<em>above</em>.</p>
<p><strong>Shape and Interpretation</strong></p>
<ul>
<li>The RCD curve starts at <strong>100%</strong> and decreases from
left to right.</li>
<li>The curve reflects the <strong>log-transformed immunogenicity
values</strong>.</li>
</ul>
<p>Different curve shapes imply different distribution
characteristics:</p>
<ul>
<li><strong>Steep Middle Section (Curve A)</strong>: Indicates a
symmetric distribution with low variability. Most subjects cluster
around the mean.</li>
<li><strong>Flatter Middle Section (Curve B)</strong>: Implies high
variability among subjects’ immune responses.</li>
<li><strong>Rectangular Shape (Curve C)</strong>: Reflects a
right-skewed distribution. Most subjects have high immune responses with
a sharp drop at the end.</li>
<li><strong>Left-Skewed Shape (Curve D)</strong>: Most subjects have low
responses; the curve declines sharply early and flattens toward the
end.</li>
</ul>
<p><strong>Comparing Groups Using RCD Curves</strong></p>
<ul>
<li><p>If one vaccine’s RCD curve lies entirely <strong>above</strong>
another’s and they do <strong>not intersect</strong> (e.g., A vs. D),
then the former induces higher immune responses at <strong>every
percentile</strong>.</p></li>
<li><p>If the curves <strong>intersect</strong>, interpretation becomes
more nuanced:</p>
<ul>
<li>One vaccine may produce <strong>more high and low
responses</strong>, while the other is more consistent.</li>
<li>If the <strong>point of intersection</strong> is <strong>above
50%</strong>, it favors the curve that descends later (e.g., B over A),
meaning more than half of the subjects in that group achieved higher
immunogenicity.</li>
</ul></li>
</ul>
<div class="float">
<img src="02_Plots/Vaccine/RCD.png"
alt="The Reverse Cumulative Distribution Plot" />
<div class="figcaption">The Reverse Cumulative Distribution Plot</div>
</div>
</div>
</div>
<div id="efficacy-analysis" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Efficacy Analysis</h1>
<div id="assessing-vaccine-efficacy" class="section level2"
number="3.1">
<h2><span class="header-section-number">3.1</span> Assessing Vaccine
Efficacy</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Definition and Formula</strong> Vaccine Efficacy (VE)
measures how well a vaccine protects individuals from disease. It is
defined as:</p>
<p><strong>VE = 1 - Risk Ratio (RR)</strong> Where:</p>
<ul>
<li><p><strong>RR (Risk Ratio)</strong> = ARV / ARU</p>
<ul>
<li><strong>ARV</strong>: Attack rate in vaccinated group</li>
<li><strong>ARU</strong>: Attack rate in unvaccinated (control/placebo)
group</li>
</ul></li>
</ul>
<p>Interpreting VE:</p>
<ul>
<li>The range of VE is from negative infinity to 1.</li>
<li>A VE of <strong>1</strong> (or 100%) means <strong>complete
protection</strong> — no cases occurred in the vaccinated group.</li>
<li>A VE of <strong>0</strong> means <strong>no effect</strong> — the
vaccine does not reduce the risk.</li>
<li>A <strong>negative VE</strong> suggests harm (higher risk in the
vaccinated group than unvaccinated).</li>
</ul></li>
<li><p><strong>Multiple Testing Approaches</strong> The choice of
statistical method to estimate VE depends on how disease occurrence is
modeled:</p>
<ul>
<li><strong>Counts</strong>: Number of events (e.g., infections) across
follow-up time.</li>
<li><strong>Proportions</strong>: Percentage of individuals infected in
each group.</li>
<li><strong>Time-to-Event</strong>: Time from vaccination to disease
onset (e.g., using survival analysis).</li>
</ul></li>
<li><p><strong>VE as Primary Endpoint</strong> In vaccine trials, VE is
typically the <strong>primary efficacy endpoint</strong>, alongside key
<strong>safety endpoints</strong>.</p>
<ul>
<li>Trials may define a “case” not just based on lab-confirmed
infection, but also by <strong>symptom presentation</strong>.</li>
<li>This is particularly relevant when mild/asymptomatic cases are
common or underreported.</li>
</ul></li>
</ol>
</div>
<div id="vaccine-efficacy-hypothesis-into-binomial-testing-framework"
class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Vaccine Efficacy
Hypothesis into Binomial Testing Framework</h2>
<p><strong>Goal:</strong> Compare two hypotheses:</p>
<ul>
<li><strong>Null Hypothesis (H₀): VE ≤ 0.3</strong> (vaccine provides
low or no protection)</li>
<li><strong>Alternative Hypothesis (H₁): VE = 0.6</strong> (vaccine
provides strong protection)</li>
</ul>
<p><strong>Step 1: Define Probabilities</strong></p>
<ul>
<li>Let <strong>πₚₗₐcₑbₒ</strong> be the probability of getting
symptomatic COVID-19 in the <strong>placebo group</strong></li>
<li>Let <strong>πᵥₐccᵢₙₑ</strong> be the same for the <strong>vaccine
group</strong></li>
</ul>
<p><strong>Step 2: Conditional Probability for Vaccine Case</strong>
Derive the probability that a <strong>random case</strong> (symptomatic
COVID-19) comes from the vaccine group, under a given hypothesis
(VE):</p>
<p><span class="math display">\[
P_{H_x}(Vaccine\ Case\ |\ Case) = \frac{π_{Placebo}(1 -
VE)}{π_{Placebo}(1 - VE) + π_{Placebo}} = \frac{0.08(1 - VE)}{0.08(1 -
VE) + 0.08}
\]</span></p>
<p><strong>Step 3: Plug in VE under each hypothesis</strong></p>
<ul>
<li><p>Under <strong>H₀ (VE = 0.3)</strong>, the conditional probability
becomes:</p>
<p><span class="math display">\[
P_{H₀} = \frac{0.08(1 - 0.3)}{0.08(1 - 0.3) + 0.08} = 0.4118
\]</span></p></li>
<li><p>Under <strong>H₁ (VE = 0.6)</strong>:</p>
<p><span class="math display">\[
P_{H₁} = \frac{0.08(1 - 0.6)}{0.08(1 - 0.6) + 0.08} = 0.2857
\]</span></p></li>
</ul>
<p><strong>Step 4: Modeling the Number of Vaccine Cases</strong>
Assume:</p>
<ul>
<li><strong>N</strong> is the total number of symptomatic cases
observed.</li>
<li><strong>V</strong> is the number of those that are in the vaccine
group.</li>
</ul>
<p>Then:</p>
<ul>
<li><strong>V ~ Binomial(N, Pₕₓ)</strong> depending on which hypothesis
is true.</li>
<li><strong>1 – β = P(V ≥ c | H₁) = 0.9</strong>, meaning we want
<strong>90% power</strong> to detect a VE of 0.6 if it’s true.</li>
</ul>
</div>
<div id="adjusting-for-imbalance-in-pre-vaccination-state"
class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Adjusting for
Imbalance in Pre-Vaccination State</h2>
<p><strong>Imbalance in Pre-Vaccination State</strong></p>
<p>In some vaccine studies, participants already have <strong>non-zero
antibody levels before vaccination</strong>—this is common in diseases
requiring repeated immunizations, such as tetanus or pertussis. When
this happens, the <strong>post-vaccination antibody level</strong>
reflects both the <strong>effect of the vaccine</strong> and the
<strong>pre-existing immunity</strong>, which complicates comparisons
between vaccine groups.</p>
<p>If there’s an imbalance in pre-vaccination levels between groups,
then <strong>any difference in post-vaccination values</strong> might
partly be due to this initial difference—not the vaccine. In such cases,
the vaccine’s effect may be <strong>overestimated</strong>.</p>
<p>A common but flawed solution is to analyze <strong>fold
increases</strong> (post/pre ratios). However, this doesn’t solve the
problem. If <strong>pre- and post-vaccination levels are positively
correlated</strong>, then higher pre-vaccination values predict
both:</p>
<ul>
<li>higher post-vaccination values, and</li>
<li><strong>lower fold increases</strong> (because the baseline is
larger).</li>
</ul>
<p>This means that analyzing fold increases can <strong>introduce
bias</strong> in the opposite direction.</p>
<p>Though <strong>statistical theory doesn’t require balanced
baselines</strong>, correcting for imbalances <strong>increases
credibility</strong>. Randomization ensures balance <em>on average</em>,
but imbalances still arise due to chance.</p>
<hr />
<p><strong>Adjusting for Baseline Imbalance</strong></p>
<p>Two major methods are:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Stratification</strong>:</p>
<ul>
<li>Participants are grouped into strata based on baseline values.</li>
<li>In clinical practice, <em>pre-randomization stratification</em> is
rare due to logistical challenges (e.g., needing two visits: one for
blood draw and one for randomization).</li>
<li>A practical alternative is <strong>post-randomization
stratification</strong>, done at the analysis stage—often using
<strong>quartiles</strong> of baseline values.</li>
</ul></li>
<li><p><strong>Analysis of Covariance (ANCOVA)</strong>:</p>
<ul>
<li>Adjusts for baseline differences by modeling post-vaccination values
<strong>as a function of pre-vaccination levels</strong>.</li>
<li>This method assumes <strong>parallel regression lines</strong> for
each group and estimates the treatment effect as the <strong>vertical
gap between these lines</strong>.</li>
</ul></li>
</ol>
<hr />
<p>ANCOVA models the relationship between:</p>
<ul>
<li><strong>Y</strong>: log-transformed post-vaccination antibody
level</li>
<li><strong>X</strong>: log-transformed pre-vaccination level</li>
</ul>
<p>Using the model: <strong>Y = β₀ + β₁x + ε</strong></p>
<p>Here:</p>
<ul>
<li>β₀ is the intercept (mean post-vaccination level for seronegative
individuals),</li>
<li>β₁ is the slope (how much the log-post level increases with each
log-pre unit),</li>
<li>ε is the error term.</li>
</ul>
<p>The log-transformed outcome is used to maintain normality and allow
interpretation in terms of <strong>geometric means</strong>.</p>
<hr />
<p><strong>Addressing Heteroscedasticity</strong></p>
<p>Usually, ANCOVA assumes <strong>constant variance</strong>
(homoscedasticity). But <strong>variance decreases as pre-vaccination
levels increase</strong>—called <strong>heteroscedasticity</strong>.
Ignoring this can lead to <strong>incorrect standard errors and
p-values</strong>.</p>
<p>Solution: Model the variance as a function of <strong>x</strong> (the
pre-vaccination value). One such model:</p>
<p><strong>σ²ₓ = σ² (1 + c₁x + c₂x²)</strong> An alternative (easier to
fit in software):</p>
<p><strong>σ²ₓ = σ² exp(C₁x + C₂x²)</strong></p>
<p>This model guarantees non-negative variance and gives stable
estimates.</p>
<p>SAS Code to model heteroscedasticity:</p>
<pre class="sas"><code>proc mixed;
model y = x / solution;
repeated / local=exp(x x2);
run;</code></pre>
<p>This outputs estimates for the regression line and the variance
model. Comparing models under homoscedasticity vs heteroscedasticity
shows:</p>
<ul>
<li><strong>Slope and intercept</strong> are slightly different.</li>
<li><strong>Standard errors</strong> are more accurate under
heteroscedasticity.</li>
<li><strong>Seronegative-only estimates</strong> are less precise due to
smaller sample size.</li>
</ul>
</div>
<div id="equivalence-and-noninferiority-vaccine-trials"
class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Equivalence and
Noninferiority Vaccine Trials</h2>
<div
id="equivalence-and-noninferiority-trials-with-a-geometric-mean-response"
class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Equivalence and
Noninferiority Trials with a Geometric Mean Response</h3>
<p>In many vaccine trials, the <strong>geometric mean</strong> of
antibody levels (titre or concentration) is used as the outcome. The
main parameter of interest is the <strong>geometric mean ratio
(GMR)</strong>:</p>
<p> <strong>θ = e₁ / e₀</strong></p>
<p>Where:</p>
<ul>
<li><strong>e₁</strong> = geometric mean in the investigational
group</li>
<li><strong>e₀</strong> = geometric mean in the control group</li>
</ul>
<p>If you log-transform the data, equivalence in terms of GMR
<strong>θ</strong> translates into equivalence in terms of the
<strong>difference Δ</strong> of the log-transformed means:</p>
<p> <strong>Δ = log(e₁) − log(e₀) = log(θ)</strong></p>
<p>So, if your equivalence range for <strong>θ</strong> is [1/δ, δ],
then your equivalence range for <strong>Δ</strong> is [−log(δ),
+log(δ)]. The <strong>FDA/CBER</strong> often recommends using
<strong>0.67 to 1.5</strong> as a standard equivalence margin for
GMRs.</p>
<hr />
<p>Example: Combination Hepatitis A and B Vaccine Trial</p>
<ul>
<li><strong>Design</strong>: Combination vaccine vs. separate monovalent
vaccines</li>
<li><strong>Sample size</strong> (per protocol): 264 (combination), 269
(monovalent)</li>
</ul>
<p><strong>Hepatitis B (Anti-HBs):</strong></p>
<ul>
<li><p><strong>Geometric means</strong>:  Combination: 2099  Monovalent:
1871  GMR = 2099 / 1871 = <strong>1.12</strong></p></li>
<li><p><strong>Standard deviation of log titres</strong>:  log(6.8) =
1.92, log(9.5) = 2.25 → pooled SD = 2.09</p></li>
<li><p><strong>Standard error of difference</strong>:  SE = 2.09 ×
√(1/264 + 1/269) = 0.18</p></li>
<li><p><strong>Test statistic for equivalence</strong>:  Equivalence
margin (log scale) = ±log(1.5) = ±0.41  Observed difference = log(2099)
− log(1871) = 7.65 − 7.53 = 0.12  Z_EQ = (0.41 − |0.12|) / 0.18 = 1.61 →
p = 0.054 (not significant)</p></li>
<li><p><strong>95% CI for log(θ)</strong>:  0.12 ± 1.964 × 0.18 =
[−0.23, 0.47]  Exp back to θ scale → CI for GMR = [0.79, 1.60] → does
<strong>not</strong> lie within [0.67, 1.5]</p></li>
</ul>
<p> <strong>Conclusion</strong>: Equivalence <strong>not</strong>
demonstrated.</p>
<hr />
<p><strong>Hepatitis A (Anti-HAV):</strong></p>
<ul>
<li><p><strong>Geometric means</strong>:  Combination: 4756  Monovalent:
2948  GMR = 1.61</p></li>
<li><p><strong>Log means</strong>:  log(4756) = 8.47, log(2948) = 7.99
 SDs: log(3.1) = 1.13, log(2.5) = 0.92 → pooled SD = 1.03  SE =
0.09</p></li>
<li><p><strong>Lower bound of CI for Δ (noninferiority test)</strong>:
 Δ = 8.47 − 7.99 = 0.48  Lower bound = 0.48 − 1.964 × 0.09 = 0.30
 Back-transformed: exp(0.30) = <strong>1.35</strong></p></li>
</ul>
<p> <strong>Conclusion</strong>: Lower bound &gt; 0.67 →
<strong>noninferiority</strong> confirmed</p>
</div>
<div id="equivalence-and-noninferiority-with-seroresponse-rates"
class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Equivalence and
Noninferiority with Seroresponse Rates</h3>
<p>When the outcome is a <strong>seroconversion rate</strong> or
<strong>seroprotection rate</strong> (i.e., a binary outcome), the
parameter of interest is the <strong>risk difference (RD)</strong>.</p>
<ul>
<li><strong>Equivalence</strong>: Two-sided CI for RD must lie within
±margin</li>
<li><strong>Noninferiority</strong>: One-sided CI lower bound must be
above −margin</li>
<li><strong>Margins</strong>: Common values are ±5% or ±10%</li>
</ul>
<hr />
<p><strong>Example (Anti-HAV Seroconversion Rates):</strong></p>
<ul>
<li><p><strong>Observed rates</strong>:  Combination: 267 / 269 = 0.993
 Monovalent: 263 / 264 = 0.996  Risk difference = 0.993 − 0.996 =
<strong>−0.003</strong></p></li>
<li><p><strong>95% CI for RD</strong>: [−0.023, +0.014]  Equivalence
margin = [−0.05, +0.05]</p></li>
</ul>
<p> <strong>Conclusion</strong>: CI lies within margin →
<strong>equivalence confirmed</strong></p>
<hr />
<p><strong>SAS Implementation</strong></p>
<p>To perform the <strong>TOST (Two One-Sided Tests)</strong> approach
using the <strong>Wilson score method</strong>, use this code:</p>
<pre class="sas"><code>proc freq;
  table vaccine*seroconverted /
  riskdiff(equivalence method=fm margin=0.05) alpha=0.025;
run;</code></pre>
<ul>
<li>The output will give <strong>two P-values</strong> (one for each
one-sided test).</li>
<li>If both are &lt;0.025, <strong>equivalence is
concluded</strong>.</li>
<li>You can use trial-and-error with <code>margin=</code> to find exact
CI bounds.</li>
</ul>
</div>
</div>
<div id="vaccine-lot-consistency-trials" class="section level2"
number="3.5">
<h2><span class="header-section-number">3.5</span> Vaccine Lot
Consistency Trials</h2>
<p>Before licensing a vaccine, <strong>regulatory agencies</strong> like
the FDA (CBER) and EMA require proof that <strong>different lots
(batches)</strong> of the vaccine induce <strong>consistent
immunogenicity</strong>. This is evaluated via <strong>lot consistency
trials</strong>, which are a type of <strong>equivalence
study</strong>.</p>
<p>Trial Design</p>
<ul>
<li>Three <strong>consecutively produced</strong> vaccine lots are
used.</li>
<li>Participants are <strong>randomly assigned</strong> to one of the
three lots.</li>
<li>After vaccination, <strong>antibody titers</strong> are
measured.</li>
<li><strong>Lot consistency is demonstrated</strong> if all
<strong>pair-wise comparisons</strong> of post-vaccination
<strong>geometric mean titers (GMTs)</strong> lie within a
<strong>predefined equivalence range</strong>.</li>
</ul>
<table>
<colgroup>
<col width="19%" />
<col width="15%" />
<col width="42%" />
<col width="21%" />
</colgroup>
<thead>
<tr class="header">
<th>Method</th>
<th>Type</th>
<th>Key Feature</th>
<th>Decision Rule</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Confidence Interval</td>
<td>Estimation</td>
<td>Requires 3 pairwise GMR CIs within margin</td>
<td>CI ∈ [lower, upper]</td>
</tr>
<tr class="even">
<td>Wiens &amp; Iglewicz</td>
<td>Hypothesis Test</td>
<td>Uses min Z-statistic across pairs</td>
<td>Zmin &gt; critical value</td>
</tr>
</tbody>
</table>
<p>Both methods are valid and often give similar conclusions. The
<strong>CI method</strong> is simpler and more common. The <strong>Wiens
&amp; Iglewicz test</strong> is more formal but needs careful
consideration of sample size to control Type I error.</p>
<div id="confidence-interval-ci-method" class="section level3"
number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Confidence Interval
(CI) Method</h3>
<p>The most common method involves calculating <strong>two-sided 95%
confidence intervals</strong> for all <strong>three pairwise geometric
mean ratios (GMRs)</strong>.</p>
<p><strong>Conclusion:</strong> If <strong>all three</strong> CIs are
entirely within the pre-specified <strong>equivalence margin</strong>,
lot consistency is declared.</p>
<p><strong>Example – Influenza Vaccine Trial</strong></p>
<ul>
<li><p><strong>Three lots</strong> (n = 123, 123, 117)</p></li>
<li><p><strong>GMTs</strong> for A-H3N2 strain:</p>
<ul>
<li>Lot #1: 192.9 (mean log titer = 5.27)</li>
<li>Lot #2: 162.2 (mean log titer = 5.02)</li>
<li>Lot #3: 202.5 (mean log titer = 5.34)</li>
</ul></li>
<li><p><strong>Pooled SD</strong>: ~1.585</p></li>
<li><p><strong>Equivalence margin</strong> (in GMR): 0.35 to
2.83</p></li>
</ul>
<p>Calculations (example: Lot #2 vs Lot #1):</p>
<ul>
<li>Difference in log means: 5.02 − 5.27 = −0.25</li>
<li>Standard error: √(1.585² × 2 / 123) ≈ 0.203</li>
<li>CI for difference: −0.25 ± 1.97 × 0.203 → [−0.648, 0.148]</li>
<li>Exponentiating → GMR CI: [0.638, 1.108]</li>
</ul>
<p><strong>Results for all pairs</strong>:</p>
<ul>
<li>Lot #2 vs Lot #1: [0.638, 1.108]</li>
<li>Lot #3 vs Lot #1: [0.796, 1.384]</li>
<li>Lot #3 vs Lot #2: [0.944, 1.651]</li>
</ul>
<p><strong>Conclusion</strong>: All three intervals fall within [0.35,
2.83] → <strong>Lot consistency confirmed</strong>.</p>
</div>
<div id="wiens-and-iglewicz-test" class="section level3" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Wiens and Iglewicz
Test</h3>
<p>This method treats lot consistency as a <strong>formal hypothesis
test</strong>:</p>
<p><strong>Hypotheses:</strong></p>
<ul>
<li><strong>H0</strong>: At least one pairwise difference in log means ≥
equivalence margin (in absolute value)</li>
<li><strong>H1</strong>: All pairwise differences in log means &lt;
equivalence margin</li>
</ul>
<p><strong>Test Statistic:</strong></p>
<p>For each pairwise comparison (i, j), define:</p>
<ul>
<li>Difference: Dij = x̄j − x̄i</li>
<li>SE of Dij: SE(Dij)</li>
<li>Equivalence margin: δ (in log-scale)</li>
</ul>
<p>Then compute:</p>
<p> <strong>Zmin = min over pairs of { (δ − |Dij|) / SE(Dij)
}</strong></p>
<ul>
<li>Zmin is a <strong>one-sided</strong> test statistic.</li>
<li>If Zmin exceeds the critical value (e.g., z₀.975 = 1.96),
<strong>reject H0</strong> → lot consistency.</li>
</ul>
<p><strong>Example (continued):</strong></p>
<ul>
<li><p>Differences:</p>
<ul>
<li>D12 = −0.25</li>
<li>D13 = 0.07</li>
<li>D23 = 0.32</li>
</ul></li>
<li><p>SEs:</p>
<ul>
<li>SE(D12) ≈ 0.202</li>
<li>SE(D13) ≈ 0.203</li>
<li>SE(D23) ≈ 0.205</li>
</ul></li>
<li><p>δ = log(2.83) ≈ 1.5</p></li>
</ul>
<p>Then:</p>
<p>Zmin = min of</p>
<ul>
<li>(1.5 − 0.25)/0.202 ≈ 6.19</li>
<li>(1.5 − 0.07)/0.203 ≈ 7.03</li>
<li>(1.5 − 0.32)/0.205 ≈ 5.76</li>
</ul>
<p><strong>Zmin = 5.76</strong> → exceeds critical value →
<strong>reject H0</strong> → lot consistency.</p>
<hr />
<p><strong>Type I Error Control</strong></p>
<p>The Wiens &amp; Iglewicz test is <strong>not conservative</strong> by
default. To ensure Type I error is controlled, the sample size must be
sufficiently large relative to δ/σ (equivalence margin to SD ratio).</p>
<p>Rule of thumb (based on equation 6.4):</p>
<p> <strong>n &gt; 50 / (δ / σ)²</strong></p>
<p>Example:</p>
<ul>
<li>If δ = 1.5, σ = 1.6 → δ/σ = 0.94</li>
<li>Then required n &gt; 50 / (0.94)² ≈ 57 → met in example</li>
</ul>
<p><strong>Simulation studies</strong> confirm:</p>
<ul>
<li>Type I error is lowest when the ratio of mean differences is 0 or
1</li>
<li>Highest when the ratio is ~0.5</li>
</ul>
</div>
</div>
</div>
<div id="sample-size-estimation" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Sample Size
Estimation</h1>
<div id="vaccine-sse-methods" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Vaccine SSE
Methods</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Confidence Interval for VE</strong></p>
<ul>
<li>Based on classical <strong>case-control</strong> or
<strong>cohort</strong> designs.</li>
<li>Reference: <strong>O’Neill (1988)</strong>.</li>
<li>Focused on estimating VE along with its confidence interval to
quantify uncertainty.</li>
</ul></li>
<li><p><strong>One Proportion Tests</strong></p>
<ul>
<li>Applied when comparing observed event proportion against a known
benchmark (e.g., historical attack rate).</li>
<li>Often used in <strong>event-driven designs</strong> (study continues
until a set number of events occurs).</li>
<li>Uses standard binomial tests.</li>
</ul></li>
<li><p><strong>Two Proportion Tests</strong></p>
<ul>
<li>Directly compares attack rates between vaccinated and control
groups.</li>
<li>Computes VE using the formula: <strong>VE = 1 – RR</strong>, and
then applies tests like the <strong>z-test for
proportions</strong>.</li>
</ul></li>
<li><p><strong>Composite Models</strong></p>
<ul>
<li>Consider multiple outcomes (e.g., disease <strong>incidence and
severity</strong>) together.</li>
<li>Reference: <strong>Callegaro (2020)</strong>.</li>
<li>Used to account for the <strong>public health relevance</strong> of
various symptoms, not just whether infection occurred.</li>
</ul></li>
<li><p><strong>Count Models (Poisson/Negative Binomial)</strong></p>
<ul>
<li>Appropriate when tracking <strong>event counts over time</strong>
(e.g., number of infections/person-time).</li>
<li>Poisson models work when events are rare and follow a constant
rate.</li>
<li>Negative Binomial (NB) used when there is overdispersion (more
variability than Poisson allows).</li>
<li>Applies both to standard and rare-event settings.</li>
</ul></li>
<li><p><strong>Cox Regression</strong></p>
<ul>
<li>A <strong>time-to-event model</strong> commonly used for survival
analysis.</li>
<li>Used in <strong>non-inferiority (NI)</strong> or <strong>superiority
(SM)</strong> hypotheses testing.</li>
<li>Can adjust for covariates and censoring (e.g., if participants drop
out or are uninfected at study end).</li>
</ul></li>
<li><p><strong>Cluster Randomized Designs</strong></p>
<ul>
<li>Applied when randomizing <strong>groups</strong> (e.g., households,
villages) instead of individuals.</li>
<li>Often used in <strong>public health</strong> or <strong>epidemic
settings</strong>.</li>
<li>Statistical methods account for intra-cluster correlation.</li>
<li>Reference: <strong>Hayes &amp; Bennett (1999)</strong>.</li>
</ul></li>
</ol>
<hr />
<p><img src="./02_Plots/Vaccine/Vaccine-COVID19Example.png" /></p>
</div>
<div id="ssd-consideration" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> SSD
Consideration</h2>
<div id="comparing-two-geometric-mean-responses" class="section level3"
number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Comparing Two
Geometric Mean Responses</h3>
<p>There are many formulas for estimating the sample size in
parallel-group trials with Normally distributed data. The simplest—but
least accurate—uses the Normal approximation and can be found in any
basic statistics textbook. The most accurate approach, which involves no
approximation, is based on the noncentral t-distribution, as shown in
Formula of Julious’s book [20]. Sample sizes using this method can be
estimated using the <code>POWER</code> procedure in SAS.</p>
<p>Two parameters must be specified:</p>
<ul>
<li><strong>Δ</strong>: the difference in log-transformed geometric
means (i.e., log(GMR)).</li>
<li><strong>σ</strong>: the within-group standard deviation of the
log-transformed immunogenicity values.</li>
</ul>
<p>The value of σ can be obtained from previous trials. Alternatively, a
conservative estimate can be calculated as:</p>
<p><strong>4σ ≈ log(largest expected immunogenicity value) − log(lowest
expected immunogenicity value)</strong></p>
<p>This is based on the rule that approximately 95% of Normally
distributed data falls within the range <strong>μ ± 2σ</strong>. To get
a less conservative estimate, the multiplier 4 can be replaced with
6.</p>
<p><strong>Example</strong>: Consider a clinical trial comparing a
licensed influenza vaccine to a new investigational vaccine. The primary
endpoint is the antibody response measured by the HI test. Suppose the
investigator expects the new vaccine to be substantially more
immunogenic, with a geometric mean ratio (GMR) of 2.0. In such trials,
HI titers &gt;5120 are rare, and the lowest measurable titer is usually
5. Therefore:</p>
<ul>
<li>Δ = logₑ(2.0) ≈ 0.69</li>
<li>σ = [logₑ(5120) − logₑ(5)] / 4 ≈ 1.73</li>
</ul>
<p>The SAS code to estimate the required number of subjects per group to
achieve a power of 0.9 is:</p>
<pre class="sas"><code>proc power;
  meandiff=0.69 stddev=1.73
  power=0.9 npergroup=.;
run;</code></pre>
<p><strong>Output</strong>: 134 subjects per group (268 total) to
achieve ~90% power.</p>
</div>
<div id="comparing-two-proportions-1" class="section level3"
number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Comparing Two
Proportions</h3>
<p>There are also many formulas for estimating the sample size required
to compare two proportions. Most are asymptotic; detailed discussions
can be found in Chapter 4 of Fleiss, Levin, and Paik [21]. The
recommended formulas—(4.14) for equal group sizes and (4.19) for unequal
sizes—are implemented in the SAS <code>POWER</code> procedure.</p>
<p>For exact power calculations for 2x2 tables, see the work by Hirji et
al. [22].</p>
<p><strong>Example</strong>: Now suppose the investigator wants to
compare seroprotection rates instead of GMTs and expects the rate to
increase from 0.85 to 0.95. The required sample size for 90% power is
computed using:</p>
<pre class="sas"><code>proc power;
  twosamplefreq test=pchi
  groupproportions=(0.85,0.95)
  power=0.9 npergroup=.;
run;</code></pre>
<p><strong>Output</strong>: 188 subjects per group to achieve ~90%
power.</p>
</div>
<div
id="sample-size-estimation-for-trials-with-multiple-co-primary-endpoints"
class="section level3" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Sample Size
Estimation for Trials with Multiple Co-Primary Endpoints</h3>
<p>Estimating power for a trial with multiple co-primary endpoints is
complex. Power is highly dependent on the <strong>correlation between
endpoints</strong>. If the endpoints are strongly correlated, power will
be higher; if weakly correlated, it will be lower. Unfortunately, these
correlations are often unknown in practice.</p>
<p>If the objective is to <strong>demonstrate significance for all k
co-primary endpoints</strong>, a commonly used lower bound for overall
power <strong>P</strong> is:</p>
<p><strong>P ≥ Π (Pᵢ), for i = 1 to k</strong> — assuming all endpoints
are non-negatively correlated.</p>
<p>If you <strong>don’t</strong> want to assume anything about
correlation, you can use this alternative bound:</p>
<p><strong>P ≥ Σ (Pᵢ) − (k − 1)</strong></p>
<p>If k is large, then each individual power Pᵢ must be close to 1.0 to
maintain a global power of ≥ 0.8. To achieve 90% overall power, each Pᵢ
should satisfy:</p>
<p><strong>Pᵢ ≥ (0.9)^(1/k)</strong></p>
<p>For example, if k = 5, each endpoint’s power should be at least
~0.979.</p>
<p><strong>Example 3.8</strong>: A comparative trial has two co-primary
endpoints: geometric mean concentration and seroprotection rate. A total
of 2,150 subjects is planned. Using SAS code from earlier sections:</p>
<ul>
<li>Power for endpoint 1 = 0.93</li>
<li>Power for endpoint 2 = 0.90</li>
</ul>
<p>Under the assumption of non-negative correlation:</p>
<p><strong>P ≥ 0.93 × 0.90 = 0.837</strong></p>
<p>Using Equation above instead:</p>
<p><strong>P ≥ 0.93 + 0.90 − 1 = 0.83</strong></p>
<hr />
<p><strong>Monte Carlo Simulation Alternative</strong> If prior trial
data are available, power can be estimated using simulation. For
example:</p>
<ul>
<li>Simulate 5,000+ trials by drawing random samples of size
<strong>n</strong> with replacement.</li>
<li>Count how many simulated trials achieve success on
<strong>both</strong> endpoints (e.g., seroprotection and seroconversion
rates exceed predefined thresholds).</li>
<li>The proportion of “successful” simulations estimates global power
for sample size <strong>n</strong>.</li>
</ul>
<p>This method offers flexibility and accounts for complex correlation
structures.</p>
</div>
</div>
<div id="comparing-two-geometric-mean-responses-1"
class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Comparing Two
Geometric Mean Responses</h2>
<p>When designing a vaccine trial where the <strong>primary
endpoint</strong> is either <strong>equivalence</strong> or
<strong>noninferiority</strong> of immunogenicity based on a
<strong>geometric mean response</strong>—such as a geometric mean titer
(GMT) or concentration (GMC)—the statistical power of the trial depends
on several factors:</p>
<ol style="list-style-type: decimal">
<li>The <strong>equivalence or noninferiority margin</strong> (denoted
as delta, δ). This defines how close the responses from the two vaccine
groups must be for the vaccines to be considered equivalent or for the
new vaccine to be considered not inferior.</li>
<li>The <strong>within-group standard deviation</strong> (denoted as
sigma, σ) of the <strong>log-transformed</strong> immunogenicity
data.</li>
<li>The <strong>true difference</strong> between the mean
log-transformed immunogenicity values of the two groups, delta = μ1 −
μ0, where μ1 is the mean for the investigational vaccine and μ0 for the
control.</li>
</ol>
<p>The power is especially sensitive to the value of this true
difference (Δ). If we assume no difference between groups (Δ = 0), the
required sample size may be underestimated if in reality there is a
small difference between them. Therefore, it’s advisable not to assume Δ
= 0 but to include plausible deviations in planning.</p>
<p><strong>Assumption and Distribution</strong></p>
<p>It is generally assumed that the <strong>log-transformed
immunogenicity values follow a Normal distribution</strong>. Power and
sample size are then calculated based on <strong>noncentral
t-distributions</strong>.</p>
<p><strong>Equivalence Trial Power Estimation (Example)</strong></p>
<p>In the hepatitis B vaccine trial example, the investigator sets:</p>
<ul>
<li>Equivalence margin on the log scale: δ = log(1.5) ≈ 0.41</li>
<li>Within-group standard deviation: σ = 2.0</li>
<li>Assumed true difference Δ = 0</li>
<li>Desired power: 0.90</li>
<li>Significance level (one-sided alpha): 0.025</li>
</ul>
<p>SAS Code (for estimating required sample size when Δ = 0):</p>
<pre><code>proc power;
twosamplemeans test=equiv_diff alpha=0.025
meandiff=0 stddev=2.0
lower=-0.41 upper=0.41
power=0.9 npergroup=.;
run;</code></pre>
<p>Result: 620 subjects per group are needed to achieve 90% power
assuming no difference between vaccines.</p>
<p><strong>Robustness Check</strong></p>
<p>To evaluate how sensitive this estimate is, the investigator also
calculates power under small deviations from equality:</p>
<ul>
<li>Case 1: Geometric mean ratio (GMR) = 0.95 → Δ = log(0.95) ≈
-0.051</li>
<li>Case 2: GMR = 0.90 → Δ = log(0.90) ≈ -0.105</li>
</ul>
<p>SAS Code:</p>
<pre><code>proc power;
twosamplemeans test=equiv_diff alpha=0.025
meandiff=0, -0.051, -0.105
stddev=2.0
lower=-0.41 upper= 0.41
npergroup=620 power=.;
run;</code></pre>
<p>Output:</p>
<ul>
<li>Power at Δ = 0: 0.900</li>
<li>Power at Δ = −0.051: 0.866</li>
<li>Power at Δ = −0.105: 0.760</li>
</ul>
<p>Interpretation: A small deviation from equality (Δ = −0.105 or GMR =
0.90) reduces the power to 76%. To maintain 90% power under this
assumption, a larger sample size would be needed—906 subjects per
group.</p>
<p><strong>Noninferiority Trial Power Estimation</strong></p>
<p>Now, suppose the investigator wants to know the sample size for
demonstrating <strong>noninferiority</strong> of the combination vaccine
(hepatitis A) to the monovalent one.</p>
<p>Given:</p>
<ul>
<li>Standard deviation: σ = 1.25</li>
<li>Assumed true difference Δ = 0</li>
<li>Noninferiority margin (lower bound): −0.41</li>
<li>Upper bound set to a large value (e.g., 999) since the focus is
one-sided</li>
<li>Desired power: 0.90</li>
</ul>
<p>SAS Code:</p>
<pre><code>proc power;
twosamplemeans test=equiv_diff alpha=0.025
meandiff=0 stddev=1.25
lower=-0.41 upper=999
power=0.9 npergroup=.;
run;</code></pre>
<p>Result: The required sample size is 197 subjects per group.</p>
<p><strong>Conclusion</strong></p>
<ul>
<li>Power calculations for equivalence and noninferiority trials
involving geometric means should <strong>not assume zero
difference</strong> blindly.</li>
<li>Assumptions about <strong>standard deviation</strong> and
<strong>expected difference</strong> (Δ) critically affect the sample
size.</li>
<li>Simulation or sensitivity analysis under multiple scenarios is
highly recommended during trial planning.</li>
</ul>
</div>
<div id="comparing-two-seroresponse-rates" class="section level2"
number="4.4">
<h2><span class="header-section-number">4.4</span> Comparing Two
Seroresponse Rates</h2>
<p>When planning a <strong>noninferiority clinical vaccine
trial</strong> with the endpoint being a <strong>seroprotection
rate</strong> or <strong>seroconversion rate</strong> (i.e., comparing
two proportions), the <strong>risk difference</strong> is often used as
the effect measure. In this context, the key objective is to demonstrate
that the investigational vaccine is not unacceptably worse than the
control vaccine by more than a specified <strong>noninferiority
margin</strong> (δ).</p>
<hr />
<p><strong>Form of the Hypothesis</strong></p>
<p>In a noninferiority setting, the null hypothesis typically tested
is:</p>
<p><strong>H₀: π₁ − π₀ ≤ −δ</strong> vs. <strong>H₁: π₁ − π₀ &gt;
−δ</strong></p>
<p>where:</p>
<ul>
<li>π₁ is the seroprotection (or seroconversion) rate of the
<strong>investigational</strong> vaccine</li>
<li>π₀ is the rate of the <strong>control</strong> vaccine</li>
<li>δ is the <strong>noninferiority margin</strong> (e.g., −0.1 means
the new vaccine cannot be worse by more than 10%)</li>
</ul>
<hr />
<p><strong>Sample Size Estimation Using SAS POWER Procedure</strong></p>
<p>To estimate the required number of subjects per group that ensures
adequate <strong>statistical power</strong>, you can use the
<code>PROC POWER</code> procedure in SAS. Here’s an example:</p>
<p><strong>Scenario:</strong></p>
<ul>
<li>Investigational vaccine seroprotection rate (π₁) = 70%</li>
<li>Control vaccine rate (π₀) = 75%</li>
<li>Noninferiority margin δ = −0.1</li>
<li>Desired power = 90%</li>
<li>One-sided alpha = 0.025</li>
</ul>
<p><strong>SAS Code:</strong></p>
<pre><code>proc power;
twosamplefreq test=pchi
alpha=0.025 sides=1
groupproportions=(0.75 0.70)
nullpdifference=-0.10
power=0.9 npergroup=.;
run;</code></pre>
<p><strong>Result:</strong> The computed required sample size is
<strong>1,674 subjects per group</strong>.</p>
<p>This result assumes that the <strong>difference in
proportions</strong> (π₁ − π₀ = −0.05) still lies <strong>within the
noninferiority bound</strong> of −0.1. If the true difference were
larger (e.g., −0.09), the required sample size would increase to
maintain the same power.</p>
<hr />
<p><strong>Equivalence Trials (Two-Sided)</strong></p>
<p>For <strong>equivalence trials</strong>, the hypothesis is
two-sided:</p>
<p><strong>H₀: |π₁ − π₀| ≥ δ</strong> vs. <strong>H₁: |π₁ − π₀| &lt;
δ</strong></p>
<p>This means the new vaccine must not differ from the control by more
than δ in either direction.</p>
<p>While SAS <code>PROC POWER</code> does <strong>not support
equivalence tests for proportions directly</strong>, the sample size can
still be calculated using <strong>Formula 12.4</strong> from Julious
(2004), or programmed manually.</p>
<hr />
<p><strong>Key Takeaways:</strong></p>
<ul>
<li><p>Sample size for comparing two seroresponse rates is highly
sensitive to:</p>
<ul>
<li>The expected rates</li>
<li>The noninferiority or equivalence margin</li>
<li>The significance level (alpha)</li>
<li>Desired power (commonly 80% or 90%)</li>
</ul></li>
<li><p>For noninferiority: <code>PROC POWER</code> with
<code>twosamplefreq</code> and <code>test=pchi</code> works
well.</p></li>
<li><p>For equivalence: Manual programming or external tools are
required, as SAS does not have a built-in equivalence test for
proportions using risk differences.</p></li>
</ul>
</div>
<div id="lot-consistency-trials" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Lot Consistency
Trials</h2>
<p>Lot consistency trials aim to demonstrate that different
manufacturing batches (lots) of a vaccine produce equivalent immune
responses. Typically, <strong>three lots</strong> are compared in a
randomized design using post-vaccination immunogenicity (e.g., geometric
mean titres of antibodies). The analysis usually involves <strong>three
pair-wise comparisons</strong>: Lot 1 vs Lot 2, Lot 1 vs Lot 3, and Lot
2 vs Lot 3.</p>
<hr />
<p><strong>Standard Approach to Power Estimation</strong></p>
<p>The most common approach is:</p>
<ol style="list-style-type: decimal">
<li><strong>Estimate power separately</strong> for each pair-wise
comparison.</li>
<li><strong>Combine the results</strong> to approximate the
<strong>overall power</strong>.</li>
</ol>
<p>Suppose each comparison is evaluated with an <strong>equivalence
range</strong>, often in the form of geometric mean ratio (e.g., from
1/1.5 to 1.5, which equals log2 bounds of ±0.585).</p>
<p>If all three confidence intervals fall within this range, lot
consistency is concluded.</p>
<hr />
<p><strong>Example Summary</strong></p>
<p>In a meningococcal vaccine lot consistency study:</p>
<ul>
<li>Assume <strong>standard deviation (SD)</strong> of log2-transformed
titres: √6.15 ≈ 2.48.</li>
<li>Sample size: <strong>500 subjects per lot</strong>.</li>
<li>Equivalence margin: ±0.585 (i.e., log2(1.5)).</li>
<li>Each pair-wise test has identical setup → individual power per
comparison: <strong>0.963</strong>.</li>
<li>Assuming <strong>independence</strong>, overall power: 0.963³ ≈
<strong>0.893</strong>.</li>
<li>More conservative combination: average power minus overlap → around
<strong>0.889</strong>.</li>
</ul>
<hr />
<p><strong>Impact of Between-Lot Differences (δ12 and δ13)</strong></p>
<p>Power drops if you assume the lots have <strong>some
variation</strong> (i.e., δ12 and δ13 are nonzero). For example:</p>
<ul>
<li><p>If δ13 = 0.10 and δ12 = 0.06:</p>
<ul>
<li>Power(Lot1 vs Lot2): 0.949</li>
<li>Power(Lot1 vs Lot3): 0.923</li>
<li>Power(Lot2 vs Lot3): 0.956</li>
</ul></li>
<li><p>Overall power:</p>
<ul>
<li>Conservative: (0.949 + 0.923 + 0.956)/2 =
<strong>0.828</strong></li>
<li>Independence assumption: 0.949 × 0.923 × 0.956 ≈
<strong>0.837</strong></li>
</ul></li>
</ul>
<hr />
<p><strong>Dependence Between Comparisons</strong></p>
<p>The three comparisons are <strong>not independent</strong>, since the
difference between Lot2 and Lot3 can be inferred if Lot1 vs Lot2 and
Lot1 vs Lot3 are known. Therefore:</p>
<ul>
<li>Assuming independence <strong>overestimates</strong> power
slightly.</li>
<li>Standard approach introduces minor bias but is often
<strong>acceptable</strong> in practice.</li>
</ul>
<hr />
<p><strong>Alternative: Monte Carlo Simulation</strong></p>
<p>To address the correlation properly:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Simulate</strong> 5,000+ trials.</p></li>
<li><p>For each simulation:</p>
<ul>
<li><p>Draw three samples of size <em>n</em>:</p>
<ul>
<li>Lot1: from N(0, σ²)</li>
<li>Lot2: from N(δ12, σ²)</li>
<li>Lot3: from N(δ13, σ²)</li>
</ul></li>
<li><p>Calculate three confidence intervals for geometric mean
ratios.</p></li>
<li><p>Check if all intervals fall within the equivalence
range.</p></li>
</ul></li>
<li><p><strong>Overall power</strong> = proportion of simulated trials
that meet all three criteria.</p></li>
</ol>
<p>For example, with:</p>
<ul>
<li>σ = 2.48</li>
<li>δ13 = 0.10, δ12 = 0.06</li>
<li>n = 500 per group</li>
</ul>
<p>The <strong>simulated power</strong> = <strong>0.887</strong></p>
<hr />
<p><strong>Step 1: Power for Each Pairwise Comparison (δ =
0)</strong></p>
<pre class="sas"><code>proc power;
  twosamplemeans test=equiv_diff
  alpha = 0.05
  meandiff = 0
  stddev = 2.48
  lower = -0.585  /* log2(1/1.5) */
  upper = 0.585   /* log2(1.5) */
  npergroup = 500
  power = .;
run;</code></pre>
<p>This gives the <strong>power for each pairwise test</strong> when
assuming no between-lot difference (δ = 0). If power ≈ 0.963, the
<strong>approximate overall power</strong> can be:</p>
<ul>
<li>Conservative: (0.963 + 0.963 + 0.963)/2 ≈ 0.889</li>
<li>Assuming independence: 0.963³ ≈ 0.893</li>
</ul>
<hr />
<p><strong>Step 2: Power for Pairwise Comparisons with Small Differences
(δ ≠ 0)</strong></p>
<pre class="sas"><code>proc power;
  twosamplemeans test=equiv_diff
  alpha = 0.05
  meandiff = 0.10 0.06 0.04
  stddev = 2.48
  lower = -0.585
  upper = 0.585
  npergroup = 500
  power = .;
run;</code></pre>
<p>This code estimates the power for:</p>
<ul>
<li>Lot1 vs Lot3: δ = 0.10</li>
<li>Lot1 vs Lot2: δ = 0.06</li>
<li>Lot2 vs Lot3: δ = 0.04</li>
</ul>
<p>Then combine the results:</p>
<ul>
<li>Conservative: (Power1 + Power2 + Power3)/2</li>
<li>Independent: Power1 × Power2 × Power3</li>
</ul>
<hr />
<p><strong>Optional: Sample Size Estimation for Fixed Power</strong></p>
<p>You can also compute the <strong>required sample size</strong> per
lot for a desired power (e.g., 90%):</p>
<pre class="sas"><code>proc power;
  twosamplemeans test=equiv_diff
  alpha = 0.05
  meandiff = 0
  stddev = 2.48
  lower = -0.585
  upper = 0.585
  power = 0.90
  npergroup = .;
run;</code></pre>
<hr />
<p><strong>Conclusion</strong></p>
<ul>
<li>Power in lot consistency trials is sensitive to assumptions about
<strong>between-lot variability</strong>.</li>
<li><strong>Assuming no variability (δ = 0)</strong> often leads to
<strong>overly optimistic power</strong>.</li>
<li>The <strong>standard approximation method</strong> (individual
pair-wise power + adjustment) is practical but not exact.</li>
<li><strong>Monte Carlo simulation</strong> is preferred when precise
power estimation is needed and between-lot differences are
expected.</li>
</ul>
</div>
</div>
<div id="vaccine-safety" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Vaccine Safety</h1>
<div id="introduction-1" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Introduction</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Focus on Adverse Events</strong> Vaccine safety revolves
around the monitoring of <strong>adverse events</strong> (AEs) that
occur after vaccination. These events could range from mild
(e.g. injection site reactions) to severe or life-threatening
(e.g. anaphylaxis or myocarditis). Because vaccines are administered to
healthy people, including children and vulnerable groups, safety is held
to an especially high standard.</p></li>
<li><p><strong>Primary Safety Endpoints</strong> During clinical trials
(especially Phase III), specific <strong>safety endpoints</strong> are
defined—typically focusing on <strong>severe adverse events
(SAEs)</strong>. These endpoints are carefully monitored in parallel
with efficacy.</p></li>
<li><p><strong>Post-Marketing Surveillance</strong> Some AEs may be too
rare to detect even in large trials (e.g., with frequencies of less than
1 in 10,000). These events are captured through <strong>post-market
surveillance</strong>, which continues after the vaccine is approved and
in widespread use.</p>
<ul>
<li>This includes <strong>Phase IV studies</strong> and ongoing data
collection from <strong>vaccine safety networks</strong> (e.g., VAERS in
the U.S.).</li>
<li>The goal is to identify <strong>unexpected or delayed
effects</strong> that only become evident with broader population
exposure.</li>
</ul></li>
<li><p><strong>Use of Self-Controlled Studies</strong></p>
<p>There’s a practical and ethical limitation post-approval:</p>
<blockquote>
<p>You can’t maintain a randomized control group once a vaccine is
proven effective, since it would be unethical to deny the vaccine to
people who need it.</p>
</blockquote>
<p>As a result, researchers rely on <strong>observational
methods</strong> that don’t require a control group — this is where
<strong>SCCS</strong> comes in.</p>
<p>Among the most important tools in post-marketing safety surveillance
are <strong>self-controlled designs</strong>, such as the
<strong>Self-Controlled Case Series (SCCS)</strong>. These are useful
for evaluating whether the rate of adverse events increases shortly
after vaccination compared to other periods in the same
individual.</p></li>
<li><p><strong>Phases of Clinical Safety Evaluation</strong>:</p>
<ul>
<li><strong>Phase I</strong>: Early exploration of safety (no
statistical testing usually).</li>
<li><strong>Phase II</strong>: Measures the frequency of
<strong>local/systemic reactions</strong> and lab abnormalities.</li>
<li><strong>Phase III</strong>: Focuses on <strong>less common but still
expected adverse events</strong>.</li>
<li><strong>Post-licensure</strong>: Attention shifts to <strong>rare
and serious adverse events</strong>, often using <strong>large
surveillance systems</strong> like the <strong>CDC’s Vaccine Safety
Datalink</strong>.</li>
<li>Even a <strong>small increase in risk</strong> can lead to a
<strong>large number of additional cases</strong> when vaccines are
given to large populations.</li>
<li>For example, if a condition naturally occurs in 0.1% of people and
the vaccine raises this to 0.2%, that means <strong>1,000 extra cases
per million people vaccinated</strong>.</li>
</ul></li>
</ol>
</div>
<div id="vaccine-safety-surveillance" class="section level2"
number="5.2">
<h2><span class="header-section-number">5.2</span> Vaccine Safety
Surveillance</h2>
<p>Section 9.2: Vaccine Safety Surveillance – Summary Explanation</p>
<p>Vaccine safety surveillance is essential after a vaccine is licensed
and released for public use. Its main function is to detect rare or
serious adverse events that may not have been identified during
pre-licensure clinical trials.</p>
<p><strong>Main concepts</strong></p>
<ol style="list-style-type: decimal">
<li>Reporting systems</li>
</ol>
<p>Vaccine manufacturers are required to report all serious adverse
events to regulatory authorities. In addition, systems like the U.S.
Vaccine Adverse Event Reporting System (VAERS) rely on voluntary
reporting from healthcare providers and the public.</p>
<p>Strength: The system can detect very rare but serious adverse events
linked to vaccines.</p>
<p>Weaknesses: Reports may be incomplete, inconsistent, or biased. There
is no unvaccinated control group. Events may be coincidental, and
causality cannot be confirmed. True incidence rates cannot be calculated
from such data due to underreporting and lack of denominator
information.</p>
<ol start="2" style="list-style-type: decimal">
<li>Coincidental events</li>
</ol>
<p>Some adverse events may occur naturally and appear to be linked to
vaccination purely by coincidence. For example, Sudden Infant Death
Syndrome (SIDS) affects about 1 in 1,300 infants. Based on timing,
50–100 SIDS cases might occur shortly after vaccination each year, even
if vaccines are not the cause. This illustrates the challenge of
distinguishing correlation from causation.</p>
<ul>
<li>Example: Menactra and Guillain-Barré Syndrome (GBS): After the
meningococcal vaccine Menactra was introduced in 2005, five cases of GBS
were reported in teens aged 17–18. GBS is a neurological condition that
may lead to temporary or long-term paralysis. While GBS can be caused by
infections or other factors, post-licensure surveillance did not
establish a causal link between Menactra and GBS.</li>
<li>Example: Rotashield and intussusception: Rotashield, a rotavirus
vaccine approved in 1998, was withdrawn shortly after approval due to
cases of intussusception, a serious bowel obstruction. VAERS reported 76
cases, 70% of which occurred after the first dose. The estimated risk
was one in every 5,000–9,500 vaccinated infants. Despite its potential
to save lives in developing countries, the vaccine’s withdrawal in the
U.S. prevented its global use.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Limitations of statistical analysis</li>
</ol>
<p>Because surveillance data are observational and subject to many
biases, standard statistical tests like p-values and confidence
intervals are not recommended for assessing vaccine risk based solely on
this data. Ideally, safety signals should be confirmed through
randomized controlled trials.</p>
<ol start="4" style="list-style-type: decimal">
<li>Self-controlled case series method</li>
</ol>
<p>This is a specialized statistical method that analyzes only
individuals who experienced an adverse event (cases). It estimates the
relative incidence of the event by comparing risk periods and non-risk
periods within the same individual. The model assumes
Poisson-distributed event counts and controls for fixed confounders like
sex or genetics.</p>
<p>Example: The method was used in the UK to confirm a link between the
Urabe strain mumps vaccine and aseptic meningitis. As a result, vaccines
using this strain were withdrawn.</p>
</div>
<div id="self-controlled-case-studies" class="section level2"
number="5.3">
<h2><span class="header-section-number">5.3</span> Self Controlled Case
Studies</h2>
<ol style="list-style-type: decimal">
<li><p><strong>Overview</strong> SCCS is a
<strong>within-person</strong> comparison method that estimates the
<strong>relative incidence (RI)</strong> of an adverse event during
defined risk windows (e.g., 0–28 days post-vaccination) versus control
periods.</p>
<ul>
<li>It uses <strong>only individuals who experienced the adverse
event</strong> (i.e., case-only design).</li>
<li>The benefit is that <strong>time-invariant confounders</strong>
(e.g., genetics, baseline health status) are automatically controlled
for, since each person serves as their own control.</li>
</ul></li>
<li><p><strong>How It Works</strong> The observation period for each
case is divided into:</p>
<ul>
<li><strong>Risk period(s)</strong>: When the event is suspected to be
more likely due to vaccination.</li>
<li><strong>Control period(s)</strong>: Baseline period when the event
risk is assumed to be unaffected by the vaccine.</li>
</ul>
<p>The <strong>RI</strong> is the ratio of event incidence in the risk
period vs. the control period.</p></li>
<li><p><strong>Statistical Methods</strong> Estimation methods
include:</p>
<ul>
<li><strong>Likelihood functions</strong> based on binomial models.</li>
<li>Signed root likelihood ratios and log-relative incidence
estimates.</li>
<li>Modifications for <strong>age effects</strong>, which are common
confounders in vaccine safety.</li>
</ul>
<p>The formula for RI without age effects is based on a likelihood
function <span class="math inline">\(l(\rho)\)</span>, where:</p>
<ul>
<li><span class="math inline">\(x\)</span>: number of events in the risk
window,</li>
<li><span class="math inline">\(n_1\)</span>: total number of
events,</li>
<li><span class="math inline">\(r\)</span>: proportion of time at
risk.</li>
</ul>
<p>With age effects, the likelihood becomes more complex, including
terms to account for age groups and event timing (e.g., <span
class="math inline">\(\delta_j\)</span> and <span
class="math inline">\(\beta\)</span> in the likelihood).</p></li>
<li><p><strong>Why SCCS Is Useful</strong> SCCS is particularly
effective for vaccine safety because:</p>
<ul>
<li>It <strong>requires only case data</strong>, which is often more
readily available.</li>
<li>It handles <strong>time-varying exposures</strong> (like vaccines)
and <strong>rare events</strong>.</li>
<li>It is robust against confounders that do not change over time (e.g.,
sex, ethnicity).</li>
</ul></li>
</ol>
<p><strong>Example</strong></p>
<p>“Miller et al. [13] studied the association between measles, mumps,
rubella (MMR) vaccine and idiopathic thrombocytopenic purpura (ITP)
(abnormal bleeding into the skin due to low blood platelet count) in
children aged 12–23 months during the period from October 1991 to
September 1994 within 42 days of receiving the vaccine.”</p>
<p>“The observation period includes the ages 366–730 days, which we
subdivide into J = 4 periods of lengths e1 = e2 = e3 = 91 days, and e4 =
92 days. We take the proportions vaccinated in each of these age
intervals to be p1 = 0.6, p2 = 0.2, p3 = 0.05, p4 = 0.05. We take the
age effects to be e1 = 1, e2 = 0.6, e3 = e4 = 0.4. The risk period is e∗
= 42 days. We set ⍴ = 3, z⍺ = 1.96 and zꞵ = 0.8416 for 80 per cent power
to detect a relative incidence of 3 at the 5 per cent significance
level.”</p>
<table>
<thead>
<tr class="header">
<th>Parameter</th>
<th>Overall</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Number of Periods</td>
<td>4</td>
</tr>
<tr class="even">
<td>Observation Periods</td>
<td>91, 91, 91, 92 days</td>
</tr>
<tr class="odd">
<td>Risk Period</td>
<td>42 days</td>
</tr>
<tr class="even">
<td>P(Exposure)</td>
<td>0.6, 0.2, 0.05, 0.05</td>
</tr>
<tr class="odd">
<td>Age Effects</td>
<td>1, 0.6, 0.4, 0.4</td>
</tr>
<tr class="even">
<td>Relative Incidence</td>
<td>3</td>
</tr>
<tr class="odd">
<td>Alpha (2-sided)</td>
<td>0.05</td>
</tr>
<tr class="even">
<td>Power</td>
<td>80%</td>
</tr>
</tbody>
</table>
</div>
<div id="safety-data-and-the-problem-of-multiplicity"
class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Safety Data and the
Problem of Multiplicity</h2>
<p><strong>Section 9.3: Safety Data and the Problem of Multiplicity –
Summary Explanation</strong></p>
<p>The analysis and interpretation of vaccine safety data is complicated
by the issue of <em>multiplicity</em>—the statistical problem that
arises when many tests are conducted simultaneously.</p>
<p><strong>1. Multiplicity Increases False Positives</strong></p>
<p>When multiple safety variables (like adverse events) are tested at
once, the chance of finding at least one statistically significant
result by random chance (type I error) increases. In large datasets,
even small, clinically irrelevant differences can become statistically
significant. However, adjusting for multiplicity to reduce false
positives tends to increase the risk of <em>false negatives</em> (type
II errors), meaning true safety signals might go undetected.</p>
<hr />
<p><strong>2. Approaches to Handle Multiplicity in Safety
Data</strong></p>
<p><strong>First approach: No correction</strong></p>
<ul>
<li>Many regulators and safety reviewers prefer not adjusting for
multiplicity in safety analyses.</li>
<li>The rationale is that false negatives are more dangerous in safety
than false positives.</li>
<li>Although the false-positive rate increases, regulatory agencies
understand this and rarely base decisions solely on such signals.</li>
<li>Post-marketing studies or further surveillance may be requested for
clarification.</li>
</ul>
<p><strong>Second approach: Control Family-Wise Error Rate
(FWER)</strong></p>
<ul>
<li>Methods like <strong>Bonferroni</strong> or <strong>Holm</strong>
corrections aim to reduce the chance of even one false positive across
all tests.</li>
<li><strong>Bonferroni</strong>: Divide the significance level α by the
number of tests (e.g., α/m).</li>
<li><strong>Holm</strong>: A step-wise method testing ordered p-values
with progressively larger thresholds.</li>
<li>These methods strongly control type I error but increase false
negatives. Hence, they are seldom used in vaccine safety analyses.</li>
</ul>
<p><strong>Third approach: Control False Discovery Rate
(FDR)</strong></p>
<ul>
<li>Introduced by <strong>Benjamini and Hochberg</strong>, this method
controls the expected proportion of false positives among all
significant results.</li>
<li>It is less strict than FWER methods and thus retains more power
(more true positives).</li>
<li>It works by ordering p-values and comparing each to a threshold that
grows with the number of hypotheses.</li>
<li>FDR strikes a balance: fewer false positives than uncorrected tests,
but more true positives than Bonferroni/Holm.</li>
</ul>
<p><strong>Example:</strong> If five p-values are all &lt;0.05:</p>
<ul>
<li><strong>Bonferroni</strong>: Only the smallest (e.g., p=0.0045) is
considered significant.</li>
<li><strong>Holm</strong>: Might flag two.</li>
<li><strong>FDR</strong>: Might flag all five as significant.</li>
</ul>
<p>FDR offers a good compromise between false positive control and
detection of real safety signals.</p>
<hr />
<p><strong>4. Fourth approach: Double FDR</strong></p>
<ul>
<li><p><strong>Mehrotra and Heyse</strong> introduced a method for
hierarchical testing:</p>
<ul>
<li><p>First, adverse events are grouped by <strong>body
systems</strong> (e.g., MedDRA categories).</p></li>
<li><p>A representative p-value is chosen per body system (e.g., the
smallest).</p></li>
<li><p>FDR is applied at two levels:</p>
<ol style="list-style-type: decimal">
<li>Across body systems (to control for multiplicity across groups)</li>
<li>Within each group (for events in the same body system)</li>
</ol></li>
</ul></li>
<li><p>This method helps account for dependencies between related events
and reduces incorrect flagging.</p></li>
</ul>
<p><strong>5. Fifth approach: Bayesian Hierarchical Model</strong></p>
<ul>
<li><strong>Berry and Berry</strong> proposed a Bayesian method as an
alternative to the double FDR.</li>
<li>It uses a three-level hierarchical mixture model, grouping adverse
events within body systems.</li>
<li>It allows sharing of information both <strong>within</strong> and
<strong>across</strong> groups.</li>
<li>While complex, it’s powerful and increasingly popular in vaccine
safety analysis.</li>
</ul>
</div>
<div id="vaccine-reactogenicity" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> Vaccine
Reactogenicity</h2>
<p><strong>Vaccine Reactogenicity Summary:</strong></p>
<p>Reactogenicity refers to <strong>common adverse reactions</strong>
that are expected or attributable to vaccination. These reactions are
either:</p>
<ul>
<li><strong>Local reactions</strong> (at injection site): pain,
tenderness, swelling, redness, etc.</li>
<li><strong>Systemic reactions</strong> (generalized responses): fever,
malaise, fatigue, headache, etc.</li>
</ul>
<p>They are generally <strong>collected by diary</strong> over a short
time (typically 3 or 7 days) after vaccination.</p>
<p><strong>Severity grading</strong> is usually based on:</p>
<ul>
<li><strong>None</strong> = 0 (no interference)</li>
<li><strong>Mild</strong> = 1 (not interfering with daily
activities)</li>
<li><strong>Moderate</strong> = 2 (interferes with daily
activities)</li>
<li><strong>Severe</strong> = 3 (prevents daily activities)</li>
</ul>
<hr />
<p><strong>Statistical Analyses of Reactogenicity:</strong></p>
<ol style="list-style-type: decimal">
<li><p><strong>Incidence Analysis</strong></p>
<ul>
<li>Calculate the <strong>proportion of subjects</strong> reporting each
reaction at least once.</li>
<li>Use <strong>Clopper–Pearson confidence intervals</strong>.</li>
<li>Example: If 3 out of 4,500 subjects report a reaction, and the upper
95% CI limit is 0.0019, we interpret it as “less than 1 in 500”
vaccinated individuals.</li>
</ul></li>
<li><p><strong>Relative Risk Comparison</strong> (between two
vaccines)</p>
<ul>
<li>Use <strong>risk ratios with confidence intervals</strong> (e.g.,
Wilson-type).</li>
<li>Limit: Cannot compute CI when one group has 0 cases.</li>
</ul></li>
<li><p><strong>Total Number of Reactions per Subject</strong></p>
<ul>
<li>When many reactions are solicited, sum the number reported per
individual.</li>
<li>Compare total counts using <strong>Wilcoxon rank-sum test</strong>
via <code>proc freq</code> and Cochran-Mantel-Haenszel (CMH) statistics
with rank scores.</li>
</ul></li>
</ol>
<p><strong>SAS Code Example for Total Number of Local
Reactions:</strong></p>
<pre class="sas"><code>proc freq;
  table vaccine*number_of_local_reactions /
  nopercent nocol cmh scores=rank;
run;</code></pre>
<ol start="4" style="list-style-type: decimal">
<li><p><strong>Severity Analysis (Ordinal Scale)</strong></p>
<ul>
<li>Reactions rated on a 0–3 scale (none to severe).</li>
<li>Compare severity across vaccine groups using <strong>Wilcoxon
rank-sum test</strong> with <strong>modified ridit scores</strong>.</li>
</ul></li>
</ol>
<p><strong>SAS Code Example for Ordinal Severity
Comparison:</strong></p>
<pre class="sas"><code>proc freq;
  table vaccine*tenderness_score /
  nopercent nocol cmh scores=modridit;
run;</code></pre>
<p>Make sure severity scores are numerically ordered (e.g., 0, 1, 2, 3)
instead of text labels, or SAS will rank them alphabetically, causing
wrong results.</p>
</div>
<div id="vaccine-durability---deferred-vaccine-designs"
class="section level2" number="5.6">
<h2><span class="header-section-number">5.6</span> Vaccine Durability -
Deferred Vaccine Designs</h2>
<p><strong>Ethical Considerations</strong> Once a vaccine has been shown
to be effective in early trials, continuing to keep participants in a
placebo group becomes ethically problematic. Therefore:</p>
<ul>
<li>Placebo recipients are typically <strong>offered the
vaccine</strong> after efficacy is established.</li>
<li>This transition must be handled carefully to <strong>preserve the
integrity of the trial</strong>, particularly in long-term (durability)
follow-up.</li>
</ul>
<hr />
<p><strong>Crossover Design</strong> To address this ethical issue while
still enabling long-term efficacy assessment, trials often use a
<strong>crossover design</strong>:</p>
<ul>
<li><p><strong>All participants eventually receive the vaccine</strong>,
either at the beginning or after the placebo phase.</p></li>
<li><p>This setup:</p>
<ul>
<li>Maintains <strong>blinding</strong> in the early phases (so
participants and investigators don’t know who has which treatment).</li>
<li>Still allows for <strong>comparisons</strong> of initial protection
and long-term durability across groups.</li>
</ul></li>
</ul>
<hr />
<p><strong>Less Precision Compared to Standard Designs</strong>
Crossover and ethical adaptations can result in <strong>less precise
efficacy estimates</strong>, especially over time:</p>
<ul>
<li>The estimates for long-term efficacy (durability) may be
<strong>less reliable</strong> than those in traditional randomized
controlled trials.</li>
<li>In durability trials, any early misestimations can undermine later
results—this is described as the study being only as <strong>strong as
its weakest link</strong>.</li>
</ul>
<hr />
<p><strong>Subgroup Safety and Harm Analysis</strong> Durability trials
also provide opportunities for <strong>subgroup analysis</strong>, such
as:</p>
<ul>
<li>Assessing whether <strong>efficacy or safety</strong> varies by
<strong>demographic or clinical characteristics</strong> (e.g., age,
gender, comorbidities).</li>
<li>These analyses help understand <strong>differential vaccine
performance</strong>, contributing to more personalized or targeted
vaccination strategies.</li>
</ul>
<p><img src="./02_Plots/Vaccine/Vaccine-Durability.png" /></p>
</div>
</div>
<div id="reference" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Reference</h1>
<p>Nauta, J. (2011). Statistics in clinical vaccine trials.
Springer-Verlag.</p>
<p>nQuery Webinar, Sample Sizes for Vaccine Trials, Trial Designs for
Assessing Vaccine Efficacy, Safety &amp; Durability, <a
href="https://www.statsols.com/guides/sample-sizes-for-vaccine-trials"
class="uri">https://www.statsols.com/guides/sample-sizes-for-vaccine-trials</a></p>
<p>nQuery Webinar, Design and Sample Size for Vaccine Trials, Case Study
of COVID-19, <a
href="https://www.statsols.com/guides/design-and-sample-size-for-vaccine-trials"
class="uri">https://www.statsols.com/guides/design-and-sample-size-for-vaccine-trials</a></p>
<p>Fleiss, J.L., Levin, B. and Paik, M.C., 2013. Statistical methods for
rates and proportions. John Wiley &amp; Sons.</p>
<p>Halloran, M.E., Longini, I.M., Struchiner, C.J. and Longini, I.M.,
2010. Design and analysis of vaccine studies (Vol. 18). New York, NY:
Springer.</p>
<p>Singh, K. and Mehta, S., 2016. The clinical development process for a
novel preventive vaccine: An overview. Journal of postgraduate medicine,
62(1), p.4.</p>
<p>O’Neill, R.T., 1988. On sample sizes to estimate the protective
efficacy of a vaccine. Statistics in Medicine, 7(12), pp.1279-1288.</p>
<p>Chan, I.S. and Bohidar, N.R., 1998. Exact power and sample size for
vaccine efficacy studies. Communications in Statistics-Theory and
Methods, 27(6), pp.1305-1322.</p>
<p>Gu, K., Ng, H.K.T., Tang, M.L. and Schucany, W.R., 2008. Testing the
ratio of two poisson rates. Biometrical Journal: Journal of Mathematical
Methods in Biosciences, 50(2), pp.283-298.</p>
<p>Schoenfeld, D.A., 1983. Sample-size formula for the
proportional-hazards regression model. Biometrics, pp.499-503.</p>
<p>Callegaro, A., Curran, D. and Matthews, S., 2020. Burden‐of‐illness
vaccine efficacy. Pharmaceutical statistics, 19(5), pp.636-645.</p>
<p>Chow, S.C., Shao, J., Wang, H. and Lokhnygina, Y., 2017. Sample size
calculations in clinical research. CRC press.</p>
<div id="vaccine-efficacy" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Vaccine Efficacy</h2>
<p>Polack, F.P., Thomas, S.J., Kitchin, N., Absalon, J., Gurtman, A.,
Lockhart, S., Perez, J.L., Pérez Marc, G., Moreira, E.D., Zerbini, C.
and Bailey, R., 2020. Safety and efficacy of the BNT162b2 mRNA Covid-19
vaccine. New England Journal of Medicine, 383(27), pp.2603-2615.</p>
<p>Baden, L.R., El Sahly, H.M., Essink, B., Kotloff, K., Frey, S.,
Novak, R., Diemert, D., Spector, S.A., Rouphael, N., Creech, C.B. and
McGettigan, J., 2021. Efficacy and safety of the mRNA-1273 SARS-CoV-2
vaccine. New England Journal of Medicine, 384(5), pp.403-416.</p>
<p>Voysey, M., Clemens, S.A.C., Madhi, S.A., Weckx, L.Y., Folegatti,
P.M., Aley, P.K., Angus, B., Baillie, V.L., Barnabas, S.L., Bhorat, Q.E.
and Bibi, S., 2021. Safety and efficacy of the ChAdOx1 nCoV-19 vaccine
(AZD1222) against SARS-CoV-2: an interim analysis of four randomised
controlled trials in Brazil, South Africa, and the UK. The Lancet,
397(10269), pp.99-111.</p>
<p>Sadoff, J., Le Gars, M., Shukarev, G., Heerwegh, D., Truyers, C., de
Groot, A.M., Stoop, J., Tete, S., Van Damme, W., Leroux-Roels, I. and
Berghmans, P.J., 2021. Interim Results of a Phase 1–2a Trial of Ad26.
COV2. S Covid-19 Vaccine. New England Journal of Medicine.</p>
<p>Logunov, D.Y., Dolzhikova, I.V., Shcheblyakov, D.V., Tukhvatulin,
A.I., Zubkova, O.V., Dzharullaeva, A.S., Kovyrshina, A.V., Lubenets,
N.L., Grousova, D.M., Erokhova, A.S. and Botikov, A.G., 2021. Safety and
efficacy of an rAd26 and rAd5 vector-based heterologous prime-boost
COVID-19 vaccine: an interim analysis of a randomised controlled phase 3
trial in Russia. The Lancet, 397(10275), pp.671-681.</p>
<p>Uttarilli, Anusha, Amalakanti, Sridhar, Kommoju, Phaneeswara-Rao,
Sharma, Srihari, Goyal, Pankaj, Manjunath, Gowrang Kasaba, Upadhayay,
Vineet, Parveen, Alisha, Tandon, Ravi, Prasad, Kumar Suranjit, Dakal,
Tikam Chand, Ben Shlomo, Izhar, Yousef, Malik, Neerathilingam, Muniasamy
and Kumar, Abhishek. “Super-rapid race for saving lives by developing
COVID-19 vaccines” Journal of Integrative Bioinformatics, vol. 18, no.
1, 2021, pp. 27-43. <a href="https://doi.org/10.1515/jib-2021-0002"
class="uri">https://doi.org/10.1515/jib-2021-0002</a></p>
</div>
<div id="vaccine-safety-1" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Vaccine Safety</h2>
<p>Petersen, I., Douglas, I. and Whitaker, H., 2016. Self controlled
case series methods: an alternative to standard epidemiological study
designs. BMJ, 354.Musonda, P., Paddy Farrington, C. and Whitaker, H.J.,
2006. Sample sizes for self‐controlled case series studies. Statistics
in Medicine, 25(15), pp.2618-2631.</p>
<p>Farrington CP. 1995. Relative incidence estimation from case series
for vaccine safety evaluation. Biometrics 51:228-235. page 228.Dodd, C.,
Andrews, N., Petousis-Harris, H., Sturkenboom, M., Omer, S.B. and Black,
S., 2021. Methodological frontiers in vaccine safety: qualifying
available evidence for rare events, use of distributed data networks to
monitor vaccine safety issues, and monitoring the safety of pregnancy
interventions. BMJ global health, 6(Suppl 2), p.e003540. Knipe, D.M.,
Levy, O., Fitzgerald, K.A. and Mühlberger, E., 2020. Ensuring vaccine
safety. Science, 370(6522), pp.1274-1275.</p>
<p>Matz, K.M., Marzi, A. and Feldmann, H., 2019. Ebola vaccine trials:
progress in vaccine safety and immunogenicity. Expert review of
vaccines, 18(12), pp.1229-1242.</p>
<p>Pormohammad, A., Zarei, M., Ghorbani, S., Mohammadi, M., Razizadeh,
M.H., Turner, D.L. and Turner, R.J., 2021. Efficacy and safety of
COVID-19 vaccines: a systematic review and meta-analysis of randomized
clinical trials. Vaccines, 9(5), p.467.</p>
</div>
<div id="vaccine-durability" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Vaccine
Durability</h2>
<p>Choe, P.G., Kang, C.K., Suh, H.J., Jung, J., Song, K.H., Bang, J.H.,
Kim, E.S., Kim, H.B., Park, S.W., Kim, N.J. and Park, W.B., 2021. Waning
antibody responses in asymptomatic and symptomatic SARS-CoV-2 infection.
Emerging infectious diseases, 27(1), p.327.</p>
<p>Nason, M. and Follmann, D., 2010. Design and analysis of crossover
trials for absorbing binary endpoints. Biometrics, 66(3),
pp.958-965.</p>
<p>Makubate, B. and Senn, S., 2010. Planning and analysis of cross‐over
trials in infertility. Statistics in medicine, 29(30), pp.3203-3210.</p>
<p>Follmann, D., 2006. Augmented designs to assess immune response in
vaccine trials. Biometrics, 62(4), pp.1161-1169.</p>
<p>Palin, A.C., Alter, G., Crotty, S., Ellebedy, A.H., Lane, M.C., Lee,
F.E.H., Locci, M., Malaspina, A., Mallia, C., McElrath, M.J. and
Pulendran, B., 2022. The persistence of memory: defining, engineering,
and measuring vaccine durability. Nature immunology, 23(12),
pp.1665-1668.Fintzi, J. and Follmann, D., 2021. Assessing vaccine
durability in randomized trials following placebo crossover. Statistics
in Medicine, 40(27), pp.5983-6007.</p>
<p>Follmann, D., Fintzi, J., Fay, M.P., Janes, H.E., Baden, L.R., El
Sahly, H.M., Fleming, T.R., Mehrotra, D.V., Carpp, L.N., Juraska, M. and
Benkeser, D., 2021. A deferred-vaccination design to assess durability
of COVID-19 vaccine effect after the placebo group is vaccinated. Annals
of internal medicine, 174(8), pp.1118-1125.</p>
<p>Lin, D.Y., Zeng, D. and Gilbert, P.B., 2021. Evaluating the long-term
efficacy of coronavirus disease 2019 (COVID-19) vaccines. Clinical
Infectious Diseases, 73(10), pp.1927-1939.</p>
<p>Lipsitch, M., 2019. Challenges of vaccine effectiveness and waning
studies. Clinical Infectious Diseases, 68(10), pp.1631-1633.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
