<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title> Forest Plot and Meta Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Zehui Bai</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="02-Clinical_Experience.html">
    <span class="fa fa-user"></span>
     
    Clinical Experience
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-sliders"></span>
     
    Sample Size
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="03-SSC-Everything-to-Know-About-Sample-Size-Determination.html">Everything to Know About Sample Size Determination</a>
    </li>
    <li>
      <a href="03-SSC-Choosing-the-Effect-Size-for-Sample-Size-Calculations.html">Choosing the Effect Size</a>
    </li>
    <li>
      <a href="03-SSC-Biosimilar-Trials.html">Statistical Considerations for the Design and Analysis of Biosimilar Trials</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-and-Power-for-Non-Parametric-Analysis.html">Sample Size and Power for Non-Parametric Analysis</a>
    </li>
    <li>
      <a href="03-SSC-Power-for-Complex-Hypotheses.html">Power for Complex Hypotheses</a>
    </li>
    <li>
      <a href="03-SSC-Alternatives-to-Power.html">Bayesian methods - Alternatives to Power</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-for-Pilot-Studies.html">Sample Size for Pilot Studies</a>
    </li>
    <li>
      <a href="03-SSC-Case-Continuous-Endpoint.html">Sample Size Determination for Continuous Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Categorical-Endpoint.html">Sample Size Determination for Categorical Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-Determination-for-Counts-and-Rates.html">Sample Size Determination for Counts and Rates</a>
    </li>
    <li>
      <a href="03-SSC-Case-Survival-Endpoint.html">Sample Size Determination for Survival Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Repeated-Measures.html">Sample Size Determination for Repeated Measures</a>
    </li>
    <li>
      <a href="03-SSC-IA-Sequential-Design.html">Statistical Considerations for Group Sequential Design</a>
    </li>
    <li>
      <a href="03-SSC-IA-Adaptive-Design.html">Statistical Considerations for Adaptive Design</a>
    </li>
    <li>
      <a href="03-SSC-Multiple-Test.html">Sample Size for Multiple Test</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    Study Design
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04-Design-General-Consideration.html">General Consideration</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Framework.html">Estimands Framework</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Practice.html">Estimands and Sensitivity Analyses</a>
    </li>
    <li>
      <a href="04-Design-Phase-I-Trials---Design-Considerations.html">Phase I Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-II-Trials---Design-Considerations.html">Phase II Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-III-Trials---Design-Considerations.html">Phase III Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-IV-Trials---Design-Considerations.html">Phase IV Trials - Design Considerations for Post Marketing Surveillance</a>
    </li>
    <li>
      <a href="04-Design-Complex-Sequential-Trials.html">Complex Sequential Analysis Trials</a>
    </li>
    <li>
      <a href="04-Design-Adaptive-Clinical-Trials.html">Adaptive Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Bayesian-Clinical-Trials.html">Bayesian Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Dose-Escalation-and-Stratification-Designs.html">Dose Escalation and Stratification Designs in Early Oncology Development</a>
    </li>
    <li>
      <a href="04-Design-Single-Arm-Clinical-Trials.html">Single Arm Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-Design-and-Evaluation.html">Diagnostic Study-Design and Evaluation</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-MRMC.html">Diagnostic Study-Multireader Multicase (MRMC)</a>
    </li>
    <li>
      <a href="04-Design-Vaccine-Design.html">Vaccine Trials</a>
    </li>
    <li>
      <a href="04-Design-Externally-Controlled-Trials.html">Externally Controlled Trials</a>
    </li>
    <li>
      <a href="04-Design-Noninferiority-Trials.html">Noninferiority Trials</a>
    </li>
    <li>
      <a href="04-Design-Bioequivalence-and-Biosimilar-Trials.html">Bioequivalence and Biosimilar Trials</a>
    </li>
    <li>
      <a href="04-Design-Exploring-Survival-Analysis-Designs-for-Clinical-Trials.html">Exploring Survival Analysis Designs for Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Hierarchical-composite-endpoints.html">Hierarchical Composite Endpoints</a>
    </li>
    <li>
      <a href="04-Design-Regulatory-Submission.html">Regulatory Submission from Stats Perspective</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Data Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="05-Plot-Time-To-Event.html">Time to Event Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-PRO-Data.html">Patient Reported Outcome Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-SSC-and-Power-Calculation.html">Sample Size and Power Calculations Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Correlation.html">Correlation Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Model-Table.html">Clinical Data and Model visualization</a>
    </li>
    <li>
      <a href="05-Plot-ScatterPlot.html">Scatter and Line Plot</a>
    </li>
    <li>
      <a href="05-Plot-BarPlot.html">Bar Chart</a>
    </li>
    <li>
      <a href="05-Plot-PieChart.html">Pie Chart</a>
    </li>
    <li>
      <a href="05-Plot-BoxPlot.html">Box Plot</a>
    </li>
    <li>
      <a href="05-Plot-Histogram.html">Histogram</a>
    </li>
    <li>
      <a href="05-Plot-Forest-Plot.html">Forest Plot</a>
    </li>
    <li>
      <a href="05-Plot-Flow-Chart.html">Flow Chart</a>
    </li>
    <li>
      <a href="05-Plot-Some-Interesting.html">Some Interesting Plots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-calculator"></span>
     
    Statistical Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06-Analysis-Linear-Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Logistic-Regression.html">Logistic Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Mixed-Model.html">Mixed Model</a>
    </li>
    <li>
      <a href="06-Analysis-MMRM.html">Mixed Model Repeated Measures</a>
    </li>
    <li>
      <a href="06-Analysis-GEE.html">Generalized Estimating Equation</a>
    </li>
    <li>
      <a href="06-Analysis-ANOVA.html">Analysis of Variance</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Analysis.html">Survival Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Competing-Risk.html">Survival Analysis - Competing Risk</a>
    </li>
    <li>
      <a href="06-Analysis-Missing-Data.html">Missing Data Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-MI-Implementation.html">Multiple Imputation Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SSD.html">Safety Signal Detection and Evaluation</a>
    </li>
    <li>
      <a href="06-Analysis-Meta-Analysis.html">Meta Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-PK-and-PD.html">PK and PD Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Time-Series-Analysis.html">Time Series Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SEM.html">Structural Equation Modeling</a>
    </li>
    <li>
      <a href="06-Analysis-Factor-Analysis.html">Factor Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Propensity-Score.html">Propensity Score Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Machine Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="07-ML-Bayesian-Theory.html">Bayesian Theory</a>
    </li>
    <li>
      <a href="07-ML-Bayesian-Analysis.html">Bayesian Analysis</a>
    </li>
    <li>
      <a href="07-ML-Regularization-Penalized-Regression.html">Regularization Penalized Regression</a>
    </li>
    <li>
      <a href="07-ML-Loss-Regression.html">Loss Functions in Machine Learning</a>
    </li>
    <li>
      <a href="07-ML-PCA.html">Principal Component Analysis</a>
    </li>
    <li>
      <a href="07-ML-KNN.html">K-Nearest Neighbors</a>
    </li>
    <li>
      <a href="07-ML-SVM.html">Support Vector Machine</a>
    </li>
    <li>
      <a href="07-ML-Tree-Models.html">Tree Models</a>
    </li>
    <li>
      <a href="07-ML-LDA.html">Linear Discriminant Analysis</a>
    </li>
    <li>
      <a href="07-ML-Cluster-Analysis.html">Cluster Analysis</a>
    </li>
    <li>
      <a href="07-ML-Neural-Networks.html">Neural Network</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-stethoscope"></span>
     
    Clinical Analysis (ADaM)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="08-Clinical-Disposition-Baseline.html">Disposition and Baseline</a>
    </li>
    <li>
      <a href="08-Clinical-Efficacy.html">Efficacy Analysis</a>
    </li>
    <li>
      <a href="08-Clinical-Pharmacokinetics-Analysis.html">Pharmacokinetics Analysis</a>
    </li>
    <li>
      <a href="08-Clinical-Adverse-Event.html">Adverse Event Visualization</a>
    </li>
    <li>
      <a href="08-Clinical-Laboratory-Data.html">Laboratory Data Visualization</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="08-CV.html">
    <span class="fa fa-file-pdf-o"></span>
     
    CV
  </a>
</li>
<li>
  <a href="mailto:zehuibai@outlook.com">
    <span class="fa fa-envelope-o"></span>
     
    Contact me
  </a>
</li>
<li>
  <a href="https://github.com/Zehuibai">
    <span class="fa fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore"><p><img src="logo.png"
style="width:3in" /><br />
Forest Plot and Meta Analysis</p></h1>

</div>


<div id="general" class="section level1">
<h1>General</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>A <strong>forest plot</strong> is a visual tool commonly used in
meta-analysis to present the results of multiple individual studies
alongside the combined, or pooled, result. It allows for a quick
understanding of the consistency and magnitude of effects across
studies.</p>
<p>The layout of a forest plot is fairly standardized, with distinct
components organized for clarity:</p>
<p><strong>1. Study names (left side):</strong> On the left side of a
forest plot, you will usually find a list of the studies included in the
meta-analysis. Each row corresponds to one study. This makes it easy to
identify which individual results are being displayed.</p>
<p><strong>2. Graphical display of effect sizes (center):</strong> In
the middle of the plot, each study is represented graphically:</p>
<ul>
<li>A <strong>square</strong> represents the point estimate of the
effect size (e.g., mean difference, odds ratio) for each study.</li>
<li>The <strong>horizontal line</strong> through the square illustrates
the <strong>confidence interval (CI)</strong> of that effect size. This
interval reflects the uncertainty or precision of the estimate.</li>
<li>The <strong>size of the square</strong> is proportional to the
weight that the study contributes to the meta-analysis. Studies with
larger sample sizes or lower variance get more weight and thus larger
squares.</li>
</ul>
<p><strong>3. Effect size data (often in a table):</strong> Next to the
graphical display, the numeric values for the point estimate and
confidence interval are usually provided. This gives readers access to
the exact data behind the plot, supporting reproducibility and
transparency.</p>
<p><strong>4. Reference line (vertical line):</strong> A vertical line
is often drawn at the value corresponding to “no effect”:</p>
<ul>
<li>For <strong>mean differences</strong>, this line is typically at
<strong>0</strong>.</li>
<li>For <strong>ratios</strong> (like risk ratios, odds ratios), it is
usually placed at <strong>1</strong>. This line allows for an easy
visual check of which studies show significant effects (i.e., their CI
does not overlap the reference line) and in what direction.</li>
</ul>
<p><strong>5. Summary estimate (bottom of the plot):</strong> The
overall or pooled effect is shown at the bottom, usually using a
<strong>diamond shape</strong>:</p>
<ul>
<li>The <strong>center of the diamond</strong> represents the pooled
point estimate.</li>
<li>The <strong>width of the diamond</strong> reflects the confidence
interval for the pooled effect size. This diamond gives a concise visual
summary of the overall findings of the meta-analysis.</li>
</ul>
<p><strong>6. Heterogeneity information (optional):</strong> Some forest
plots include statistics like <strong>I²</strong> or
<strong>τ²</strong>, which quantify heterogeneity:</p>
<ul>
<li><strong>I²</strong> indicates the percentage of total variation
across studies that is due to heterogeneity rather than chance.</li>
<li><strong>τ²</strong> is the between-study variance in a
random-effects model.</li>
</ul>
<p><strong>7. X-axis scale (linear or logarithmic):</strong> Effect
sizes are typically plotted along a linear x-axis. However, if the
effect size is a <strong>ratio</strong> (such as an odds ratio or risk
ratio), a <strong>logarithmic scale</strong> is often used:</p>
<ul>
<li>In a log scale, values close to 1 are closer together than values
far from 1.</li>
<li>This is appropriate because ratio metrics are multiplicative. For
example, the reciprocal of 0.5 is 2 (not 1.5), so a log scale makes this
symmetry clearer.</li>
<li>The reference line is then placed at 1, since that represents no
effect for ratios.</li>
</ul>
<p>In summary, forest plots bring together both
<strong>numerical</strong> and <strong>graphical</strong> summaries of
multiple studies in one visual, helping researchers quickly assess:</p>
<ul>
<li>The individual and overall effects</li>
<li>The precision of those effects</li>
<li>The consistency or variability between studies</li>
</ul>
<p>By clearly showing each study’s contribution and the combined result,
forest plots are an essential tool for interpreting and communicating
findings in evidence synthesis.</p>
</div>
<div id="annotated-forest-plots-alternative-1" class="section level2">
<h2>Annotated Forest Plots Alternative 1</h2>
<pre class="r"><code>## Example data frame
dat &lt;- data.frame(
  Index = c(1, 2, 3, 4), ## This provides an order to the data
  label = c(&quot;Age (65 and older versus &lt;65)&quot;, &quot;Male versus Female&quot;, &quot;High income versus Low income&quot;, &quot;High school or higher versus No High school&quot;),
  OR = c(1.00, 2.00, 3.00, 0.50),
  LL = c(0.25, 0.90, 2.25, 0.2),
  UL = c(1.75, 3.10, 3.75, 0.8),
  CI = c(&quot;0.25, 1.75&quot;, &quot;0.90, 3.10&quot;, &quot;2.25, 3.75&quot;, &quot;0.20, 0.80&quot;)
)
dat</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Index"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["label"],"name":[2],"type":["chr"],"align":["left"]},{"label":["OR"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["LL"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["UL"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["CI"],"name":[6],"type":["chr"],"align":["left"]}],"data":[{"1":"1","2":"Age (65 and older versus <65)","3":"1.0","4":"0.25","5":"1.75","6":"0.25, 1.75"},{"1":"2","2":"Male versus Female","3":"2.0","4":"0.90","5":"3.10","6":"0.90, 3.10"},{"1":"3","2":"High income versus Low income","3":"3.0","4":"2.25","5":"3.75","6":"2.25, 3.75"},{"1":"4","2":"High school or higher versus No High school","3":"0.5","4":"0.20","5":"0.80","6":"0.20, 0.80"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>plot1 &lt;- ggplot(dat, aes(y = Index, x = OR)) +
  geom_point(shape = 18, size = 5) +  
  geom_errorbarh(aes(xmin = LL, xmax = UL), height = 0.25) +
  geom_vline(xintercept = 1, color = &quot;red&quot;, linetype = &quot;dashed&quot;, cex = 1, alpha = 0.5) +
  scale_y_continuous(name = &quot;&quot;, breaks=1:4, labels = dat$label, trans = &quot;reverse&quot;) +
  xlab(&quot;Odds Ratio (95% CI)&quot;) + 
  ylab(&quot; &quot;) + 
  theme_bw() +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(), 
        axis.line = element_line(colour = &quot;black&quot;),
        axis.text.y = element_text(size = 12, colour = &quot;black&quot;),
        axis.text.x.bottom = element_text(size = 12, colour = &quot;black&quot;),
        axis.title.x = element_text(size = 12, colour = &quot;black&quot;))


## Create the table-base pallete
table_base &lt;- ggplot(dat, aes(y=label)) +
  ylab(NULL) + xlab(&quot;  &quot;) + 
  theme(plot.title = element_text(hjust = 0.5, size=12), 
        axis.text.x = element_text(color=&quot;white&quot;, hjust = -3, size = 25), ## This is used to help with alignment
        axis.line = element_blank(),
        axis.text.y = element_blank(), 
        axis.ticks = element_blank(),
        axis.title.y = element_blank(), 
        legend.position = &quot;none&quot;,
        panel.background = element_blank(), 
        panel.border = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), 
        plot.background = element_blank())

## OR point estimate table
tab1 &lt;- table_base + 
  labs(title = &quot;space&quot;) +
  geom_text(aes(y = rev(Index), x = 1, label = sprintf(&quot;%0.1f&quot;, round(OR, digits = 1))), size = 4) + ## decimal places
  ggtitle(&quot;OR&quot;)

## 95% CI table
tab2 &lt;- table_base +
  geom_text(aes(y = rev(Index), x = 1, label = CI), size = 4) + 
  ggtitle(&quot;95% CI&quot;)

## Merge tables with plot 
grid.arrange(plot1, tab1, tab2, 
             layout_matrix = matrix(c(1,1,1,1,1,1,1,1,1,1,2,3,3), nrow = 1))</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-1-1.png" width="1152" /></p>
</div>
<div id="annotated-forest-plots-alternative-2" class="section level2">
<h2>Annotated Forest Plots Alternative 2</h2>
<pre class="r"><code># library(&quot;gt&quot;)
res_log &lt;- read_csv(&quot;https://raw.githubusercontent.com/kathoffman/steroids-trial-emulation/main/output/res_log.csv&quot;)
res &lt;- read_csv(&quot;https://raw.githubusercontent.com/kathoffman/steroids-trial-emulation/main/output/res.csv&quot;)

# res &lt;- res_log |&gt;
#       rename_with(~str_c(&quot;log.&quot;, .), estimate:conf.high) |&gt;
#   select(-p.value) |&gt;
#   full_join(res)
ForestData &lt;- res_log %&gt;%
      rename_with(~str_c(&quot;log.&quot;, .), estimate:conf.high) %&gt;%
  select(-p.value) %&gt;%
  full_join(res)
## Get a glimpse of your data
## glimpse(ForestData)
ForestData %&gt;% kbl() %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;))</code></pre>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
log.estimate
</th>
<th style="text-align:right;">
log.conf.low
</th>
<th style="text-align:right;">
log.conf.high
</th>
<th style="text-align:right;">
estimate
</th>
<th style="text-align:right;">
conf.low
</th>
<th style="text-align:right;">
conf.high
</th>
<th style="text-align:right;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
-0.6846879
</td>
<td style="text-align:right;">
-0.8947661
</td>
<td style="text-align:right;">
-0.4746097
</td>
<td style="text-align:right;">
0.5042476
</td>
<td style="text-align:right;">
0.4087032
</td>
<td style="text-align:right;">
0.6221278
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:right;">
-0.0525578
</td>
<td style="text-align:right;">
-0.4201151
</td>
<td style="text-align:right;">
0.3149994
</td>
<td style="text-align:right;">
0.9487994
</td>
<td style="text-align:right;">
0.6569712
</td>
<td style="text-align:right;">
1.3702585
</td>
<td style="text-align:right;">
0.7792783
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
-0.0864096
</td>
<td style="text-align:right;">
-0.4567781
</td>
<td style="text-align:right;">
0.2839589
</td>
<td style="text-align:right;">
0.9172184
</td>
<td style="text-align:right;">
0.6333208
</td>
<td style="text-align:right;">
1.3283783
</td>
<td style="text-align:right;">
0.6474743
</td>
</tr>
<tr>
<td style="text-align:left;">
D
</td>
<td style="text-align:right;">
-0.1214702
</td>
<td style="text-align:right;">
-0.5869119
</td>
<td style="text-align:right;">
0.3439714
</td>
<td style="text-align:right;">
0.8856174
</td>
<td style="text-align:right;">
0.5560418
</td>
<td style="text-align:right;">
1.4105383
</td>
<td style="text-align:right;">
0.6089951
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:right;">
-0.4216403
</td>
<td style="text-align:right;">
-0.8819218
</td>
<td style="text-align:right;">
0.0386411
</td>
<td style="text-align:right;">
0.6559699
</td>
<td style="text-align:right;">
0.4139865
</td>
<td style="text-align:right;">
1.0393974
</td>
<td style="text-align:right;">
0.0725863
</td>
</tr>
<tr>
<td style="text-align:left;">
F
</td>
<td style="text-align:right;">
-0.2575052
</td>
<td style="text-align:right;">
-0.5093796
</td>
<td style="text-align:right;">
-0.0056308
</td>
<td style="text-align:right;">
0.7729776
</td>
<td style="text-align:right;">
0.6008682
</td>
<td style="text-align:right;">
0.9943850
</td>
<td style="text-align:right;">
0.0450936
</td>
</tr>
<tr>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.0526079
</td>
<td style="text-align:right;">
-0.2671835
</td>
<td style="text-align:right;">
0.3723992
</td>
<td style="text-align:right;">
1.0540163
</td>
<td style="text-align:right;">
0.7655326
</td>
<td style="text-align:right;">
1.4512123
</td>
<td style="text-align:right;">
0.7471289
</td>
</tr>
<tr>
<td style="text-align:left;">
H
</td>
<td style="text-align:right;">
0.0427286
</td>
<td style="text-align:right;">
-0.2871902
</td>
<td style="text-align:right;">
0.3726475
</td>
<td style="text-align:right;">
1.0436546
</td>
<td style="text-align:right;">
0.7503690
</td>
<td style="text-align:right;">
1.4515726
</td>
<td style="text-align:right;">
0.7996192
</td>
</tr>
<tr>
<td style="text-align:left;">
I
</td>
<td style="text-align:right;">
-0.4624971
</td>
<td style="text-align:right;">
-0.7340078
</td>
<td style="text-align:right;">
-0.1909864
</td>
<td style="text-align:right;">
0.6297092
</td>
<td style="text-align:right;">
0.4799815
</td>
<td style="text-align:right;">
0.8261438
</td>
<td style="text-align:right;">
0.0008419
</td>
</tr>
<tr>
<td style="text-align:left;">
J
</td>
<td style="text-align:right;">
0.0798845
</td>
<td style="text-align:right;">
-0.2224750
</td>
<td style="text-align:right;">
0.3822440
</td>
<td style="text-align:right;">
1.0831619
</td>
<td style="text-align:right;">
0.8005350
</td>
<td style="text-align:right;">
1.4655697
</td>
<td style="text-align:right;">
0.6045771
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>## Make point and line range section of the plot
p_mid &lt;- ForestData |&gt;
  ## 1. Reverse order of factor levels
  ggplot(aes(y = fct_rev(model))) + 
  theme_classic() +
  ## 2. Show all of our information (point estimate and 95% confidence interval) on the graph
  geom_point(aes(x=log.estimate), shape=15, size=3) +
  geom_linerange(aes(xmin=log.conf.low, xmax=log.conf.high)) +
  ## 3. Add a vertical line at 0 and rename the x axis, zoom to the exact height and width
  geom_vline(xintercept = 0, linetype=&quot;dashed&quot;) +
  labs(x=&quot;Log Hazard Ratio&quot;, y=&quot;&quot;) +
  coord_cartesian(ylim=c(1,11), xlim=c(-1, .5)) +
  ## 4. Add text about protective vs. harmful using the annotate layer
  annotate(&quot;text&quot;, x = -.32, y = 11, label = &quot;Corticosteroids protective&quot;) +
  annotate(&quot;text&quot;, x = .3, y = 11, label = &quot;Corticosteroids harmful&quot;) +
  ## 5. remove everything on the y axis
  theme(axis.line.y = element_blank(),
        axis.ticks.y= element_blank(),
        axis.text.y= element_blank(),
        axis.title.y= element_blank())
p_mid</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-2-1.png" width="1152" /></p>
<pre class="r"><code>## Create estimate annotations plot
# wrangle results into pre-plotting table form
# res_plot &lt;- ForestData |&gt;
#   # round estimates and 95% CIs to 2 decimal places for journal specifications
#   mutate(across(
#     c(estimate, conf.low, conf.high),
#     ~ str_pad(
#       round(.x, 2),
#       width = 4,
#       pad = &quot;0&quot;,
#       side = &quot;right&quot;
#     )
#   ),
#   # add an &quot;-&quot; between HR estimate confidence intervals
#   estimate_lab = paste0(estimate, &quot; (&quot;, conf.low, &quot;-&quot;, conf.high, &quot;)&quot;)) |&gt;
#   # round p-values to two decimal places, except in cases where p &lt; .001
#   mutate(p.value = case_when(
#     p.value &lt; .001 ~ &quot;&lt;0.001&quot;,
#     round(p.value, 2) == .05 ~ as.character(round(p.value,3)),
#     p.value &lt; .01 ~ str_pad( # if less than .01, go one more decimal place
#       as.character(round(p.value, 3)),
#       width = 4,
#       pad = &quot;0&quot;,
#       side = &quot;right&quot;
#     ),
#     TRUE ~ str_pad( # otherwise just round to 2 decimal places and pad string so that .2 reads as 0.20
#       as.character(round(p.value, 2)),
#       width = 4,
#       pad = &quot;0&quot;,
#       side = &quot;right&quot;
#     )
#   )) |&gt;
#   # add a row of data that are actually column names which will be shown on the plot in the next step
#   bind_rows(
#     data.frame(
#       model = &quot;Model&quot;,
#       estimate_lab = &quot;Hazard Ratio (95% CI)&quot;,
#       conf.low = &quot;&quot;,
#       conf.high = &quot;&quot;,
#       p.value = &quot;p-value&quot;
#     )
#   ) |&gt;
#   mutate(model = fct_rev(fct_relevel(model, &quot;Model&quot;)))
# saveRDS(res_plot, file = &quot;res_plot.rds&quot;)
res_plot &lt;- readRDS(&quot;./01_Datasets/res_plot.rds&quot;)

res_plot %&gt;% kbl() %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;))</code></pre>
<table class="table table-striped table-hover table-condensed" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
model
</th>
<th style="text-align:right;">
log.estimate
</th>
<th style="text-align:right;">
log.conf.low
</th>
<th style="text-align:right;">
log.conf.high
</th>
<th style="text-align:left;">
estimate
</th>
<th style="text-align:left;">
conf.low
</th>
<th style="text-align:left;">
conf.high
</th>
<th style="text-align:left;">
p.value
</th>
<th style="text-align:left;">
estimate_lab
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
A
</td>
<td style="text-align:right;">
-0.6846879
</td>
<td style="text-align:right;">
-0.8947661
</td>
<td style="text-align:right;">
-0.4746097
</td>
<td style="text-align:left;">
0.50
</td>
<td style="text-align:left;">
0.41
</td>
<td style="text-align:left;">
0.62
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
0.50 (0.41-0.62)
</td>
</tr>
<tr>
<td style="text-align:left;">
B
</td>
<td style="text-align:right;">
-0.0525578
</td>
<td style="text-align:right;">
-0.4201151
</td>
<td style="text-align:right;">
0.3149994
</td>
<td style="text-align:left;">
0.95
</td>
<td style="text-align:left;">
0.66
</td>
<td style="text-align:left;">
1.37
</td>
<td style="text-align:left;">
0.78
</td>
<td style="text-align:left;">
0.95 (0.66-1.37)
</td>
</tr>
<tr>
<td style="text-align:left;">
C
</td>
<td style="text-align:right;">
-0.0864096
</td>
<td style="text-align:right;">
-0.4567781
</td>
<td style="text-align:right;">
0.2839589
</td>
<td style="text-align:left;">
0.92
</td>
<td style="text-align:left;">
0.63
</td>
<td style="text-align:left;">
1.33
</td>
<td style="text-align:left;">
0.65
</td>
<td style="text-align:left;">
0.92 (0.63-1.33)
</td>
</tr>
<tr>
<td style="text-align:left;">
D
</td>
<td style="text-align:right;">
-0.1214702
</td>
<td style="text-align:right;">
-0.5869119
</td>
<td style="text-align:right;">
0.3439714
</td>
<td style="text-align:left;">
0.89
</td>
<td style="text-align:left;">
0.56
</td>
<td style="text-align:left;">
1.41
</td>
<td style="text-align:left;">
0.61
</td>
<td style="text-align:left;">
0.89 (0.56-1.41)
</td>
</tr>
<tr>
<td style="text-align:left;">
E
</td>
<td style="text-align:right;">
-0.4216403
</td>
<td style="text-align:right;">
-0.8819218
</td>
<td style="text-align:right;">
0.0386411
</td>
<td style="text-align:left;">
0.66
</td>
<td style="text-align:left;">
0.41
</td>
<td style="text-align:left;">
1.04
</td>
<td style="text-align:left;">
0.07
</td>
<td style="text-align:left;">
0.66 (0.41-1.04)
</td>
</tr>
<tr>
<td style="text-align:left;">
F
</td>
<td style="text-align:right;">
-0.2575052
</td>
<td style="text-align:right;">
-0.5093796
</td>
<td style="text-align:right;">
-0.0056308
</td>
<td style="text-align:left;">
0.77
</td>
<td style="text-align:left;">
0.60
</td>
<td style="text-align:left;">
0.99
</td>
<td style="text-align:left;">
0.045
</td>
<td style="text-align:left;">
0.77 (0.60-0.99)
</td>
</tr>
<tr>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
0.0526079
</td>
<td style="text-align:right;">
-0.2671835
</td>
<td style="text-align:right;">
0.3723992
</td>
<td style="text-align:left;">
1.05
</td>
<td style="text-align:left;">
0.77
</td>
<td style="text-align:left;">
1.45
</td>
<td style="text-align:left;">
0.75
</td>
<td style="text-align:left;">
1.05 (0.77-1.45)
</td>
</tr>
<tr>
<td style="text-align:left;">
H
</td>
<td style="text-align:right;">
0.0427286
</td>
<td style="text-align:right;">
-0.2871902
</td>
<td style="text-align:right;">
0.3726475
</td>
<td style="text-align:left;">
1.04
</td>
<td style="text-align:left;">
0.75
</td>
<td style="text-align:left;">
1.45
</td>
<td style="text-align:left;">
0.80
</td>
<td style="text-align:left;">
1.04 (0.75-1.45)
</td>
</tr>
<tr>
<td style="text-align:left;">
I
</td>
<td style="text-align:right;">
-0.4624971
</td>
<td style="text-align:right;">
-0.7340078
</td>
<td style="text-align:right;">
-0.1909864
</td>
<td style="text-align:left;">
0.63
</td>
<td style="text-align:left;">
0.48
</td>
<td style="text-align:left;">
0.83
</td>
<td style="text-align:left;">
&lt;0.001
</td>
<td style="text-align:left;">
0.63 (0.48-0.83)
</td>
</tr>
<tr>
<td style="text-align:left;">
J
</td>
<td style="text-align:right;">
0.0798845
</td>
<td style="text-align:right;">
-0.2224750
</td>
<td style="text-align:right;">
0.3822440
</td>
<td style="text-align:left;">
1.08
</td>
<td style="text-align:left;">
0.80
</td>
<td style="text-align:left;">
1.47
</td>
<td style="text-align:left;">
0.60
</td>
<td style="text-align:left;">
1.08 (0.80-1.47)
</td>
</tr>
<tr>
<td style="text-align:left;">
Model
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:right;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
p-value
</td>
<td style="text-align:left;">
Hazard Ratio (95% CI)
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>p_left &lt;-
  res_plot  |&gt;
  ggplot(aes(y = model)) +
  geom_text(aes(x = 0, label = model), hjust = 0, fontface = &quot;bold&quot;) +
  geom_text(
    aes(x = 1, label = estimate_lab),
    hjust = 0,
    fontface = ifelse(res_plot$estimate_lab == &quot;Hazard Ratio (95% CI)&quot;, &quot;bold&quot;, &quot;plain&quot;) 
  ) +
  ## Remove the background and edit the sizing
  theme_void() +
  coord_cartesian(xlim = c(0, 4))
p_left</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-2-2.png" width="1152" /></p>
<pre class="r"><code>## Create p-value annotations
p_right &lt;-
  res_plot  |&gt;
  ggplot() +
  geom_text(
    aes(x = 0, y = model, label = p.value),
    hjust = 0,
    fontface = ifelse(res_plot$p.value == &quot;p-value&quot;, &quot;bold&quot;, &quot;plain&quot;)
  ) +
  theme_void() 
p_right</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-2-3.png" width="1152" /></p>
<pre class="r"><code>## Merge tables with plot
# library(&quot;gridExtra&quot;)
grid.arrange(p_left, p_mid, p_right, 
             layout_matrix = matrix(c(1,1,1,2,2,2,2,2,2,2,2,2,3,3), nrow = 1))</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-2-4.png" width="1152" /></p>
<pre class="r"><code>## ggsave(&quot;forest-plot.eps&quot;, width=9, height=4)</code></pre>
</div>
</div>
<div id="meta-package" class="section level1">
<h1>meta package</h1>
<p>The most common way to visualize meta-analyses is through forest
plots. Such plots provide a graphical display of the observed effect,
confidence interval, and usually also the weight of each study. They
also display the pooled effect we have calculated in a meta-analysis.
Overall, this allows others to quickly examine the precision and spread
of the included studies, and how the pooled effect relates to the
observed effect sizes.</p>
<p>The {meta} package has an in-built function which makes it very easy
to produce beautiful forest plots directly in R. The function has a
broad functionality and allows one to change the appearance of the plot
as desired.</p>
<div id="general-1" class="section level2">
<h2>General</h2>
<pre class="r"><code>## https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/forest.html
# library(&quot;meta&quot;)
data(Fleiss93)
Fleiss93 %&gt;%
  kable(&quot;html&quot;, caption = &quot;Fleiss93 Data from the irr Package&quot;) %&gt;%
  kable_styling(bootstrap_options = c(&quot;striped&quot;, &quot;hover&quot;, &quot;condensed&quot;, &quot;responsive&quot;), 
                full_width = FALSE, 
                position = &quot;left&quot;)</code></pre>
<table class="table table-striped table-hover table-condensed table-responsive" style="width: auto !important; ">
<caption>
Fleiss93 Data from the irr Package
</caption>
<thead>
<tr>
<th style="text-align:left;">
study
</th>
<th style="text-align:right;">
year
</th>
<th style="text-align:right;">
event.e
</th>
<th style="text-align:right;">
n.e
</th>
<th style="text-align:right;">
event.c
</th>
<th style="text-align:right;">
n.c
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
MRC-1
</td>
<td style="text-align:right;">
1974
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
615
</td>
<td style="text-align:right;">
67
</td>
<td style="text-align:right;">
624
</td>
</tr>
<tr>
<td style="text-align:left;">
CDP
</td>
<td style="text-align:right;">
1976
</td>
<td style="text-align:right;">
44
</td>
<td style="text-align:right;">
758
</td>
<td style="text-align:right;">
64
</td>
<td style="text-align:right;">
771
</td>
</tr>
<tr>
<td style="text-align:left;">
MRC-2
</td>
<td style="text-align:right;">
1979
</td>
<td style="text-align:right;">
102
</td>
<td style="text-align:right;">
832
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:right;">
850
</td>
</tr>
<tr>
<td style="text-align:left;">
GASP
</td>
<td style="text-align:right;">
1979
</td>
<td style="text-align:right;">
32
</td>
<td style="text-align:right;">
317
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:right;">
309
</td>
</tr>
<tr>
<td style="text-align:left;">
PARIS
</td>
<td style="text-align:right;">
1980
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:right;">
810
</td>
<td style="text-align:right;">
52
</td>
<td style="text-align:right;">
406
</td>
</tr>
<tr>
<td style="text-align:left;">
AMIS
</td>
<td style="text-align:right;">
1980
</td>
<td style="text-align:right;">
246
</td>
<td style="text-align:right;">
2267
</td>
<td style="text-align:right;">
219
</td>
<td style="text-align:right;">
2257
</td>
</tr>
<tr>
<td style="text-align:left;">
ISIS-2
</td>
<td style="text-align:right;">
1988
</td>
<td style="text-align:right;">
1570
</td>
<td style="text-align:right;">
8587
</td>
<td style="text-align:right;">
1720
</td>
<td style="text-align:right;">
8600
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>metaresult&lt;-metabin(event.e, n.e,event.c,n.c,data=Fleiss93,sm=&quot;RR&quot;,
                    studlab=paste(study, year),random=FALSE)
m.gen &lt;- metagen(TE, seTE, studlab, n.e = n.e + n.c, data = metaresult, sm = &quot;RR&quot;)

meta::forest(metaresult)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-3-1.png" width="1152" /></p>
<pre class="r"><code>meta::forest(m.gen, 
             sortvar = TE,
             prediction = TRUE, 
             print.tau2 = FALSE,
             leftlabs = c(&quot;Author&quot;, &quot;g&quot;, &quot;SE&quot;))</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-3-2.png" width="1152" /></p>
</div>
<div id="layout-types" class="section level2">
<h2>Layout Types</h2>
<p>The meta::forest function has two “pre-packaged” layouts, which we
can use to bring our forest plot into a specific format without having
to specify numerous arguments. One of them is the “JAMA” layout, which
gives us a forest plot according to the guidelines of the Journal of the
American Medical Association. This layout may be used if you want to
publish your meta-analysis in a medical journal.</p>
<pre class="r"><code>meta::forest(m.gen, layout = &quot;JAMA&quot;)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-4-1.png" width="1152" /></p>
<p>The other layout is “RevMan5”, which produces a forest plot similar
to the ones generated by Cochrane’s Review Manager 5.</p>
<pre class="r"><code>meta::forest(m.gen, layout = &quot;RevMan5&quot;)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-5-1.png" width="1152" /></p>
</div>
<div id="saving-the-forest-plots" class="section level2">
<h2>Saving the Forest Plots</h2>
<p>Forest plots generated by meta::forest can be saved as a PDF, PNG, or
scalable vector graphic (SVG) file. In contrast to other plots generated
through base R or the {ggplot2} package, the output of meta::forest is
not automatically re-scaled when we save it as a file. This means that
forest plots are sometimes cut off on two or four sides, and we have to
adjust the width and height manually so that everything is visible.</p>
<p>The pdf, png and svg function can be used to save plots via R code.
We have to start with a call to one of these functions, which tells R
that the output of the following code should be saved in the document.
Then, we add our call to the meta::forest function. In the last line, we
have to include dev.off(), which will save the generated output to the
file we specified above.</p>
<p>All three functions require us to specify the file argument, which
should contain the name of the file. The file is then automatically
saved in the working directory under that name. Additionally, we can use
the width and height argument to control the size of the plot, which can
be helpful when the output is cut off.</p>
<p>Assuming we want to save our initial forest plot under the name
“forestplot”, we can use the following code to generate a PDF, PNG and
SVG file.</p>
<p><strong>PDF</strong></p>
<pre class="r"><code>pdf(file = &quot;forestplot.pdf&quot;, width = 8, height = 7)

meta::forest(m.gen, 
             sortvar = TE,
             prediction = TRUE, 
             print.tau2 = FALSE,
             leftlabs = c(&quot;Author&quot;, &quot;g&quot;, &quot;SE&quot;))

dev.off()</code></pre>
<p><strong>PNG</strong></p>
<pre class="r"><code>png(file = &quot;forestplot.png&quot;, width = 2800, height = 2400, res = 300)

meta::forest(m.gen, 
             sortvar = TE,
             prediction = TRUE, 
             print.tau2 = FALSE,
             leftlabs = c(&quot;Author&quot;, &quot;g&quot;, &quot;SE&quot;))

dev.off()</code></pre>
<p><strong>SVG</strong></p>
<pre class="r"><code>svg(file = &quot;forestplot.svg&quot;, width = 8, height = 7)

meta::forest(m.gen, 
             sortvar = TE,
             prediction = TRUE, 
             print.tau2 = FALSE,
             leftlabs = c(&quot;Author&quot;, &quot;g&quot;, &quot;SE&quot;))

dev.off()</code></pre>
</div>
<div id="drapery-plots" class="section level2">
<h2>Drapery plots</h2>
<p>Drapery plots are an alternative method for visualizing meta-analysis
results. While forest plots are the standard tool for summarizing and
displaying meta-analysis findings, drapery plots provide a different
perspective that can be useful, especially in discussions involving
statistical significance and confidence interpretation.</p>
<p>A forest plot typically shows a 95% confidence interval for each
study, along with a pooled effect. However, this is based on a fixed
significance threshold, usually p &lt; 0.05. Because of this, forest
plots visually separate results into “significant” and “not significant”
based on that threshold. This binary distinction can be misleading or
overly simplistic.</p>
<p>In recent years, there has been growing concern about over-reliance
on p-values and the traditional threshold of significance. Critics argue
that this has contributed to reproducibility problems in scientific
research. To address this issue, researchers have proposed the use of
p-value functions, which are visualized through drapery plots.</p>
<p>A p-value function shows the relationship between the assumed p-value
and the corresponding confidence interval. Instead of showing a single
interval (like the 95% CI), it shows a range of intervals corresponding
to all possible significance levels. As a result, a drapery plot shows
how robust the evidence is across different thresholds.</p>
<p>In a drapery plot:</p>
<ul>
<li>The x-axis represents the effect size.</li>
<li>The y-axis represents either the p-value or the z-value, depending
on the setting.</li>
<li>Each study is represented by a curve that looks like an inverted V.
The peak of the V indicates the point estimate of the effect size.</li>
<li>The pooled estimate is shown as a thicker curve.</li>
<li>The shaded area around the pooled estimate indicates the prediction
interval, which shows the expected range of effects in future
studies.</li>
<li>Dashed horizontal lines may indicate standard significance
thresholds (e.g., p = 0.05, p = 0.01), helping to visualize when the
estimate becomes statistically significant.</li>
</ul>
<p>Interpreting the plot:</p>
<ul>
<li>The highest point of each curve represents the study’s point
estimate.</li>
<li>As you move downward (i.e., smaller p-values), the confidence
intervals widen.</li>
<li>The closer a study’s curve gets to the x-axis without touching the
zero line, the stronger the evidence that the effect is different from
zero.</li>
<li>If the thick pooled curve reaches the vertical line at zero only
when p is already very small (e.g., p &lt; 0.01), this implies strong
evidence against the null hypothesis.</li>
</ul>
<p>Despite their strengths, drapery plots are not meant to replace
forest plots. They do not show raw effect size estimates or confidence
intervals in the structured, tabular way that forest plots do.
Therefore, they are best used as a complement to forest plots, not as a
substitute.</p>
<p>In summary, drapery plots enhance the interpretability of
meta-analysis results by showing a continuous spectrum of uncertainty.
They help reduce over-dependence on arbitrary p-value thresholds and
allow a more nuanced view of statistical evidence.</p>
<pre class="r"><code>drapery(m.gen, 
        labels = &quot;studlab&quot;,
        type = &quot;pval&quot;, 
        legend = FALSE)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-6-1.png" width="1152" /></p>
</div>
</div>
<div id="ggforestplot-package-based-on-se" class="section level1">
<h1>ggforestplot package (based on SE)</h1>
<p>The R package ggforestplot allows to plot vertical forest plots,
a.k.a. blobbograms, and it’s based on ggplot2, see more click (here)[<a
href="https://nightingalehealth.github.io/ggforestplot/articles/ggforestplot.html"
class="uri">https://nightingalehealth.github.io/ggforestplot/articles/ggforestplot.html</a>]</p>
<div id="basic-forestplot" class="section level3">
<h3>Basic Forestplot</h3>
<pre class="r"><code># devtools::install_github(&quot;NightingaleHealth/ggforestplot&quot;)
# library(&quot;ggforestplot&quot;)

df &lt;-
  ggforestplot::df_linear_associations %&gt;%
  filter(
    trait == &quot;BMI&quot;,
    dplyr::row_number() &lt;= 30
  )

ggforestplot::forestplot(
  df = df,
  name = name,
  estimate = beta,
  se = se,
  pvalue = pvalue,
  psignif = 0.002,
  xlab = &quot;1-SD increment in BMI\nper 1-SD increment in biomarker concentration&quot;,
  title = &quot;Associations of blood biomarkers to BMI&quot;
)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-7-1.png" width="1152" /></p>
</div>
<div id="comparing-several-traits" class="section level3">
<h3>Comparing several traits</h3>
<pre class="r"><code># Extract the biomarker names
selected_bmrs &lt;- df %&gt;% pull(name)

# Filter the demo dataset for the biomarkers above and all three traits:
# BMI, HOMA-IR and fasting glucose
df_compare_traits &lt;-
  ggforestplot::df_linear_associations %&gt;%
  filter(name %in% selected_bmrs) %&gt;%
  # Set class to factor to set order of display.
  mutate(
    trait = factor(
      trait,
      levels = c(&quot;BMI&quot;, &quot;HOMA-IR&quot;, &quot;Fasting glucose&quot;)
    )
  )

# Draw a forestplot of cross-sectional, linear associations
# Notice how the df variable &#39;trait&#39; is used here to color the points
ggforestplot::forestplot(
  df = df_compare_traits,
  estimate = beta,
  pvalue = pvalue,
  psignif = 0.002,
  xlab = &quot;1-SD increment in cardiometabolic trait\nper 1-SD increment in biomarker concentration&quot;,
  title = &quot;Biomarker associations to metabolic traits&quot;,
  colour = trait
)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-8-1.png" width="1152" /></p>
</div>
<div id="grouping-the-biomarkers" class="section level3">
<h3>Grouping the biomarkers</h3>
<pre class="r"><code># library(&quot;ggforce&quot;)
# Filter df_NG_biomarker_metadata, that contain the groups, for only the 30
# biomarkers under discussion
df_grouping &lt;-
  df_NG_biomarker_metadata %&gt;%
  filter(name %in% df_compare_traits$name)
# Join the association data frame df_compare_traits with group data
df_compare_traits_groups &lt;-
  df_compare_traits %&gt;%
  # use right_join, with df_grouping on the right, to preserve the order of
  # biomarkers it specifies.
  dplyr::right_join(., df_grouping, by = &quot;name&quot;) %&gt;%
  dplyr::mutate(
    group = factor(.data$group, levels = unique(.data$group))
  )

# Draw a forestplot of cross-sectional, linear associations.
forestplot(
  df = df_compare_traits_groups,
  estimate = beta,
  pvalue = pvalue,
  psignif = 0.002,
  xlab = &quot;1-SD increment in cardiometabolic trait\nper 1-SD increment in biomarker concentration&quot;,
  colour = trait
) +
  ggforce::facet_col(
    facets = ~group,
    scales = &quot;free_y&quot;,
    space = &quot;free&quot;
  )</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-9-1.png" width="1152" /></p>
</div>
<div id="ggplot2" class="section level2">
<h2>ggplot2</h2>
<pre class="r"><code>Master_Cancer_D01 &lt;- read_excel(&quot;./01_Datasets/HR_Ratio_2024.02.09.xlsx&quot;)

Breast_Cancer_Factor &lt;- c(
  &quot;3. Impact of Adjuvant Chemotherapy on Breast Cancer Survival&quot;,
  &quot;3. Breast Cancer Diagnosis to Surgery Time&quot;,
  &quot;2. Platinum Chemotherapy for Early Triple-Negative Breast Cancer&quot;,
  &quot;2. Breast Cancer Diagnosis to Surgery Time&quot;,
  &quot;1. Trastuzumab Regimen in HER2+ Early-Stage Breast Cancer: Meta-Analysis&quot;,
  &quot;1. Breast Cancer Diagnosis to Surgery Time&quot;
  )
 
Breast_Cancer_Label&lt;- c(
  &quot; Impact of Adjuvant Chemotherapy on Breast Cancer Survival&quot;,
  &quot; Breast Cancer Diagnosis to Surgery Time&quot;,
  &quot;  Platinum Chemotherapy for Early Triple-Negative Breast Cancer&quot;,
  &quot;  Breast Cancer Diagnosis to Surgery Time&quot;,
  &quot;   Trastuzumab Regimen in HER2+ Early-Stage Breast Cancer: Meta-Analysis&quot;,
  &quot;   Breast Cancer Diagnosis to Surgery Time&quot;
  ) 
 
Breast_Cancer_Label &lt;- paste0(Breast_Cancer_Label,
                              &quot;\n&quot;,
                              rev(Master_Cancer_D01[which(Master_Cancer_D01$CancerType == &quot;Breast Cancer&quot;),]$Cite))

Breast_Cancer_Data &lt;- Master_Cancer_D01 %&gt;%
  filter(CancerType == &quot;Breast Cancer&quot;) %&gt;%
  mutate(Category = factor(Category, levels = Breast_Cancer_Factor, labels=Breast_Cancer_Label))
  
Breast_Cancer_Plot &lt;- 
  Breast_Cancer_Data %&gt;%
  ggplot(aes(x = Estimate, y = Category)) + 
  scale_x_continuous(trans = &quot;log10&quot;, 
                breaks = c(0.6, 0.8, 1.0, 1.2, 1.5),
                limits = c(0.5, 1.5)) + 
  theme_forest() + 
  scale_colour_ng_d() + 
  scale_fill_ng_d() + 
  geom_stripes() + 
  geom_vline(xintercept = 1, linetype = &quot;dashed&quot;, size = 0.5, colour = &quot;black&quot;) + 
  geom_hline(yintercept = 4.5, linetype = &quot;solid&quot;, size = 1.2, colour = &quot;black&quot;) + 
  geom_hline(yintercept = 2.5, linetype = &quot;solid&quot;, size = 1.2, colour = &quot;black&quot;) + 
  geom_effect(ggplot2::aes(xmin = conf.low, xmax = conf.high), 
        position = ggstance::position_dodgev(height = 0.5)) + 
        ggplot2::scale_shape_manual(values = c(21L, 22L, 23L, 
            24L, 25L)) + guides(colour = guide_legend(reverse = TRUE), 
        shape = guide_legend(reverse = TRUE)) +
  annotate(&quot;text&quot;, x = 0.65, y = Inf, label = &quot;Relative Improvement in\nOverall Survival From Treatment&quot;, 
           hjust = 0, vjust = 1, colour = &quot;black&quot;,size = 5) +
  annotate(&quot;text&quot;, x = 1.08, y = Inf, label = &quot;Relative Decrease in\nOverall Survival From Delay&quot;, 
           hjust = 0, vjust = 1, colour = &quot;black&quot;,size = 5) +
  geom_textbox(aes(label = paste0(format(Estimate, digits = 3),
                             &quot; [&quot;,
                             format(conf.low, digits = 3),&quot;,&quot;,
                             format(conf.high, digits = 3),
                             &quot;]&quot;), 
              x = Estimate), hjust = 0.5, vjust = 1.3, width = 0.132)+ 
  labs(title = &quot;Figure 1: Breast Cancer&quot;) +
  labs(subtitle = &quot;&quot;) +
  labs(x = &quot;Hazard Ratio&quot;) +
  labs(y = &quot;&quot;) +
  theme(text = element_text(size = 16)) 

png(&#39;./02_Plots/Visualization/ForestPlot/HR_Breast_Cancer.png&#39;,width=16, height=8,unit=&quot;in&quot;, res=600)
Breast_Cancer_Plot
dev.off()</code></pre>
<pre><code>## cairo_pdf 
##         2</code></pre>
<pre class="r"><code>knitr::include_graphics(&quot;./02_Plots/Visualization/ForestPlot/HR_Breast_Cancer.png&quot;)</code></pre>
<p><img src="02_Plots/Visualization/ForestPlot/HR_Breast_Cancer.png" width="4800" /></p>
</div>
</div>
<div id="alternative-to-forest-plot-dlqi-data" class="section level1">
<h1>Alternative to Forest Plot DLQI Data</h1>
<div id="errorbar" class="section level2">
<h2>Errorbar</h2>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_1.png" /></p>
<p><strong>Source: Figure 2 in Muntyanu A, Gabrielli S, Donovan J,
Gooderham M, Guenther L, Hanna S, et al. The burden of alopecia areata:
A scoping review focusing on quality of life, mental health and work
productivity. J Eur Acad Dermatol Venereol. 2023; 37: 1490–1520. <a
href="https://doi.org/10.1111/jdv.18926"
class="uri">https://doi.org/10.1111/jdv.18926</a></strong></p>
<pre class="r"><code># Example data frame
data &lt;- data.frame(
  Study_Number = 1:12,
  Mean_DLQI = c(4.8, 3.5, 2.1, 10.69, 6.1, 7.9, 7.7, 8.1636, 7.21, 5.8, 6.8, 6.4),
  Lower_CI = c(2, 1, 1, 9, 4, 6, 5, 7, 6, 4, 5, 5),  # Example lower CI values
  Upper_CI = c(8, 6, 4, 12, 8, 10, 10, 10, 9, 8, 9, 8)  # Example upper CI values
)
 

ggplot(data, aes(x = factor(Study_Number), y = Mean_DLQI)) +
  geom_col(fill = &quot;salmon&quot;, width = 0.7) +  # Bars with salmon fill
  geom_errorbar(aes(ymin = Lower_CI, ymax = Upper_CI), width = 0.2, color = &quot;black&quot;) +  # Error bars
  geom_label(aes(label = round(Mean_DLQI, 2)), vjust = -1.5, fill = &quot;white&quot;, color = &quot;black&quot;, 
             fontface = &quot;bold&quot;, label.size = 0.5) +  # Label with white background
  labs(
    title = &quot;DLQI in Alopecia Areata Studies&quot;,
    x = &quot;Study Number&quot;,
    y = &quot;DLQI Score&quot;
  ) +
  theme_minimal() +
  theme(
    plot.title = element_text(hjust = 0.5),  # Centering the title
    axis.title.x = element_text(face = &quot;bold&quot;),
    axis.title.y = element_text(face = &quot;bold&quot;),
    axis.text.x = element_text(angle = 90, hjust = 1)  # Angle x axis text for readability
  )</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-11-1.png" width="1152" /></p>
</div>
<div id="mean-dlqi" class="section level2">
<h2>Mean DLQI</h2>
<p>This is a simple, but clean visualisation with logical ordering of
studies based on mean DLQI score. As per some of the other submissions,
the axes have been flipped here compared to the original, published
visualisation. This allows for the author name and sample size to be
displayed on the label, as well as sample size. However, sample size
could more effectively be encoded within the plot itself, to save the
reader from having to study the axis labels at length. There is a
telling title added with colours nicely included to link the title to
the plot itself. Additional information is deferred to a footnote, and
here shading is used to display the meaningful DLQI categories (again,
these categories are for DLQI scores at a total level). These categories
motivate a panel discussion on how useful it is to display these
categories, corresponding to individual patients, for a presentation of
the mean. The issue here is that, due to DLQI only taking integer scores
for each patient, the meaningful categories have gaps between their
extremes. The author has chosen to leave these as blank space in the
plot, which could be confusing for a reader who is not familiar with
DLQI scores or the categories. An alternative would be to use the
midpoints (1.5 and 5.5 for the displayed categories) as the values to
change the shading at. This submission also lead to a nice discussion on
minimising white space, while avoiding confusion. The user only takes
the x-axis as far as a score of 10, which captures all the observed
data, while presenting it in the maximum amount of space to allow
comparison between the categories. However, it could mislead the reader
that several studies have a mean DLQI score close to the upper limit of
those which can be taken by the tool (which is not the case, DLQI scores
can be as high as 30). Taking the x-axis as far as 30 here would give an
easier interpretation of the overall experience of patients with
alopecia areata, compared with what is possible.</p>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_2.png" /></p>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_2b.png" /></p>
<pre class="r"><code># Sample data
data &lt;- data.frame(
  Study = c(&quot;Abedini et al.&quot;, &quot;Liu et al.&quot;, &quot;Shi et al.&quot;, &quot;Ghajarzadeh et al.&quot;,
            &quot;Velez-Muniz et al.&quot;, &quot;Qi et al.&quot;, &quot;Ito et al.&quot;, &quot;Ferentinos et al.&quot;, &quot;Andersen et al.&quot;),
  N = c(176, 383, 532, 100, 126, 698, 400, 52, 1494),
  Mean_DLQI = c(1.5, 2.5, 3.0, 4.0, 5.5, 4.5, 6.0, 6.5, 7.0),
  Lower_CI = c(0.5, 1.5, 2.5, 3.5, 4.0, 3.5, 5.5, 6.0, 6.0),
  Upper_CI = c(2.5, 3.5, 3.5, 4.5, 7.0, 5.5, 6.5, 8.0, 10.0)
)

data &lt;- data %&gt;%
  arrange(Mean_DLQI) %&gt;%
  mutate(Study = factor(Study, levels = Study))

# Determine the number of studies to set the y position for annotations
num_studies &lt;- nrow(data)

ggplot(data, aes(y = Study, x = Mean_DLQI, xmin = Lower_CI, xmax = Upper_CI)) +
  geom_rect(aes(xmin = 0, xmax = 1, ymin = -Inf, ymax = Inf), fill = &quot;#FFD885&quot;, alpha = 0.2) +  # None
  geom_rect(aes(xmin = 2, xmax = 5, ymin = -Inf, ymax = Inf), fill = &quot;#E4D383&quot;, alpha = 0.2) +  # Small
  geom_rect(aes(xmin = 6, xmax = 10, ymin = -Inf, ymax = Inf), fill = &quot;#9BBC93&quot;, alpha = 0.2) +  # Moderate
  geom_pointrange(size = 0.5, color = &quot;black&quot;) +
  annotate(&quot;text&quot;, x = 1, y = num_studies + 0.5, label = &quot;None&quot;, vjust = 0) +
  annotate(&quot;text&quot;, x = 3.5, y = num_studies + 0.5, label = &quot;Small&quot;, vjust = 0) +
  annotate(&quot;text&quot;, x = 7.5, y = num_studies + 0.5, label = &quot;Medium&quot;, vjust = 0) +
  scale_y_discrete(limits = c(levels(data$Study), &quot; &quot;), expand = c(0, 0.1)) +  # Extend y-axis
  scale_x_continuous(&quot;Mean DLQI Score&quot;, breaks = 0:10) +
  labs(y = &quot;Study&quot;, title = &quot;Across Studies, alopecia areata had on average a small or moderate impact on patients&#39; lives&quot;) +
  theme_minimal() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-12-1.png" width="1152" /></p>
<pre class="r"><code># Clear environment and read in libraries
# rm(list = ls()) 

# Simulate 4 studies with different treatments/sample sizes
subjects &lt;- as.character(seq(1, 850))

trts &lt;- c(sample(c(&#39;Placebo&#39;, &#39;A&#39;, &#39;B&#39;),
                 size = 300, replace = T),
          sample(c(&#39;Placebo&#39;, &#39;C&#39;),
                 size = 100, replace = T),
          sample(c(&#39;B&#39;, &#39;C&#39;),
                 size = 300, replace = T),
          sample(c(&#39;B&#39;, &#39;D&#39;),
                 size = 150, replace = T))

studies &lt;- c(rep(&#39;Study 1&#39;, 300),
             rep(&#39;Study 2&#39;, 100),
             rep(&#39;Study 3&#39;, 300),
             rep(&#39;Study 4&#39;, 150))

visits &lt;- c(rep(seq(0, 100, length.out = 10), each = 300),
            rep(seq(0, 75,  length.out  = 8), each = 100),
            rep(seq(0, 100, length.out = 10), each = 300),
            rep(seq(0, 50,  length.out = 6),  each = 150))

# Random study effects
study_effs &lt;- c(rep(rnorm(1, sd = 3), 300),
                rep(rnorm(1, sd = 3), 100),
                rep(rnorm(1, sd = 3), 300),
                rep(rnorm(1, sd = 3), 150))

# Random subject effects
sub_effs &lt;- rnorm(850, 0, 3)

# Repeat values for each visit e.g. in study 1, each subject needs 10 records
split &lt;- function(vec){
  return(
    c(rep(vec[1:300], 10),
      rep(vec[301:400], 8),
      rep(vec[401:700], 10),
      rep(vec[701:850], 6)))
}

# Create study dataframe
df &lt;- data.frame(subject   = split(subjects),
                 study     = split(studies),
                 trt       = split(trts),
                 study_eff = split(study_effs),
                 sub_eff   = split(sub_effs),
                 time      = as.character(round(visits, 0)))

# Find each subject/treatment combo - will need to merge this on later
subject_df &lt;- df %&gt;%
              select(subject, trt) %&gt;%
              mutate(trt = if_else(trt != &#39;Placebo&#39;,
                                   paste0(&#39;Treatment &#39;, trt),
                                   trt)) %&gt;%
              unique()

# Coefficients for treatment variables
beta0 &lt;- 15
betaA &lt;- 0.2
betaB &lt;- -0.1
betaC &lt;- 0.1
betaD &lt;- 0.05

# Coefficients for interaction variables
gamma &lt;- 0.01
gammaA &lt;- 0.025
gammaB &lt;- -0.1
gammaC &lt;- 0.0001
gammaD &lt;- -0.05


# Find DLQI score for each 
dlqi_calc &lt;- function(n, A, B, C, D, time, study_eff, sub_eff){
  
  dlqi &lt;- beta0 + betaA*A + betaB*B +
          betaC*C + betaD*D + 
          gamma*time + gammaA*A*time + gammaB*B*time +
          gammaC*C*time + gammaD*D*time +
          sub_eff + study_eff
  
  return(rnorm(n, dlqi, 3))
}


dlqi_df &lt;- df %&gt;%
  
           # Create variables needed for dlqi_calc
           pivot_wider(id_cols = c(subject, study, study_eff, sub_eff, time),
                       names_from = trt,
                       values_from = trt,
                       values_fn = length,
                       values_fill = 0) %&gt;%
           mutate(time = as.numeric(time),
                  dlqi = dlqi_calc(nrow(.), A, B, C, D, time, study_eff, sub_eff),
                  
                  # Apply cut-off thresholds
                  dlqi = case_when(
                                   dlqi &lt; 0 ~ 0,
                                   dlqi &gt; 30 ~ 30,
                                   T ~ dlqi)) %&gt;%
           left_join(subject_df, by = &#39;subject&#39;) %&gt;%
           group_by(trt, study, time) %&gt;%
           mutate(mean = mean(dlqi),
                  
                  # Add some jitter so plot points are not over-layed
                  time = case_when(study == &#39;Study 1&#39; &amp; trt == &#39;Placebo&#39; ~ time - 2,
                          study == &#39;Study 1&#39; &amp; trt == &#39;Treatment B&#39; ~ time + 2,
                          study == &#39;Study 2&#39; &amp; trt == &#39;Placebo&#39; ~ time - 1,
                          study == &#39;Study 2&#39; &amp; trt == &#39;Treatment C&#39; ~ time + 1,
                          study == &#39;Study 3&#39; &amp; trt == &#39;Treatment B&#39; ~ time - 1,
                          study == &#39;Study 3&#39; &amp; trt == &#39;Treatment C&#39; ~ time + 1,
                          study == &#39;Study 4&#39; &amp; trt == &#39;Treatment D&#39; ~ time + 1,
                          study == &#39;Study 4&#39; &amp; trt == &#39;Treatment B&#39; ~ time - 1,
                          T ~ time))
  
# Color the subscript treatment labels
color_label &lt;- function(color, trt){
    paste0(&quot;N&quot;,
    &quot;&lt;sub&gt;&lt;span style = &#39;color:&quot;, color, &quot;;&#39;&gt;&quot;, trt, &quot;&lt;/span&gt;&lt;/sub&gt;&lt;/span&gt;&quot;)
}

nP &lt;- color_label(hue_pal()(5)[1], &#39;P&#39;)
nA &lt;- color_label(hue_pal()(5)[2], &#39;A&#39;)
nB &lt;- color_label(hue_pal()(5)[3], &#39;B&#39;)
nC &lt;- color_label(hue_pal()(5)[4], &#39;C&#39;)
nD &lt;- color_label(hue_pal()(5)[5], &#39;D&#39;)

# Create the desired facet labels
study.labs &lt;- c(paste0(&#39;Study 1 &lt;br&gt;&lt;br&gt;&#39;,
                       nP, &#39; = &#39;, nA, &#39; = &#39;, nB, &#39; = 100&#39;),
                paste0(&#39;Study 2 &lt;br&gt;&lt;br&gt;&#39; ,
                       nP, &#39; = &#39;, nC, &#39; = &#39;, &#39;50&#39;),
                paste0(&#39;Study 3 &lt;br&gt;&lt;br&gt;&#39;,
                       nB, &#39; = &#39;, nC, &#39; = &#39;, &#39;100&#39;),
                paste0(&#39;Study 4 &lt;br&gt;&lt;br&gt;&#39;,
                       nB, &#39; = &#39;, nD, &#39; = &#39;, &#39;75&#39;))

names(study.labs) &lt;- c(&#39;Study 1&#39;, &#39;Study 2&#39;, &#39;Study 3&#39;, &#39;Study 4&#39;)

# Create the plot
png(&quot;./02_Plots/Visualization/ForestPlot/DLQI_5.png&quot;, res = 500, height = 5, width = 7, units = &#39;in&#39;)

ggplot(data = dlqi_df) +
  
  geom_point(aes(x = time,
                 y = dlqi,
                 color = trt), alpha = 0.1, size = 0.3) +
  
  facet_wrap(~study, ncol = 1, strip.position = &#39;right&#39;,
             labeller = labeller(study = study.labs)) +
  
  geom_line(aes(x = time,
                y = mean,
                group = trt,
                color = trt),
            size = 0.5, alpha = 0.75) +
  
  theme(panel.background       = element_rect(fill = &#39;gray10&#39;),
        panel.border           = element_rect(color = &#39;gray20&#39;, 
                                              size  = unit(1, &#39;in&#39;), 
                                              fill  = NA),
        plot.margin            = margin(20, 20, 20, 20),
        panel.grid.minor       = element_blank(),
        plot.background        = element_rect(fill  = &#39;gray10&#39;, 
                                              color = &#39;gray10&#39;),
        panel.grid.major       = element_line(color =&#39;gray13&#39;),

        legend.position        = c(0.43, 1.17),
        legend.justification   = &#39;top&#39;,
        legend.background      = element_rect(fill = &#39;gray10&#39;),
        legend.title           = element_blank(),
        legend.key             = element_rect(fill = &#39;gray10&#39;),
        legend.text.align      = 0,
        legend.text            = element_text(color = &#39;gray90&#39;,
                                              size = 8),
        
        strip.background       = element_rect(fill = &#39;gray20&#39;),
        panel.spacing          = unit(0.15, &#39;in&#39;),
        strip.text.y           = element_markdown(angle = 0, 
                                                  color = &#39;white&#39;,
                                                  face  = &#39;bold&#39;, 
                                                  size  = 8),

        axis.text             = element_text(color = &#39;gray70&#39;),
        axis.title.x          = element_text(hjust  = 0, 
                                             color  = &#39;white&#39;,
                                             face   = &#39;bold&#39;,
                                             margin = margin(t = 12.5),
                                             size   = 8),
        plot.title            = element_text(color = &#39;white&#39;, 
                                             hjust = 0,
                                             face = &#39;bold&#39;,
                                             margin = margin(b = 7),
                                             size = 12),
        plot.subtitle         = element_text(color = &#39;gray90&#39;, 
                                             face = &#39;italic&#39;,
                                             size = 7, 
                                            margin = margin(b = 32)),
        plot.title.position   = &#39;plot&#39;,
        plot.caption.position = &#39;plot&#39;,
        plot.caption          = element_text(size = 7, 
                                             color = &#39;white&#39;, 
                                             face = &#39;bold.italic&#39;)) +
  
  
  coord_cartesian(ylim = c(0, 30), clip = &#39;off&#39;) +
  
  guides(color = guide_legend(nrow = 1)) +
  
  labs(x        = &#39;Relative Day&#39;,
       y        = NULL,
       title    = &#39;Mean DLQI Score over Time for Four Different Studies&#39;,
       caption  = &#39;Huw Wilson&#39;,
       subtitle = &#39;Lower DLQI scores indicate less severe symptoms.&#39;)

dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_5.png" /></p>
</div>
<div id="informative-panel" class="section level2">
<h2>Informative Panel</h2>
<p>This is an informative submission which can be broken down into its
different panels. These panels are supplemented by additional links and
informative subheadings to aid understanding. The user is first
presented with some general information about the studies (demographics,
as well as sample size). The cells presenting sample size are shaded in
accordance with the sample size, and these sample sizes are used to
provide a logical ordering of the studies. The choice of the intense
shade for the largest sample size immediately draws our attention to
that study. The next panel then displays small 5-number summaries for
the total DLQI in each study (presenting mean, medians, quartiles and
SD). One panellist did pick out that the choice of display for the
median and quartiles could confuse a reader who does not take care to
read the legend, as they could interpret the display as a boxplot (which
would have the quartiles displayed differently). A shaded is applied
within this column, which is intuitive with ‘clearance’ of issues and
the intense, dark red shade depicting the largest (negative) impact. It
could be argued that the select shades should be based on the DLQI score
range of 0 to 30, as opposed to the observed means for the selected
studies (could mislead that the highest mean of 10.7, here, is towards
the upper extreme of possible DLQI scores). The final panel presents
density dot plots for simulated patient level data. While the individual
points are a little small to make out (meaning we don’t see a lot for
some of the smaller studies), this nicely highlights how summary stats
may not capture the experience of many patients, per other submissions
which have presented simulated patient level data. The mean is added as
a reference line to aid interpretation, and the x-axis presents the full
range of 0 to 30 here, indicating that across studies, the bulk of
patients have DLQI scores towards the lower end of the DLQI range
(mitigating against misunderstanding which could have been caused by the
shading in the previous panel, as well as driving home just how
different some of the sample sizes are)</p>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_3.png" /></p>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_3b.png" /></p>
</div>
<div id="individual-study-dlqi" class="section level2">
<h2>Individual Study DLQI</h2>
<p>The visualisation had simulated the DLQI scores of individual
patients, to provide mean and standard deviations equal to those
published for each of the studies. These individual data points are then
plotted for each of the studies, with the corresponding densities
displayed. This nicely highlights, in this case, how a simple
presentation of mean and standard deviation may not capture the
experience of many of the individual patients within the studies. The
mean for each study is still displayed as a diamond, and has its size
set proportional to the sample size for the respective study. Such a
display is effectively used in some of the other submissions from this
webinar, but here does sometimes cause the mean to be somewhat difficult
to spot, particularly for small studies with a mean value close to an
integer score (where it overlaps with individual data points – see for
instance, study 11). Reference lines are added to display the meaningful
score categories which exist for DLQI total score at a patient level.
These in turn dictate the colours used for each study, with a telling
title added to provide an interpretation of the overall patient
experience. A more convincing message could have been delivered by
applying a logical sorting based on e.g. mean score, rather than study
number.</p>
<p><img src="02_Plots/Visualization/ForestPlot/DLQI_4.png" /></p>
</div>
<div id="double-dot-plot" class="section level2">
<h2>Double Dot Plot</h2>
<p><img src="02_Plots/Visualization/ForestPlot/DoubleDot.png" /></p>
</div>
<div id="lollipopforest-plot" class="section level2">
<h2>Lollipop/forest plot</h2>
<pre class="r"><code># library(RCurl) 
# library(ggrepel)
# library(cowplot)
# library(ggtext)

y &lt;- read.csv(&quot;./01_Datasets/CGI_S_3_groups_csv.csv&quot;) %&gt;%
   dplyr::rename(Group = CGI)

l &lt;- y %&gt;%
  pivot_longer(cols = X1:X7,
               names_to = &quot;Category&quot;,
               values_to = &quot;n&quot;) %&gt;%
  mutate(Category = as.numeric(gsub(&quot;X&quot;, &quot;&quot;, Category))) %&gt;%
  dplyr::group_by(Week, Group) %&gt;%
  arrange(Category) %&gt;%
  mutate(
    CumN = cumsum(n),
    Week = paste(&quot;Week&quot;, Week),
    CumFreq = CumN / Total.sample.size,
    Freq = n / Total.sample.size,
    `Cumulative %` = round(CumFreq * 100, 1),
    `%` = round(Freq * 100, 1)
  ) %&gt;%
  ungroup() %&gt;%
  dplyr::group_by(Week, Category) %&gt;%
  arrange(Group) %&gt;%
  mutate(
    Odds = CumFreq / (1 - CumFreq),
    OR = ifelse(Group == &quot;Active&quot;, round(Odds / Odds[2], 2), NA),
    selogOR = ifelse(Group == &quot;Active&quot;, sqrt(
      1 / CumN + 1 / (Total.sample.size - CumN) + 1 / CumN[2] + 1 / (Total.sample.size[2] -
                                                                       CumN[2])
    ), NA),
    ORlower95CI = ifelse(Group == &quot;Active&quot;, round(exp(log(OR) - 1.96 * selogOR), 2), NA),
    ORupper95CI = ifelse(Group == &quot;Active&quot;, round(exp(log(OR) + 1.96 * selogOR), 2), NA)
  )
l</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Week"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Group"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Total.sample.size"],"name":[3],"type":["int"],"align":["right"]},{"label":["Category"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["n"],"name":[5],"type":["int"],"align":["right"]},{"label":["CumN"],"name":[6],"type":["int"],"align":["right"]},{"label":["CumFreq"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["Freq"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["Cumulative %"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["%"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["Odds"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["OR"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["selogOR"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["ORlower95CI"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["ORupper95CI"],"name":[15],"type":["dbl"],"align":["right"]}],"data":[{"1":"Week 12","2":"Active","3":"108","4":"1","5":"18","6":"30","7":"0.27777778","8":"0.166666667","9":"27.8","10":"16.7","11":"0.38461538","12":"0.10","13":"0.3184287","14":"0.05","15":"0.19"},{"1":"Week 52","2":"Active","3":"108","4":"1","5":"35","6":"85","7":"0.78703704","8":"0.324074074","9":"78.7","10":"32.4","11":"3.69565217","12":"1.00","13":"0.3323942","14":"0.52","15":"1.92"},{"1":"Week 12","2":"Active","3":"108","4":"2","5":"32","6":"152","7":"1.40740741","8":"0.296296296","9":"140.7","10":"29.6","11":"-3.45454545","12":"-0.93","13":"0.1977236","14":"NaN","15":"NaN"},{"1":"Week 52","2":"Active","3":"108","4":"2","5":"29","6":"223","7":"2.06481481","8":"0.268518519","9":"206.5","10":"26.9","11":"-1.93913043","12":"-0.52","13":"0.2259018","14":"NaN","15":"NaN"},{"1":"Week 12","2":"Active","3":"108","4":"3","5":"44","6":"356","7":"3.29629630","8":"0.407407407","9":"329.6","10":"40.7","11":"-1.43548387","12":"-0.39","13":"0.2324214","14":"NaN","15":"NaN"},{"1":"Week 52","2":"Active","3":"108","4":"3","5":"35","6":"471","7":"4.36111111","8":"0.324074074","9":"436.1","10":"32.4","11":"-1.29752066","12":"-0.35","13":"0.2336906","14":"NaN","15":"NaN"},{"1":"Week 12","2":"Active","3":"108","4":"4","5":"10","6":"539","7":"4.99074074","8":"0.092592593","9":"499.1","10":"9.3","11":"-1.25058005","12":"-0.34","13":"0.2340472","14":"NaN","15":"NaN"},{"1":"Week 52","2":"Active","3":"108","4":"4","5":"5","6":"598","7":"5.53703704","8":"0.046296296","9":"553.7","10":"4.6","11":"-1.22040816","12":"-0.33","13":"0.2342528","14":"NaN","15":"NaN"},{"1":"Week 12","2":"Active","3":"108","4":"5","5":"3","6":"618","7":"5.72222222","8":"0.027777778","9":"572.2","10":"2.8","11":"-1.21176471","12":"-0.33","13":"0.2343081","14":"NaN","15":"NaN"},{"1":"Week 52","2":"Active","3":"108","4":"5","5":"3","6":"632","7":"5.85185185","8":"0.027777778","9":"585.2","10":"2.8","11":"-1.20610687","12":"-0.33","13":"0.2343434","14":"NaN","15":"NaN"},{"1":"Week 12","2":"Active","3":"108","4":"6","5":"1","6":"637","7":"5.89814815","8":"0.009259259","9":"589.8","10":"0.9","11":"-1.20415879","12":"-0.33","13":"0.2343554","14":"NaN","15":"NaN"},{"1":"Week 52","2":"Active","3":"108","4":"6","5":"1","6":"643","7":"5.95370370","8":"0.009259259","9":"595.4","10":"0.9","11":"-1.20186916","12":"-0.33","13":"0.2343694","14":"NaN","15":"NaN"},{"1":"Week 12","2":"Active","3":"108","4":"7","5":"0","6":"645","7":"5.97222222","8":"0.000000000","9":"597.2","10":"0.0","11":"-1.20111732","12":"-0.33","13":"0.2343740","14":"NaN","15":"NaN"},{"1":"Week 52","2":"Active","3":"108","4":"7","5":"0","6":"650","7":"6.01851852","8":"0.000000000","9":"601.9","10":"0.0","11":"-1.19926199","12":"-0.32","13":"0.2343852","14":"NaN","15":"NaN"},{"1":"Week 12","2":"Comparator","3":"109","4":"1","5":"10","6":"12","7":"0.11009174","8":"0.091743119","9":"11.0","10":"9.2","11":"0.12371134","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"1","5":"15","6":"50","7":"0.45871560","8":"0.137614679","9":"45.9","10":"13.8","11":"0.84745763","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Comparator","3":"109","4":"2","5":"30","6":"120","7":"1.10091743","8":"0.275229358","9":"110.1","10":"27.5","11":"-10.90909091","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"2","5":"33","6":"194","7":"1.77981651","8":"0.302752294","9":"178.0","10":"30.3","11":"-2.28235294","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Comparator","3":"109","4":"3","5":"59","6":"312","7":"2.86238532","8":"0.541284404","9":"286.2","10":"54.1","11":"-1.53694581","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"3","5":"50","6":"436","7":"4.00000000","8":"0.458715596","9":"400.0","10":"45.9","11":"-1.33333333","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Comparator","3":"109","4":"4","5":"8","6":"529","7":"4.85321101","8":"0.073394495","9":"485.3","10":"7.3","11":"-1.25952381","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"4","5":"9","6":"593","7":"5.44036697","8":"0.082568807","9":"544.0","10":"8.3","11":"-1.22520661","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Comparator","3":"109","4":"5","5":"2","6":"615","7":"5.64220183","8":"0.018348624","9":"564.2","10":"1.8","11":"-1.21541502","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"5","5":"1","6":"629","7":"5.77064220","8":"0.009174312","9":"577.1","10":"0.9","11":"-1.20961538","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Comparator","3":"109","4":"6","5":"0","6":"636","7":"5.83486239","8":"0.000000000","9":"583.5","10":"0.0","11":"-1.20683112","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"6","5":"1","6":"642","7":"5.88990826","8":"0.009174312","9":"589.0","10":"0.9","11":"-1.20450281","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Comparator","3":"109","4":"7","5":"0","6":"645","7":"5.91743119","8":"0.000000000","9":"591.7","10":"0.0","11":"-1.20335821","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Comparator","3":"109","4":"7","5":"0","6":"650","7":"5.96330275","8":"0.000000000","9":"596.3","10":"0.0","11":"-1.20147874","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"1","5":"2","6":"2","7":"0.01851852","8":"0.018518519","9":"1.9","10":"1.9","11":"0.01886792","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"1","5":"5","6":"35","7":"0.32407407","8":"0.046296296","9":"32.4","10":"4.6","11":"0.47945205","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"2","5":"5","6":"90","7":"0.83333333","8":"0.046296296","9":"83.3","10":"4.6","11":"5.00000000","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"2","5":"9","6":"161","7":"1.49074074","8":"0.083333333","9":"149.1","10":"8.3","11":"-3.03773585","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"3","5":"30","6":"253","7":"2.34259259","8":"0.277777778","9":"234.3","10":"27.8","11":"-1.74482759","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"3","5":"30","6":"386","7":"3.57407407","8":"0.277777778","9":"357.4","10":"27.8","11":"-1.38848921","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"4","5":"50","6":"521","7":"4.82407407","8":"0.462962963","9":"482.4","10":"46.3","11":"-1.26150121","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"4","5":"45","6":"584","7":"5.40740741","8":"0.416666667","9":"540.7","10":"41.7","11":"-1.22689076","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"5","5":"15","6":"613","7":"5.67592593","8":"0.138888889","9":"567.6","10":"13.9","11":"-1.21386139","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"5","5":"10","6":"628","7":"5.81481481","8":"0.092592593","9":"581.5","10":"9.3","11":"-1.20769231","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"6","5":"4","6":"636","7":"5.88888889","8":"0.037037037","9":"588.9","10":"3.7","11":"-1.20454545","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"6","5":"4","6":"641","7":"5.93518519","8":"0.037037037","9":"593.5","10":"3.7","11":"-1.20262664","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 12","2":"Placebo","3":"108","4":"7","5":"2","6":"645","7":"5.97222222","8":"0.018518519","9":"597.2","10":"1.9","11":"-1.20111732","12":"NA","13":"NA","14":"NA","15":"NA"},{"1":"Week 52","2":"Placebo","3":"108","4":"7","5":"5","6":"650","7":"6.01851852","8":"0.046296296","9":"601.9","10":"4.6","11":"-1.19926199","12":"NA","13":"NA","14":"NA","15":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>k = 3
labs &lt;- c(&quot;Normal, not at all ill -   1&quot;,
          &quot;Normal to borderline ill - &lt;=2&quot;,
          &quot;Normal to mildly ill - &lt;=3&quot;)

p1 &lt;-
  ggplot(data = l[l$Category &lt;= k,], aes(x = `Cumulative %`, y = Category, col =
                                           Group)) +
  geom_point(size = 5) +
  facet_grid(rows = vars(Week)) +
  labs(title = &quot;Cumulative percentage of response&quot;, y = &quot;&quot;, x = &quot;&quot;) +
  scale_y_continuous(breaks = 1:k, labels = labs) +
  scale_x_continuous(limits = c(0, 100)) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;,
        strip.text = element_blank()) +
  geom_text_repel(aes(label = paste(sprintf(
    &quot;%.1f&quot;, `Cumulative %`
  ), &quot;%&quot;)),
  nudge_y = 0.1)
p1</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-14-1.png" width="1152" /></p>
<pre class="r"><code>legend_b &lt;- get_legend(p1 +
                         guides(color = guide_legend(nrow = 1)) +
                         theme(legend.position = &quot;bottom&quot;,
                               legend.title = element_blank(),
                               legend.text = element_text(size = 12)))
p2 &lt;-
  ggplot(data = l[l$Category &lt;= k &amp;
                    !is.na(l$OR),], aes(x = OR, y = Category)) +
  geom_point(size = 3, pch = 15) +
  geom_linerange(aes(xmin = ORlower95CI, xmax = ORupper95CI)) +
  geom_vline(xintercept = 1, linetype = 3) +
  facet_grid(rows = vars(Week)) +
  scale_y_continuous(breaks = 1:k, labels = labs) +
  scale_x_continuous(limits = c(0, 6)) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;,
        axis.text.y = element_blank()) +
  labs(x = &quot;&quot;, y=&quot;&quot;, title = &quot;OR (Active vs. Comparator)&quot;) +
  geom_text_repel(aes(label = sprintf(&quot;%.2f&quot;, OR)),
                  nudge_y = 0.1) +
  theme(strip.text = element_text(size=12))
p2</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-14-2.png" width="1152" /></p>
<pre class="r"><code>plot_row &lt;- plot_grid(p1, p2, nrow = 1, rel_widths = c(2, 1))

title &lt;- ggdraw() +
  draw_label(
    &quot;Active treatment is more effective than comparator at Week 52&quot;,
    fontface = &#39;bold&#39;,
    x = 0,
    hjust = 0,
    size = 24,
  ) +
  theme(plot.margin = margin(0, 0, 0, 7))

p &lt;- plot_grid(title,
               plot_row,
               legend_b,
               ncol = 1,
               rel_heights = c(0.1, 1, 0.05))
p</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-14-3.png" width="1152" /></p>
</div>
</div>
<div id="visualising-ranking-data" class="section level1">
<h1>Visualising Ranking Data</h1>
<p>To visualise ranking data resulting from a network meta-analysis
(NMA). The NMA synthesised the results of 11 trials comparing 12
treatments for major depression. The provided dataset consisted of the
probabilities that each treatment ranked in each position from Rank 1
(best) to Rank 12 (worst).</p>
<div id="lolly-plot" class="section level2">
<h2>Lolly plot</h2>
<p><img src="02_Plots/Visualization/ForestPlot/Rank.PNG" /></p>
<pre><code>alpha &lt;- 0.10
dta &lt;- read.csv(&quot;./01_Datasets/bayesian_ranking.csv&quot;) %&gt;%
  pivot_longer(cols = -&quot;Treatment&quot;) %&gt;%
  mutate(
    rank = gsub(&quot;^Rank(.*)$&quot;, &quot;\\1&quot;, name),
    rank = factor(rank, levels = unique(rank[order(as.numeric(rank))])),
    rankx = as.numeric(rank)
  ) %&gt;%
  group_by(Treatment) %&gt;%
  mutate(
    lci = cumsum(value) &gt;= !!alpha / 2,
    uci = rev(cumsum(rev(value))) &gt;= !!alpha / 2,
    raw_value = value,
    value = (function(x) (x - min(x)) / (max(x) - min(x)))(value)
  ) %&gt;%
  mutate(avg = as.numeric(rank)[which.max(value)]) %&gt;%
  ungroup() %&gt;%
  mutate(
    Treatment = paste0(stringr::str_to_title(Treatment), &quot;:&quot;),
    Treatment = factor(Treatment, levels = unique(Treatment[order(avg)]))
  )

ci &lt;- dta %&gt;%
  group_by(Treatment) %&gt;%
  summarise(min = min(rankx[lci]), max = max(rankx[uci])) %&gt;%
  ungroup()

gg &lt;- ggplot(
  dta,
  aes(rankx, value, color = value, alpha = value, size = value)
) +
  geom_ribbon(
    aes(ymin = 0, ymax = value),
    fill = &quot;gray90&quot;, color = &quot;transparent&quot;, alpha = 1, size = 1
  ) +
  geom_hline(yintercept = 0, color = &quot;gray80&quot;, lwd = .1) +
  geom_segment(aes(xend = rankx, yend = 0), lwd = 1) +
  geom_point(shape = 21, fill = &quot;white&quot;, stroke = .5) +
  geom_point(shape = 21, fill = &quot;white&quot;, stroke = 0, alpha = 1) +
  geom_segment(
    data = ci,
    mapping = aes(x = min - 0.2, xend = max + 0.2, y = 0, yend = 0),
    color = &quot;gray20&quot;,
    alpha = 1,
    size = .25,
    arrow = arrow(angle = 90, ends = &quot;both&quot;, length = unit(0.1, &quot;lines&quot;))
  ) +
  scale_color_viridis_c(direction = -1, option = &quot;D&quot;, begin = .2, end = .8) +
  scale_x_continuous(breaks = 1:12) +
  scale_alpha_continuous(range = c(.3, 1)) +
  scale_size_continuous(range = c(.5, 2)) +
  xlab(&quot;Ranks&quot;) +
  facet_grid(Treatment ~ ., switch = &quot;y&quot;, scales = &quot;free&quot;) +
  coord_cartesian(clip = &quot;off&quot;) +
  guides(color = &quot;none&quot;, alpha = &quot;none&quot;, size = &quot;none&quot;) +
  theme_minimal() +
  theme(
    panel.grid = element_blank(),
    strip.text.y.left = element_text(
      margin = margin(b = 2, r = 5),
      angle = 0, vjust = 0,
      face = &quot;bold&quot;,
      color = &quot;gray30&quot;,
      size = 7
    ),
    panel.spacing.y = unit(.5, &quot;lines&quot;),
    axis.text.y = element_blank(),
    axis.text.x = element_text(margin = margin(t = 5, b = 5), size = 7),
    axis.title.x = element_text(color = &quot;gray40&quot;, size = 9),
    axis.title.y = element_blank()
  )

gg</code></pre>
<p>This plot uses lollipops to display the probability of a given
treatment ranking in each position. Rank is again displayed on the
x-axis and probability on the y-axis, although a separate sub-plot is
now produced for each treatment and stacked vertically. The ordering of
these subplots is logical, from the most effective treatment at the top
to the least effective treatment at the bottom, allowing the reader to
quickly establish the relative performance of each treatment. Lots of
visual elements are utilised in the plot, with the colour-gradient,
y-axis, point size and transparency all scaled to the treatment-wise
min-max-normalised NMA rank probabilities. The panel suggested that this
could result in some redundancy in the plot and that a subset of these
visual elements could be used to display key probability scales of
interest. Whilst the submitter acknowledges they have been produced by a
somewhat questionable algorithm in this instance, the panel appreciated
the inclusion of the credible intervals (shown by the horizontal black
lines under the lollipops), which add useful information to such plots
without adding significant clutter. Likewise, the inclusion of the
shaded grey background nicely displays how underlying distributions can
be displayed in such plots. The panel were extremely complimentary of
the aesthetics of the plot and all the little details that had been
considered to minimise clutter and produce an extremely clean display.
The main suggestion for improvement was to incorporate the absolute
probabilities somehow, perhaps with one of the several visual elements
used. Whilst the current scaled display makes within-treatment
comparison straightforward, it makes comparisons across treatments more
difficult. For instance, the rank 12 lollipop for Reboxetine is of
similar height to the Rank 11 lollipops for the three treatments above
it, despite us knowing from submission 1 that this probability is
clearly the largest observed for any treatment/rank.</p>
</div>
<div id="use-of-color-intensity" class="section level2">
<h2>Use of color intensity</h2>
<p>In the first, rank is displayed on the x-axis and treatment on the
y-axis, with the intensity of the colour used to display the probability
(ranging from 0=white to 1=red). The left-hand plot displays absolute
probability, whilst the right displays cumulative probability. The panel
appreciated the two different probabilities each being displayed,
suggesting they compliment one another. The cumulative probability plot
gives an area-under-the-curve like interpretation, providing a notion of
overall performance of the treatment. This could be further enhanced by
a logical ordering of the treatments, like seen in submission 2.
Similarly, some of the ‘cleaning’ seen in submission 2 would help remove
clutter from this plot, such as the background and gridlines. Whilst
aesthetically pleasing, the use of a colour gradient could cause
confusion as to the discrete nature of the ranks, since a smooth
gradient is displayed between each successive rank’s colour. This could
be circumvented with the use of a heatmap style display, with a distinct
cell displayed for each treatment and rank. The second half of the
submission displays the same information, but rather than having rank on
the x-axis, each rank corresponds to a concentric ring (with faceting
applied to distinguish treatments). This gives an aesthetically pleasing
plot that would be great for grabbing attention, although the
interpretation is less obvious than the previous plots. That being said,
the cumulative probability plots once again provide an
area-under-the-curve like interpretation which helps in assessing
relative performance (although once more a logical ordering of
treatments would support this). The panel highlighted that the main
challenge with such plots is that outer rings occupy much more space
than inner ones. This makes the worst performing treatment Reboxetine
stand out far more on the left-hand plot than the best performing
Mirtazapine. Inverting the ring ordering could be considered if wanting
to emphasise the best performing treatment.</p>
<pre class="r"><code># Read in packages
# require(gganimate)
# library(&quot;ggpubr&quot;)
# library(modelr)
# library(broom)
# library(grid)
# library(gridExtra)
# library(&quot;transformr&quot;)
# library(ggtext)
# library(&#39;ggforce&#39;)

#data processing

# Set working directory using setwd()
# rnk1 &lt;-  read.csv(&quot;./01_Datasets/bayesian_ranking.csv&quot;)
# rnk2&lt;-tibble(rnk1)
# rnk3 &lt;- mutate(rnk2, rowid_to_column(rnk2, &quot;ID&quot;))
# 
# rnk4 &lt;- rnk3 %&gt;%
#   tidyr::pivot_longer(!c(Treatment,ID), names_to = &quot;ranks&quot;, values_to = &quot;prop&quot;)
# rnk5 &lt;- mutate(rnk4, ranker=str_replace_all(string=ranks, pattern=&quot;Rank&quot;, replacement=&quot;&quot;)) 
# rnk6 &lt;- mutate(rnk5, rnk=as.numeric(ranker), r1=rnk-0.5, r2=rnk+0.5, p2=prop*12, rnkr=1/rnk, clr=letters[4])
# rnk &lt;- rnk6 %&gt;% group_by(Treatment) %&gt;% mutate(cp=cumsum(prop))
# saveRDS(rnk, file = &quot;01_Datasets/rnk.rds&quot;)

#drawing
rnk &lt;- readRDS(&quot;01_Datasets/rnk.rds&quot;)

p&lt;-
ggplot() +
  ggforce::geom_link2(data = rnk,
                      aes(x = rnk, y = Treatment, color=clr, alpha = prop, size=5)) +
  #scale_colour_manual(values = &quot;green&quot;) +
  theme(legend.position=&quot;none&quot;) + 
  scale_x_discrete(name =&quot;Rank&quot;, limits=c(&#39;1&#39;, &#39;2&#39;, &#39;3&#39;, &#39;4&#39;, &#39;5&#39;, &#39;6&#39;, &#39;7&#39;, &#39;8&#39;, &#39;9&#39;, &#39;10&#39;, &#39;11&#39;, &#39;12&#39;))
print(p)     </code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-15-1.png" width="1152" /></p>
<pre class="r"><code>q&lt;-
ggplot() +
  ggforce::geom_link2(data = rnk,
                      aes(x = rnk, y = Treatment, colour=&#39;green&#39;, alpha = cp, size=5))  +
  #scale_colour_manual(values = &quot;green&quot;) +
  theme(legend.position=&quot;none&quot;) + 
  scale_x_discrete(name =&quot;Rank&quot;, limits=c(&#39;1&#39;, &#39;2&#39;, &#39;3&#39;, &#39;4&#39;, &#39;5&#39;, &#39;6&#39;, &#39;7&#39;, &#39;8&#39;, &#39;9&#39;, &#39;10&#39;, &#39;11&#39;, &#39;12&#39;))
print(q)     </code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-15-2.png" width="1152" /></p>
<pre class="r"><code>r1 &lt;- 
  ggplot(data = rnk, aes(x = factor(1), group=factor(rnkr), fill=p2)) +
  geom_bar(width = 1) +
  scale_fill_continuous(low=&#39;white&#39;, high=&#39;green1&#39;, guide=&quot;none&quot;) +
  facet_wrap(~Treatment, ncol=4) + theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
  axis.text.x = element_blank(),
  axis.text.y = element_blank(),
  axis.ticks = element_blank())  
r1</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-15-3.png" width="1152" /></p>
<pre class="r"><code>r &lt;- r1 + coord_polar(theta = &quot;x&quot;)
print(r) </code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-15-4.png" width="1152" /></p>
<pre class="r"><code>s1 &lt;- ggplot(data = rnk, aes(x = factor(1), group=factor(rnkr), fill=cp)) +
  geom_bar(width = 1) +
  scale_fill_continuous(low=&#39;white&#39;, high=&#39;green1&#39;, guide=&quot;none&quot;) +
  facet_wrap(~Treatment, ncol=4) + theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank())
s1</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-15-5.png" width="1152" /></p>
<pre class="r"><code>s&lt;- s1 + coord_polar(theta = &quot;x&quot;)
print(s)</code></pre>
<p><img src="05-Plot-Forest-Plot_files/figure-html/unnamed-chunk-15-6.png" width="1152" /></p>
</div>
<div id="treatment-ranking-dashboard" class="section level2">
<h2>Treatment Ranking Dashboard</h2>
<p><a href="https://huw-wilson.shinyapps.io/RankingDashboard/">App
1</a></p>
<p>On the left-hand side, a horizontal bar chart is displayed,
representing the SUCRA (surface under the cumulative ranking). A
description of this metric is provided, and this metric is used
consistently throughout to order plots for individual treatments from
best to worst, which the panel complimented. The user can select a
treatment of interest from a drop-down box, and this treatment is
highlighted in blue in all plots, with other treatments displayed in
grey and pushed to the background. This nicely highlights a treatment of
interest in a way that could be considered for enhancing submission 1.
On the right-hand side of the app, the user can explore a range of
plots. The first are line graphs and the user can select to display
either absolute or cumulative probabilities. By hovering over the plot,
the user can view the probability that each treatment ranks in each
position, meaning the entire underlying dataset can be recovered. This
feature can also be used to identify an interesting treatment from the
plot and subsequently select it from the drop-down box. The user can
either view all lines on a single plot select to display faceted plots
by treatment to make it easier to follow the line for a given treatment.
More could be done on these plots to emphasise the discrete nature of
the ranks, although the ability to hover and see the discrete
probabilities does go some way towards doing this. The user can also
select to view bar charts. These have probability on the x-axis and
allow the user to display whether to stack by treatment or by rank.
Depending on this selection, either treatment or rank is displayed on
the y-axis, with the other used to fill the stacked bars. This better
emphasises the discrete nature of the ranks than the line graphs and
allows the user to make both treatment-wise and rank-wise comparisons.
Small updates could further enhance this submission, such as widening
the extremes of the colour gradient on the bar charts. This would make
individual sections of the stacked bars easier to distinguish in the
presence of many treatments/ranks. Similarly, the consistent use of
colouring applied throughout the plots (blue for the selected
treatments, grey for the others), could be extended to legends and
labels to further strengthen the connection.</p>
</div>
</div>
<div id="reference" class="section level1">
<h1>Reference</h1>
<ul>
<li><p>SIG (2024, Feb. 8). VIS-SIG Blog: Wonderful Wednesdays February
2024. Retrieved from <a
href="https://graphicsprinciples.github.io/posts/2024-09-01-wonderful-wednesdays-february-2024/"
class="uri">https://graphicsprinciples.github.io/posts/2024-09-01-wonderful-wednesdays-february-2024/</a></p></li>
<li><p>Warham (2023, March 8). VIS-SIG Blog: Wonderful Wednesdays March
2023. Retrieved from <a
href="https://graphicsprinciples.github.io/posts/2023-04-16-wonderful-wednesdays-march-2023/"
class="uri">https://graphicsprinciples.github.io/posts/2023-04-16-wonderful-wednesdays-march-2023/</a></p></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
