<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title> Survival Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Zehui Bai</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="02-Clinical_Experience.html">
    <span class="fa fa-user"></span>
     
    Clinical Experience
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-sliders"></span>
     
    Sample Size
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="03-SSC-Everything-to-Know-About-Sample-Size-Determination.html">Everything to Know About Sample Size Determination</a>
    </li>
    <li>
      <a href="03-SSC-Choosing-the-Effect-Size-for-Sample-Size-Calculations.html">Choosing the Effect Size</a>
    </li>
    <li>
      <a href="03-SSC-Biosimilar-Trials.html">Statistical Considerations for the Design and Analysis of Biosimilar Trials</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-and-Power-for-Non-Parametric-Analysis.html">Sample Size and Power for Non-Parametric Analysis</a>
    </li>
    <li>
      <a href="03-SSC-Power-for-Complex-Hypotheses.html">Power for Complex Hypotheses</a>
    </li>
    <li>
      <a href="03-SSC-Alternatives-to-Power.html">Bayesian methods - Alternatives to Power</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-for-Pilot-Studies.html">Sample Size for Pilot Studies</a>
    </li>
    <li>
      <a href="03-SSC-Case-Continuous-Endpoint.html">Sample Size Determination for Continuous Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Categorical-Endpoint.html">Sample Size Determination for Categorical Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Sample-Size-Determination-for-Counts-and-Rates.html">Sample Size Determination for Counts and Rates</a>
    </li>
    <li>
      <a href="03-SSC-Case-Survival-Endpoint.html">Sample Size Determination for Survival Endpoint</a>
    </li>
    <li>
      <a href="03-SSC-Case-Repeated-Measures.html">Sample Size Determination for Repeated Measures</a>
    </li>
    <li>
      <a href="03-SSC-IA-Sequential-Design.html">Statistical Considerations for Group Sequential Design</a>
    </li>
    <li>
      <a href="03-SSC-IA-Adaptive-Design.html">Statistical Considerations for Adaptive Design</a>
    </li>
    <li>
      <a href="03-SSC-Multiple-Test.html">Sample Size for Multiple Test</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-pencil-square-o"></span>
     
    Study Design
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04-Design-Estimands-Framework.html">Estimands Framework</a>
    </li>
    <li>
      <a href="04-Design-Estimands-Practice.html">Estimands and Sensitivity Analyses</a>
    </li>
    <li>
      <a href="04-Design-Phase-I-Trials---Design-Considerations.html">Phase I Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-II-Trials---Design-Considerations.html">Phase II Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-III-Trials---Design-Considerations.html">Phase III Trials - Design Considerations</a>
    </li>
    <li>
      <a href="04-Design-Phase-IV-Trials---Design-Considerations.html">Phase IV Trials - Design Considerations for Post Marketing Surveillance</a>
    </li>
    <li>
      <a href="04-Design-Complex-Sequential-Trials.html">Complex Sequential Analysis Trials</a>
    </li>
    <li>
      <a href="04-Design-Adaptive-Clinical-Trials.html">Adaptive Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Bayesian-Clinical-Trials.html">Bayesian Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Dose-Escalation-and-Stratification-Designs.html">Dose Escalation and Stratification Designs in Early Oncology Development</a>
    </li>
    <li>
      <a href="04-Design-Single-Arm-Clinical-Trials.html">Single Arm Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-Design-and-Evaluation.html">Diagnostic Study-Design and Evaluation</a>
    </li>
    <li>
      <a href="04-Design-Diagnostic-Study-MRMC.html">Diagnostic Study-Multireader Multicase (MRMC)</a>
    </li>
    <li>
      <a href="04-Design-Hierarchical-composite-endpoints.html">Hierarchical Composite Endpoints</a>
    </li>
    <li>
      <a href="04-Design-Externally-Controlled-Trials.html">Considerations for the Design and Conduct of Externally Controlled Trials</a>
    </li>
    <li>
      <a href="04-Design-Noninferiority-Trials.html">Noninferiority Trials</a>
    </li>
    <li>
      <a href="04-Design-Bioequivalence-and-Biosimilar-Trials.html">Bioequivalence and Biosimilar Trials</a>
    </li>
    <li>
      <a href="04-Design-Exploring-Survival-Analysis-Designs-for-Clinical-Trials.html">Exploring Survival Analysis Designs for Clinical Trials</a>
    </li>
    <li>
      <a href="04-Design-Projecting-How-Long-Your-Trial-Will-Take.html">Projecting How Long Your Trial Will Take</a>
    </li>
    <li>
      <a href="04-Design-Regulatory-Submission.html">Regulatory Submission from Stats Perspective</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-line-chart"></span>
     
    Data Visualization
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="05-Plot-Adverse-Event.html">Adverse Event Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Laboratory-Data.html">Laboratory Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Time-To-Event.html">Time to Event Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-PRO-Data.html">Patient Reported Outcome Data Visualization</a>
    </li>
    <li>
      <a href="05-Plot-SSC-and-Power-Calculation.html">Sample Size and Power Calculations Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Correlation.html">Correlation Visualization</a>
    </li>
    <li>
      <a href="05-Plot-Model-Table.html">Clinical Data and Model visualization</a>
    </li>
    <li>
      <a href="05-Plot-ScatterPlot.html">Scatter and Line Plot</a>
    </li>
    <li>
      <a href="05-Plot-BarPlot.html">Bar Chart</a>
    </li>
    <li>
      <a href="05-Plot-PieChart.html">Pie Chart</a>
    </li>
    <li>
      <a href="05-Plot-BoxPlot.html">Box Plot</a>
    </li>
    <li>
      <a href="05-Plot-Histogram.html">Histogram</a>
    </li>
    <li>
      <a href="05-Plot-Forest-Plot.html">Forest Plot</a>
    </li>
    <li>
      <a href="05-Plot-Flow-Chart.html">Flow Chart</a>
    </li>
    <li>
      <a href="05-Plot-Some-Interesting.html">Some Interesting Plots</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-calculator"></span>
     
    Statistical Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06-Analysis-Linear-Regression.html">Linear Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Logistic-Regression.html">Logistic Regression</a>
    </li>
    <li>
      <a href="06-Analysis-Mixed-Model.html">Mixed Model</a>
    </li>
    <li>
      <a href="06-Analysis-MMRM.html">Mixed Model Repeated Measures</a>
    </li>
    <li>
      <a href="06-Analysis-GEE.html">Generalized Estimating Equation</a>
    </li>
    <li>
      <a href="06-Analysis-ANOVA.html">Analysis of Variance</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Analysis.html">Survival Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Survival-Competing-Risk.html">Survival Analysis - Competing Risk</a>
    </li>
    <li>
      <a href="06-Analysis-Missing-Data.html">Missing Data Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-MI-Implementation.html">Multiple Imputation Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SSD.html">Safety Signal Detection and Evaluation</a>
    </li>
    <li>
      <a href="06-Analysis-Meta-Analysis.html">Meta Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-PK-and-PD.html">PK and PD Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-Time-Series-Analysis.html">Time Series Analysis</a>
    </li>
    <li>
      <a href="06-Analysis-SEM.html">Structural Equation Modeling</a>
    </li>
    <li>
      <a href="06-Analysis-Factor-Analysis.html">Factor Analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-bar-chart"></span>
     
    Machine Learning
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="07-ML-Bayesian-Theory.html">Bayesian Theory</a>
    </li>
    <li>
      <a href="07-ML-Bayesian-Analysis.html">Bayesian Analysis</a>
    </li>
    <li>
      <a href="07-ML-Regularization-Penalized-Regression.html">Regularization Penalized Regression</a>
    </li>
    <li>
      <a href="07-ML-Loss-Regression.html">Loss Functions in Machine Learning</a>
    </li>
    <li>
      <a href="07-ML-PCA.html">Principal Component Analysis</a>
    </li>
    <li>
      <a href="07-ML-KNN.html">K-Nearest Neighbors</a>
    </li>
    <li>
      <a href="07-ML-SVM.html">Support Vector Machine</a>
    </li>
    <li>
      <a href="07-ML-Tree-Models.html">Tree Models</a>
    </li>
    <li>
      <a href="07-ML-LDA.html">Linear Discriminant Analysis</a>
    </li>
    <li>
      <a href="07-ML-Cluster-Analysis.html">Cluster Analysis</a>
    </li>
    <li>
      <a href="07-ML-Neural-Networks.html">Neural Network</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="08-CV.html">
    <span class="fa fa-file-pdf-o"></span>
     
    CV
  </a>
</li>
<li>
  <a href="mailto:zehuibai@outlook.com">
    <span class="fa fa-envelope-o"></span>
     
    Contact me
  </a>
</li>
<li>
  <a href="https://github.com/Zehuibai">
    <span class="fa fa-github"></span>
     
    GitHub
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore"><p><img src="logo.png"
style="width:3in" /><br />
Survival Analysis</p></h1>

</div>


<div id="preliminary" class="section level1">
<h1>Preliminary</h1>
<p>Survival analysis mainly focuses on processing a special kind of time
data, and the time data may be partially censored. The shortcomings of
the ordinary least squares regression method are that the event
occurrence time is usually not normally distributed, and the model
cannot handle Censor. The non-parametric method can simply and quickly
check the survival experience, while the Cox proportional hazard
regression model is still the main Analytical method.</p>
<div id="probability-density-function" class="section level2">
<h2>Probability density function</h2>
<p>The function that describes likelihood of observing Time at time t
relative to all other survival times is known as the probability density
function the probability of observing a survival time within the
interval <span class="math inline">\([a, b]\)</span> is <span
class="math display">\[
\operatorname{Pr}(a \leq \text { Time } \leq b)=\int_{a}^{b} f(t) d
t=\int_{a}^{b} \lambda e^{-\lambda t} d t
\]</span> ## Cumulative distribution function</p>
<p>Describes the probability of observing Time less than or equal to
some timet <span class="math inline">\(t\)</span> <span
class="math inline">\(\operatorname{Pr}(\)</span> Time <span
class="math inline">\(\leq t)\)</span> <span class="math display">\[
\begin{array}{c}
F(t)=\int_{0}^{t} f(t) d t \\
f(t)=\frac{d F(t)}{d t}
\end{array}
\]</span> In SAS, we can graph an estimate of the cdf using proc
univariate.</p>
<pre><code>proc univariate data = whas500(where=(fstat=1));
    var lenfol;
    cdfplot lenfol;
run;</code></pre>
</div>
<div id="survival-function" class="section level2">
<h2>Survival function</h2>
<p><span class="math display">\[
S(t)=P(T&gt;t)=1-F(t)
\]</span> The survival function gives the probability that a person
survives longer than some specified time <span
class="math inline">\(t\)</span> : that is, <span
class="math inline">\(S(t)\)</span> gives the probability that the
random variable <span class="math inline">\(T\)</span> exceeds the
specified time <span class="math inline">\(t .\)</span> And here, some
important characteristics:</p>
<ul>
<li>It is nonincreasing; that is, it heads downward as <span
class="math inline">\(t\)</span> increases.</li>
<li>At time <span class="math inline">\(t=0, S(t)=S(0)=1\)</span>; that
is, at the start of the study, since no one has gotten the event yet,
the probability of surviving past time zero is one.</li>
<li>At time <span class="math inline">\(t=\inf , S(t)=S(\mathrm{inf})=0
;\)</span> that is, theoretically, if the study period increased without
limit. eventually nobody would survive, so the survival curve must
eventually fall to zero.</li>
</ul>
<p>Here we can use proc lifetest to graph <span
class="math inline">\(S_(t)\)</span>.</p>
<pre><code>proc lifetest data=whas500(where=(fstat=1)) plots=survival(atrisk);
time lenfol*fstat(0);
run; </code></pre>
</div>
<div id="hazard-function" class="section level2">
<h2>Hazard function</h2>
<p>The hazard function <span class="math inline">\(h(t)\)</span>, is
given by the formula: <span class="math display">\[
h(t)=\lim _{\Delta_{t} \rightarrow 0} \frac{P(t \leq T&lt;t+\Delta t
\mid T \geq t)}{\Delta t}
\]</span> We could say that the hazard function is the probability that
if you survive to time <span class="math inline">\(t\)</span>, you will
experience the event in the next instant, or in other words, the hazard
function gives the instantaneous potential per unit time for the event
to occur, given that the individual has survived up to time <span
class="math inline">\(t\)</span>. Because of the given sign here, the
hazard function is sometimes called a conditional failure rate.</p>
<p>We can estimate the hazard function is SAS as well using proc
lifetest</p>
<pre><code>proc lifetest data=whas500(where=(fstat=1)) plots=hazard(bw=200);
    time lenfol*fstat(0);
run;</code></pre>
</div>
<div id="cumulative-hazard-function" class="section level2">
<h2>Cumulative hazard function</h2>
<p>Calculated by integrating the hazard function over an interval of
time: <span class="math display">\[H(t) = \int_0^th(u)du\]</span></p>
<p>The cumulative hazard function H(t) and the survival function S(t)
have a simple monotonic relationship. Therefore, when the survival
function reaches the maximum at the beginning of the analysis time, the
cumulative hazard function is the smallest. As time goes by, the
survival function advances toward its minimum, and the cumulative hazard
function advances toward its maximum. You can use proc lifetest to
estimate the cumulative hazard function, and then send the results to
proc sgplot for plotting.</p>
<pre><code>ods output ProductLimitEstimates = ple;
proc lifetest data=whas500(where=(fstat=1))  nelson outs=outwhas500;
    time lenfol*fstat(0);
run;
proc sgplot data = ple;
    series x = lenfol y = CumHaz;
run;</code></pre>
</div>
<div id="mean-residual-life" class="section level2">
<h2>Mean Residual Life</h2>
<p><span class="math display">\[r(t)=E(T-t \mid T \geq
t)=\frac{\int_{t}^{\infty} S(u) d u}{S(t)},\]</span></p>
<p>Intuitively, this is as simple as when I know that I have lived to
the time point t and how many years I have to live.</p>
</div>
<div id="relation-between-functions" class="section level2">
<h2>Relation between functions</h2>
<p>The survival function can be ascertained from the probability density
function by integrating over the probability density function from time
<span class="math inline">\(t\)</span> to infinity, or by calculating
the difference between one and the cumulative distribution function
<span class="math inline">\(F(t)\)</span>. The hazard can then be found
by dividing the negative derivative of the survival function by the
survival function. Note that the functions <span
class="math inline">\(f(t), F(t), h(t)\)</span>, and <span
class="math inline">\(H(t)\)</span> are all related.</p>
<p><strong>Assume that <span class="math inline">\(T\)</span> is
non-negative and continuos:</strong></p>
<ul>
<li>Probability density function: <span class="math display">\[
f(t)=F^{\prime}(t)=\frac{d F(t)}{d t}
\]</span></li>
<li>Cumulative distribution function: <span class="math display">\[
F(t)=P(T \leq t)=\int_{0}^{t} f(u) d u
\]</span></li>
<li>Survival function <span class="math display">\[
\begin{array}{l}
S(t)=1-F(t)\\
S(t)=P(T&gt;t)=\int_{t}^{+\infty} f(u) d u \\
S(t)=\exp \left(-\int_{0}^{t} h(u) d u\right) \\
S(t)=\exp (-H(t))
\end{array}
\]</span></li>
<li>Hazard function <span class="math display">\[h(t) = \frac{
f(t)}{S(t)}= \frac{ -d[S(t)]/dt}{S(t)}\]</span></li>
<li>Cumulative hazard function o Cumulative hazard function <span
class="math display">\[
H(t)=\int_{0}^{t} h(u) d u
\]</span></li>
</ul>
<p><strong>Assume that <span class="math inline">\(T\)</span> is
non-negative and discrete,</strong></p>
<ul>
<li>Probability mass function: <span class="math display">\[
\begin{aligned}
p\left(t_{i}\right) &amp;=P\left(T=t_{i}\right) \\
p\left(t_{i}\right) &amp;=S\left(t_{i-1}\right)-S\left(t_{i}\right) \\
p\left(t_{i}\right) &amp;=F\left(t_{i}\right)-F\left(t_{i-1}\right)
\end{aligned}
\]</span></li>
<li>Cumulative distribution function: <span class="math display">\[
F(t)=P(T \leq t)=\sum_{t_{i} \leq t} p\left(t_{i}\right)
\]</span></li>
<li>Survival function <span class="math display">\[
S(t)=\prod_{t_{i} \leq t}\left(1-h\left(t_{i}\right)\right)
\]</span></li>
<li>Hazard function <span class="math display">\[
\begin{aligned}
h(t)
&amp;=\frac{p\left(t_{i}\right)}{S\left(t_{i-1}\right)}=\frac{-d[S(t)] /
d t}{S(t)} \\
h(t) &amp;=1-\frac{S\left(t_{i}\right)}{S\left(t_{i-1}\right)}
\end{aligned}
\]</span></li>
<li>Cumulative hazard function <span class="math display">\[
H(t)=\sum_{t_{i} \leq t} h\left(t_{i}\right)
\]</span></li>
</ul>
</div>
</div>
<div id="kaplan-meier-estimator" class="section level1">
<h1>Kaplan-Meier Estimator</h1>
<div id="kaplan-meier-and-nelson-aalen" class="section level2">
<h2>Kaplan-Meier and Nelson-Aalen</h2>
<p><span class="math display">\[\hat S(t)=\prod_{t_i\leq t}\frac{n_i –
d_i}{n_i},\]</span></p>
<ul>
<li><span class="math inline">\(n_{i}\)</span> is the number of subjects
at risk</li>
<li><span class="math inline">\(d_{i}\)</span> is the number of subjects
who fail, both at time <span class="math inline">\(t_{i}\)</span>, the
number who failed out of <span class="math inline">\(n_{i}\)</span></li>
</ul>
<p><strong>Nelson-Aalen estimator of the cumulative hazard
function</strong></p>
<p>Since it and the survival function <span
class="math inline">\(S(t)=e^{-H(t)}\)</span> Nelson-Aalen estimator is
a non-parametric estimator of the cumulative hazard function</p>
<p><span class="math display">\[
\hat{H}(t)=\sum_{t_{i} l e q t} \frac{d_{i}}{n_{i}}
\]</span></p>
<ul>
<li>The Nelson-Aalen estimator is requested in SAS through the nelson
option on the proc lifetest statement. SAS will output both Kaplan Meier
estimates of the survival function and Nelson-Aalen estimates of the
cumulative hazard function in one table.</li>
<li>Quartile Estimates: Calculating median, mean, and other survival
times</li>
</ul>
<pre><code>proc lifetest data=whas500 atrisk nelson;
    time lenfol*fstat(0);
run;</code></pre>
</div>
<div id="survival-analysis-in-sas" class="section level2">
<h2>Survival Analysis in SAS</h2>
<pre><code>proc lifetest data=whas500 atrisk plots=survival(cb) outs=outwhas500;
  time lenfol*fstat(0);
run;</code></pre>
<p><strong>Graphing the Kaplan-Meier estimate</strong></p>
<ul>
<li>By default, proc lifetest graphs the Kaplan Meier estimate, even
without the plot= option on the proc lifetest statement, so we could
have used the same code from above that produced the table of
Kaplan-Meier estimates to generate the graph.</li>
<li>However, we would like to add confidence bands and the number at
risk to the graph, so we add plots=survival(atrisk cb)</li>
</ul>
<blockquote>
<p>The blue shaded area around the survival curve represents the 95%
confidence band, here the Hall-Wellner confidence band. This confidence
band is computed for the entire survival function, and must be wider
than the pointwise confidence interval (the confidence interval around a
single interval) at any given time interval to ensure that 95% of all
pointwise confidence intervals are contained within the belt. Many
transformations of the survivor function are available for alternatives
to computing confidence intervals via the conftype option</p>
</blockquote>
<p><strong>Life Table method</strong></p>
<p>The life table approach will have many unique event times if the
number of observations is large, and if the event times are measured
precisely. The KM method then produces longer tables that can be
difficult to present and interpret.</p>
<ol style="list-style-type: decimal">
<li>One way to solve this problem is to use the TIMELIST option (in the
PROC LIFEREG statement), which only reports KM estimates at specified
points in time.</li>
<li>Another solution is to switch to the life-table method, where event
times are grouped into intervals that can be set arbitrarily. In
addition, the life table method (also known as the actuarial method) can
generate estimates and plots of the hazard function. The disadvantage of
the life table approach is that the choice of intervals is usually
somewhat arbitrary, leading to arbitrariness in the results and
uncertainty in how the intervals are chosen. Inevitably some information
is also lost. Note, however, that PROC LIFETEST computes the log-rank
and Wilcoxon statistics (and optionally other test statistics) on
ungrouped data (if available), so they are not affected by the choice of
intervals of the life table method.</li>
</ol>
<pre><code>ODS GRAPHICS ON;
PROC LIFETEST DATA=recid METHOD=LIFE PLOTS=(S,H);
 TIME week*arrest(0);
RUN;
ODS GRAPHICS OFF;

    Interval    Number   Number    Sample   Probability   Standard
 [Lower, Upper) Failed   Censored  Size     of Failure    Error
  0      10     14       0         432.0    0.0324        0.00852
 10      20     21       0         418.0    0.0502        0.0107
 20      30     23       0         397.0    0.0579        0.0117
 30      40     23       0         374.0    0.0615        0.0124
 40      50     26       0         351.0    0.0741        0.0140
 50      60      7       318       166.0    0.0422        0.0156</code></pre>
<div id="tests-of-equality-of-the-survival-function"
class="section level3">
<h3>Tests of equality of the survival function</h3>
<p>The log-rank test, is a hypothesis test to compare the survival
distributions of two samples. It is a nonparametric test and appropriate
to use when the data are right-skewed and censored (technically, the
censoring must be non-informative).</p>
<p>The logrank test is based on the same assumptions as the Kaplan-Meier
survival curve—namely, that censoring is unrelated to prognosis, the
survival probabilities are the same for subjects recruited early and
late in the study, and the events happened at the times specified.
Deviations from these assumptions matter most if they are satisfied
differently in the groups being compared, for example if censoring is
more likely in one group than another</p>
<p>The calculation of the statistic for the nonparametric “Log-Rank” and
“Wilcoxon” tests is given by :</p>
<p><span class="math display">\[Q = \frac{\bigg[\sum\limits_{i=1}^m
w_j(d_{ij}-\hat e_{ij})\bigg]^2}{\sum\limits_{i=1}^m w_j^2\hat
v_{ij}},\]</span></p>
<ul>
<li><span class="math inline">\(d_{i j}\)</span> is the observed number
of failures in stratum <span class="math inline">\(i\)</span> at time
<span class="math inline">\(t_{i j}\)</span></li>
<li><span class="math inline">\(\hat{e}_{i j}\)</span> is the expected
number of failures in stratum <span class="math inline">\(i\)</span> at
time <span class="math inline">\(t_{i j}\)</span></li>
<li><span class="math inline">\(\hat{v}_{i j}\)</span> is the estimator
of the variance of <span class="math inline">\(d_{i j}\)</span></li>
<li><span class="math inline">\(w_{j}\)</span> is the weight of the
difference at time <span class="math inline">\(t_{j}\)</span></li>
<li>The log-rank or Mantel-Haenzel test uses <span
class="math inline">\(w_{j}=1\)</span></li>
<li>The Wilcoxon test uses <span
class="math inline">\(w_{j}=n_{j}\)</span>, so that differences are
weighted by the number at risk at time <span
class="math inline">\(t_{j}\)</span></li>
</ul>
<p><strong>Specified, for group 1, the log-rank statistic can be written
as</strong></p>
<p><span class="math display">\[
\sum_{j=1}^{r}\left(d_{1 j}-e_{1 j}\right)
\]</span> where the summation is over all unique event times (in both
groups), and there are a total of <span class="math inline">\(r\)</span>
such times. <span class="math inline">\(d_{1 j}\)</span> is the number
of deaths that occur in group 1 at time <span
class="math inline">\(j\)</span>, and <span class="math inline">\(e_{1
j}\)</span> is the expected number of events in group 1 at time <span
class="math inline">\(j\)</span>. The expected number is given by <span
class="math inline">\(n_{1 j} d_{j} / n_{j}\)</span>, where <span
class="math inline">\(n_{j}\)</span> is the total number of cases that
are at risk just prior to time <span class="math inline">\(j, n_{1
j}\)</span> is the number at risk just prior to time <span
class="math inline">\(j\)</span> in group 1, and <span
class="math inline">\(d_{j}\)</span> is the total number of deaths at
time <span class="math inline">\(j\)</span> in both groups.</p>
<p><strong>The Wilcoxon statistic, given by</strong></p>
<p><span class="math display">\[\sum_{j=1}^{r} n_{j}\left(d_{1 j}-e_{1
j}\right)\]</span></p>
<p><strong>strata statement in proc lifetest in SAS</strong></p>
<pre><code>proc lifetest data=whas500 atrisk plots=survival(atrisk cb) outs=outwhas500;
strata gender;
time lenfol*fstat(0);
run;</code></pre>
<div class="figure" style="text-align: center">
<img src="02_Plots/Survival/Proc_lifetest_strata.png" alt="Figure: Strata statement in proc lifetest" width="100%" />
<p class="caption">
Figure: Strata statement in proc lifetest
</p>
</div>
<table>
<thead>
<tr class="header">
<th>Test of Equality over Strata</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Test</td>
<td>Chi-Square</td>
<td>DF</td>
<td>Pr &gt; Chi-Square</td>
</tr>
<tr class="even">
<td>Log-Rank</td>
<td>7.7911</td>
<td>1</td>
<td>0.0053</td>
</tr>
<tr class="odd">
<td>Wilcoxon</td>
<td>5.5370</td>
<td>1</td>
<td>0.0186</td>
</tr>
<tr class="even">
<td>-2Log(LR)</td>
<td>10.5120</td>
<td>1</td>
<td>0.0012</td>
</tr>
</tbody>
</table>
<p><strong>log-rank test v.s. Wilcoxon test</strong></p>
<ul>
<li>If the log-rank test is meaningful but the Wilcoxon test is
meaningless, it indicates that the long-term difference may be large,
but not necessarily in the early stage, and the difference may not be
large.</li>
<li>If the log-rank test is meaningless and the Wilcoxon test is
meaningful, it indicates that there is a large difference in early
survival and little difference in long-term survival. Because the
Wilcoxon test is more important for early weights than for late weights
(nj does not increase with time), it is not as sensitive as log-rank
test to differences between groups that occur at later time points. In
other words, although the two statistics test the same null hypothesis,
they have different sensitivities to various deviations from the
hypothesis.</li>
</ul>
<blockquote>
<p>If the log-rank test is meaningful and the Wilcoxon test is
meaningless, it indicates that the difference may be large in the long
run, but not necessarily in the early stages, where the difference may
not be large. If the log-rank test is insignificant but the Wilcoxon
test is significant, it indicates that there is a large difference in
early survival and little difference in long-term survival. Because the
Wilcoxon test is more important for earlier weights than for heavier
weights (nj does not increase over time)</p>
</blockquote>
<p>In particular, the log-rank test is more powerful for detecting
differences of the form <span
class="math display">\[S_{1}(t)=\left[S_{2}(t)\right]^{\gamma}, \gamma
&gt; 1\]</span></p>
<p>This equation defines a proportional hazards model, the log-rank test
is closely related to tests for differences between two groups that are
performed within the framework of Cox’s proportional hazards model</p>
<p>In contrast, the Wilcoxon test is more powerful than the log-rank
test in situations where event times have log-normal distributions
(discussed in the next chapter) with a common variance but with
different means in the two groups.</p>
</div>
<div id="other-nonparametric-tests-for-strata-statement"
class="section level3">
<h3>Other nonparametric tests for STRATA statement</h3>
<p>In addition to the log-rank and Wilcoxon tests, which are produced by
default, the STRATA statement also has options for four other
nonparametric tests:</p>
<ul>
<li>Tarone-Ware,</li>
<li>Peto-Peto,</li>
<li>modified Peto-Peto,</li>
<li>Fleming-Harrington.</li>
</ul>
<p>Like the Wilcoxon and log-rank tests, all of these can be represented
as a weighted sum of observed and expected numbers of events:</p>
<pre><code>STRATA treat / TESTS=ALL;</code></pre>
<p>For the Tarone-Ware test, <span class="math inline">\(w j\)</span> is
the square root of <span class="math inline">\(n i\)</span>, so this
test behaves much like the Wilcoxon test in being more sensitive to
differences at earlier rather than later times. <span
class="math display">\[
\sum_{j=1}^{r} w_{j}\left(d_{1 j}-e_{1 j}\right)
\]</span> That’s also true of the two Peto tests, for which <span
class="math inline">\(w_{j}\)</span> is a function of the survivor
function itself. <span class="math display">\[
\sum_{j=1}^{r} \hat{S}\left(t_{j}\right)\left(d_{1 j}-e_{1 j}\right)
\]</span> Fleming-Harrington is actually a family of tests in which the
weights depend on two parameters, <span class="math inline">\(p\)</span>
and <span class="math inline">\(q\)</span> which can be chosen by the
user: <span
class="math display">\[w_{j}=\hat{S}\left(t_{j}\right)^{p}\left[1-\hat{S}\left(t_{j}\right)\right]^{q}\]</span></p>
<ul>
<li>When both <span class="math inline">\(\mathrm{p}\)</span> and <span
class="math inline">\(\mathrm{q}\)</span> are 0 , you get the log-rank
test.</li>
<li>When <span class="math inline">\(\mathrm{p}\)</span> is 1 and <span
class="math inline">\(\mathrm{q}\)</span> is 0 , you get something very
close to the Peto-Peto test.</li>
<li>When <span class="math inline">\(\mathrm{q}\)</span> is 1 and <span
class="math inline">\(\mathrm{p}\)</span> is 0 , wi increases with time,
unlike any of the other tests.</li>
</ul>
</div>
<div id="multiple-comparisons" class="section level3">
<h3>Multiple Comparisons</h3>
<p>When more than one strata, the ADJUST option tells PROC LIFETEST to
produce p-values for all six pairwise comparisons of the four strata and
then to report p-values that have been adjusted for multiple comparisons
using Tukey’s method (other methods are also available):</p>
<pre><code>PROC LIFETEST DAta=my.recid;
 TIME week*arrest(0);
 STRATA wexp paro / ADJUST=TUKEY;
RUN;</code></pre>
<p>For numeric variables, you can use the STRATA statement to define
groups by intervals rather than by unique values.</p>
<pre><code>PROC LIFETEST DAta=my.recid;
 TIME week*arrest(0);
 STRATA age(21 24 28) / ADJUST=BON;
RUN;

     age &lt; 21
21 ≤ age &lt; 24
24 ≤ age &lt; 28
28 ≤ age </code></pre>
</div>
<div id="comparing-survival-functions-using-log-rank-hr"
class="section level3">
<h3>Comparing Survival Functions using Log-Rank HR</h3>
<p>One way to compare two survival curves is to calculate the hazard
ratio (HR)</p>
<pre><code>proc lifetest data=whas500 atrisk plots=hazard(bw=200) outs=outwhas500;
    strata bmi(15,18.5,25,30,40);
    time lenfol*fstat(0);
run;</code></pre>
<p>作为Kaplan-Meier计算的一部分，计算每个组中观察到的事件 (通常为死亡)
的数量, 以及假设生存时间 没有差异的零假设的预期事件的数量。危险比为: -
Treatment A and Treatment B. the observed and expected deaths are
summed, to give - <span class="math inline">\(O_{A}=\Sigma O_{A t},
O_{B}=\Sigma O_{B t}, E_{A}=\Sigma E_{A t}\)</span>, and <span
class="math inline">\(E_{B}=\Sigma E_{B t}\)</span> - Finally the
Logrank statistic is calculated as <span class="math display">\[
\chi_{\text {Logrank
}}^{2}=\frac{\left(O_{A}-E_{A}\right)^{2}}{E_{A}}+\frac{\left(O_{B}-E_{B}\right)^{2}}{E_{B}}
\]</span> - It can be written as <span class="math display">\[
\chi_{\text {Logrank
}}^{2}=\left(O_{A}-E_{A}\right)^{2}\left(\frac{1}{E_{A}}+\frac{1}{E_{B}}\right)
\]</span> - since <span
class="math inline">\(\left(O_{A}-E_{A}\right)^{2}=\left(O_{B}-E_{B}\right)^{2}\)</span>.
- We can obtain the ratio <span class="math inline">\(O_{A} /
E_{A}\)</span> and <span class="math inline">\(\left.O_{B} /
E_{B}\right)\)</span>, we can calculate the hazard ratio HR, defined as
the ratio of these two relative death rates; that is, <span
class="math display">\[H R=\frac{O_{A} / E_{A}}{O_{B} /
E_{B}}\]</span></p>
<p><strong>Confidence Interval of HR</strong></p>
<p>In calculating <span class="math inline">\(C I s\)</span>, it is
convenient if the statistic under consideration can be assumed to follow
an approximately Normal distribution. However, the estimate of the <span
class="math inline">\(H R\)</span> is not normally distributed. In
particular it has a possible range of values from 0 to <span
class="math inline">\(\infty\)</span>, with the null hypothesis value of
unity not located at the centre of this interval. To make the scale
symmetric and to enable us to calculate <span class="math inline">\(C I
s\)</span>, we transform the estimate to make it approximately normally
distributed. We do this by using log <span class="math inline">\(H
R\)</span>, rather than <span class="math inline">\(H R\)</span> itself,
as the basis for our calculation. It is possible to show that a general
<span class="math inline">\(100(1-\alpha) \% C I\)</span> for the <span
class="math inline">\(\log H R\)</span> is Convert the estimated value
to an approximately normal distribution. For this, we use logarithmic HR
<span class="math display">\[
\log H R-\left[z_{1-\alpha / 2} \times S E(\log H R)\right] \text { to }
\log H R+\left[z_{1-\alpha / 2} \times S E(\log H R)\right]
\]</span> HR itself is then <span class="math display">\[
\exp \left[\operatorname { l o g } H R - [ z _ { 1 - \alpha / 2 } \times
S E ( \operatorname { l o g } H R ) ] \text { to } \operatorname { e x p
} \left[\log H R+\left[z_{1-\alpha / 2} \times S E(\log H
R)\right]\right.\right.
\]</span> In both these expressions <span class="math display">\[S
E(\log H
R)=\sqrt{\left(\frac{1}{E_{A}}+\frac{1}{E_{B}}\right)}\]</span></p>
</div>
<div id="km-plots" class="section level3">
<h3>KM Plots</h3>
<p>Suppose we need confidence bands that can be interpreted by saying
that we are <span class="math inline">\(95 \%\)</span> confident that
the entire survivor function falls within the upper curve and the lower
curve. More complex methods are needed to produce such bands, and PROC
LIFETEST offers two: the HallWellner method and the equal precision (EP)
method. There is 95% confidence that the entire survivor function falls
between the upper and lower curves. Generating such bands requires more
sophisticated methods, and PROC LIFETEST offers two methods: the
HallWellner method and the equal precision (EP) method.</p>
<pre><code>+ The EP method tends to produce confidence bands that are more stable in the tails. To implement this method, the option becomes PLOTS $=\mathrm{S}(\mathrm{CB}=$EP) . To get both pointwise and EP bands, use the option PLOTS $=\mathrm{S}(\mathrm{CL} \mathrm{CB}=\mathrm{EP})$ Confidence bands are always wider than pointwise confidence limits
+ Other transformations are available, the most attractive being the logit $\log [\hat{S}(t) /(1-\hat{S}(t))]$ To switch to this transform, include the CONFTYPE=LOGIT option in the PROC statement.</code></pre>
<pre><code>proc lifetest data=whas500 atrisk  PLOTS=S(NOCENSOR ATRISK CL) OUTSURV=outwhas500;
  time lenfol*fstat(0);
run;</code></pre>
<p><strong>Test in plot</strong></p>
<pre><code>ODS GRAPHICS ON;
PROC LIFETEST DATA=myel PLOTS=S(TEST);
 TIME dur*status(0);
 STRATA treat;
RUN;
ODS GRAPHICS OFF;</code></pre>
<p><strong>log-log survival plots</strong></p>
<p>LS produces a plot of <span class="math inline">\(-\log
\hat{S}(t)\)</span> versus <span class="math inline">\(t\)</span>. <span
class="math display">\[
-\log S(t)=\int_{0}^{t} h(u) d u
\]</span></p>
<p>The LLS keyword produces a plot of <span class="math inline">\(\log
[-\log \hat{S}(t)]\)</span> versus <span class="math inline">\(\log
t\)</span>. If survival times follow a Weibull distribution, which has a
hazard given by <span class="math inline">\(\log h(t)=\alpha+\beta \log
t\)</span>, then the log-log survival plot (log cumulative hazard plot)
should be a straight line with a slope of <span
class="math inline">\(\beta\)</span>. If the hazards are proportional,
the log-log survivor functions should be strictly parallel.</p>
<pre><code>PROC LIFETEST DATA=COMBINE PLOTS=LLS;
 TIME years*event(0);
 STRATA type;
RUN; </code></pre>
<p><strong>Hazard plots</strong></p>
<p>Examine smoothed hazard plots using the kernel smoothing option</p>
<pre><code>ODS GRAPHICS ON;
PROC LIFETEST DATA=combine PLOTS=H(BW=10);
 TIME years*event(0);
 STRATA type;
RUN;
ODS GRAPHICS OFF;

# confidence limits around the hazard function;
PROC LIFETEST DATA=recid PLOTS=H(CL);</code></pre>
</div>
</div>
<div id="survival-analysis-survfit-in-r" class="section level2">
<h2>Survival Analysis (survfit) in R</h2>
<div id="fit-model" class="section level3">
<h3>Fit Model</h3>
<p>Survival function <span class="math inline">\(S(t) = 1-F(t)\)</span>,
hazard function</p>
<p><span class="math display">\[
\begin{equation*}
\lambda(t) = \lim_{\Delta_{t} \to 0^{+}} \frac{P(t \leq T &lt; t +
\Delta_{t} \, | \, T \geq t)}{\Delta_{t}} = \frac{P(t \leq T &lt; t +
\Delta_{t}) / \Delta_{t}}{P(T &gt; t)} = \frac{f(t)}{S(t)}, \quad t \geq
0
\end{equation*}
\]</span></p>
<pre class="r"><code># Compute survival curves
data(&quot;lung&quot;)
(fit &lt;- survfit(Surv(time, status) ~ 1, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung))</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ 1, data = lung, type = &quot;kaplan-meier&quot;, 
##     conf.type = &quot;log&quot;)
## 
##        n events median 0.95LCL 0.95UCL
## [1,] 228    165    310     285     363</code></pre>
<pre class="r"><code># Stratification
(fit2 &lt;- survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung))</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung, type = &quot;kaplan-meier&quot;, 
##     conf.type = &quot;log&quot;)
## 
##         n events median 0.95LCL 0.95UCL
## sex=1 138    112    270     212     310
## sex=2  90     53    426     348     550</code></pre>
</div>
<div id="convert-personal-level-to-personal-period-in-r"
class="section level3">
<h3>Convert Personal-level to Personal-period in R</h3>
<p>In studies of survival or modeling discrete-time events, one compact
way to store data is in what may be called, “person-level” or generally
“observation-level”. For example, you could have three variables, one
indicating the observation, one indicating the time period the event
occurred or the last follow-up period and one indicating whether the
observation was censored.</p>
<p>The PLPP function takes five arguments. The first, data is the data
set to be converted. The second, id is the name of the variable
containing the identifier for each observation. The third, period is the
name of the variable that indicates how many periods the person or
observation was in. The fourth, event is the name of the variable that
indicates whether the event occurred or not or whether the observation
was censored (depending on which direction you are converting). The
fifth, direction indicates whether the function should go from
person-level to person-period or from person-period to person-level.
There are two options, “period” to go to person-period or “level” to go
to person-level. Now let’s try it out. For the examples that follow to
work, you need to source the function into R.</p>
<pre class="r"><code># Person-Level Person-Period Converter Function
PLPP &lt;- function(data, id, period, event, direction = c(&quot;period&quot;, &quot;level&quot;)){
  # Data Checking and Verification Steps
  stopifnot(is.matrix(data) || is.data.frame(data))
  stopifnot(c(id, period, event) %in% c(colnames(data), 1:ncol(data)))

  if (any(is.na(data[, c(id, period, event)]))) {
    stop(&quot;PLPP cannot currently handle missing data in the id, period, or event variables&quot;)
  }
   # Do the conversion
  switch(match.arg(direction),
    period = {
      index &lt;- rep(1:nrow(data), data[, period])
      idmax &lt;- cumsum(data[, period])
      reve &lt;- !data[, event]
      dat &lt;- data[index, ]
      dat[, period] &lt;- ave(dat[, period], dat[, id], FUN = seq_along)
      dat[, event] &lt;- 0
      dat[idmax, event] &lt;- reve},
    level = {
      tmp &lt;- cbind(data[, c(period, id)], i = 1:nrow(data))
      index &lt;- as.vector(by(tmp, tmp[, id],
        FUN = function(x) x[which.max(x[, period]), &quot;i&quot;]))
      dat &lt;- data[index, ]
      dat[, event] &lt;- as.integer(!dat[, event])
  })

  rownames(dat) &lt;- NULL
  return(dat)
}

# Read in the person-level dataset
teachers &lt;- read.csv(&quot;./01_Datasets/teachers.csv&quot;)
# Look at a subset of the cases
subset(teachers, id %in% c(20, 126, 129))</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["id"],"name":[1],"type":["int"],"align":["right"]},{"label":["t"],"name":[2],"type":["int"],"align":["right"]},{"label":["censor"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"20","2":"3","3":"0","_rn_":"19"},{"1":"126","2":"12","3":"0","_rn_":"125"},{"1":"129","2":"12","3":"1","_rn_":"128"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code># Uses PLPP to convert to person-period and store in object, &#39;tpp&#39;
tpp &lt;- PLPP(data = teachers, id = &quot;id&quot;, period = &quot;t&quot;, event = &quot;censor&quot;, direction = &quot;period&quot;)
# Look at a subset of the cases
subset(tpp, id %in% c(20, 126, 129))</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["id"],"name":[1],"type":["int"],"align":["right"]},{"label":["t"],"name":[2],"type":["int"],"align":["right"]},{"label":["censor"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"20","2":"1","3":"0","_rn_":"95"},{"1":"20","2":"2","3":"0","_rn_":"96"},{"1":"20","2":"3","3":"1","_rn_":"97"},{"1":"126","2":"1","3":"0","_rn_":"740"},{"1":"126","2":"2","3":"0","_rn_":"741"},{"1":"126","2":"3","3":"0","_rn_":"742"},{"1":"126","2":"4","3":"0","_rn_":"743"},{"1":"126","2":"5","3":"0","_rn_":"744"},{"1":"126","2":"6","3":"0","_rn_":"745"},{"1":"126","2":"7","3":"0","_rn_":"746"},{"1":"126","2":"8","3":"0","_rn_":"747"},{"1":"126","2":"9","3":"0","_rn_":"748"},{"1":"126","2":"10","3":"0","_rn_":"749"},{"1":"126","2":"11","3":"0","_rn_":"750"},{"1":"126","2":"12","3":"1","_rn_":"751"},{"1":"129","2":"1","3":"0","_rn_":"760"},{"1":"129","2":"2","3":"0","_rn_":"761"},{"1":"129","2":"3","3":"0","_rn_":"762"},{"1":"129","2":"4","3":"0","_rn_":"763"},{"1":"129","2":"5","3":"0","_rn_":"764"},{"1":"129","2":"6","3":"0","_rn_":"765"},{"1":"129","2":"7","3":"0","_rn_":"766"},{"1":"129","2":"8","3":"0","_rn_":"767"},{"1":"129","2":"9","3":"0","_rn_":"768"},{"1":"129","2":"10","3":"0","_rn_":"769"},{"1":"129","2":"11","3":"0","_rn_":"770"},{"1":"129","2":"12","3":"0","_rn_":"771"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>## Convert person-period to person-level
# Read in person-period dataset
teachers.pp &lt;- read.csv(&quot;./01_Datasets/teachers_pp.csv&quot;)
# Look at a subset of the cases
subset(teachers.pp, id %in% c(20, 126, 129))</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["id"],"name":[1],"type":["int"],"align":["right"]},{"label":["period"],"name":[2],"type":["int"],"align":["right"]},{"label":["event"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"20","2":"1","3":"0","_rn_":"95"},{"1":"20","2":"2","3":"0","_rn_":"96"},{"1":"20","2":"3","3":"1","_rn_":"97"},{"1":"126","2":"1","3":"0","_rn_":"740"},{"1":"126","2":"2","3":"0","_rn_":"741"},{"1":"126","2":"3","3":"0","_rn_":"742"},{"1":"126","2":"4","3":"0","_rn_":"743"},{"1":"126","2":"5","3":"0","_rn_":"744"},{"1":"126","2":"6","3":"0","_rn_":"745"},{"1":"126","2":"7","3":"0","_rn_":"746"},{"1":"126","2":"8","3":"0","_rn_":"747"},{"1":"126","2":"9","3":"0","_rn_":"748"},{"1":"126","2":"10","3":"0","_rn_":"749"},{"1":"126","2":"11","3":"0","_rn_":"750"},{"1":"126","2":"12","3":"1","_rn_":"751"},{"1":"129","2":"1","3":"0","_rn_":"760"},{"1":"129","2":"2","3":"0","_rn_":"761"},{"1":"129","2":"3","3":"0","_rn_":"762"},{"1":"129","2":"4","3":"0","_rn_":"763"},{"1":"129","2":"5","3":"0","_rn_":"764"},{"1":"129","2":"6","3":"0","_rn_":"765"},{"1":"129","2":"7","3":"0","_rn_":"766"},{"1":"129","2":"8","3":"0","_rn_":"767"},{"1":"129","2":"9","3":"0","_rn_":"768"},{"1":"129","2":"10","3":"0","_rn_":"769"},{"1":"129","2":"11","3":"0","_rn_":"770"},{"1":"129","2":"12","3":"0","_rn_":"771"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="summary-lifer-table" class="section level3">
<h3>Summary Lifer Table</h3>
<pre class="r"><code>## Alternative 1: function survfit() returns a list of variables,components can be accessed as follow:
fit &lt;- survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung)
d &lt;- data.frame(time = fit$time,
                  n.risk = fit$n.risk,
                  n.event = fit$n.event,
                  n.censor = fit$n.censor,
                  surv = fit$surv,
                  upper = fit$upper,
                  lower = fit$lower
                  )
head(d)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["time"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["n.risk"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n.event"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n.censor"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["surv"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["upper"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["lower"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"11","2":"138","3":"3","4":"0","5":"0.9782609","6":"1.0000000","7":"0.9542301","_rn_":"1"},{"1":"12","2":"135","3":"1","4":"0","5":"0.9710145","6":"0.9994124","7":"0.9434235","_rn_":"2"},{"1":"13","2":"134","3":"2","4":"0","5":"0.9565217","6":"0.9911586","7":"0.9230952","_rn_":"3"},{"1":"15","2":"132","3":"1","4":"0","5":"0.9492754","6":"0.9866017","7":"0.9133612","_rn_":"4"},{"1":"26","2":"131","3":"1","4":"0","5":"0.9420290","6":"0.9818365","7":"0.9038355","_rn_":"5"},{"1":"30","2":"130","3":"1","4":"0","5":"0.9347826","6":"0.9768989","7":"0.8944820","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>## Alternative 2: For Summary table from Survfit with Strata
res &lt;- summary( survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung))
cols &lt;- lapply(c(2:6, 8:11) , function(x) res[x])
tbl &lt;- do.call(data.frame, cols)
head(tbl)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["time"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["n.risk"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n.event"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n.censor"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["surv"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["cumhaz"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["std.chaz"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["strata"],"name":[8],"type":["fct"],"align":["left"]},{"label":["type"],"name":[9],"type":["chr"],"align":["left"]}],"data":[{"1":"11","2":"138","3":"3","4":"0","5":"0.9782609","6":"0.02173913","7":"0.01255109","8":"sex=1","9":"right","_rn_":"1"},{"1":"12","2":"135","3":"1","4":"0","5":"0.9710145","6":"0.02914654","7":"0.01457394","8":"sex=1","9":"right","_rn_":"2"},{"1":"13","2":"134","3":"2","4":"0","5":"0.9565217","6":"0.04407191","7":"0.01799397","8":"sex=1","9":"right","_rn_":"3"},{"1":"15","2":"132","3":"1","4":"0","5":"0.9492754","6":"0.05164767","7":"0.01952371","8":"sex=1","9":"right","_rn_":"4"},{"1":"26","2":"131","3":"1","4":"0","5":"0.9420290","6":"0.05928126","7":"0.02096299","8":"sex=1","9":"right","_rn_":"5"},{"1":"30","2":"130","3":"1","4":"0","5":"0.9347826","6":"0.06697356","7":"0.02232976","8":"sex=1","9":"right","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>## Alternative 3: Kaplan-Meier life table: summary of survival curves summary(fit)
res.sum &lt;- surv_summary(fit)
head(res.sum)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["time"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["n.risk"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n.event"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n.censor"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["surv"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["std.err"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["upper"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["lower"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["strata"],"name":[9],"type":["fct"],"align":["left"]},{"label":["sex"],"name":[10],"type":["fct"],"align":["left"]}],"data":[{"1":"11","2":"138","3":"3","4":"0","5":"0.9782609","6":"0.01268978","7":"1.0000000","8":"0.9542301","9":"sex=1","10":"1","_rn_":"1"},{"1":"12","2":"135","3":"1","4":"0","5":"0.9710145","6":"0.01470747","7":"0.9994124","8":"0.9434235","9":"sex=1","10":"1","_rn_":"2"},{"1":"13","2":"134","3":"2","4":"0","5":"0.9565217","6":"0.01814885","7":"0.9911586","8":"0.9230952","9":"sex=1","10":"1","_rn_":"3"},{"1":"15","2":"132","3":"1","4":"0","5":"0.9492754","6":"0.01967768","7":"0.9866017","8":"0.9133612","9":"sex=1","10":"1","_rn_":"4"},{"1":"26","2":"131","3":"1","4":"0","5":"0.9420290","6":"0.02111708","7":"0.9818365","8":"0.9038355","9":"sex=1","10":"1","_rn_":"5"},{"1":"30","2":"130","3":"1","4":"0","5":"0.9347826","6":"0.02248469","7":"0.9768989","8":"0.8944820","9":"sex=1","10":"1","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>## Alternative 4: use the ggfortify library and can plot your survival curves with ggplot.
library(&quot;ggfortify&quot;)
# extract results to a data.frame
res &lt;- fortify(fit)
head(res)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["time"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["n.risk"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n.event"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["n.censor"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["surv"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["std.err"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["upper"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["lower"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["strata"],"name":[9],"type":["fct"],"align":["left"]}],"data":[{"1":"11","2":"138","3":"3","4":"0","5":"0.9782609","6":"0.01268978","7":"1.0000000","8":"0.9542301","9":"1","_rn_":"1"},{"1":"12","2":"135","3":"1","4":"0","5":"0.9710145","6":"0.01470747","7":"0.9994124","8":"0.9434235","9":"1","_rn_":"2"},{"1":"13","2":"134","3":"2","4":"0","5":"0.9565217","6":"0.01814885","7":"0.9911586","8":"0.9230952","9":"1","_rn_":"3"},{"1":"15","2":"132","3":"1","4":"0","5":"0.9492754","6":"0.01967768","7":"0.9866017","8":"0.9133612","9":"1","_rn_":"4"},{"1":"26","2":"131","3":"1","4":"0","5":"0.9420290","6":"0.02111708","7":"0.9818365","8":"0.9038355","9":"1","_rn_":"5"},{"1":"30","2":"130","3":"1","4":"0","5":"0.9347826","6":"0.02248469","7":"0.9768989","8":"0.8944820","9":"1","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>options(warn = -1)
ggplot(data = res, aes(x = time, y = surv, color = strata)) +
       geom_line() + 
       # plot censor marks
       geom_point(aes(shape = factor(ifelse(n.censor &gt;= 1, 1, NA)))) + 
       # format censor shape as &quot;+&quot;
       scale_shape_manual(values = 3) + 
       # hide censor legend 
       guides(shape = &quot;none&quot;) </code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>options(warn = 0)</code></pre>
</div>
<div id="arbitrary-quantiles" class="section level3">
<h3>Arbitrary quantiles</h3>
<pre class="r"><code># Alternative 1: Access to the sort summary statistics table
summary(fit)$table</code></pre>
<pre><code>##       records n.max n.start events    rmean se(rmean) median 0.95LCL 0.95UCL
## sex=1     138   138     138    112 326.0841  22.91156    270     212     310
## sex=2      90    90      90     53 460.6473  34.68985    426     348     550</code></pre>
<pre class="r"><code>quantile(fit, probs=c(0.25, 0.5, 0.75), conf.int=TRUE)</code></pre>
<pre><code>## $quantile
##        25  50  75
## sex=1 144 270 457
## sex=2 226 426 687
## 
## $lower
##        25  50  75
## sex=1 107 212 387
## sex=2 186 348 550
## 
## $upper
##        25  50  75
## sex=1 177 310 574
## sex=2 340 550  NA</code></pre>
<pre class="r"><code>## Estimated mean survival time
print(fit, print.rmean=TRUE)</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung, type = &quot;kaplan-meier&quot;, 
##     conf.type = &quot;log&quot;)
## 
##         n events rmean* se(rmean) median 0.95LCL 0.95UCL
## sex=1 138    112    326      22.9    270     212     310
## sex=2  90     53    461      34.7    426     348     550
##     * restricted mean with upper limit =  1022</code></pre>
<pre class="r"><code>## 50-day and 100-day survival including point-wise confidence interval.
summary(fit, times=c(50, 100))</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung, type = &quot;kaplan-meier&quot;, 
##     conf.type = &quot;log&quot;)
## 
##                 sex=1 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##    50    128      10    0.928  0.0221        0.885        0.972
##   100    114      14    0.826  0.0323        0.765        0.892
## 
##                 sex=2 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##    50     89       1    0.989  0.0110        0.967        1.000
##   100     82       6    0.922  0.0283        0.868        0.979</code></pre>
<pre class="r"><code># Alternative 2: The surv_summary object also has an attribute called &quot;table&quot; that contains information about the survival curves, including the survival median with confidence intervals and the total number of subjects and events in each curve
res.sum &lt;- surv_summary(fit)
attr(res.sum, &quot;table&quot;)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["records"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["n.max"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["n.start"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["events"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["rmean"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["se(rmean)"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["median"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["0.95LCL"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["0.95UCL"],"name":[9],"type":["dbl"],"align":["right"]}],"data":[{"1":"138","2":"138","3":"138","4":"112","5":"326.0841","6":"22.91156","7":"270","8":"212","9":"310","_rn_":"sex=1"},{"1":"90","2":"90","3":"90","4":"53","5":"460.6473","6":"34.68985","7":"426","8":"348","9":"550","_rn_":"sex=2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code># Alternative 3: from gtsummary package
library(&quot;gtsummary&quot;)
tbl_survfit(
  survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung),
  probs = c(0.25, 0.5, 0.75),
  label_header = &quot;**{prob} Percentile**&quot;
)</code></pre>
<div id="jdyndhdukx" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jdyndhdukx table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jdyndhdukx thead, #jdyndhdukx tbody, #jdyndhdukx tfoot, #jdyndhdukx tr, #jdyndhdukx td, #jdyndhdukx th {
  border-style: none;
}

#jdyndhdukx p {
  margin: 0;
  padding: 0;
}

#jdyndhdukx .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jdyndhdukx .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jdyndhdukx .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jdyndhdukx .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jdyndhdukx .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jdyndhdukx .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jdyndhdukx .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jdyndhdukx .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jdyndhdukx .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jdyndhdukx .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jdyndhdukx .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jdyndhdukx .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jdyndhdukx .gt_spanner_row {
  border-bottom-style: hidden;
}

#jdyndhdukx .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jdyndhdukx .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jdyndhdukx .gt_from_md > :first-child {
  margin-top: 0;
}

#jdyndhdukx .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jdyndhdukx .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jdyndhdukx .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jdyndhdukx .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jdyndhdukx .gt_row_group_first td {
  border-top-width: 2px;
}

#jdyndhdukx .gt_row_group_first th {
  border-top-width: 2px;
}

#jdyndhdukx .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdyndhdukx .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jdyndhdukx .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jdyndhdukx .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jdyndhdukx .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdyndhdukx .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jdyndhdukx .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jdyndhdukx .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jdyndhdukx .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jdyndhdukx .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jdyndhdukx .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdyndhdukx .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jdyndhdukx .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdyndhdukx .gt_left {
  text-align: left;
}

#jdyndhdukx .gt_center {
  text-align: center;
}

#jdyndhdukx .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jdyndhdukx .gt_font_normal {
  font-weight: normal;
}

#jdyndhdukx .gt_font_bold {
  font-weight: bold;
}

#jdyndhdukx .gt_font_italic {
  font-style: italic;
}

#jdyndhdukx .gt_super {
  font-size: 65%;
}

#jdyndhdukx .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jdyndhdukx .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jdyndhdukx .gt_indent_1 {
  text-indent: 5px;
}

#jdyndhdukx .gt_indent_2 {
  text-indent: 10px;
}

#jdyndhdukx .gt_indent_3 {
  text-indent: 15px;
}

#jdyndhdukx .gt_indent_4 {
  text-indent: 20px;
}

#jdyndhdukx .gt_indent_5 {
  text-indent: 25px;
}

#jdyndhdukx .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#jdyndhdukx div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_1"><span class='gt_from_md'><strong>0.25 Percentile</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_2"><span class='gt_from_md'><strong>0.5 Percentile</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_3"><span class='gt_from_md'><strong>0.75 Percentile</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">sex</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td>
<td headers="stat_3" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    1</td>
<td headers="stat_1" class="gt_row gt_center">144 (107, 177)</td>
<td headers="stat_2" class="gt_row gt_center">270 (212, 310)</td>
<td headers="stat_3" class="gt_row gt_center">457 (387, 574)</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    2</td>
<td headers="stat_1" class="gt_row gt_center">226 (186, 340)</td>
<td headers="stat_2" class="gt_row gt_center">426 (348, 550)</td>
<td headers="stat_3" class="gt_row gt_center">687 (550, —)</td></tr>
  </tbody>
  
  
</table>
</div>
<pre class="r"><code>list(
    survfit(Surv(time, status) ~ 1, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung),
    survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung)
  ) %&gt;% 
  tbl_survfit(probs = c(0.25, 0.5, 0.75),
  label_header = &quot;**{prob} Percentile**&quot;)</code></pre>
<div id="ithkeyffbr" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#ithkeyffbr table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#ithkeyffbr thead, #ithkeyffbr tbody, #ithkeyffbr tfoot, #ithkeyffbr tr, #ithkeyffbr td, #ithkeyffbr th {
  border-style: none;
}

#ithkeyffbr p {
  margin: 0;
  padding: 0;
}

#ithkeyffbr .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#ithkeyffbr .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#ithkeyffbr .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#ithkeyffbr .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#ithkeyffbr .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ithkeyffbr .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ithkeyffbr .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#ithkeyffbr .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#ithkeyffbr .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#ithkeyffbr .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#ithkeyffbr .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#ithkeyffbr .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#ithkeyffbr .gt_spanner_row {
  border-bottom-style: hidden;
}

#ithkeyffbr .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#ithkeyffbr .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#ithkeyffbr .gt_from_md > :first-child {
  margin-top: 0;
}

#ithkeyffbr .gt_from_md > :last-child {
  margin-bottom: 0;
}

#ithkeyffbr .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#ithkeyffbr .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#ithkeyffbr .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#ithkeyffbr .gt_row_group_first td {
  border-top-width: 2px;
}

#ithkeyffbr .gt_row_group_first th {
  border-top-width: 2px;
}

#ithkeyffbr .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ithkeyffbr .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#ithkeyffbr .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#ithkeyffbr .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ithkeyffbr .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#ithkeyffbr .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#ithkeyffbr .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#ithkeyffbr .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#ithkeyffbr .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#ithkeyffbr .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ithkeyffbr .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ithkeyffbr .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#ithkeyffbr .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#ithkeyffbr .gt_left {
  text-align: left;
}

#ithkeyffbr .gt_center {
  text-align: center;
}

#ithkeyffbr .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#ithkeyffbr .gt_font_normal {
  font-weight: normal;
}

#ithkeyffbr .gt_font_bold {
  font-weight: bold;
}

#ithkeyffbr .gt_font_italic {
  font-style: italic;
}

#ithkeyffbr .gt_super {
  font-size: 65%;
}

#ithkeyffbr .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#ithkeyffbr .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#ithkeyffbr .gt_indent_1 {
  text-indent: 5px;
}

#ithkeyffbr .gt_indent_2 {
  text-indent: 10px;
}

#ithkeyffbr .gt_indent_3 {
  text-indent: 15px;
}

#ithkeyffbr .gt_indent_4 {
  text-indent: 20px;
}

#ithkeyffbr .gt_indent_5 {
  text-indent: 25px;
}

#ithkeyffbr .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#ithkeyffbr div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_1"><span class='gt_from_md'><strong>0.25 Percentile</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_2"><span class='gt_from_md'><strong>0.5 Percentile</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_3"><span class='gt_from_md'><strong>0.75 Percentile</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">Overall</td>
<td headers="stat_1" class="gt_row gt_center">170 (145, 197)</td>
<td headers="stat_2" class="gt_row gt_center">310 (285, 363)</td>
<td headers="stat_3" class="gt_row gt_center">550 (460, 654)</td></tr>
    <tr><td headers="label" class="gt_row gt_left">sex</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td>
<td headers="stat_3" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    1</td>
<td headers="stat_1" class="gt_row gt_center">144 (107, 177)</td>
<td headers="stat_2" class="gt_row gt_center">270 (212, 310)</td>
<td headers="stat_3" class="gt_row gt_center">457 (387, 574)</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    2</td>
<td headers="stat_1" class="gt_row gt_center">226 (186, 340)</td>
<td headers="stat_2" class="gt_row gt_center">426 (348, 550)</td>
<td headers="stat_3" class="gt_row gt_center">687 (550, —)</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<div id="confidence-interval" class="section level3">
<h3>Confidence Interval</h3>
<pre class="r"><code># Confidence Interval type
# One of &quot;none&quot;, &quot;plain&quot;, &quot;log&quot; (the default), &quot;log-log&quot;, &quot;logit&quot; or &quot;arcsin&quot;.
 # The plain option standard interval curven  +-k *se(curve)
 # The log option calculates intervals based on the cumulative hazard or log(survival). 
 # The log-log option bases the intervals on the log hazard or log(-log(survival)),
 # The logit option on log(survival/(1- survival)) and arcsin on arcsin(survival).
# 1) normal distribution assumption  S(t) +- SE(S(t)) * u_(1-alpha/2)
survfit(Surv(time, status) ~ sex, data = lung, conf.type=&quot;plain&quot;)</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung, conf.type = &quot;plain&quot;)
## 
##         n events median 0.95LCL 0.95UCL
## sex=1 138    112    270     212     306
## sex=2  90     53    426     345     524</code></pre>
<pre class="r"><code># 2) log-Methode,nicht in Vorlesung
survfit(Surv(time, status) ~ sex, data = lung, conf.type=&quot;log&quot;)</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung, conf.type = &quot;log&quot;)
## 
##         n events median 0.95LCL 0.95UCL
## sex=1 138    112    270     212     310
## sex=2  90     53    426     348     550</code></pre>
<pre class="r"><code># 3) log-log-Methode 
# S(t)^(exp(+-SE(log(-logS(t)))*u_(1-alpha/2))), dabei muss SE(log(-logS(t))) mit Formel (2.16) berechnet werden 
 # ( nicht: S(t)^(exp(+-SE *(log(-logS(t)))*u_(1-alpha/2))) )
survfit(Surv(time, status) ~ sex, data = lung, conf.type=&quot;log-log&quot;)</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung, conf.type = &quot;log-log&quot;)
## 
##         n events median 0.95LCL 0.95UCL
## sex=1 138    112    270     210     306
## sex=2  90     53    426     345     524</code></pre>
</div>
<div id="survival-probability" class="section level3">
<h3>Survival Probability</h3>
<pre class="r"><code>tbl_survfit(
  survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung),
  times = c(180, 240),
  label_header = &quot;**{time} Days**&quot;
)</code></pre>
<div id="vaqfzhtkeq" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#vaqfzhtkeq table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#vaqfzhtkeq thead, #vaqfzhtkeq tbody, #vaqfzhtkeq tfoot, #vaqfzhtkeq tr, #vaqfzhtkeq td, #vaqfzhtkeq th {
  border-style: none;
}

#vaqfzhtkeq p {
  margin: 0;
  padding: 0;
}

#vaqfzhtkeq .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#vaqfzhtkeq .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#vaqfzhtkeq .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#vaqfzhtkeq .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#vaqfzhtkeq .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vaqfzhtkeq .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vaqfzhtkeq .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#vaqfzhtkeq .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#vaqfzhtkeq .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#vaqfzhtkeq .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#vaqfzhtkeq .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#vaqfzhtkeq .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#vaqfzhtkeq .gt_spanner_row {
  border-bottom-style: hidden;
}

#vaqfzhtkeq .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#vaqfzhtkeq .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#vaqfzhtkeq .gt_from_md > :first-child {
  margin-top: 0;
}

#vaqfzhtkeq .gt_from_md > :last-child {
  margin-bottom: 0;
}

#vaqfzhtkeq .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#vaqfzhtkeq .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#vaqfzhtkeq .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#vaqfzhtkeq .gt_row_group_first td {
  border-top-width: 2px;
}

#vaqfzhtkeq .gt_row_group_first th {
  border-top-width: 2px;
}

#vaqfzhtkeq .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vaqfzhtkeq .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#vaqfzhtkeq .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#vaqfzhtkeq .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vaqfzhtkeq .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#vaqfzhtkeq .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#vaqfzhtkeq .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#vaqfzhtkeq .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#vaqfzhtkeq .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#vaqfzhtkeq .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vaqfzhtkeq .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#vaqfzhtkeq .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#vaqfzhtkeq .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#vaqfzhtkeq .gt_left {
  text-align: left;
}

#vaqfzhtkeq .gt_center {
  text-align: center;
}

#vaqfzhtkeq .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#vaqfzhtkeq .gt_font_normal {
  font-weight: normal;
}

#vaqfzhtkeq .gt_font_bold {
  font-weight: bold;
}

#vaqfzhtkeq .gt_font_italic {
  font-style: italic;
}

#vaqfzhtkeq .gt_super {
  font-size: 65%;
}

#vaqfzhtkeq .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#vaqfzhtkeq .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#vaqfzhtkeq .gt_indent_1 {
  text-indent: 5px;
}

#vaqfzhtkeq .gt_indent_2 {
  text-indent: 10px;
}

#vaqfzhtkeq .gt_indent_3 {
  text-indent: 15px;
}

#vaqfzhtkeq .gt_indent_4 {
  text-indent: 20px;
}

#vaqfzhtkeq .gt_indent_5 {
  text-indent: 25px;
}

#vaqfzhtkeq .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#vaqfzhtkeq div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_1"><span class='gt_from_md'><strong>180 Days</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_2"><span class='gt_from_md'><strong>240 Days</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">sex</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    1</td>
<td headers="stat_1" class="gt_row gt_center">64% (57%, 73%)</td>
<td headers="stat_2" class="gt_row gt_center">53% (45%, 62%)</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    2</td>
<td headers="stat_1" class="gt_row gt_center">84% (77%, 92%)</td>
<td headers="stat_2" class="gt_row gt_center">73% (64%, 83%)</td></tr>
  </tbody>
  
  
</table>
</div>
<pre class="r"><code>list(
    survfit(Surv(time, status) ~ 1, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung),
    survfit(Surv(time, status) ~ sex, type=&quot;kaplan-meier&quot;,
               conf.type=&quot;log&quot;, data = lung)
  ) %&gt;%
  tbl_survfit(times = c(180, 240),
  label_header = &quot;**{time} Days**&quot;)</code></pre>
<div id="dyhthqayif" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#dyhthqayif table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#dyhthqayif thead, #dyhthqayif tbody, #dyhthqayif tfoot, #dyhthqayif tr, #dyhthqayif td, #dyhthqayif th {
  border-style: none;
}

#dyhthqayif p {
  margin: 0;
  padding: 0;
}

#dyhthqayif .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#dyhthqayif .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#dyhthqayif .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#dyhthqayif .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#dyhthqayif .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dyhthqayif .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dyhthqayif .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#dyhthqayif .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#dyhthqayif .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#dyhthqayif .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#dyhthqayif .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#dyhthqayif .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#dyhthqayif .gt_spanner_row {
  border-bottom-style: hidden;
}

#dyhthqayif .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#dyhthqayif .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#dyhthqayif .gt_from_md > :first-child {
  margin-top: 0;
}

#dyhthqayif .gt_from_md > :last-child {
  margin-bottom: 0;
}

#dyhthqayif .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#dyhthqayif .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#dyhthqayif .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#dyhthqayif .gt_row_group_first td {
  border-top-width: 2px;
}

#dyhthqayif .gt_row_group_first th {
  border-top-width: 2px;
}

#dyhthqayif .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dyhthqayif .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#dyhthqayif .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#dyhthqayif .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dyhthqayif .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#dyhthqayif .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#dyhthqayif .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#dyhthqayif .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#dyhthqayif .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#dyhthqayif .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dyhthqayif .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dyhthqayif .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#dyhthqayif .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#dyhthqayif .gt_left {
  text-align: left;
}

#dyhthqayif .gt_center {
  text-align: center;
}

#dyhthqayif .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#dyhthqayif .gt_font_normal {
  font-weight: normal;
}

#dyhthqayif .gt_font_bold {
  font-weight: bold;
}

#dyhthqayif .gt_font_italic {
  font-style: italic;
}

#dyhthqayif .gt_super {
  font-size: 65%;
}

#dyhthqayif .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#dyhthqayif .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#dyhthqayif .gt_indent_1 {
  text-indent: 5px;
}

#dyhthqayif .gt_indent_2 {
  text-indent: 10px;
}

#dyhthqayif .gt_indent_3 {
  text-indent: 15px;
}

#dyhthqayif .gt_indent_4 {
  text-indent: 20px;
}

#dyhthqayif .gt_indent_5 {
  text-indent: 25px;
}

#dyhthqayif .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#dyhthqayif div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_1"><span class='gt_from_md'><strong>180 Days</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="stat_2"><span class='gt_from_md'><strong>240 Days</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">Overall</td>
<td headers="stat_1" class="gt_row gt_center">72% (67%, 78%)</td>
<td headers="stat_2" class="gt_row gt_center">61% (55%, 68%)</td></tr>
    <tr><td headers="label" class="gt_row gt_left">sex</td>
<td headers="stat_1" class="gt_row gt_center"><br /></td>
<td headers="stat_2" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    1</td>
<td headers="stat_1" class="gt_row gt_center">64% (57%, 73%)</td>
<td headers="stat_2" class="gt_row gt_center">53% (45%, 62%)</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    2</td>
<td headers="stat_1" class="gt_row gt_center">84% (77%, 92%)</td>
<td headers="stat_2" class="gt_row gt_center">73% (64%, 83%)</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<div id="log-rank-test" class="section level3">
<h3>Log-Rank Test</h3>
<pre class="r"><code># Survival curves comparing
# When rho=0, log-rank test or Mantel-Haenszel test is performed.
# When rho=1, the Peto correction test of Gehan-Wilcoxon is performed, which gives a greater weight to the early outcome events. But this is not the Wilcoxon test 
(surv_diff &lt;- survdiff(Surv(time, status) ~ sex, data = lung, rho=0))</code></pre>
<pre><code>## Call:
## survdiff(formula = Surv(time, status) ~ sex, data = lung, rho = 0)
## 
##         N Observed Expected (O-E)^2/E (O-E)^2/V
## sex=1 138      112     91.6      4.55      10.3
## sex=2  90       53     73.4      5.68      10.3
## 
##  Chisq= 10.3  on 1 degrees of freedom, p= 0.001</code></pre>
<pre class="r"><code>(surv_diff &lt;- survdiff(Surv(time, status) ~ sex, data = lung, rho=1))</code></pre>
<pre><code>## Call:
## survdiff(formula = Surv(time, status) ~ sex, data = lung, rho = 1)
## 
##         N Observed Expected (O-E)^2/E (O-E)^2/V
## sex=1 138     70.4     55.6      3.95      12.7
## sex=2  90     28.7     43.5      5.04      12.7
## 
##  Chisq= 12.7  on 1 degrees of freedom, p= 4e-04</code></pre>
<pre class="r"><code># Extracting information from a survdiff object 
1 - pchisq(surv_diff$chisq, length(surv_diff$n) - 1)</code></pre>
<pre><code>## [1] 0.0003628989</code></pre>
</div>
<div id="km-survival-plots-in-r" class="section level3">
<h3>KM Survival Plots in R</h3>
<pre class="r"><code># Change color, linetype by strata, risk.table color by strata
ggsurvplot(fit,
          pval = TRUE, conf.int = TRUE,
          risk.table = TRUE,               # Add risk table
          risk.table.col = &quot;strata&quot;,       # Change risk table color by groups
          linetype = &quot;strata&quot;,             # Change line type by groups
          surv.median.line = &quot;hv&quot;,         # Specify median survival
          ggtheme = theme_bw(),            # Change ggplot2 theme
          palette = c(&quot;#E7B800&quot;, &quot;#2E9FDF&quot;))</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/Visualize%20survival%20curves-1.png" width="672" /></p>
<pre class="r"><code># The plot can be further customized
ggsurvplot(
   fit,                     # survfit object with calculated statistics.
   pval = TRUE,             # show p-value of log-rank test.
   conf.int = TRUE,         # show confidence intervals for 
                            # point estimaes of survival curves.
   conf.int.style = &quot;step&quot;,  # customize style of confidence intervals
   xlab = &quot;Time in days&quot;,   # customize X axis label.
   break.time.by = 200,     # break X axis in time intervals by 200.
   ggtheme = theme_light(), # customize plot and risk table with a theme.
   risk.table = &quot;abs_pct&quot;,  # absolute number and percentage at risk.
  risk.table.y.text.col = T,# colour risk table text annotations.
  risk.table.y.text = FALSE,# show bars instead of names in text annotations
                            # in legend of risk table.
  ncensor.plot = TRUE,      # plot the number of censored subjects at time t
  surv.median.line = &quot;hv&quot;,  # add the median survival pointer.
  legend.labs = 
    c(&quot;Male&quot;, &quot;Female&quot;),    # change legend labels.
  palette = 
    c(&quot;#E7B800&quot;, &quot;#2E9FDF&quot;) # custom color palettes.
)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/Visualize%20survival%20curves-2.png" width="672" /></p>
<pre class="r"><code>## Plot cumulative events
 # fun = &quot;event&quot; cumulative events 
 # fun = &quot;log&quot;: log transformation 
 # fun = &quot;cumhaz&quot;: cumulative hazard

# plot cumulative events
ggsurvplot(fit,
          conf.int = TRUE,
          risk.table.col = &quot;strata&quot;,          # Change risk table color by groups
          ggtheme = theme_bw(),               # Change ggplot2 theme
          palette = c(&quot;#E7B800&quot;, &quot;#2E9FDF&quot;),
          fun = &quot;event&quot;)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/Visualize%20survival%20curves-3.png" width="672" /></p>
<pre class="r"><code># Plot log survival curve
ggsurvplot(fit,
           conf.int = TRUE,
           risk.table.col = &quot;strata&quot;, # Change risk table color by groups
           ggtheme = theme_bw(), # Change ggplot2 theme
           palette = c(&quot;#E7B800&quot;, &quot;#2E9FDF&quot;),
           fun = &quot;log&quot;)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/Visualize%20survival%20curves-4.png" width="672" /></p>
<pre class="r"><code># Plot cummulative hazard
# The cummulative hazard is commonly used to estimate the hazard probability
ggsurvplot(fit,
          conf.int = TRUE,
          risk.table.col = &quot;strata&quot;, # Change risk table color by groups
          ggtheme = theme_bw(), # Change ggplot2 theme
          palette = c(&quot;#E7B800&quot;, &quot;#2E9FDF&quot;),
          fun = &quot;cumhaz&quot;)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/Visualize%20survival%20curves-5.png" width="672" /></p>
<pre class="r"><code># Complex survival curves in grouping
fit2 &lt;- survfit( Surv(time, status) ~ sex + rx + adhere,
                 data = colon )
# Plot survival curves by sex and facet by rx and adhere
ggsurv &lt;- ggsurvplot(fit2, fun = &quot;event&quot;, conf.int = TRUE,
                     ggtheme = theme_bw())

ggsurv$plot +theme_bw() + 
  theme (legend.position = &quot;right&quot;)+
  facet_grid(rx ~ adhere)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/Visualize%20survival%20curves-6.png" width="672" /></p>
</div>
</div>
</div>
<div id="accelerated-failure-time-aft-model" class="section level1">
<h1>Accelerated Failure Time (AFT) Model</h1>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>The accelerated failure time model (AFT model) is a parametric model
that can replace the commonly used proportional hazard model. The
proportional hazard model assumes that the role of the covariate is to
multiply the risk by a certain constant, while the AFT model assumes
that the role of the covariate is to <strong>accelerate or slow down the
life course of the disease by a certain constant.</strong> The AFT model
describes the relationship between the survivor functions of any two
individuals in its most general form.</p>
<p>Assuming that <span class="math inline">\(T\)</span> is the failure
time and <span class="math inline">\(x\)</span> is the covariate, the
assumption of the accelerated failure time model is that the survival
time of a person is equal to the baseline survival time of the
population, the acceleration factor of this person, and its mathematical
form is as follows: <span class="math display">\[
T=t * e^{\theta \cdot x}, t=e^{\mu+\sigma * W}
\]</span> Common forms also include: <span class="math display">\[
S(t \mid x)=S_{0}\left(t * e^{\theta \cdot x}\right)
\]</span> <span class="math display">\[
Y=\log (T)=\mu+\gamma \cdot x+\sigma * W
\]</span> Among them, the baseline survival time <span
class="math inline">\(t\)</span> of the population obeys a certain
probability distribution, and the <span class="math inline">\(W\)</span>
is also a random variable that obeys a certain probability
distribution.</p>
<p><strong>PH &amp; AFT</strong></p>
<ul>
<li>AFT Model is based on time T modeling, and PH Model is based on
Hazard modeling</li>
<li>Weibull regression is the only regression that satisfies both model
assumptions</li>
</ul>
<p>For PH Model: <span class="math display">\[\quad
\lambda_{i}(t)=\lambda_{0}(t) \exp \left(x_{i}^{T} \beta\right)\]</span>
For AFT Model: <span
class="math display">\[\lambda_{i}(t)=\lambda_{0}\left(\exp
\left(-x_{i}^{T} \beta\right) t\right) \exp \left(-x_{i}^{T}
\beta\right)\]</span></p>
<p>这是由于 <span class="math inline">\(T_{i}=\exp \left(x_{i}^{T}
\beta\right) T_{0}\)</span>, 因此 <span class="math display">\[
S_{i}(t)=P\left(T_{i} \geq t\right)=P\left(T_{0} \geq \exp
\left(-x_{i}^{T} \beta\right) t\right)=S_{0}\left(\exp \left(-x_{i}^{T}
\beta\right) t\right)
\]</span> <span class="math display">\[
f_{i}(t)=f_{0}\left(\exp \left(-x_{i}^{T} \beta\right) t\right) \exp
\left(-x_{i}^{T} \beta\right)
\]</span> <span class="math display">\[
\lambda_{i}(t)=\exp \left(-x_{i}^{T} \beta\right) \lambda_{0}\left(\exp
\left(-x_{i}^{T} \beta\right) t\right)
\]</span></p>
<p>Intuitively, the larger the <span class="math inline">\(x_{i}^{T}
\beta\)</span>, the longer the survival time. Since S is a monotonic
non-decreasing function, the survival curve is higher. From a
logarithmic perspective</p>
<p>PH Model: <span class="math display">\[
\log \lambda_{i}(t)=\log \lambda_{0}(\exp (\log t))+x_{i}^{T} \beta
\]</span> AFT Model: <span class="math display">\[
\log \lambda_{i}(t)=\log \lambda_{0}\left(\exp \left(\log t-x_{i}^{T}
\beta\right)\right)-x_{i}^{T} \beta
\]</span></p>
</div>
<div id="parametric-model-in-sas" class="section level2">
<h2>Parametric Model in SAS</h2>
<div id="introduction-of-proc-lifereg" class="section level3">
<h3>Introduction of Proc Lifereg</h3>
<p>The LIFEREG program uses the maximum likelihood method to generate
the estimated value of the parameter regression model using censored
survival data [method of maximum likelihood]. To some extent, PROREG
LIFEREG has been obscured by the PHREG program, which uses a method
known as partial likelihoo for semiparametric regression analysis
[method known as partial likelihoo].</p>
<p>PROC LIFEREG is by no means obsolete. It can do some better things
than PROC PHREG, and it can do other things that PROC PHREG simply
cannot do:</p>
<ul>
<li>PROC LIFEREG accommodates left censoring and interval censoring.
PROC PHREG allows only right censoring.</li>
<li>With PROC LIFEREG, you can test certain hypotheses about the shape
of the hazard function. PROC PHREG gives you only nonparametric
estimates of the survivor function, which can be difficult to
interpret.</li>
<li>If the shape of the survival distribution is known, PROC LIFEREG
produces more efficient estimates (with smaller standard errors) than
PROC PHREG.</li>
<li>PROC LIFEREG can easily generate predicted event times for any
specified set of covariate values. This is more difficult with PHREG,
and often impossible.</li>
<li>PROC LIFEREG’s greatest limitation is that it does not handle
time-dependent covariates, something at which PROC PHREG excels.</li>
</ul>
<!-- LIFEREG程序使用最大似然方法使用删失的生存数据生成参数回归模型的估计值 [method of maximum likelihood]。在某种程度上，PROREG LIFEREG已被PHREG程序所遮盖，该程序使用称为偏似然的方法进行半参数回归分析 [method known as partial likelihoo]。 -->
<!-- PROC LIFEREG绝不是过时的。它可以做一些比PROC PHREG更好的事情，它可以做PROC PHREG根本无法做的其他事情： -->
<!-- PROC LIFEREG可以容纳左检查和间隔检查。 -->
<!-- 使用PROC LIFEREG，您可以检验关于危害函数形状的某些假设。 PROC PHREG仅给您幸存者功能的非参数估计，这可能很难解释。 -->
<!-- 如果已知生存分布的形状，则PROC LIFEREG会比PROC PHREG产生更有效的估计（标准误差较小）。  -->
<!-- PROC LIFEREG可以轻松地为任何指定的一组协变量值生成预测事件时间。对于PHREG，这更困难，而且通常是不可能的。  -->
<!-- PROC LIFEREG的最大局限性在于它不能处理与时间相关的协变量，而PROC PHREG擅长于此。 -->
<p>What PROC LIFEREG actually estimates is a special case of this model
that is quite similar in form to an ordinary linear regression model.
Let <span class="math inline">\(T_{i}\)</span> be a random variable
denoting the event time for the ith individual in the sample, and let
<span class="math inline">\(x_{i 1}, \ldots, x_{i k}\)</span> be the
values of <span class="math inline">\(k\)</span> covariates for that
same individual. The model is then <span class="math display">\[
\log T_{i}=\beta_{0}+\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i k}+\sigma
\varepsilon_{i}
\]</span> where <span class="math inline">\(\varepsilon_{i}\)</span> is
a random disturbance term, and <span class="math inline">\(\beta_{0},
\ldots, \beta_{k}\)</span>, and <span
class="math inline">\(\sigma\)</span> are parameters to be estimated.
Exponentiating both sides gives an alternative way of expressing the
model: <span class="math display">\[
T_{i}=\exp \left(\beta_{0}+\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i
k}+\sigma \varepsilon_{i}\right)
\]</span> In regression analysis, <span class="math inline">\(-\log
(T)\)</span> represents fixed effects, and <span
class="math inline">\(\epsilon\)</span> represents noise. Different
<span class="math inline">\(\epsilon\)</span> distribution means
different distribution of <span
class="math inline">\(T_{0}\)</span>.</p>
<p>According to a simple transformation, we have <span
class="math inline">\(T_{i}=\exp \left(x_{i}^{T} \beta\right) T_{0},
T_{0}=\exp (W)\)</span>. Therefore, the AFT model parameter <span
class="math inline">\(\beta\)</span> represents: when the jth dimension
changes in <span class="math inline">\(\Delta_{j}\)</span>, the survival
time will change <span class="math inline">\(\exp \left(\Delta_{j}
\beta_{j}\right )\)</span></p>
</div>
<div id="residuum-distribution" class="section level3">
<h3>Residuum distribution</h3>
<p>Assuming <span class="math inline">\(Y=\log T\)</span>, the AFT Model
can be expressed as <span class="math display">\[
Y_{i}=x_{i}^{T} \beta+W_{i}
\]</span> <span class="math inline">\(W_{i} \sim f\)</span> represents
the <strong>independent residual</strong></p>
<ul>
<li>If <span class="math inline">\(W \sim N\left(0,
\sigma^{2}\right)\)</span>, the model at this time is called
<strong>Lognormal-AFT Model</strong>. Because the assumption of
normality for <span class="math inline">\(Y\)</span> is equivalent to
the assumption of lognormality. At this time, if there is no censorship,
we can directly use the LSE (least squares) method to solve the problem
(in fact, there is rarely such a possibility). Survival data usually has
at least some vetted observations, and it is difficult to process these
observations with <strong>OLS (Ordinary Least Squares)</strong>.
Instead, we can use <strong>maximum likelihood estimation</strong>.</li>
<li>If <span class="math inline">\(W \sim W
\operatorname{libull}(\lambda, \gamma)\)</span>, the model at this time
is called <strong>Weibull regression model</strong>. It is a special AFT
Model.</li>
<li>Following surmmaries the model</li>
</ul>
<p><span class="math display">\[
\begin{array}{c|l}
\text { Distribution of } \varepsilon &amp; \text { Distribution of } T
\\
\hline \text { extreme value (2 par.) } &amp; \text { Weibull } \\
\text { extreme value (1 par.) } &amp; \text { exponential } \\
\text { log-gamma } &amp; \text { gamma } \\
\text { logistic } &amp; \text { log-logistic } \\
\text { normal } &amp; \text { log-normal }
\end{array}
\]</span></p>
<p><strong>Specify Model in SAS</strong></p>
<pre><code>*** Lognormal-AFT Model;
PROC LIFEREG DATA=recid;
 MODEL week*arrest(0)=fin age race wexp mar paro prio / DISTRIBUTION=LNORMAL;
RUN;</code></pre>
</div>
<div id="exponential-model" class="section level3">
<h3>Exponential Model</h3>
<p>The simplest model estimated by PROC LIFEREG is the exponential
model, which is called by DISTRIBUTION = EXPONENTIAL in the MODEL
statement. The model specifies that <span
class="math inline">\(\varepsilon\)</span> has a standard extreme-value
distribution and constrains <span
class="math inline">\(\sigma=1\)</span>. If <span
class="math inline">\(\varepsilon\)</span> has an extreme value
distribution, then the logarithm T also has an extreme value
distribution, provided that the covariate is the condition. Constant
hazard function, expresses the exponential regression model as <span
class="math display">\[
\log h(t)=\beta_{0}^{\bullet}+\beta_{1}^{\bullet}
x_{1}+\ldots+\beta_{k}^{\bullet} x_{k}
\]</span></p>
<p><strong>Interpretation</strong></p>
<p><span class="math display">\[\lambda_{i}(t)=\lambda(t) \exp
\left(x_{i}^{T} \beta\right), \lambda(t) \geq 0\]</span> Where <span
class="math inline">\(\lambda(t)=\lambda\)</span> <span
class="math display">\[\frac{\lambda_{1}(t)}{\lambda_{2}(t)}=\exp
\left(\Delta_{j} \beta_{j}\right), \Delta_{j}=x_{1 j}-x_{2 j}\]</span>
Therefore, the meaning of the parameter <span
class="math inline">\(\beta_{j}\)</span> is: when the data <span
class="math inline">\(x_{j}\)</span> changes with the size of <span
class="math inline">\(\Delta_{j}\)</span>, the Hazard Ratio of the model
will change <span class="math inline">\(\exp \left(\Delta_{ j}
\beta_{j}\right)\)</span></p>
</div>
<div id="weibull-model" class="section level3">
<h3>Weibull Model</h3>
<p>The Weibull model is a slight modification of the exponential model
with great consequences. By specifying DISTRIBUTION = WEIBULL in the
MODEL statement, we retain the assumption that <span
class="math inline">\(\varepsilon\)</span> has a standard extreme value
distribution, but <strong>relax the assumption of <span
class="math inline">\(\sigma=1\)</span>.</strong></p>
<ul>
<li>When <span class="math inline">\(\sigma\)</span>&gt; 1, the danger
decreases with time.</li>
<li>When <span class="math inline">\(0.5&lt;\sigma&lt;1\)</span>, the
risk is increasing at a reduced rate.</li>
<li>When <span class="math inline">\(0&lt;\sigma&lt;0.5\)</span>, the
danger is increasing at an increasing rate.</li>
<li>-When <span class="math inline">\(\sigma=0.5\)</span>, the hazard
function is a straight line and its starting point is 0.</li>
</ul>
<p><strong>Weibull Distribution</strong></p>
<p><span class="math display">\[
\begin{array}{l}
\lambda(t)=\lambda \gamma(t \lambda)^{\gamma-1} \\
\Lambda(t)=(t \lambda)^{\gamma} \\
S(t)=\exp \left[-(t \lambda)^{\gamma}\right] \\
f(t)=\lambda \gamma(t \lambda)^{\gamma-1} \exp \left((t
\lambda)^{\gamma}\right)
\end{array}
\]</span></p>
<p><strong>Connection with exponential distribution</strong></p>
<p>If <span class="math inline">\(X \sim
\operatorname{Exp}(\tau)\)</span>, then <span
class="math inline">\(T=X^{\sigma} \sim\)</span> Weibull <span
class="math inline">\(\left(\gamma=1 / \sigma, \lambda=\tau^{\
sigma}\right)\)</span> also, - For exponential distribution, <span
class="math inline">\(\quad \log S(t)=(-\log t)+C_(1)\)</span> - For
Weibull distribution, <span class="math inline">\(\quad \log
S(t)=\gamma(-\log t)+C_{2}\)</span></p>
<p>T has a Weibull distribution conditioned on covariates.it has a
relatively simple survival function, which is easy to manipulate
mathematically: (where <span
class="math inline">\(\mathbf{x}_{i}\)</span> is a vector of the
covariate values and $ $ is a vector of coefficients.) <span
class="math display">\[
S_{i}(t)=\exp \left\{-\left[t_{i} e^{-\beta
\mathbf{x}_{i}}\right]^{\frac{1}{\sigma}}\right\}
\]</span> Secondly, in addition to the AFT model, the Weibull model is
also a proportional hazard model. This means that its coefficient (after
proper conversion) can be interpreted as a relative hazard ratio. In
fact, the Weibull model (and its special case, the exponential model) is
the only model that belongs to both categories. Like the exponential
model, there is a completely equivalent relationship between the
logarithmic risk forms of the model <span class="math display">\[
\log h(t)=\alpha \log t+\beta_{0}^{\bullet}+\beta_{1}^{\bullet}
x_{1}+\ldots+\beta_{k}^{\bullet} x_{k}
\]</span> and the log-survival time model <span class="math display">\[
\log T_{i}=\beta_{0}+\beta_{1} x_{1}+\ldots+\beta_{k} x_{k}+\sigma
\varepsilon
\]</span> However, the relationship between the parameters is slightly
more complicated. Specifically, for the Weibull model <span
class="math display">\[
\beta_{j}^{\bullet}=\frac{-\beta_{j}}{\sigma} \text { for } j=1, \ldots,
k
\]</span> <span class="math display">\[\alpha=(1 /
\sigma)-1.\]</span></p>
<pre><code># Output
# σˆ (labeled “Scale” 0.7124  in Output) is between 0 and 1
# calculate αˆ = (1/.7124)–1=0.4037, which is the coefficient for log t 
   in the log-hazard model. 
# “Weibull Shape.” This is not an independent parameter but merely 
   the reciprocal of the Shape parameter (that is, 1/σˆ ). </code></pre>
</div>
<div id="log-normal-model" class="section level3">
<h3>Log-Normal Model</h3>
<p>Unlike the Weibull model, the lognormal model has a non-monotonic
hazard function. When t=0, the danger is 0. It rises to a peak, and then
drops to 0 as t approaches infinity. The log normal is not a
proportional hazard model, and its risk function cannot be expressed in
a closed form. However, it can be expressed as a regression model, where
the dependent variable is the log hazard. <span class="math display">\[
\log h(t)=\log h_{0}\left(t e^{-\beta x}\right)-\beta \mathbf{x}
\]</span> where <span class="math inline">\(h_{0}\)</span> (.) can be
interpreted as the hazard function for an individual with <span
class="math inline">\(\mathbf{x}=\mathbf{0}\)</span>. (personal hazard
function)</p>
</div>
<div id="log-logistic-model" class="section level3">
<h3>Log-Logistic Model</h3>
<p>An inverted U-shaped hazard is the log-logistic model, which assumes
that ε has a logistic distribution with <span class="math display">\[
f(\varepsilon)=\frac{e^{\varepsilon}}{\left(1+e^{\varepsilon}\right)^{2}}
\]</span></p>
<p>DISTRIBUTION=LLOGISTIC, The logical distribution is symmetrical, with
an average value of 0, and is very similar in shape to the normal
distribution. If <span class="math inline">\(\varepsilon\)</span> has a
logarithmic distribution, then the logarithm <span
class="math inline">\(T\)</span> (although the mean is non-zero) has a
logarithmic distribution. The loglogistic hazard function is <span
class="math display">\[
h(t)=\frac{\lambda \gamma(\lambda t)^{\gamma-1}}{1+(\lambda t)^{\gamma}}
\]</span> where <span class="math inline">\(\gamma=1 / \sigma\)</span>,
and <span class="math display">\[
\lambda=\exp \left\{-\left[\beta_{0}+\beta_{1} x_{1}+\ldots+\beta_{k}
x_{k}\right]\right\}
\]</span></p>
<ul>
<li>When <span class="math inline">\(\sigma&lt;1\)</span>, the
logarithmic logic hazard is similar to the lognormal hazard: starting
from 0, rising to the peak, and then falling to 0.</li>
<li>When <span class="math inline">\(\sigma&gt;1\)</span>, the hazard
behaves like a diminishing Weibull hazard: it starts at infinity and
then decreases towards 0.</li>
<li>When <span class="math inline">\(\sigma=1\)</span>, the hazard value
at t <span class="math inline">\(=0\)</span> is <span
class="math inline">\(\lambda\)</span>, and then decreases toward 0 as t
tends to infinity.</li>
</ul>
<p>Although the logarithmic logic model has the complexity of the hazard
function, it has a fairly simple survival function <span
class="math display">\[
S(t)=\frac{1}{1+(\lambda t)^{\gamma}}
\]</span> As before, <span class="math inline">\(\gamma=1 /
\sigma\)</span> and <span class="math inline">\(\lambda=\exp
\left\{-\left[\beta_{0}+\beta_{1} x_{1}+\ldots+\beta_{k}
x_{k}\right]\right\} .\)</span> A little algebra shows that this can be
written as <span class="math display">\[
\log
\left[\frac{S(t)}{1-S(t)}\right]=\beta_{0}^{\bullet}+\beta_{1}^{\bullet}
x_{1}+\ldots+\beta_{k}^{\bullet} x_{k}-\gamma \log t
\]</span> Where <span
class="math inline">\(\beta_{i}^{\bullet}=\beta_{i} / \sigma\)</span>
for <span class="math inline">\(i=1, \ldots, k\)</span>. 这只是事件
<span class="math inline">\(t\)</span>
之前发生的概率的逻辑回归模型。</p>
</div>
<div id="fit-statistics-for-model-comparsion" class="section level3">
<h3>Fit statistics for model comparsion</h3>
<p><strong>The statistical information largely depends on the sample
size.</strong></p>
<ul>
<li>Akaike’s information criterion (AIC), is a modification of the <span
class="math inline">\(-2\)</span> log-likelihood that penalizes models
for having more covariates. where <span
class="math inline">\(\mathrm{k}\)</span> is the number of covariates.
<span class="math display">\[
A I C=-2 \log L+2 k
\]</span></li>
<li>(AICC) is a “corrected” version of the AIC that may have better
behavior in small samples: <span class="math display">\[
A I C C=A I C+\frac{2 k(k+1)}{n-k-1}
\]</span></li>
<li>Bayesian information criterion (also known as Schwarz’s criterion)
gives a more severe penalization for additional covariates
(对其他协变量进行更严厉的惩罚) <span class="math display">\[
B I C=-2 \log L+k \log n
\]</span></li>
</ul>
</div>
<div
id="graphical-method-for-distinguishing-different-distributions-in-sas"
class="section level3">
<h3>Graphical method for distinguishing different distributions in
SAS</h3>
<ul>
<li>the PLOTS=LS option produces a plot of -log and t. If the true
distribution is an exponential distribution, the graph should produce a
straight line with an origin of 0</li>
<li>The LLS option produces a plot of −loglog versus log t, if the true
distribution is Weibull, it should be a straight line;</li>
</ul>
</div>
<div id="prediction-and-hazard-function" class="section level3">
<h3>Prediction and hazard function</h3>
<p>After fitting the model with PROC LIFEREG, sometimes it is desirable
to generate predicted survival times for the observations in the data
set. If a single point estimate is to be made for each person, the
predicted median survival time may be the best. You can easily get this
information using the OUTPUT statement.</p>
<p>P in the OUTPUT statement = option request percentile. By default,
PROC LIFEREG calculates the 50th percentile (ie median). You can use the
QUANTILE keyword to request other percentiles, as described in the PROC
LIFEREG documentation. The STD keyword requests the standard error of
the median.</p>
<pre><code>PROC LIFEREG DATA=recid;
 MODEL week*arrest(0)=fin age race wexp mar paro prio
 / D=WEIBULL;
 OUTPUT OUT=a P=median STD=s;
RUN;</code></pre>
<!-- 还可以获取不在原始数据集中的一组协变量值的预测值。在估算模型之前，只需将具有所需协变量值且事件时间设置为缺失的其他观测值附加到数据集即可。
这些添加的观察值不用于估计模型，但是将为它们生成预测值。研究人员通常不希望预测生存时间，而是要预测生存到指定时间的概率（例如5年生存概率）。
虽然这些不是由PROC LIFEREG直接计算的，但计算起来却相当简单。这是通过将可由OUTPUT语句生成的线性预测值替换为幸存函数的公式来实现的 %PREDICT. -->
<!-- 使用OUTEST =选项将参数估计值写入数据集。接下来，在OUTPUT语句中使用XBETA =选项请求将线性预测变量包含在第二个数据集中。最后，调用宏，指示两个数据集的名称，分配给线性预测变量的名称以及计算生存概率的时间。 -->
<p><strong>Predict the probability of survival to a specified
time</strong></p>
<pre><code>PROC LIFEREG DATA=recid OUTEST=a;
 MODEL week*arrest(0) = fin age race wexp mar paro prio
 / D=WEIBULL;
 OUTPUT OUT=b XBETA=lp;
RUN;
%PREDICT(OUTEST=a,OUT=b,XBETA=lp,TIME=30)


%macro predict (outest=, out=_last_,xbeta=,time=);
 data _pred_;
 _p_=1;
 set &amp;outest point=_p_;
 set &amp;out;
 lp=&amp;xbeta;
 t=&amp;time;
 gamma=1/_scale_;
 alpha=exp(-lp*gamma);
 prob=0;
 _dist_=upcase(_dist_);
 if _dist_=&#39;WEIBULL&#39; or _dist_=&#39;EXPONENTIAL&#39; or _dist_=&#39;EXPONENT&#39; then prob=exp(-
 alpha*t**gamma);
 if _dist_=&#39;LOGNORMAL&#39; or _dist_=&#39;LNORMAL&#39; then prob=1-probnorm((log(t)-lp)/_scale_);
 if _dist_=&#39;LLOGISTIC&#39; or _dist_=&#39;LLOGISTC&#39; then prob=1/(1+alpha*t**gamma);
 if _dist_=&#39;GAMMA&#39; then do;
  d=_shape1_;
  k=1/(d*d);
  u=(t*exp(-lp))**gamma;
  prob=1-probgam(k*u**d,k);
  if d lt 0 then prob=1-prob;
  end;
 drop lp gamma alpha _dist_ _scale_ intercept
  _shape1_ _model_ _name_ _type_ _status_ _prob_ _lnlike_ d k u;
 run;
 proc print data=_pred_;
 run;
%mend predict;

*** %LIFEHAZ generated a graph of hazard versus time;
PROC LIFEREG DATA=recid OUTEST=a;
 MODEL week*arrest(0) = fin age race wexp mar paro prio
 / D=WEIBULL;
 OUTPUT OUT=b XBETA=lp;
RUN;
%LIFEHAZ(OUTEST=a,OUT=b,XBETA=lp)

%macro lifehaz(outest=,out=,obsno=0,xbeta=lp);
data;
 set &amp;outest;
 call symput(&#39;time&#39;,_NAME_);
run;
proc means data=&amp;out noprint;
 var &amp;time &amp;xbeta;
 output out=_c_ min(&amp;time)=min max(&amp;time)=max mean(&amp;xbeta)=mean;
run;
data;
 set &amp;outest;
 call symput(&#39;model&#39;,_dist_);
 s=_scale_;
 d=_shape1_;
 _y_=&amp;obsno;
 set _c_ (keep=min max mean);
 if _y_=0 then m=mean;
 else do;
 set &amp;out (keep=&amp;xbeta) point=_y_;
 m=&amp;xbeta;
 end;
 inc=(max-min)/300;
 g=1/s;
 alph=exp(-m*g);
 _dist_=upcase(_dist_);
if _dist_=&#39;LOGNORMAL&#39; or _dist_=&#39;LNORMAL&#39; then do;
 do t=min to max by inc;
 z=(log(t)-m)/s;
 f=exp(-z*z/2)/(t*s*sqrt(2*3.14159));
 Surv=1-probnorm(z);
 h=f/Surv;
 output;
 end;
end;
else if _dist_=&#39;GAMMA&#39; then do;
 k=1/(d*d);
 do t=min to max by inc;
 u=(t*exp(-m))**(1/s);
 f=abs(d)*(k*u**d)**k*exp(-k*u**d)/(s*gamma(k)*t);
 Surv=1-probgam(k*u**d,k);
 if d lt 0 then Surv=1-Surv;
 h=f/Surv;
 output;
 end;
end;
else if _dist_=&#39;WEIBULL&#39; or _dist_=&#39;EXPONENTIAL&#39; or _dist_=&#39;EXPONENT&#39; then do;
 do t=min to max by inc;
 h=g*alph*t**(g-1);
 output;
 end;
end;
else if _dist_=&#39;LLOGISTIC&#39; or _dist_=&#39;LLOGISTC&#39; then do;
 do t=min to max by inc;
 h=g*alph*t**(g-1)/(1+alph*t**g);
 output;
 end;
end;
else put &#39;ERROR:DISTRIBUTION NOT FITTED BY LIFEREG&#39;;
run;
proc gplot;
 plot h*t / haxis=axis2 vaxis=axis1 vzero;
 symbol1 i=join v=none c=black;
 axis1 label=(f=titalic angle=90 &#39;Hazard&#39;);
 axis2 label=(f=titalic justify=c &#39;time&#39; f=titalic justify=c &quot;&amp;model&quot;);
run; quit;
%mend lifehaz;</code></pre>
</div>
<div id="left-censoring-and-interval-censoring" class="section level3">
<h3>Left Censoring and Interval Censoring</h3>
<p>PROC LIFEREG excludes any observations with a time of 0 or less,
because it must take the logarithm of the event time as the first step.
One way is to assign some arbitrarily chosen number between 0 and 1. A
more elegant solution is to treat this situation as being censored at
time 1.</p>
<!-- PROC LIFEREG排除时间为0或更短的任何观测值，因为它必须将事件时间的对数作为第一步。一种方法是在0到1之间分配一些任意选择的数字。一种更优雅的解决方案是将这种情况视为在时间1处被审查。 -->
<ul>
<li>For uncensored observations, LOWER and UPPER have the same
value.</li>
<li>For observations with YEARS=0, we set LOWER=. and UPPER=1.</li>
<li>For right-censored observations (including those with events other
than the one of interest), UPPER=. and LOWER=YEARS.</li>
<li>If both UPPER and LOWER are missing (which happens for individuals
with YEARS=0 who have events other than the one of interest), the
observation is excluded. T</li>
</ul>
<pre><code>DATA leaders2;
 SET leaders;
 lower=years;
 upper=years;
 IF years=0 THEN DO;
 lower=.;
 upper=1;
 END;
 IF lost IN (0,1,2) THEN upper=.;
RUN;

*** fit an exponential model;
PROC LIFEREG DATA=leaders2;
 CLASS region;
 MODEL (lower,upper)= manner start military age conflict
 loginc literacy region / D=EXPONENTIAL;
RUN;</code></pre>
</div>
</div>
<div id="parametric-model-in-r" class="section level2">
<h2>Parametric Model in R</h2>
<div id="fit-weibull-model" class="section level3">
<h3>Fit Weibull-model</h3>
<p>Shape parameter <span class="math inline">\(a &gt; 0\)</span>, scale
parameter <span class="math inline">\(b &gt; 0\)</span>, such that <span
class="math inline">\(f(t) = \lambda(t) \, S(t)\)</span> with
hazard-function <span class="math inline">\(\lambda(t) = \frac{a}{b}
\left(\frac{t}{b}\right)^{a-1}\)</span> and survival-function <span
class="math inline">\(S(t) = \exp(-(\frac{t}{b})^{a})\)</span>.
Cumulative hazard-function is <span class="math inline">\(\Lambda(t) =
(\frac{t}{b})^{a}\)</span> with inverse <span
class="math inline">\(\Lambda^{-1}(t) = (b \, t)^{\frac{1}{a}}\)</span>.
<span class="math inline">\(E(T) = b \, \Gamma(1 +
\frac{1}{a})\)</span>. The exponential distribution is a special case
for <span class="math inline">\(a = 1\)</span>.</p>
<p>Alternative parametrization:</p>
<p>Setting <span class="math inline">\(\lambda =
\frac{1}{b^{a}}\)</span>, such that <span
class="math inline">\(\lambda(t) = \lambda \, a \, t^{a-1}\)</span>,
<span class="math inline">\(S(t) = \exp(-\lambda \, t^{a})\)</span> and
<span class="math inline">\(\Lambda(t) = \lambda \, t^{a}\)</span>. The
influence of covariates <span class="math inline">\(X_{j}\)</span>
causes <span class="math inline">\(\lambda&#39; = \lambda \, e^{\bf{X}
\bf{\beta}}\)</span>.</p>
<p>Assumptions (note that <span class="math inline">\(\bf{X}
\bf{\beta}\)</span> does not include an intercept <span
class="math inline">\(\beta_{0}\)</span>)</p>
<p><span class="math display">\[
\begin{equation*}
\begin{array}{rclcl}
\lambda(t)     &amp;=&amp; \frac{a}{b} \left(\frac{t}{b}\right)^{a-1} \,
e^{\bf{X} \bf{\beta}} &amp;=&amp; \lambda \, a \, t^{a-1} \, e^{\bf{X}
\bf{\beta}}\\
\ln \lambda(t) &amp;=&amp; \ln \left(\frac{a}{b}
\left(\frac{t}{b}\right)^{a-1}\right) + \bf{X} \bf{\beta} &amp;=&amp;
\ln \lambda + \ln a + (a-1) \, \ln t + \bf{X} \bf{\beta}\\
S(t)           &amp;=&amp; \exp\left(-(\frac{t}{b})^{a} \, e^{\bf{X}
\bf{\beta}}\right) &amp;=&amp; \exp\left(-\lambda \, t^{a} \, e^{\bf{X}
\bf{\beta}}\right)\\
\Lambda(t)     &amp;=&amp; (\frac{t}{b})^{a} \, e^{\bf{X} \bf{\beta}}
&amp;=&amp; \lambda \, t^{a} \, e^{\bf{X} \bf{\beta}}
\end{array}
\end{equation*}
\]</span></p>
<p><strong>Simulated right-censored event times with Weibull
distribution</strong></p>
<pre class="r"><code>set.seed(123)
N      &lt;- 180                  # number of observations
P      &lt;- 3                    # number of groups
sex    &lt;- factor(sample(c(&quot;f&quot;, &quot;m&quot;), N, replace=TRUE))  # stratification factor
X      &lt;- rnorm(N, 0, 1)       # continuous covariate
IV     &lt;- factor(rep(LETTERS[1:P], each=N/P))  # factor covariate
IVeff  &lt;- c(0, -1, 1.5)        # effects of factor levels (1 -&gt; reference level)
Xbeta  &lt;- 0.7*X + IVeff[unclass(IV)] + rnorm(N, 0, 2)
weibA  &lt;- 1.5                  # Weibull shape parameter
weibB  &lt;- 100                  # Weibull scale parameter
U      &lt;- runif(N, 0, 1)       # uniformly distributed RV
eventT &lt;- ceiling((-log(U)*weibB*exp(-Xbeta))^(1/weibA))   # simulated event time

# censoring due to study end after 120 days
obsLen &lt;- 120                  # length of observation time
censT  &lt;- rep(obsLen, N)       # censoring time = end of study
obsT   &lt;- pmin(eventT, censT)  # observed censored event times
status &lt;- eventT &lt;= censT      # has event occured?
dfSurv &lt;- data.frame(obsT, status, sex, X, IV)          # data frame

## Survival data in counting process (start-stop) notation.
dfSurvCP &lt;- survSplit(dfSurv, cut=seq(30, 90, by=30), end=&quot;obsT&quot;,
                      event=&quot;status&quot;, start=&quot;start&quot;, id=&quot;ID&quot;, zero=0)
## Plot simulated data
plot(ecdf(eventT), xlim=c(0, 200), main=&quot;Cumulative survival distribution&quot;,
     xlab=&quot;t&quot;, ylab=&quot;F(t)&quot;, cex.lab=1.4)
abline(v=obsLen, col=&quot;blue&quot;, lwd=2)
text(obsLen-5, 0.2, adj=1, labels=&quot;end of study&quot;, cex=1.4)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>## Fit Weibull-model
fitWeib &lt;- survreg(Surv(obsT, status) ~ X + IV, dist=&quot;weibull&quot;, data=dfSurv)
summary(fitWeib)</code></pre>
<pre><code>## 
## Call:
## survreg(formula = Surv(obsT, status) ~ X + IV, data = dfSurv, 
##     dist = &quot;weibull&quot;)
##               Value Std. Error     z       p
## (Intercept)  3.1672     0.1779 17.80 &lt; 2e-16
## X           -0.6585     0.1138 -5.79 7.1e-09
## IVB          1.0942     0.2634  4.15 3.3e-05
## IVC         -0.5840     0.2432 -2.40 0.01631
## Log(scale)   0.2393     0.0625  3.83 0.00013
## 
## Scale= 1.27 
## 
## Weibull distribution
## Loglik(model)= -673.9   Loglik(intercept only)= -703.6
##  Chisq= 59.38 on 3 degrees of freedom, p= 8e-13 
## Number of Newton-Raphson Iterations: 5 
## n= 180</code></pre>
<pre class="r"><code>fitWeib %&gt;%
  tbl_regression(exponentiate = TRUE)</code></pre>
<div id="peudwcojgz" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#peudwcojgz table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#peudwcojgz thead, #peudwcojgz tbody, #peudwcojgz tfoot, #peudwcojgz tr, #peudwcojgz td, #peudwcojgz th {
  border-style: none;
}

#peudwcojgz p {
  margin: 0;
  padding: 0;
}

#peudwcojgz .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#peudwcojgz .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#peudwcojgz .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#peudwcojgz .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#peudwcojgz .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#peudwcojgz .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#peudwcojgz .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#peudwcojgz .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#peudwcojgz .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#peudwcojgz .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#peudwcojgz .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#peudwcojgz .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#peudwcojgz .gt_spanner_row {
  border-bottom-style: hidden;
}

#peudwcojgz .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#peudwcojgz .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#peudwcojgz .gt_from_md > :first-child {
  margin-top: 0;
}

#peudwcojgz .gt_from_md > :last-child {
  margin-bottom: 0;
}

#peudwcojgz .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#peudwcojgz .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#peudwcojgz .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#peudwcojgz .gt_row_group_first td {
  border-top-width: 2px;
}

#peudwcojgz .gt_row_group_first th {
  border-top-width: 2px;
}

#peudwcojgz .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#peudwcojgz .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#peudwcojgz .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#peudwcojgz .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#peudwcojgz .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#peudwcojgz .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#peudwcojgz .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#peudwcojgz .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#peudwcojgz .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#peudwcojgz .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#peudwcojgz .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#peudwcojgz .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#peudwcojgz .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#peudwcojgz .gt_left {
  text-align: left;
}

#peudwcojgz .gt_center {
  text-align: center;
}

#peudwcojgz .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#peudwcojgz .gt_font_normal {
  font-weight: normal;
}

#peudwcojgz .gt_font_bold {
  font-weight: bold;
}

#peudwcojgz .gt_font_italic {
  font-style: italic;
}

#peudwcojgz .gt_super {
  font-size: 65%;
}

#peudwcojgz .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#peudwcojgz .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#peudwcojgz .gt_indent_1 {
  text-indent: 5px;
}

#peudwcojgz .gt_indent_2 {
  text-indent: 10px;
}

#peudwcojgz .gt_indent_3 {
  text-indent: 15px;
}

#peudwcojgz .gt_indent_4 {
  text-indent: 20px;
}

#peudwcojgz .gt_indent_5 {
  text-indent: 25px;
}

#peudwcojgz .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#peudwcojgz div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="estimate"><span class='gt_from_md'><strong>exp(Beta)</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="conf.low"><span class='gt_from_md'><strong>95% CI</strong></span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="p.value"><span class='gt_from_md'><strong>p-value</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">X</td>
<td headers="estimate" class="gt_row gt_center">-0.66</td>
<td headers="conf.low" class="gt_row gt_center">-0.88, -0.44</td>
<td headers="p.value" class="gt_row gt_center"><0.001</td></tr>
    <tr><td headers="label" class="gt_row gt_left">IV</td>
<td headers="estimate" class="gt_row gt_center"><br /></td>
<td headers="conf.low" class="gt_row gt_center"><br /></td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    A</td>
<td headers="estimate" class="gt_row gt_center">—</td>
<td headers="conf.low" class="gt_row gt_center">—</td>
<td headers="p.value" class="gt_row gt_center"><br /></td></tr>
    <tr><td headers="label" class="gt_row gt_left">    B</td>
<td headers="estimate" class="gt_row gt_center">1.09</td>
<td headers="conf.low" class="gt_row gt_center">0.58, 1.61</td>
<td headers="p.value" class="gt_row gt_center"><0.001</td></tr>
    <tr><td headers="label" class="gt_row gt_left">    C</td>
<td headers="estimate" class="gt_row gt_center">-0.58</td>
<td headers="conf.low" class="gt_row gt_center">-1.06, -0.11</td>
<td headers="p.value" class="gt_row gt_center">0.016</td></tr>
  </tbody>
  
  <tfoot class="gt_footnotes">
    <tr>
      <td class="gt_footnote" colspan="4"><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span> <span class='gt_from_md'>CI = Confidence Interval</span></td>
    </tr>
  </tfoot>
</table>
</div>
<pre class="r"><code>citation(&quot;gtsummary&quot;)</code></pre>
<pre><code>## To cite gtsummary in publications use:
## 
##   Sjoberg DD, Whiting K, Curry M, Lavery JA, Larmarange J. Reproducible
##   summary tables with the gtsummary package. The R Journal
##   2021;13:570–80. https://doi.org/10.32614/RJ-2021-053.
## 
## A BibTeX entry for LaTeX users is
## 
##   @Article{gtsummary,
##     author = {Daniel D. Sjoberg and Karissa Whiting and Michael Curry and Jessica A. Lavery and Joseph Larmarange},
##     title = {Reproducible Summary Tables with the gtsummary Package},
##     journal = {{The R Journal}},
##     year = {2021},
##     url = {https://doi.org/10.32614/RJ-2021-053},
##     doi = {10.32614/RJ-2021-053},
##     volume = {13},
##     issue = {1},
##     pages = {570-580},
##   }</code></pre>
<p>Transform AFT parameter <span class="math inline">\(\hat{\gamma}_{j}
= - \hat{\beta}_{j} \cdot \hat{a}\)</span> to parameter <span
class="math inline">\(\hat{\beta}_{j}\)</span> a Cox PH model.</p>
<pre class="r"><code>## AFT parameters
(betaHat &lt;- -coef(fitWeib) / fitWeib$scale)</code></pre>
<pre><code>## (Intercept)           X         IVB         IVC 
##  -2.4932013   0.5183936  -0.8613157   0.4597344</code></pre>
<pre class="r"><code>## Cox PH parameters
fitCox &lt;- coxph(Surv(obsT, status) ~ X + IV, data=dfSurv)
coef(fitCox)</code></pre>
<pre><code>##          X        IVB        IVC 
##  0.4930364 -0.8224490  0.3766960</code></pre>
</div>
<div id="fit-exponential-model" class="section level3">
<h3>Fit Exponential Model</h3>
<p>Exponential model = restricted Weibull model with shape parameter
<span class="math inline">\(a = 1\)</span>.</p>
<pre class="r"><code>fitExp &lt;- survreg(Surv(obsT, status) ~ X + IV, dist=&quot;exponential&quot;, data=dfSurv)
coef(fitExp)</code></pre>
<pre><code>## (Intercept)           X         IVB         IVC 
##   3.2765279  -0.6491499   1.0059347  -0.5709446</code></pre>
</div>
<div id="model-comparisons" class="section level3">
<h3>Model Comparisons</h3>
<pre class="r"><code>anova(fitExp, fitWeib)               # model comparison</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Terms"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Resid. Df"],"name":[2],"type":["int"],"align":["right"]},{"label":["-2*LL"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Test"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Df"],"name":[5],"type":["int"],"align":["right"]},{"label":["Deviance"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Pr(>Chi)"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"X + IV","2":"176","3":"1364.383","4":"","5":"NA","6":"NA","7":"NA","_rn_":"1"},{"1":"X + IV","2":"175","3":"1347.808","4":"=","5":"1","6":"16.575","7":"4.676351e-05","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<pre class="r"><code>## Test significance of factor IV (associated with multiple effect parameters) as a whole by doing a model comparison against the restricted model without factor IV.
## restricted model without IV
fitR &lt;- survreg(Surv(obsT, status) ~ X, dist=&quot;weibull&quot;, data=dfSurv)
anova(fitR, fitWeib)                 </code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Terms"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Resid. Df"],"name":[2],"type":["int"],"align":["right"]},{"label":["-2*LL"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["Test"],"name":[4],"type":["chr"],"align":["left"]},{"label":["Df"],"name":[5],"type":["int"],"align":["right"]},{"label":["Deviance"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["Pr(>Chi)"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"X","2":"177","3":"1390.316","4":"","5":"NA","6":"NA","7":"NA","_rn_":"1"},{"1":"X + IV","2":"175","3":"1347.808","4":"+IV","5":"2","6":"42.5078","7":"5.882312e-10","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="predict-using-new-data" class="section level3">
<h3>Predict using New Data</h3>
<p>Apply fit to new data with specific observations. Result from
<code>predict()</code> is the estimated distribution function <span
class="math inline">\(\hat{F}^{-1}(t)\)</span> for given percentiles
<code>p</code>.</p>
<pre class="r"><code>dfNew &lt;- data.frame(sex=factor(c(&quot;m&quot;, &quot;m&quot;), levels=levels(dfSurv$sex)),
                      X=c(0, 0),
                     IV=factor(c(&quot;A&quot;, &quot;C&quot;), levels=levels(dfSurv$IV)))
percs &lt;- (1:99)/100
FWeib &lt;- predict(fitWeib, newdata=dfNew, type=&quot;quantile&quot;, p=percs, se=TRUE)

matplot(cbind(FWeib$fit[1, ],
              FWeib$fit[1, ] - 2*FWeib$se.fit[1, ],
              FWeib$fit[1, ] + 2*FWeib$se.fit[1, ]), 1-percs,
        type=&quot;l&quot;, main=expression(paste(&quot;Weibull-Fit &quot;, hat(S)(t), &quot; mit SE&quot;)),
        xlab=&quot;t&quot;, ylab=&quot;Survival&quot;, lty=c(1, 2, 2), lwd=2, col=&quot;blue&quot;)
matlines(cbind(FWeib$fit[2, ],
               FWeib$fit[2, ] - 2*FWeib$se.fit[2, ],
               FWeib$fit[2, ] + 2*FWeib$se.fit[2, ]), 1-percs, col=&quot;red&quot;, lwd=2)
legend(x=&quot;topright&quot;, lwd=2, lty=c(1, 2, 1, 2), col=c(&quot;blue&quot;, &quot;blue&quot;, &quot;red&quot;, &quot;red&quot;),
       legend=c(&quot;sex=m, X=0, IV=A&quot;, &quot;+- 2*SE&quot;, &quot;sex=m, X=0, IV=C&quot;, &quot;+- 2*SE&quot;))</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="cox-proportional-hazards-model" class="section level1">
<h1>Cox Proportional Hazards Model</h1>
<div id="introduction-1" class="section level2">
<h2>Introduction</h2>
<p>The basic model that does not include time-dependent covariates or
nonproportional hazards. The model is usually written as <span
class="math display">\[h_{i}(t)=\lambda_{0}(t) \exp \left(\beta_{1} x_{i
1}+\ldots+\beta_{k} x_{i k}\right)\]</span> Taking the logarithm of both
sides, we can rewrite the model as <span class="math display">\[\log
h_{i}(t)=\alpha(t)+\beta_{1} x_{i 1}+\ldots+\beta_{k} x_{i k}\]</span>
Take the ratio of the hazards for two individuals <span
class="math inline">\(i\)</span> and <span
class="math inline">\(j\)</span> <span
class="math display">\[\frac{h_{i}(t)}{h_{j}(t)}=\exp
\left\{\beta_{1}\left(x_{i 1}-x_{j}\right)+\ldots+\beta_{k}\left(x_{i
k}-x_{j k}\right)\right\}\]</span></p>
<div id="partial-likelihood" class="section level3">
<h3>Partial Likelihood</h3>
<p>If censoring is not considered, <span class="math inline">\(\quad
j\)</span> observations <span class="math inline">\(T_{1} \sim
\lambda_{1}(t), T_{2} \sim \lambda_{2}(t), \ldots T_{j } \sim
\lambda_{j}(t)\)</span> satisfies under the assumption of Cox Regression
<span class="math display">\[
P\left(T_{1}&lt;T_{2}&lt;\ldots&lt;T_{J}\right)=\prod_{i=1}^{J}
\frac{\lambda_{j}(t)}{\sum_{k=j}^{J} \lambda_{k}(t)}=\prod_{i=1}^{J}
\frac{\exp \left(x_{j}^{T} \beta\right)}{\sum_{k=j}^{J} \exp
\left(x_{k}^{T} \beta\right)}
\]</span> Furthermore, when the censor be considered, the time point of
death event is denoted as <span class="math inline">\(t_{j}\)</span>,
then the <span class="math inline">\(\mathrm{P}(death)\)</span> <span
class="math display">\[
\mathrm{P}(death)=\frac{\exp \left(x_{j}^{T} \beta\right)}{\sum_{k \in
R\left(t_{j}\right)} \exp \left(x_{k} \beta\right)}
\]</span> given the subject’s covariates <span
class="math inline">\(\beta\)</span>; the joint probability of observing
each subject’s failure time can be calculated given the subject’s
covariates values <span class="math inline">\(x_{j}\)</span> likelihood
<span class="math inline">\(L(\beta)\)</span> could be expressed as
<span class="math display">\[
L(\beta)=\prod_{j} \frac{\exp \left(x_{j}^{T} \beta\right)}{\sum_{k \in
R\left(t_{j}\right)} \exp \left(x_{k} \beta\right)}
\]</span> <span class="math inline">\(k \in R\left(t_{j}\right)\)</span>
represents at time <span class="math inline">\(t_{j}-\)</span> the
subject <span class="math inline">\(k\)</span> is still alive. Here we
temporarily assume that there is only one death observation at time
<span class="math inline">\(t_{j}\)</span>.</p>
<p>Strictly speaking, this is not a likelihood function. <strong>Because
the likelihood function should represent the continuous multiplication
of the distribution function under the given data</strong>, but the
above formula is obviously not. Even speaking, the continuous
multiplication should be about the continuous multiplication of the
individual, and the above formula is about the point in time. Cox (1975)
called this form of likelihood Partial Likelihood, and proved that all
its properties are the same as likelihood.</p>
</div>
<div id="nr-iteration-for-parameter-estimate" class="section level3">
<h3>NR Iteration for Parameter Estimate</h3>
<p>The iteration form is <span class="math display">\[
\hat{\beta}^{(r+1)}=\hat{\beta}^{(r)}+\left(X^{T} WX\right)^{-1} X^{T}
(dP d)
\]</span></p>
<ul>
<li><span class="math inline">\(w_{i}=\exp \left(x_{i}^{T}
\beta\right)\)</span></li>
<li><span class="math inline">\(Y_{i}\left(t_{j}\right)\)</span> 1 if
the individual is still alive at the moment <span
class="math inline">\(t_{j}\)</span>, <span
class="math inline">\(Y_{i}\left(t_{j}\right)\)</span> Is the index of
an individual’s survival at a certain time <span
class="math inline">\(t_{j}\)</span></li>
<li><span class="math inline">\(\pi_{ij}=Y_{i}\left(t_{j}\right)
\frac{w_{i}}{\sum_{k \in R\left(t_{j}\right)} w_
{k}}=Y_{i}\left(t_{j}\right) \frac{w_{i}}{\sum_{k=1}^{n}
Y_{k}\left(t_{j} \right) w_{k}}\)</span></li>
<li><span class="math inline">\(P=\left\{\pi_{i j}\right\}_{i,
j}\)</span>, represents the relative death risk of the individual <span
class="math inline">\(i\)</span> at time <span
class="math inline">\(t_{j}\)</span>. So we only consider individuals
who are still alive</li>
<li><span class="math inline">\(W_{k k}=-\sum_{i} \delta_{i} \pi_{k
i}\left(1-\pi_{k i}\right), W_{k k}\)</span> represents the weighting of
the individual. Whenever one of the remaining individuals <span
class="math inline">\(i\)</span> dies, if <span
class="math inline">\(k\)</span> is still alive, its weight increases by
<span class="math inline">\(\pi(1-\pi)\)</span></li>
<li><span class="math inline">\(W_{k j}=\sum_{i} \delta_{i} \pi_{k i}
\pi_{j i}, k j\)</span> represents the interaction of the individual
<span class="math inline">\(k\)</span>. Whenever one of the remaining
individuals dies, if <span class="math inline">\(k, j\)</span> are still
alive <span class="math inline">\((\)</span> Y will Set), the
interaction weight increases <span class="math inline">\(\pi_(k)
\pi_(j)\)</span></li>
</ul>
</div>
</div>
<div id="parameter-test" class="section level2">
<h2>Parameter test</h2>
<p>In standard likelihood theory, one can derive three forms of the test
of <span class="math inline">\(H_0: {\beta}=0\)</span>: the Wald test,
the score test, and the likelihood ratio test. In survival analysis, we
may use the partial likelihood to derive these three tests, although the
underlying statistical theory for the partial likelihood is far more
complex than that for standard likelihood theory.</p>
<p>To develop the tests, we need two functions derived from the partial
log likelihood. The first, the score function, is the first derivative
of the log likelihood, <span
class="math inline">\(S(\beta)=l^{\prime}(\beta)\)</span>. The second
function, the information, is minus the derivative of the score
function, or equivalently minus the second derivative of the
log-likelihood, <span
class="math inline">\(I(\beta)=-S^{\prime}(\beta)=-l^{\prime
\prime}(\beta)\)</span>. The second derivative <span
class="math inline">\(l^{\prime \prime}(\beta)\)</span> is also known as
the Hessian. With the substitution of the parameter estimate <span
class="math inline">\(\hat{\beta}\)</span> into the information, we
obtain the observed information.</p>
<div id="wald-test" class="section level3">
<h3>Wald Test</h3>
<p><span class="math display">\[\quad \hat{\beta} \sim
N\left(\beta,\left(X^{T} W X\right)^{-1}\right)\]</span></p>
<p>The Wald test is perhaps the most commonly used test, and carrying it
out is straightforward from computer output. The test statistic is of
the form <span class="math inline">\(Z=\)</span> <span
class="math inline">\(\hat{\beta} /\)</span> s.e. <span
class="math inline">\((\hat{\beta})\)</span>, where “s.e.” stands for
“standard error.” In the previous section, we saw that <span
class="math inline">\(\hat{\beta}\)</span> was the value of <span
class="math inline">\(\beta\)</span> that maximizes <span
class="math inline">\(l(\beta)\)</span>. We know from basic differential
calculus that we may find the maximum by solving <span
class="math inline">\(S(\beta)=l^{\prime}(\beta)=0\)</span> for <span
class="math inline">\(\beta\)</span>. Ordinarily this is a non-linear
function that must be solved numerically using an iterative procedure.
To find the variance of <span
class="math inline">\(\hat{\beta}\)</span>, we evaluate the information,
<span class="math inline">\(I(\hat{\beta})=\)</span> <span
class="math inline">\(-l^{\prime \prime}(\hat{\beta})\)</span>. That is,
the information (technically, the “observed information”) is minus the
second derivative of the partial likelihood, evaluated at <span
class="math inline">\(\hat{\beta} . I(\hat{\beta})\)</span> is a measure
of the curvature of the likelihood at <span
class="math inline">\(\hat{\beta}\)</span>. Intuitively, higher values
of the curvature reflect a sharper curve, more “information”, and lower
variance. Lower curvatures, by contrast, corresponds to flatter curves
and higher variance. The variance of <span
class="math inline">\(\hat{\beta}\)</span> is approximately <span
class="math inline">\(1 / I(\hat{\beta})\)</span>, and the standard
error is <span class="math inline">\(\operatorname{s.e} .(\hat{\beta})=1
/ \sqrt{I(\hat{\beta})}\)</span>. We may use this to construct a
normalized test statistic <span class="math inline">\(Z_w=\hat{\beta} /
\operatorname{se}(\hat{\beta})\)</span>, and reject <span
class="math inline">\(H_0: \beta=0\)</span> if <span
class="math inline">\(\left|Z_w\right|&gt;z_{\alpha / 2}\)</span>. We
can also construct a <span class="math inline">\(1-\alpha\)</span>
confidence interval, <span class="math inline">\(\hat{\beta} \pm
z_{\alpha / 2} \cdot\)</span> s.e. <span
class="math inline">\((\hat{\beta})\)</span>. Equivalently, we can use
the fact that the square of a standard normal random variable has a
chi-square distribution with one degree of freedom, and reject the null
hypothesis if <span class="math inline">\(Z_w^2&gt;\chi_{\alpha,
1}^2\)</span>.</p>
</div>
<div id="likelihood-ratio-test" class="section level3">
<h3>Likelihood Ratio Test</h3>
<p>The likelihood ratio test uses the result from statistical theory
that <span
class="math inline">\(2[l(\beta=\hat{\beta})-l(\beta=0)]\)</span>
follows approximately a chi-square distribution with one degree of
freedom. The key advantage of this test over the other two is that it is
invariant to monotonic transformations of <span
class="math inline">\(\beta\)</span>. For example, whether the test is
computed in terms of <span class="math inline">\(\beta\)</span> or in
terms of <span class="math inline">\(\psi=e^\beta\)</span> has no effect
at all on the p-value for testing <span class="math inline">\(H_0:
\beta=0\)</span>.</p>
</div>
<div id="score-test" class="section level3">
<h3>Score Test</h3>
<p>The score function is the first derivative of the partial
log-likelihood, <span class="math inline">\(S(\beta)=\)</span> <span
class="math inline">\(l^{\prime}(\beta)\)</span>. The variance of the
score statistic is <span class="math inline">\(I(\beta)\)</span>. We
evaluate the score and information at the null hypothesis value of <span
class="math inline">\(\beta\)</span>, normally <span
class="math inline">\(\beta=0\)</span>. The test statistic is <span
class="math inline">\(Z_s=S(\beta=0) / \sqrt{I(\beta=0)}\)</span>, and
we reject <span class="math inline">\(H_0: \beta=0\)</span> if <span
class="math inline">\(\left|Z_s\right|&gt;z_{\alpha / 2}\)</span>, or
equivalently if <span class="math inline">\(Z_s^2&gt;\chi_{\alpha,
1}^2\)</span>. The score test is equivalent to the log-rank test, as we
saw in the previous section. This test can be carried out without
finding the maximum likelihood estimate <span
class="math inline">\(\hat{\beta}\)</span>.</p>
</div>
</div>
<div id="check-proportional-hazards" class="section level2">
<h2>Check Proportional Hazards</h2>
<p>A central assumption of Cox regression is that covariate effects on
the hazard rate, namely <strong>hazard ratios, are constant over
time</strong>. For example, if males have twice the hazard rate of
females 1 day after followup, the Cox model assumes that males have
twice the hazard rate at 1000 days after follow up as well. Violations
of the proportional hazard assumption may cause bias in the estimated
coefficients as well as incorrect inference regarding significance of
effects.</p>
<div id="graphing-check" class="section level3">
<h3>Graphing Check</h3>
<p>Graphing Kaplan-Meier survival function estimates to assess
proportional hazards for categorical covariates</p>
<ol style="list-style-type: decimal">
<li><p>In the case of categorical covariates, graphs of the Kaplan-Meier
estimates of the survival function provide quick and easy checks of
proportional hazards. If proportional hazards holds, <strong>the graphs
of the survival function should look “parallel”</strong>, in the sense
that they should have basically the same shape, should not cross, and
should start close and then diverge slowly through follow up
time.</p></li>
<li><p>Another graphical methods for checking proportional hazards is to
plot log(-log(S(t))) vs. t or log(t) and look for parallelism. This can
be done only for categorical covariates</p>
<pre><code>proc lifetest data=whas500 atrisk plots=survival(atrisk cb) outs=outwhas500;
strata gender;
time lenfol*fstat(0);
run;</code></pre>
<h3 id="schoenfeld-residuals">Schoenfeld Residuals</h3></li>
</ol>
<p>Plotting scaled <strong>Schoenfeld residuals</strong> vs functions of
time to assess proportional hazards of a continuous covariate + <span
class="math display">\[E(s^\star_{kp}) + \hat{\beta}_p \approx
\beta_j(t_k)\]</span> + <span
class="math inline">\(s^\star_{kp}\)</span> is the scaled Schoenfeld
residual of the covariate <span class="math inline">\(p\)</span> at time
<span class="math inline">\(k\)</span>, <span
class="math inline">\(\beta_p\)</span> is the time-invariant
coefficient, and <span class="math inline">\(\beta_j(tk)\)</span> is the
time-varying coefficient. In other words, the average value of the
Schoenfeld residual of the coefficient <span
class="math inline">\(k\)</span> at time <span
class="math inline">\(k\)</span> estimates the coefficient change at
time <span class="math inline">\(k\)</span>. Therefore, if the average
value of the whole time is 0, it means that the coefficient <span
class="math inline">\(p\)</span> will not change with time, and the
proportional hazard assumption applies to the covariate <span
class="math inline">\(p\)</span>.</p>
<!-- SAS Implementation  -->
<pre><code>```
proc phreg data=whas500;
class gender;
model lenfol*fstat(0) = gender|age bmi|bmi hr;
output out=schoen ressch=schgender schage schgenderage
   schbmi schbmibmi schhr;
run;

data schoen;
   set schoen;
   loglenfol = log(lenfol);
run;

proc loess data = schoen;
   model schage=lenfol / smooth=(0.2 0.4 0.6 0.8);
run;
proc loess data = schoen;
   model schage=loglenfol / smooth=(0.2 0.4 0.6 0.8);
run;
```</code></pre>
<!-- R Implementation  -->
<pre class="r"><code>res.cox &lt;- coxph(Surv(time, status) ~ age + sex + wt.loss, data =  lung)
test.ph &lt;- cox.zph(res.cox)
test.ph</code></pre>
<pre><code>##          chisq df    p
## age     0.5077  1 0.48
## sex     2.5489  1 0.11
## wt.loss 0.0144  1 0.90
## GLOBAL  3.0051  3 0.39</code></pre>
<pre class="r"><code># The test of each covariate is not statistically significant, and the global test is not statistically significant. Therefore, we can take proportional risk.

# In principle, Schoenfeld residuals have nothing to do with time. A chart showing a non-random graph showing time changes over time indicates a violation of the PH assumption

ggcoxzph(test.ph)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<pre class="r"><code># The solid line is the smooth spline fit of the curve, and the dashed line represents the +/- 2 standard error band around the fit. System deviation from the horizontal line indicates non-proportional hazard, because proportional hazard assumes that the estimated values β1, β2, and β3 do not change much over time.</code></pre>
</div>
<div id="ph-option-in-sas" class="section level3">
<h3>ph Option in SAS</h3>
<p>Option ph tells SAS that in addition to checking the functional form,
we also want to evaluate the proportional hazard. As before, we specify
the resampling option to request the highest test of the null hypothesis
that the proportional hazard holds.</p>
<p><strong>These tests calculate the proportion of simulated score
processes that produce a maximum score greater than the observed maximum
score process. A small percentage (p-value) indicates that the
percentage is violated</strong></p>
<pre><code>```
proc phreg data=whas500;
    class gender;
    model lenfol*fstat(0) = gender|age bmi|bmi hr;
    assess var=(age bmi bmi*bmi hr) ph / resample ;
run;
```</code></pre>
<table>
<colgroup>
<col width="43%" />
<col width="20%" />
<col width="12%" />
<col width="9%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th>Supremum Test for Proportionals Hazards Assumptio</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Variable</td>
<td>Maximum Absolute Value</td>
<td>Replications</td>
<td>Seed</td>
<td>Pr &gt; MaxAbsVal</td>
</tr>
<tr class="even">
<td>GENDERFemale</td>
<td>0.6394</td>
<td>1000</td>
<td>778428000</td>
<td>0.7680</td>
</tr>
<tr class="odd">
<td>AGE</td>
<td>0.4965</td>
<td>1000</td>
<td>778428000</td>
<td>0.9600</td>
</tr>
<tr class="even">
<td>BMI</td>
<td>5.9813</td>
<td>1000</td>
<td>778428000</td>
<td>0.2890</td>
</tr>
<tr class="odd">
<td>BMIBMI</td>
<td>5.9350</td>
<td>1000</td>
<td>778428000</td>
<td>0.3160</td>
</tr>
<tr class="even">
<td>HR</td>
<td>0.8861</td>
<td>1000</td>
<td>778428000</td>
<td>0.3080</td>
</tr>
</tbody>
</table>
</div>
<div id="dealing-with-nonproportionality" class="section level3">
<h3>Dealing with nonproportionality</h3>
<p>If nonproportional hazards are detected, the researcher has many
options with how to address the violation (Therneau &amp; Grambsch,
2000):</p>
<p>Ignore the nonproportionality if it appears the changes in the
coefficient over time are very small or if it appears the outliers are
driving the changes in the coefficient. In large datasets, very small
departures from proportional hazards can be detected. If, say, a
regression coefficient changes only by 1% over time, it is unlikely that
any overarching conclusions of the study would be affected.
Additionally, a few heavily influential points may be causing
nonproportional hazards to be detected, so it is important to use
graphical methods to ensure this is not the case.</p>
<p><strong>Stratify</strong> the model by the nonproportional covariate.
Stratification allows each stratum to have its own baseline hazard,
which solves the problem of nonproportionality. However, one cannot test
whether the stratifying variable itself affects the hazard rate
significantly. Additionally, although stratifying by a categorical
covariate works naturally, it is often difficult to know how to best
discretize a continuous covariate. This can be easily accomplished in
proc phreh with the strata statement.</p>
<p>Include <strong>covariate interactions with time</strong> as
predictors in the Cox model. This can be accomplished through
programming statements in proc phreg, as these interactions are
time-varying covariates themselves. Indeed, including such an
interaction has been used as a test of proportional hazards — a
significant interaction indicates violation of the assumption. Below, we
provide code that shows how to include a covariate interaction with time
in the model.</p>
</div>
</div>
<div id="model-diagnostics" class="section level2">
<h2>Model Diagnostics</h2>
<div id="cox-snell-residuals" class="section level3">
<h3>Cox-Snell Residuals</h3>
<p>The residual represents the part that the explanatory variable cannot
explain, in the case of censoring <span
class="math display">\[\hat{e}_{i}=\hat{\Lambda}_{0}\left(t_{i}\right)
\exp \left(x_{i}^{T} \beta\right)\]</span></p>
<p><span class="math inline">\(\hat(\Lambda)\)</span> can be estimated
by <span class="math inline">\(KM\)</span> (or <span
class="math inline">\(NA\)</span>). Compare Cox-Snell Residuals with the
standard exponential distribution, because <span
class="math inline">\(Y=\Lambda(t) \sim \operatorname(Exp)(1)]\)</span>
, Get the residual plot. However, this has a little flaw, because it is
not easy for us to observe the situation close to the origin. Therefore,
we use standardized difference for transformation. <span
class="math display">\[
(\hat{\Lambda}(e)-\Lambda(e)) / S E=(\hat{\Lambda}(e)-e) / S E
\]</span> However, the Cox-Snell residual is flawed: it has no way to
tell us where the assumptions of the model are wrong.</p>
</div>
<div id="martingale-residuals" class="section level3">
<h3>Martingale Residuals</h3>
<p>An important tool for assessing the goodness of fit of a model is to
compare the censoring indicator ( 0 for censored, 1 for death) for each
subject to the expected value of that indicator under the proportional
hazards Cox model. If there are no time dependent covariates and if the
survival times are right-censored, this is given by <span
class="math display">\[
m_i=\delta_i-\hat{H}_0\left(t_i\right) \exp \left(z_i^{\prime}
\hat{\beta}\right)
\]</span></p>
<p>It represents the difference between the observed value of the sample
death index <span class="math inline">\(\left(\delta_{i}\right)\)</span>
and the average value <span
class="math inline">\(\left(\hat{e}_{i}\right)\)</span>. Among them, the
average is about the average over time. Positive and negative are
meaningful. Positive, it means that the patient died earlier than the
expected time; negative, it means that the patient lived longer (or
censored). But it is asymmetrical. The maximum value of <span
class="math inline">\(\quad \hat{m}_{i}\)</span> is 1, but the minimum
value is infinite.</p>
</div>
<div id="deviance-residuals" class="section level3">
<h3>Deviance Residuals</h3>
<p><span
class="math display">\[\hat{d}_{i}=\operatorname{sign}\left(\hat{m}_{i}\right)
\sqrt{2\left(\tilde{l}_{i}-l_{i}\right)},\]</span></p>
<p><span class="math inline">\(\hat{m}_{i}=\delta_{i}-\hat{e}_{i}, \quad
l_{i}\)</span> is the corresponding likelihood of i sample (given
model), <span class="math inline">\(\tilde{l }_{i}\)</span> represents
the maximum likelihood (generally the full model). It retains the
positive and negative nature of the residual error, and adds the idea of
likelihood (in fact, many tests are based on deviance), which is
symmetric.</p>
</div>
<div id="schoenfeld-residuals-1" class="section level3">
<h3>Schoenfeld Residuals</h3>
<p>The Schoenfeld residual of the observation j and the covariate p is
defined as the difference between the covariate p of the observation j
and the weighted average of the covariate values of all subjects who are
still at risk when the event occurs on the observation j. <span
class="math display">\[E(s^\star_{kp}) + \hat{\beta}_p \approx
\beta_j(t_k)\]</span></p>
<p>Schoenfeld Residuals are used to validate the above assumptions made
by the Cox model. The Cox model makes three assumptions:</p>
<ol style="list-style-type: decimal">
<li>Common baseline hazard rate <span
class="math inline">\(\lambda(t)\)</span> : At any time <span
class="math inline">\(t\)</span>, all individuals are assumed to
experience the same baseline hazard <span
class="math inline">\(\lambda(t)\)</span>. For example, if a study
consists of males and females belonging to different races and age
groups, then at any time <span class="math inline">\(t\)</span> during
the study, white males who entered the study when they were in the 18-34
years age range are assumed to experience the same baseline hazard <span
class="math inline">\(\lambda(t)\)</span> as Asian females who entered
the study when they <span class="math inline">\(40-64\)</span> year old.
This is clearly a strong assumption and one must validate it before
accepting results of the Cox model for your data set.</li>
<li>Time invariant co-variates <span class="math inline">\(X\)</span> :
The effect of the regression variables <span
class="math inline">\(\boldsymbol{X}\)</span> (a.k.a. co-variates) on
the instantaneous hazard experienced by an individual is assumed to
remain constant over time. For example, if a volunteer enters a study of
cancer risk to smokers, and if the participant’s genetic makeup is a
co-variate, then the effect of their genes on the hazard of contracting
cancer is assumed to remain constant throughout the study.
Notation-wise, <span class="math inline">\(\boldsymbol{X}\left(t_{-}
i\right)=\boldsymbol{X}\)</span> for all <span
class="math inline">\(t_{-} i\)</span>. This is again a rather strong
assumption that should be validated. For example, gene expression can
vary with time in response to a number of factors.</li>
<li>Time invariant regression coefficients <span
class="math inline">\(\beta:\)</span> The regression coefficients <span
class="math inline">\(\boldsymbol{\beta}\)</span> do not vary with time.
In other words, <span
class="math inline">\(\boldsymbol{\beta}\left(\mathrm{t}_{-}
i\right)=\boldsymbol{\beta}\)</span> for all <span
class="math inline">\(t_{-} i\)</span>. One can see that assumptions (2)
and (3) are closely related. For example, the effect of gene expression
increasing with time can be expressed as the coefficient of the gene
regression variable increasing with time.</li>
</ol>
</div>
<div id="influential-observations" class="section level3">
<h3>Influential Observations</h3>
<p>To test influential observations or outliers, we can visualize
either:</p>
<ul>
<li>the deviance residuals or</li>
<li>the dfbeta values</li>
</ul>
<pre class="r"><code># ggcoxdiagnostics(fit, type = , linear.predictions = TRUE)
# type: the type of residuals to present on Y axis. Allowed values include one of 
# c(“martingale”, “deviance”, “score”, “schoenfeld”, “dfbeta”, “dfbetas”, “scaledsch”, “partial”).
ggcoxdiagnostics(res.cox, type = &quot;dfbeta&quot;,
                 linear.predictions = FALSE, ggtheme = theme_bw())</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<pre class="r"><code>ggcoxdiagnostics(res.cox, type = &quot;deviance&quot;,
                 linear.predictions = FALSE, ggtheme = theme_bw())</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-14-2.png" width="672" /></p>
</div>
</div>
<div id="cox-ph-model-in-sas" class="section level2">
<h2>Cox PH Model in SAS</h2>
<div id="model-fitting" class="section level3">
<h3>Model fitting</h3>
<pre><code>proc phreg data = whas500;
    class gender;
    model lenfol*fstat(0) = gender age;
run;

# Output
# “Ties Handling: BRESLOW.” 
   This line refers to the default method for handling ties—two or more 
   observations that have exactly the same event time. Although Breslow’s method 
   is nearly universal

# Analysis of Maximum Likelihood Estimates
   
# Model Fit Statistics
# -2 LOG L
# AIC
# SBC: Schwartz’s Bayesian criterion (SBC) statistics 
   The main purpose is to punish models with more covariates, allowing people to compare non-nested models
   
# Testing Global Null Hypothesis
# Displays test of hypothesis that all coefficients in the model are 0 
  # Likelihood Ratio
  # Socre
  # Wald</code></pre>
</div>
<div id="graphs-of-the-survival-and-baseline-hazard-function"
class="section level3">
<h3>Graphs of the survival and baseline hazard function</h3>
<p><strong>plots=survival</strong></p>
<p>When only plots=survival is specified on the proc phreg statement,
SAS will produce one graph, a “reference curve” of the survival function
at the reference level of all categorical predictors and at the mean of
all continuous predictors.</p>
<pre><code>proc phreg data=whas500 plots=survival;
    class gender;
    model lenfol*fstat(0) = gender age;;
run;</code></pre>
<p><strong>Use the baseline statement to generate survival plots by
group</strong></p>
<pre><code>***Whether it is a survival curve or a hazad curve, it is necessary to use 
   the baseline description and create a small data set of covariate values 
   to estimate the curve of interest;
proc format;
    value gender 0 = &quot;male&quot; 1 = &quot;female&quot;;
run;
data covs;
    format gender gender.;
    input gender age;
    datalines;
    0 69.845947
    1 69.845947
    ;
run;

*** covariates = option in baseline statement to specify the dataset 
*** rowid = If a variable is specified after the option, SAS will create a separate 
    line for each level of the variable in the same graph;

proc phreg data = whas500 plots(overlay)=(survival);
    class gender;
    model lenfol*fstat(0) = gender age;
    baseline covariates=covs out=base / rowid=gender;
run;</code></pre>
<p><strong>hazardratio statement</strong></p>
<p>Using the hazardratio statement and graphs to interpret effects,
particularly interactions. Below, we show how to use the hazardratio
statement to request that SAS estimate 3 hazard ratios at specific
levels of our covariates</p>
<ul>
<li>After the keyword hazardratio, we can optionally apply a label, then
we specify the variable whose levels are to be compared in the hazard,
and finally after the option keyword at we tell SAS at which level of
our other covariates to evaluate this hazard ratio. If the variable
whose hazard rates are to computed is not involved in an interaction,
specification of additional covariates is unncessary since the hazard
ratio is constant across levels of all other covariates (a main
effect).</li>
<li>We calculate the hazard ratio describing a one-unit increase in age,
or <strong>HR(age+1) HR(age)</strong>, for both genders. Notice the =ALL
following gender, which is used only with class variables to request the
hazard ratio at all levels of the class variable.</li>
<li>We also calculate the hazard ratio between females and males, or
<strong>HR(gender=1)HR(gender=0)</strong> at ages 0, 20, 40, 60, and
80.</li>
<li>Finally, we calculate the hazard ratio describing a 5-unit increase
in bmi, or <strong>HR(bmi+5) HR(bmi)</strong>, at clinically revelant
BMI scores. Notice the additional option units=5. BMI classes are
typically separated by about 5 points, so we would like to see how the
hazard ratio between (approximately) adjacent BMI classes changes as bmi
increases.</li>
</ul>
<pre><code>proc phreg data = whas500;
    class gender;
    model lenfol*fstat(0) = gender|age bmi|bmi hr ;
    hazardratio &#39;Effect of 1-unit change in age by gender&#39; age / at(gender=ALL);
    hazardratio &#39;Effect of gender across ages&#39; gender / at(age=(0 20 40 60 80));
    hazardratio &#39;Effect of 5-unit change in bmi across bmi&#39; bmi / at(bmi = (15 18.5 25 30 40)) units=5;
run;</code></pre>
<table>
<colgroup>
<col width="48%" />
<col width="13%" />
<col width="22%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th>Effect of gender across ages: Hazard Ratios for GENDER</th>
<th>Title</th>
<th>Title</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Description</td>
<td>Point Estimate</td>
<td>95% Wald CI Lower Limits</td>
<td>CI Upper Limits</td>
</tr>
<tr class="even">
<td>GENDER Female vs Male At AGE=0</td>
<td>8.247</td>
<td>1.177</td>
<td>57.783</td>
</tr>
<tr class="odd">
<td>GENDER Female vs Male At AGE=20</td>
<td>4.594</td>
<td>1.064</td>
<td>19.841</td>
</tr>
<tr class="even">
<td>GENDER Female vs Male At AGE=40</td>
<td>2.559</td>
<td>0.955</td>
<td>6.857</td>
</tr>
<tr class="odd">
<td>GENDER Female vs Male At AGE=60</td>
<td>1.426</td>
<td>0.837</td>
<td>2.429</td>
</tr>
<tr class="even">
<td>GENDER Female vs Male At AGE=80</td>
<td>0.794</td>
<td>0.601</td>
<td>1.049</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="cox-ph-model-in-r" class="section level2">
<h2>Cox PH Model in R</h2>
<div id="model-fitting-1" class="section level3">
<h3>Model fitting</h3>
<p>Model Assumptions</p>
<p><span class="math display">\[
\begin{equation*}
\begin{array}{rclcl}
\ln \lambda(t) &amp;=&amp; \ln \lambda_{0}(t) + \beta_{1} X_{1} + \dots
+ \beta_{p} X_{p}      &amp;=&amp; \ln \lambda_{0}(t) + \bf{X}
\bf{\beta}\\
    \lambda(t) &amp;=&amp;     \lambda_{0}(t) \, e^{\beta_{1} X_{1} +
\dots + \beta_{p} X_{p}} &amp;=&amp; \lambda_{0}(t) \, e^{\bf{X}
\bf{\beta}}\\
S(t)           &amp;=&amp; S_{0}(t)^{\exp(\bf{X} \bf{\beta})}
&amp;=&amp; \exp\left(-\Lambda_{0}(t) \, e^{\bf{X} \bf{\beta}}\right)\\
\Lambda(t)     &amp;=&amp; \Lambda_{0}(t) \, e^{\bf{X} \bf{\beta}} &amp;
&amp;
\end{array}
\end{equation*}
\]</span></p>
<pre class="r"><code>data(&quot;lung&quot;)
res.cox &lt;- coxph(Surv(time, status) ~ age + sex + ph.ecog, data =  lung)
summary(res.cox)</code></pre>
<pre><code>## Call:
## coxph(formula = Surv(time, status) ~ age + sex + ph.ecog, data = lung)
## 
##   n= 227, number of events= 164 
##    (1 observation deleted due to missingness)
## 
##              coef exp(coef)  se(coef)      z Pr(&gt;|z|)    
## age      0.011067  1.011128  0.009267  1.194 0.232416    
## sex     -0.552612  0.575445  0.167739 -3.294 0.000986 ***
## ph.ecog  0.463728  1.589991  0.113577  4.083 4.45e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
##         exp(coef) exp(-coef) lower .95 upper .95
## age        1.0111     0.9890    0.9929    1.0297
## sex        0.5754     1.7378    0.4142    0.7994
## ph.ecog    1.5900     0.6289    1.2727    1.9864
## 
## Concordance= 0.637  (se = 0.025 )
## Likelihood ratio test= 30.5  on 3 df,   p=1e-06
## Wald test            = 29.93  on 3 df,   p=1e-06
## Score (logrank) test = 30.5  on 3 df,   p=1e-06</code></pre>
<pre class="r"><code>res.cox %&gt;%
  tbl_regression(exponentiate = TRUE)</code></pre>
<div id="zmdowysmlv" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#zmdowysmlv table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#zmdowysmlv thead, #zmdowysmlv tbody, #zmdowysmlv tfoot, #zmdowysmlv tr, #zmdowysmlv td, #zmdowysmlv th {
  border-style: none;
}

#zmdowysmlv p {
  margin: 0;
  padding: 0;
}

#zmdowysmlv .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#zmdowysmlv .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#zmdowysmlv .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#zmdowysmlv .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#zmdowysmlv .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zmdowysmlv .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zmdowysmlv .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zmdowysmlv .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#zmdowysmlv .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#zmdowysmlv .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#zmdowysmlv .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#zmdowysmlv .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#zmdowysmlv .gt_spanner_row {
  border-bottom-style: hidden;
}

#zmdowysmlv .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#zmdowysmlv .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#zmdowysmlv .gt_from_md > :first-child {
  margin-top: 0;
}

#zmdowysmlv .gt_from_md > :last-child {
  margin-bottom: 0;
}

#zmdowysmlv .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#zmdowysmlv .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#zmdowysmlv .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#zmdowysmlv .gt_row_group_first td {
  border-top-width: 2px;
}

#zmdowysmlv .gt_row_group_first th {
  border-top-width: 2px;
}

#zmdowysmlv .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zmdowysmlv .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#zmdowysmlv .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#zmdowysmlv .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zmdowysmlv .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zmdowysmlv .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#zmdowysmlv .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#zmdowysmlv .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#zmdowysmlv .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zmdowysmlv .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zmdowysmlv .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zmdowysmlv .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zmdowysmlv .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zmdowysmlv .gt_left {
  text-align: left;
}

#zmdowysmlv .gt_center {
  text-align: center;
}

#zmdowysmlv .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#zmdowysmlv .gt_font_normal {
  font-weight: normal;
}

#zmdowysmlv .gt_font_bold {
  font-weight: bold;
}

#zmdowysmlv .gt_font_italic {
  font-style: italic;
}

#zmdowysmlv .gt_super {
  font-size: 65%;
}

#zmdowysmlv .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#zmdowysmlv .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#zmdowysmlv .gt_indent_1 {
  text-indent: 5px;
}

#zmdowysmlv .gt_indent_2 {
  text-indent: 10px;
}

#zmdowysmlv .gt_indent_3 {
  text-indent: 15px;
}

#zmdowysmlv .gt_indent_4 {
  text-indent: 20px;
}

#zmdowysmlv .gt_indent_5 {
  text-indent: 25px;
}

#zmdowysmlv .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#zmdowysmlv div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="estimate"><span class='gt_from_md'><strong>HR</strong></span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="conf.low"><span class='gt_from_md'><strong>95% CI</strong></span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="p.value"><span class='gt_from_md'><strong>p-value</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">age</td>
<td headers="estimate" class="gt_row gt_center">1.01</td>
<td headers="conf.low" class="gt_row gt_center">0.99, 1.03</td>
<td headers="p.value" class="gt_row gt_center">0.2</td></tr>
    <tr><td headers="label" class="gt_row gt_left">sex</td>
<td headers="estimate" class="gt_row gt_center">0.58</td>
<td headers="conf.low" class="gt_row gt_center">0.41, 0.80</td>
<td headers="p.value" class="gt_row gt_center"><0.001</td></tr>
    <tr><td headers="label" class="gt_row gt_left">ph.ecog</td>
<td headers="estimate" class="gt_row gt_center">1.59</td>
<td headers="conf.low" class="gt_row gt_center">1.27, 1.99</td>
<td headers="p.value" class="gt_row gt_center"><0.001</td></tr>
  </tbody>
  
  <tfoot class="gt_footnotes">
    <tr>
      <td class="gt_footnote" colspan="4"><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span> <span class='gt_from_md'>HR = Hazard Ratio, CI = Confidence Interval</span></td>
    </tr>
  </tfoot>
</table>
</div>
</div>
<div id="stratified-cox-regression" class="section level3">
<h3>Stratified Cox regression</h3>
<pre class="r"><code>res.cox.strata &lt;- coxph(Surv(time, status) ~ age + strata(sex), data =  lung)
res.cox.strata %&gt;%
  tbl_regression(exponentiate = TRUE)</code></pre>
<div id="smromnwrlr" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#smromnwrlr table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#smromnwrlr thead, #smromnwrlr tbody, #smromnwrlr tfoot, #smromnwrlr tr, #smromnwrlr td, #smromnwrlr th {
  border-style: none;
}

#smromnwrlr p {
  margin: 0;
  padding: 0;
}

#smromnwrlr .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#smromnwrlr .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#smromnwrlr .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#smromnwrlr .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#smromnwrlr .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#smromnwrlr .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#smromnwrlr .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#smromnwrlr .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#smromnwrlr .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#smromnwrlr .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#smromnwrlr .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#smromnwrlr .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#smromnwrlr .gt_spanner_row {
  border-bottom-style: hidden;
}

#smromnwrlr .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#smromnwrlr .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#smromnwrlr .gt_from_md > :first-child {
  margin-top: 0;
}

#smromnwrlr .gt_from_md > :last-child {
  margin-bottom: 0;
}

#smromnwrlr .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#smromnwrlr .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#smromnwrlr .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#smromnwrlr .gt_row_group_first td {
  border-top-width: 2px;
}

#smromnwrlr .gt_row_group_first th {
  border-top-width: 2px;
}

#smromnwrlr .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#smromnwrlr .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#smromnwrlr .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#smromnwrlr .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#smromnwrlr .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#smromnwrlr .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#smromnwrlr .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#smromnwrlr .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#smromnwrlr .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#smromnwrlr .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#smromnwrlr .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#smromnwrlr .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#smromnwrlr .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#smromnwrlr .gt_left {
  text-align: left;
}

#smromnwrlr .gt_center {
  text-align: center;
}

#smromnwrlr .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#smromnwrlr .gt_font_normal {
  font-weight: normal;
}

#smromnwrlr .gt_font_bold {
  font-weight: bold;
}

#smromnwrlr .gt_font_italic {
  font-style: italic;
}

#smromnwrlr .gt_super {
  font-size: 65%;
}

#smromnwrlr .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#smromnwrlr .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#smromnwrlr .gt_indent_1 {
  text-indent: 5px;
}

#smromnwrlr .gt_indent_2 {
  text-indent: 10px;
}

#smromnwrlr .gt_indent_3 {
  text-indent: 15px;
}

#smromnwrlr .gt_indent_4 {
  text-indent: 20px;
}

#smromnwrlr .gt_indent_5 {
  text-indent: 25px;
}

#smromnwrlr .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#smromnwrlr div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="label"><span class='gt_from_md'><strong>Characteristic</strong></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="estimate"><span class='gt_from_md'><strong>HR</strong></span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="conf.low"><span class='gt_from_md'><strong>95% CI</strong></span><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="p.value"><span class='gt_from_md'><strong>p-value</strong></span></th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="label" class="gt_row gt_left">age</td>
<td headers="estimate" class="gt_row gt_center">1.02</td>
<td headers="conf.low" class="gt_row gt_center">1.00, 1.03</td>
<td headers="p.value" class="gt_row gt_center">0.078</td></tr>
  </tbody>
  
  <tfoot class="gt_footnotes">
    <tr>
      <td class="gt_footnote" colspan="4"><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;line-height:0;"><sup>1</sup></span> <span class='gt_from_md'>HR = Hazard Ratio, CI = Confidence Interval</span></td>
    </tr>
  </tfoot>
</table>
</div>
</div>
<div id="baseline-survival-curve" class="section level3">
<h3>Baseline Survival Curve</h3>
<p>An estimate of the baseline hazard function is given by <span
class="math display">\[
h_0\left(t_i\right)=\frac{d_i}{\sum_{j \in R_j} \exp \left(z_j
\hat{\beta}\right)} .
\]</span></p>
<pre class="r"><code># Visualizing the estimated distribution of survival times
# Plot the baseline survival function
ggsurvplot(survfit(res.cox), data=lung, 
           palette = &quot;#2E9FDF&quot;,
           ggtheme = theme_minimal())</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<pre class="r"><code># Assess the impact of gender on the estimated survival rate
sex_df &lt;- with(lung,
               data.frame(sex = c(1, 2), 
                          age = rep(mean(age, na.rm = TRUE), 2),
                          ph.ecog = c(1, 1)
                          )
               )

# Survival curves
fit &lt;- survfit(res.cox, newdata = sex_df)
ggsurvplot(fit, data=sex_df, conf.int = TRUE, 
           legend.labs=c(&quot;Sex=1&quot;, &quot;Sex=2&quot;),
           ggtheme = theme_minimal())</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-17-2.png" width="672" /></p>
<p>Using Residuals</p>
</div>
<div id="assessing-goodness-of-fit" class="section level3">
<h3>Assessing Goodness of Fit</h3>
<ul>
<li>MartingaleResidual</li>
<li>AIC</li>
<li>Model comparisons</li>
</ul>
<pre class="r"><code>res.cox1 &lt;- coxph(Surv(time, status) ~ age + sex, data =  lung)

## Martingale Residuals
rr.0 &lt;- residuals(res.cox1, type=&quot;martingale&quot;)
par(mfrow=c(1,2))
plot(rr.0 ~ lung$age)
logAge &lt;- log(lung$age)
plot(rr.0 ~ logAge)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<pre class="r"><code>par(mfrow=c(1,1))

## AIC
## library(survival)
extractAIC(res.cox1)</code></pre>
<pre><code>## [1]    2.000 1489.696</code></pre>
<pre class="r"><code>## Model comparisons
res.cox2 &lt;- coxph(Surv(time, status) ~ age, data =  lung)
anova(res.cox2, res.cox1)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["loglik"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["Chisq"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["Df"],"name":[3],"type":["int"],"align":["right"]},{"label":["Pr(>|Chi|)"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"-747.7894","2":"NA","3":"NA","4":"NA","_rn_":"1"},{"1":"-742.8482","2":"9.882213","3":"1","4":"0.001668841","_rn_":"2"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="univariate-coxph-function-to-multiple-covariate"
class="section level3">
<h3>Univariate coxph function to multiple covariate</h3>
<pre class="r"><code>#  apply the univariate coxph function to multiple covariates at once
covariates &lt;- c(&quot;age&quot;, &quot;sex&quot;,  &quot;ph.karno&quot;, &quot;ph.ecog&quot;, &quot;wt.loss&quot;)
univ_formulas &lt;- sapply(covariates,
                        function(x) as.formula(paste(&#39;Surv(time, status)~&#39;, x)))
univ_models &lt;- lapply( univ_formulas, function(x){coxph(x, data = lung)})

# Surv(time, status) ~ age
# Surv(time, status) ~ sex
# Surv(time, status) ~ ph.karno
# Surv(time, status) ~ ph.ecog
# Surv(time, status) ~ wt.loss
# Surv(time, status) ~ sex
 
# Extract data 
univ_results &lt;- lapply(univ_models,
                       function(x){ 
                          x &lt;- summary(x)
                          p.value&lt;-signif(x$wald[&quot;pvalue&quot;], digits=2)
                          wald.test&lt;-signif(x$wald[&quot;test&quot;], digits=2)
                          beta&lt;-signif(x$coef[1], digits=2);#coeficient beta
                          HR &lt;-signif(x$coef[2], digits=2);#exp(beta)
                          HR.confint.lower &lt;- signif(x$conf.int[,&quot;lower .95&quot;], 2)
                          HR.confint.upper &lt;- signif(x$conf.int[,&quot;upper .95&quot;],2)
                          HR &lt;- paste0(HR, &quot; (&quot;, 
                                       HR.confint.lower, &quot;-&quot;, HR.confint.upper, &quot;)&quot;)
                          res&lt;-c(beta, HR, wald.test, p.value)
                          names(res)&lt;-c(&quot;beta&quot;, &quot;HR (95% CI for HR)&quot;, &quot;wald.test&quot;, 
                                        &quot;p.value&quot;)
                          return(res)
                          # return(exp(cbind(coef(x),confint(x)))) # For KI
                         })

univ_results %&gt;% 
  as.data.frame(check.names = FALSE) %&gt;%
  t() %&gt;%
  kable(caption = &quot;Univariate coxph function to multiple covariate&quot;, format = &quot;html&quot;) %&gt;%
  kable_styling(latex_options = &quot;striped&quot;)</code></pre>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
Univariate coxph function to multiple covariate
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
HR (95% CI for HR)
</th>
<th style="text-align:left;">
wald.test
</th>
<th style="text-align:left;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
age
</td>
<td style="text-align:left;">
0.019
</td>
<td style="text-align:left;">
1 (1-1)
</td>
<td style="text-align:left;">
4.1
</td>
<td style="text-align:left;">
0.042
</td>
</tr>
<tr>
<td style="text-align:left;">
sex
</td>
<td style="text-align:left;">
-0.53
</td>
<td style="text-align:left;">
0.59 (0.42-0.82)
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
0.0015
</td>
</tr>
<tr>
<td style="text-align:left;">
ph.karno
</td>
<td style="text-align:left;">
-0.016
</td>
<td style="text-align:left;">
0.98 (0.97-1)
</td>
<td style="text-align:left;">
7.9
</td>
<td style="text-align:left;">
0.005
</td>
</tr>
<tr>
<td style="text-align:left;">
ph.ecog
</td>
<td style="text-align:left;">
0.48
</td>
<td style="text-align:left;">
1.6 (1.3-2)
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
2.7e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
wt.loss
</td>
<td style="text-align:left;">
0.0013
</td>
<td style="text-align:left;">
1 (0.99-1)
</td>
<td style="text-align:left;">
0.05
</td>
<td style="text-align:left;">
0.83
</td>
</tr>
</tbody>
</table>
</div>
<div id="model-diagnostics-1" class="section level3">
<h3>Model diagnostics</h3>
<div id="proportional-hazards-assumption" class="section level4">
<h4>Proportional hazards assumption</h4>
<p>Complementary log-log plot for categorized predictor Sex.</p>
<pre class="r"><code># Plot the survival curves using ggsurvplot
fit &lt;- survfit(res.cox, newdata = sex_df)
ggsurvplot(fit, data=sex_df, conf.int = F, 
           legend.labs=c(&quot;Sex=1&quot;, &quot;Sex=2&quot;),
           ggtheme = theme_minimal(),
           fun = &quot;cloglog&quot;) # This argument transforms the survival probability</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="scaled-schoenfeld-residuals" class="section level4">
<h4>Scaled Schoenfeld-residuals</h4>
<pre class="r"><code>fitCPH &lt;- coxph(Surv(time, status) ~ age + sex + wt.loss, data =  lung)
(czph &lt;- cox.zph(fitCPH))</code></pre>
<pre><code>##          chisq df    p
## age     0.5077  1 0.48
## sex     2.5489  1 0.11
## wt.loss 0.0144  1 0.90
## GLOBAL  3.0051  3 0.39</code></pre>
<pre class="r"><code>par(mfrow=c(2, 2), cex.main=1.4, cex.lab=1.4)
plot(czph)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
</div>
<div id="influential-observations-1" class="section level4">
<h4>Influential observations</h4>
<pre class="r"><code>dfbetas &lt;- residuals(fitCPH, type=&quot;dfbetas&quot;)

par(mfrow=c(2, 2), cex.main=1.4, cex.lab=1.4)
plot(dfbetas[ , 1], type=&quot;h&quot;, main=&quot;DfBETAS for age&quot;,    ylab=&quot;DfBETAS&quot;, lwd=2)
plot(dfbetas[ , 2], type=&quot;h&quot;, main=&quot;DfBETAS for Sex&quot;, ylab=&quot;DfBETAS&quot;, lwd=2)
plot(dfbetas[ , 3], type=&quot;h&quot;, main=&quot;DfBETAS for wt.loss&quot;, ylab=&quot;DfBETAS&quot;, lwd=2)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
<div id="linearity-of-log-hazard" class="section level4">
<h4>Linearity of log hazard</h4>
<pre class="r"><code>fitCPHAge &lt;- coxph(Surv(time, status) ~ age, data =  lung)
resMart &lt;- residuals(fitCPHAge, type=&quot;martingale&quot;)
par(mfrow=c(1, 1), cex.main=1.4, cex.lab=1.4)
plot(lung$age, resMart, main=&quot;Martingale-residuals for age&quot;,
     xlab=&quot;age&quot;, ylab=&quot;Residuen&quot;, pch=20)
lines(loess.smooth(lung$age, resMart), lwd=2, col=&quot;blue&quot;)
legend(x=&quot;bottomleft&quot;, col=&quot;blue&quot;, lwd=2, legend=&quot;LOESS fit&quot;, cex=1.2)</code></pre>
<p><img src="06-Analysis-Survival-Analysis_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="frailty-model" class="section level2">
<h2>Frailty Model</h2>
<p>Ordinary survival models deal with the simplest case of independent
and identically distributed data. This is based on the assumption that
the study population is homogeneous. But it is a basic observation of
medical statistics that individuals differ greatly. So do the effects of
a drug, or the influence of various explanatory variables. This
heterogeneity is often referred to as variability and it is generally
recognized as one of the most important sources of variability in
medical and biological applications. This heterogeneity may be difficult
to assess, but it is nevertheless of great importance.</p>
<p>The key idea of ‘frailty models’ models is that individuals have
different frailties, and that the most frail will die earlier than the
lesser frail. Consequently, systematic selection of robust individuals
takes place, which distorts what is observed. When mortality rates are
estimated, one may be interested in how they change over time or age.
Quite often, they rise at the beginning of the observation period, reach
a maximum and then decline (unimodal intensity) or level off. This, for
example, is typical for death rates of cancer patients, meaning that the
longer the patient lives beyond a certain time, the better his or her
chances of survival are.</p>
<p>Two sources of variability in duration data are useful to
consider</p>
<ol style="list-style-type: decimal">
<li>variability accounted for by observable risk factors (which is thus
theoretically predictable)</li>
<li>heterogeneity caused by unknown covariates and which is therefore
theoretically unpredictable even when all relevant information at that
time is known.</li>
</ol>
<p>It is the latter which is of specific interest here, and the subject
of observable covariates is treated here only for completeness. As
Hougaard (1991) pointed out, there are advantages in considering these
two sources of variability separately: heterogeneity explains some
‘unexpected’ results or gives an alternative explanation of some
results, for example non-proportional or decreasing hazards. If some
individuals experience a higher risk of failure, then the remaining
individuals at risk tend to form a more or less selected group with
lower risk. An estimate of the individual hazard rate without taking
into account the unobserved frailty will thus underestimate the hazard
function to an increasingly greater extent as time goes by.</p>
<p>To address the problem of heterogeneity in a population resulting
from unobserved covariates, Vaupel et al. (1979) suggested a random
effects model for durations. They introduced the notion of frailty and
applied it to population data. The classical and mostly applied frailty
model assumes a proportional hazards model that is conditional on the
random effect (frailty). To be more specific, the hazard of an
individual depends additionally on an unobservable, age-independent
random variable Z, which acts multiplicatively on the baseline hazard
function λ0, where λ(t, Z) = Zλ0(t)</p>
<p>Here, Z is considered as a random mixture variable, varying across
the population. Note that a scale factor common to all subjects in the
population maybe absorbed into the baseline hazard function λ0(t), so
that frailty distributions are standardized to EZ = 1. The variance
parameter σ 2 = V(Z) is interpretable as a measure of heterogeneity
across the population in baseline risk. When σ 2 is small, then the
values of Z are closely concentrated around one. If σ 2 is large, then
values of Z are more dispersed, inducing greater heterogeneity in the
individual hazards Zλ0(t). Frailty increases the individual risk and is
sometimes called liability or susceptibility in other settings. All
individuals, apart from an individual constant Z are assumed to follow
the same mortality pattern. What may be observed in a population is not
the individual hazard, but the net result for a number of individuals
with different values of the random variable Z.</p>
</div>
</div>
<div id="restricted-mean-survival-time" class="section level1">
<h1>Restricted Mean Survival Time</h1>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>Restricted mean survival time (RMST) is a useful statistical measure
in survival analysis, particularly when dealing with censored data,
which is common in clinical trials and other medical studies.</p>
<ol style="list-style-type: decimal">
<li><p><strong>Definition</strong>: RMST represents the average expected
survival time up to a specified time point, <span
class="math inline">\(t\)</span>, known as the truncation or horizon
time. It is calculated as the area under the survival curve from the
beginning of observation until time <span
class="math inline">\(t\)</span>.</p></li>
<li><p><strong>Calculation</strong>: Mathematically, RMST is defined as:
<span class="math display">\[
\text{RMST}(t) = \int_0^t S(u) \, du
\]</span> where <span class="math inline">\(S(u)\)</span> is the
survival function at time <span class="math inline">\(u\)</span>. This
integral represents the expected time a subject remains in the study up
to time <span class="math inline">\(t\)</span>, without experiencing the
event of interest (like death or disease recurrence).</p></li>
<li><p><strong>Usefulness</strong>: RMST is particularly helpful when
the proportional hazards assumption in a Cox model does not hold,
providing a straightforward and easily interpretable summary of the
survival experience of a cohort. It effectively compares the average
survival times between different treatment groups up to time <span
class="math inline">\(t\)</span>, making it a direct measure of the
“life years gained” due to a treatment in a clinical setting.</p></li>
<li><p><strong>Advantages</strong>:</p>
<ul>
<li>RMST is valid under any of the time-to-event distributions</li>
<li>It’s a measure of survival time (easier to interpret)</li>
<li>Non-dependency on PH: RMST does not rely on the PH assumption and is
valid under any survival distribution.</li>
<li>Provides a more direct interpretation relative to hazard ratios, as
it can be expressed in units of time.</li>
</ul></li>
<li><p><strong>Applications</strong>: RMST can be used to:</p>
<ul>
<li>Compare the effectiveness of different treatments.</li>
<li>Estimate the impact of covariates on survival time.</li>
<li>Provide a summary measure of survival data in health economic
evaluations.</li>
</ul></li>
</ol>
</div>
<div id="problems-with-proportional-hazards-ph" class="section level2">
<h2>Problems with Proportional Hazards (PH)</h2>
<p><strong>Background on Hazard Ratio (HR)</strong></p>
<ul>
<li><strong>Definition</strong>: HR is commonly used in RCTs to compare
the hazard functions between a treatment and a control group, aiming for
values less than 1 to signify beneficial treatment effects.</li>
<li><strong>Assumption</strong>: For HR to be valid, it assumes that the
PH condition holds, meaning the ratio of the hazards (risk of an event
per time unit) between two groups remains constant over time.</li>
<li><strong>Issues</strong>: When PH is violated, such as in cases where
treatments have different short-term and long-term effects (e.g.,
surgical vs. medical treatment), HR can become a misleading indicator of
treatment effect. This is because it might not accurately represent the
change in risk over the entire period of observation.</li>
</ul>
<p><strong>Problems with Proportional Hazards (PH)</strong></p>
<ul>
<li><strong>Violation of PH</strong>: In real-world scenarios, such as
in cancer trials or those involving advanced therapeutics, the
assumption of constant hazard ratios over time often fails. This can be
due to varying biological responses to treatments or differing
subpopulations within the study.</li>
<li><strong>Implications</strong>: When PH is violated, using HR can
lead to incorrect interpretations and decisions, potentially impacting
clinical recommendations and patient care.</li>
</ul>
<p><strong>Advocating for RMST</strong></p>
<ul>
<li><strong>Concept</strong>: RMST measures the expected time until an
event within a specified time frame, integrating the survival curve up
to a truncation time point, <span class="math inline">\(t^*\)</span>.
This can provide a direct measure of the “life expectancy” impact of a
treatment.</li>
<li><strong>Benefits</strong>:
<ul>
<li><strong>Non-dependency on PH</strong>: RMST does not rely on the PH
assumption and is valid under any survival distribution.</li>
<li><strong>Interpretability</strong>: It gives a straightforward,
time-based measure of treatment effect, which can be more clinically
meaningful than HR.</li>
<li><strong>Flexibility</strong>: It accommodates varying survival
distributions, making it suitable for complex clinical scenarios.</li>
</ul></li>
</ul>
<p>The uploaded content appears to define and discuss the Restricted
Mean Survival Time (RMST) and its related concepts in detail. Let’s
break it down for clearer understanding:</p>
</div>
<div id="definition-and-calculation" class="section level2">
<h2>Definition and Calculation</h2>
<div id="rmst" class="section level3">
<h3>RMST</h3>
<ol style="list-style-type: decimal">
<li><strong>Restricted Mean Survival Time (RMST)</strong>:
<ul>
<li><strong>Definition</strong>: RMST is the mean survival time of a
random variable <span class="math inline">\(T\)</span> up to a specified
time <span class="math inline">\(t^*\)</span>. It represents the
expected value of the survival times truncated at <span
class="math inline">\(t^*\)</span>, which is mathematically expressed as
the area under the survival curve <span
class="math inline">\(S(t)\)</span> from time 0 to <span
class="math inline">\(t^*\)</span>.</li>
<li><strong>Formula</strong>: <span class="math display">\[
\mu = E(X) = E[\min(T, t^*)] = \int_0^{t^*} S(t) dt
\]</span></li>
<li>This concept provides a practical measure of life expectancy up to
<span class="math inline">\(t^*\)</span>, accounting for all subjects as
if their survival times were censored at <span
class="math inline">\(t^*\)</span>.</li>
</ul></li>
<li><strong>Difference in RMST (<span
class="math inline">\(\Delta\)</span>)</strong>:
<ul>
<li>When comparing two groups (e.g., a treatment group and a control
group with survival functions <span
class="math inline">\(S_1(t)\)</span> and <span
class="math inline">\(S_0(t)\)</span>), the difference in RMST between
the groups is represented by: <span class="math display">\[
\Delta = \int_0^{t^*} (S_1(t) - S_0(t)) dt
\]</span></li>
<li>This difference <span class="math inline">\(\Delta\)</span> can be
interpreted as the area between the two survival curves over the
interval from 0 to <span class="math inline">\(t^*\)</span>, effectively
quantifying the treatment effect in terms of survival time.</li>
</ul></li>
</ol>
</div>
<div id="restricted-standard-deviation-of-survival-time-rsdst"
class="section level3">
<h3>Restricted Standard Deviation of Survival Time (RSDST)</h3>
<ol style="list-style-type: decimal">
<li><strong>Calculation of RSDST</strong>:
<ul>
<li><strong>Variance of Restricted Survival Time <span
class="math inline">\(X\)</span></strong>: The variance, denoted as
<span class="math inline">\(\text{var}(X)\)</span>, is necessary to
compute the standard error of the difference in RMST. It is derived from
the second moment of <span class="math inline">\(X\)</span>: <span
class="math display">\[
E(X^2) = t^*{}^2 [1 - S(t^*)] - 2 \int_0^{t^*} t [1 - S(t)] dt + 2
\int_0^{t^*} t S(t) dt
\]</span></li>
<li><strong>Formula for Variance</strong>: <span class="math display">\[
\text{var}(X) = E(X^2) - [E(X)]^2 = 2 \int_0^{t^*} t S(t) dt -
\left(\int_0^{t^*} S(t) dt \right)^2
\]</span></li>
<li><strong>Restricted Standard Deviation (RSDST)</strong>: It is the
square root of the variance: <span class="math display">\[
\text{RSDST} = \sqrt{\text{var}(X)}
\]</span></li>
</ul></li>
</ol>
</div>
</div>
<div id="integrating-rmst-into-analysis" class="section level2">
<h2>Integrating RMST into Analysis</h2>
<ol style="list-style-type: decimal">
<li><strong>Transition from HR to RMST</strong>:
<ul>
<li>Instead of relying solely on the HR as the summary statistic, which
may not adequately represent the trial results under non-proportional
hazards, RMST offers a more comprehensive measure. RMST directly
quantifies the average survival time up to a specified time point <span
class="math inline">\(t^*\)</span>, making it a more tangible measure of
treatment effect.</li>
</ul></li>
<li><strong>Sample Size Calculation and Results Presentation</strong>:
<ul>
<li>Traditional ART sample size calculations are based on expected
differences in HRs tested via the logrank test. By incorporating RMST,
the sample size calculation can focus on detecting a significant
difference in the mean survival times between the arms up to <span
class="math inline">\(t^*\)</span>.</li>
</ul></li>
<li><strong>Defining Time Horizon <span
class="math inline">\(t^*\)</span></strong>:
<ul>
<li>A critical component of using RMST is defining a suitable time
horizon <span class="math inline">\(t^*\)</span> for evaluating the
treatment effect. This horizon should be chosen based on clinical
relevance and expected treatment dynamics over time.</li>
</ul></li>
<li><strong>Flexibility in Trial Design</strong>:
<ul>
<li>RMST allows for greater flexibility in addressing varied survival
distributions and accommodating trials where the proportional hazards
assumption does not hold.</li>
</ul></li>
<li><strong>Improved Interpretability</strong>:
<ul>
<li>RMST provides results that are easily interpretable by clinicians
and policymakers, offering a clear measure of how long, on average,
patients benefit from the treatment up to <span
class="math inline">\(t^*\)</span>, which can be more meaningful than
the proportional risk reduction implied by HR.</li>
</ul></li>
</ol>
</div>
</div>
<div id="reference" class="section level1">
<h1>Reference</h1>
<p>Royston, P., Parmar, M.K. Restricted mean survival time: an
alternative to the hazard ratio for the design and analysis of
randomized trials with a time-to-event outcome. BMC Med Res Methodol 13,
152 (2013). <a href="https://doi.org/10.1186/1471-2288-13-152"
class="uri">https://doi.org/10.1186/1471-2288-13-152</a></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
